{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Installation de bibliotheques \n",
    "\n",
    "**pip install beautifulsoup4**\n",
    "\n",
    "**pip install requests**\n",
    "\n",
    "**pip install PyPDF2**\n",
    "\n",
    "**pip install pdfplumber**\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import sqlite3\n",
    "import time\n",
    "from datetime import datetime\n",
    "\n",
    "import pdfplumber\n",
    "import re, io\n",
    "import PyPDF2\n",
    "from io import BytesIO\n",
    "import os\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Scraper les articles bioRxiv liés à la bioinformatique "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Parcourir les articles de la collection Bioinformatics contenant au total 2153 pages d'articles, ainsi scraper les 10 pages à chaque exécution en enregistrant la dernière page scrappée et leurs informations dans la database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import sqlite3\n",
    "import time\n",
    "from datetime import datetime\n",
    "\n",
    "url = 'https://www.biorxiv.org'\n",
    "\n",
    "# créer la base de données SQLite \"bioinformatics_articles\" \n",
    "database = sqlite3.connect('bioinformatics_articles.db')\n",
    "c = database.cursor() \n",
    "\n",
    "# créer la table 'articles' si elle n'existe pas \n",
    "c.execute('''CREATE TABLE IF NOT EXISTS articles\n",
    "             (id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "             title TEXT, \n",
    "             link TEXT,\n",
    "             doi TEXT UNIQUE,\n",
    "             date TEXT,\n",
    "             pdf_link TEXT,\n",
    "             abstract TEXT)''')\n",
    "\n",
    "# Créer une table pour stocker la dernière page scrappée\n",
    "c.execute('''CREATE TABLE IF NOT EXISTS scrap_info (\n",
    "             last_page INTEGER)''')\n",
    "\n",
    "# Récupérer la dernière page scrappée\n",
    "c.execute(\"SELECT last_page FROM scrap_info\")\n",
    "result = c.fetchone()\n",
    "last_page = result[0] if result else 0  # Si aucune page n'a été scrappée, commencer à la page 1\n",
    "start_page = last_page + 1\n",
    "\n",
    "## Nombre de pages à scraper (exemple : 10 pages à la fois)\n",
    "pages_to_scrap_per_run = 10\n",
    "total_pages = min(start_page + pages_to_scrap_per_run - 1, 2153)  # Limiter à 2153 pages\n",
    "\n",
    "# boucle qui parcourt chaque page de la collection Bioinformatics\n",
    "for page in range(start_page, total_pages + 1):\n",
    "    print(f\"Scraping page {page} sur {total_pages}\")\n",
    "\n",
    "    # requête qui récupère la page bioinfo actuelle\n",
    "    response = requests.get(f'{url}/collection/bioinformatics?page={page}')\n",
    "    soup = BeautifulSoup(response.content, 'html.parser')\n",
    "\n",
    "    # Trouver les liens vers les articles sur cette page\n",
    "    articles = soup.find_all('a', class_='highwire-cite-linked-title')\n",
    "    print(f'Nombre total d\\'articles trouvés sur la page {page} : {len(articles)}')\n",
    "\n",
    "    # Parcourir tous les articles de la page et enregistrer leurs titres et liens\n",
    "    for article in articles:\n",
    "        title = article.text.strip()\n",
    "        link = url + article['href']\n",
    "\n",
    "        # Requête pour accéder à la page de l'article\n",
    "        article_response = requests.get(link)\n",
    "        article_soup = BeautifulSoup(article_response.content, 'html.parser')\n",
    "\n",
    "        # extraire le DOI\n",
    "        doi_tag = article_soup.find('meta', {'name': 'citation_doi'})\n",
    "        doi = doi_tag['content'] if doi_tag else 'DOI non disponible'\n",
    "\n",
    "        # extraire la date de publication\n",
    "        date_tag = article_soup.find('meta', {'name': 'citation_publication_date'})\n",
    "        date = date_tag['content'] if date_tag else 'Date non disponible'\n",
    "        if date != 'Date non disponible':\n",
    "            date = date.replace('/', '-')\n",
    "            date = datetime.strptime(date, '%Y-%m-%d').date().isoformat()\n",
    "\n",
    "        # extraire le lien vers le fichier PDF\n",
    "        pdf_links = article_soup.find_all('a')\n",
    "        pdf_link = next((url + pdf['href'] for pdf in pdf_links if 'PDF' in pdf.text and pdf['href'].endswith('.pdf')), 'Lien PDF non disponible')\n",
    "\n",
    "        # Extraire l'abstract\n",
    "        abstract_tag = article_soup.find('meta', {'name': 'citation_abstract'})\n",
    "        clean_abstract = abstract_tag['content'] if abstract_tag else 'Abstract non disponible'\n",
    "\n",
    "        # Vérifier si l'article existe déjà dans la base de données\n",
    "        c.execute(\"SELECT * FROM articles WHERE doi = ? AND date = ?\", (doi, date))\n",
    "        result = c.fetchone()\n",
    "\n",
    "        if result:\n",
    "            print(f\"L'article avec DOI {doi} existe déjà, ignoré.\")\n",
    "        else:\n",
    "            # Insérer l'article dans la base de données\n",
    "            c.execute(\"INSERT INTO articles (title, link, doi, date, pdf_link, abstract) VALUES (?, ?, ?, ?, ?, ?)\", \n",
    "                      (title, link, doi, date, pdf_link, clean_abstract))\n",
    "            print(f\"Article ajouté : {title}\")\n",
    "\n",
    "        time.sleep(1)  # pause d'une seconde pour éviter de surcharger le serveur\n",
    "\n",
    "    # Sauvegarder les modifications dans la base de données après chaque page\n",
    "    database.commit()\n",
    "\n",
    "    # Mettre à jour la dernière page scrappée dans la table 'scrap_info'\n",
    "    c.execute(\"UPDATE scrap_info SET last_page = ?\", (page,))\n",
    "    if c.rowcount == 0:  \n",
    "        c.execute(\"INSERT INTO scrap_info (last_page) VALUES (?)\", (page,))\n",
    "\n",
    "# Sauvegarder la dernière page dans la base de données\n",
    "database.commit()\n",
    "\n",
    "# Fermer la connexion à la base de données\n",
    "database.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Extraction des URLs des Dépôts Logiciels"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "1.\tVérification et ajout des colonnes nécessaires dans la base de données.\n",
    "\n",
    "2.\tValidation des liens pour s’assurer qu’ils sont valides et pertinents.\n",
    "\n",
    "3.\tVérification de l’accessibilité des URLs via des requêtes HTTP.\n",
    "\n",
    "4.\tExtraction des URLs des dépôts à partir des contenus des PDF et des abstracts.\n",
    "\n",
    "5.\tTraitement des nouveaux articles non encore traités pour l’extraction.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Connexion à la base de données SQLite\n",
    "database = sqlite3.connect('bioinformatics_articles.db')\n",
    "c = database.cursor()\n",
    "\n",
    "# Vérifier et ajouter les colonnes software_links et processed si elles n'existent pas\n",
    "try:\n",
    "    c.execute(\"ALTER TABLE articles ADD COLUMN software_links TEXT\")\n",
    "except sqlite3.OperationalError:\n",
    "    pass  # La colonne existe déjà\n",
    "\n",
    "try:\n",
    "    c.execute(\"ALTER TABLE articles ADD COLUMN processed INTEGER DEFAULT 0\")\n",
    "except sqlite3.OperationalError:\n",
    "    pass  # La colonne existe déjà\n",
    "\n",
    "# Fonction pour vérifier si un lien est valide et s'il est un lien GitHub ou GitLab\n",
    "def is_valid_link(link):\n",
    "    return link.startswith('http://') or link.startswith('https://')\n",
    "\n",
    "def is_github_or_gitlab(link):\n",
    "    return 'github.com' in link or 'gitlab.com' in link\n",
    "\n",
    "# Fonction pour nettoyer les liens partiels ou incorrects\n",
    "def clean_link(link):\n",
    "    link = link.strip(').;,[]')  # Supprimer les caractères indésirables\n",
    "    if is_valid_link(link) and len(link) > len('https://github.com/'):\n",
    "        return link\n",
    "    return None\n",
    "\n",
    "# Fonction pour tester la validité des URLs via une requête HTTP\n",
    "def check_url_validity(link):\n",
    "    try:\n",
    "        response = requests.get(link, timeout=10)\n",
    "        if response.status_code == 200:\n",
    "            print(f\"Le lien est valide : {link}\")\n",
    "            return True\n",
    "        else:\n",
    "            print(f\"Le lien a retourné un code de statut {response.status_code} pour : {link}\")\n",
    "            return False\n",
    "    except requests.RequestException:\n",
    "        print(f\"Erreur lors de la vérification du lien : {link}\")\n",
    "        return False\n",
    "\n",
    "# Fonction pour extraire les URLs des dépôts logiciels\n",
    "def extract_repository_urls(text):\n",
    "    url_pattern = r'(https?://[^\\s]+)'\n",
    "    urls = re.findall(url_pattern, text)\n",
    "    repo_urls = []\n",
    "\n",
    "    for url in urls:\n",
    "        cleaned_url = clean_link(url)\n",
    "        if cleaned_url and is_github_or_gitlab(cleaned_url):\n",
    "            if check_url_validity(cleaned_url):  # Vérification de l'URL avant de l'ajouter\n",
    "                repo_urls.append(cleaned_url)\n",
    "\n",
    "    return repo_urls\n",
    "\n",
    "# Fonction pour extraire les liens de dépôt logiciel\n",
    "def extract_software_links(article_id, pdf_link, abstract):\n",
    "    software_links_set = set()\n",
    "\n",
    "    if pdf_link and pdf_link != 'Lien PDF non disponible':\n",
    "        try:\n",
    "            print(f\"Ouverture du PDF pour l'article {article_id}: {pdf_link}\")\n",
    "            response = requests.get(pdf_link)\n",
    "            response.raise_for_status()\n",
    "\n",
    "            with pdfplumber.open(io.BytesIO(response.content)) as pdf:\n",
    "                print(f\"Le PDF a été ouvert avec succès pour l'article {article_id}. Recherche de liens...\")\n",
    "                for page in pdf.pages:\n",
    "                    text = page.extract_text()\n",
    "                    if text:\n",
    "                        links_from_pdf = extract_repository_urls(text)\n",
    "                        if links_from_pdf:\n",
    "                            software_links_set.update(links_from_pdf)\n",
    "                            print(f\"Liens trouvés dans le PDF pour l'article {article_id} : {links_from_pdf}\")\n",
    "\n",
    "        except Exception as e:\n",
    "            print(f\"Erreur lors de l'ouverture du PDF pour l'article {article_id}: {e}\")\n",
    "\n",
    "    if abstract and abstract != 'Abstract non disponible':\n",
    "        print(f\"Vérification de l'abstract pour l'article {article_id}.\")\n",
    "        links_from_abstract = extract_repository_urls(abstract)\n",
    "        if links_from_abstract:\n",
    "            software_links_set.update(links_from_abstract)\n",
    "            print(f\"Liens trouvés dans l'abstract pour l'article {article_id} : {links_from_abstract}\")\n",
    "\n",
    "    valid_links = [clean_link(link) for link in software_links_set if clean_link(link) and is_github_or_gitlab(link)]\n",
    "\n",
    "    if valid_links:\n",
    "        print(f\"Liens de dépôt logiciel valides pour l'article {article_id} : {valid_links}\")\n",
    "        c.execute(\"UPDATE articles SET software_links = ?, processed = 1 WHERE id = ?\", (', '.join(valid_links), article_id))\n",
    "        print(f\"Liens ajoutés à la base de données pour l'article {article_id}.\")\n",
    "    else:\n",
    "        print(f\"Aucun lien de dépôt logiciel trouvé pour l'article {article_id}.\")\n",
    "     \n",
    "    # Enregistrer les modifications dans la base de données\n",
    "    database.commit()\n",
    "\n",
    "    \n",
    "    time.sleep(1)\n",
    "\n",
    "# Fonction pour traiter uniquement les nouveaux articles\n",
    "def process_new_articles():\n",
    "    c.execute(\"SELECT id, title, pdf_link, abstract FROM articles WHERE processed = 0\")\n",
    "    articles_to_process = c.fetchall()\n",
    "\n",
    "    for article in articles_to_process:\n",
    "        article_id, title, pdf_link, abstract = article\n",
    "        print(f\"Traitement de l'article : {title} (ID: {article_id})...\")\n",
    "        extract_software_links(article_id, pdf_link, abstract)\n",
    "\n",
    "# Appel de la fonction principale\n",
    "try:\n",
    "    process_new_articles()\n",
    "except KeyboardInterrupt:\n",
    "    print(\"Traitement interrompu. Les données sont sauvegardées.\")\n",
    "finally:\n",
    "    database.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  FOR ME --> SUPPRIMER PLUS TARD : Affichage de mes articles dans la base de données "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#le faire en cas de soucis and need tout recommencer \n",
    "#c.execute(\"DROP TABLE IF EXISTS articles\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "id | title | link | doi | date | pdf_link | abstract | software_links | processed\n",
      "------------------------------------\n",
      "1 | Deep Learning-Based Motif Discovery in Major Histocompatibility Complex: A Primer | https://www.biorxiv.org/content/10.1101/2024.10.03.616544v1 | 10.1101/2024.10.03.616544 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.03.616544v1.full.pdf | In this work, we present a deep learning-based approach for motif discovery in the Major Histocompatibility Complex (MHC) system, which plays a key role in the immune response. We explore the use of convolutional neural networks (CNNs) to identify peptide binding motifs for different MHC alleles. By training models on data from specific MHC Class I and II molecules, we demonstrate how 1-dimensional convolutional filters can effectively capture motifs and binding preferences. Our study introduces a method for extracting motif logos directly from the trained models, providing insights into how internal neural network representations align with known biological motifs. The results show significant alignment with experimental binding motifs, underscoring the utility of deep learning in immunological research and the potential for improving vaccine design and immunotherapy. | None | 0\n",
      "2 | A browser-based platform for storage, visualization, and analysis of large-scale 3D images in HPC environments | https://www.biorxiv.org/content/10.1101/2024.10.04.616591v1 | 10.1101/2024.10.04.616591 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.04.616591v1.full.pdf | High-throughput microscopy necessitates 3D image storage, visualization, and analysis in terabyte to petabyte scales. Centralized high-performance computing (HPC) infrastructure provides the resources, but interfacing software is limited. We developed an extensible platform with a scalable image storage format (SISF) and content delivery network (CDN) to enable fast, random access to compressed data. Additionally, we built a cloud-based nTracer2 software to annotate neuron morphology from SISF images in a web browser. | https://github.com/Cai-Lab-at-University-of-Michigan/pySISF, https://github.com/Cai-Lab-at-University-of-Michigan/SISF_CDN, https://github.com/Cai-Lab-at-University-of-Michigan/nTracer2 | 1\n",
      "3 | Computational design and evaluation of optimal bait sets for scalable proximity proteomics | https://www.biorxiv.org/content/10.1101/2024.10.03.616533v1 | 10.1101/2024.10.03.616533 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.03.616533v1.full.pdf | The spatial organization of proteins in eukaryotic cells can be explored by identifying nearby proteins using proximity-dependent biotinylation approaches like BioID. BioID defines the localization of thousands of endogenous proteins in human cells when used on hundreds of bait proteins. However, this high bait number restricts the approach9s usage and gives these datasets limited scalability for context-dependent spatial profiling. To make subcellular proteome mapping across different cell types and conditions more practical and cost-effective, we developed a comprehensive benchmarking platform and multiple metrics to assess how well a given bait subset can reproduce an original BioID dataset. We also introduce GENBAIT, which uses a genetic algorithm to optimize bait subset selection, to derive bait subsets predicted to retain the structure and coverage of two large BioID datasets using less than a third of the original baits. This flexible solution is poised to improve the intelligent selection of baits for contextual studies. | https://github.com/camlab-bioml/genbait_reproducibility, https://github.com/camlab-bioml/genbait | 1\n",
      "4 | Computational design of developable therapeutic antibodies: efficient traversal of binder landscapes and rescue of escape mutations | https://www.biorxiv.org/content/10.1101/2024.10.03.616038v1 | 10.1101/2024.10.03.616038 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.03.616038v1.full.pdf | Developing therapeutic antibodies is a challenging endeavour, often requiring large-scale screening to produce initial binders, that still often require optimisation for developability. We present a computational pipeline for the discovery and design of therapeutic antibody candidates, which incorporates physics- and AI-based methods for the generation, assessment, and validation of developable candidate antibodies against diverse epitopes, via efficient few-shot experimental screens. We demonstrate that these orthogonal methods can lead to promising designs. We evaluated our approach by experimentally testing a small number of candidates against multiple SARS-CoV-2 variants in three different tasks: (i) traversing sequence landscapes of binders, we identify highly sequence dissimilar antibodies that retain binding to the Wuhan strain, (ii) rescuing binding from escape mutations, we show up to 54% of designs gain binding affinity to a new subvariant and (iii) improving developability characteristics of antibodies while retaining binding properties. These results together demonstrate an end-to-end antibody design pipeline with applicability across a wide range of antibody design tasks. We experimentally characterised binding against different antigen targets, developability profiles, and cryo-EM structures of designed antibodies. Our work demonstrates how combined AI and physics computational methods improve productivity and viability of antibody designs. | https://github.com/Exscientia/ab-characterisation | 1\n",
      "5 | More cells, more doublets in highly multiplexed single-cell data | https://www.biorxiv.org/content/10.1101/2024.10.03.616596v1 | 10.1101/2024.10.03.616596 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.03.616596v1.full.pdf | Sample barcoding allows deconvolution of multiplets in multiplexed droplet-based single-cell RNA-sequencing experiments. However, this is only possible when each cell comes from a different sample. As the number of cells in a droplet increases, the probability of two or more cells coming from the same sample increases rapidly. We show that the number of these unresolvable multiplets is greater than previously appreciated in the 10X Flex scRNA-seq protocol, and provide a formula for estimating the fraction of multiplets in a data set given a measured average droplet occupancy and number of unique samples in a pool. We also show that existing doublet detection tools should be applied to Flex data to identify these multiplets, and demonstrate that filtering out barcodes identified by these tools improves downstream analysis. | None | 0\n",
      "6 | Mosaic Regulation of Stress Pathways Underlies Senescent Cell Heterogeneity | https://www.biorxiv.org/content/10.1101/2024.10.03.616489v1 | 10.1101/2024.10.03.616489 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.03.616489v1.full.pdf | Cellular senescence (CS) and quiescence (CQ) are stress responses characterised by persistent and reversible cell cycle arrest, respectively. These phenotypes are heterogeneous, dependent on the cell type arrested and the insult inciting arrest. Because a universal biomarker for CS has yet to be identified, combinations of senescence-associated biomarkers linked to various biological stress responses including lysosomal activity (β-galactosidase staining), inflammation (senescence-associated secretory phenotypes, SASPs), and apoptosis (senescent cell anti-apoptotic pathways) are used to identify senescent cells.\n",
      "Using in vitro human bulk RNA-seq datasets, we find that senescent states enrich for various stress responses in a cell-type, temporal, and insult-dependent manner. We further demonstrate that various gene signatures used to identify senescent cells in the literature also enrich for stress responses, and are inadequate for universally and exclusively identifying senescent samples.\n",
      "Genes regulating stress responses - including transcription factors and genes controlling chromatin accessibility - are contextually differentially expressed, along with key enzymes involved in metabolism across arrest phenotypes. Additionally, significant numbers of SASP proteins can be predicted from senescent cell transcriptomes and also heterogeneously enrich for various stress responses in a context-dependent manner.\n",
      "We propose that 9senescence9 cannot be meaningfully defined due to the lack of underlying preserved biology across senescent states, and CS is instead a mosaic of stress-induced phenotypes regulated by various factors, including metabolism, TFs, and chromatin accessibility. We introduce the concept of Stress Response Modules, clusters of genes modulating stress responses, and present a new model of CS and CQ induction conceptualised as the differential activation of these clusters. | https://github.com/avelar-ageing/senescence_stress | 1\n",
      "7 | Enhanced Pooling for Weakly Supervised Gigapixel WSI Training Improves Classification and Lesion Localization | https://www.biorxiv.org/content/10.1101/2024.10.03.616445v1 | 10.1101/2024.10.03.616445 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.03.616445v1.full.pdf | A robust artificial intelligence-assisted workflow for tumor assessment in pathology requires not only accurate classification but also precise lesion localization. While current weakly supervised learning methods significantly reduce the need for extensive annotations and leverage large quantities of annotation-free whole-slide images (WSIs) to enhance classification robustness, they often fall short in segmentation accuracy. We attribute this limitation to the optimization goals in classification, which tend to focus solely on the most representative features-an approach that is particularly inefficient for WSIs with gigapixel resolution. To address this challenge, we introduce a novel approach based on streaming convolution, an end-to-end method for WSI training. Our contributions include the Rectified LogSumExp pooling method and adaptive pseudo annotation generation for self-training, both designed to encourage models to learn from sub-representative features. Using only slide-level annotations from the CAMELYON16 dataset, our method achieves a significant improvement in metastasis localization, with a recall from 49.85% to 71.33% at a precision of 90%. This conclusion also holds true for a 3,024-LN dataset used in the assessment of lung cancer lymph node metastasis with a recall improved from 28.31% to 50.82%. | None | 0\n",
      "8 | multimedia: Multimodal Mediation Analysis of Microbiome Data | https://www.biorxiv.org/content/10.1101/2024.03.27.587024v2 | 10.1101/2024.03.27.587024 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.03.27.587024v2.full.pdf | Mediation analysis has emerged as a versatile tool for answering mechanistic questions in microbiome research because it provides a statistical framework for attributing treatment effects to alternative causal pathways. Using a series of linked regressions, this analysis quantifies how complementary data relate to one another and respond to treatments.  Despite these advances, existing software9s rigid assumptions often result in users viewing mediation analysis as a black box.  We designed the multimedia R package to make advanced mediation analysis techniques accessible, ensuring that statistical components are interpretable and adaptable. The package provides a uniform interface to direct and indirect effect estimation, synthetic null hypothesis testing, bootstrap confidence interval construction, and sensitivity analysis, enabling experimentation with various mediator and outcome models while maintaining a simple overall workflow. The software includes modules for regularized linear, compositional, random forest, hierarchical, and hurdle modeling, making it well-suited to microbiome data. We illustrate the package through two case studies. The first re-analyzes a study of the microbiome and metabolome of Inflammatory Bowel Disease patients, uncovering potential mechanistic interactions between the microbiome and disease-associated metabolites, not found in the original study. The second analyzes new data about the influence of mindfulness practice on the microbiome. The mediation analysis highlights shifts in taxa previously associated with depression that cannot be explained indirectly by diet or sleep behaviors alone. A gallery of examples and further documentation can be found at https://go.wisc.edu/830110. | https://github.com/rstudio/shiny | 1\n",
      "9 | Particular sequence characteristics induce bias in the detection of polymorphic transposable element insertions | https://www.biorxiv.org/content/10.1101/2024.09.25.614865v2 | 10.1101/2024.09.25.614865 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.25.614865v2.full.pdf | Transposable elements (TEs) have an important role in genome evolution but are challenging for bioinformatics detection due to their repetitive nature and ability to move and replicate within genomes. New sequencing technologies now enable the characterization of nucleotide and structural variations within species. Among them, TE polymorphism is critical to identify as it may influence species adaptation or trigger diseases. Despite the development of numerous bioinformatic programs, identifying the most effective tool is challenging due to non-overlapping results and varying efficiency across studies. Benchmarking efforts have highlighted the limitations of these tools, often evaluated on either real or simulated data. However, real data may be incomplete or contain unannotated TEs, while simulated data may not accurately reflect real genomes. This study introduces a simulation method generating data based on real genomes to control all genomic parameters. Evaluating several TE polymorphic detection tools using data from Drosophila melanogaster and Arabidopsis thaliana, our study investigates factors like copy size, sequence divergence, and GC content that influence detection efficiency. Our results indicate that only a few programs perform satisfactorily and that all are sensitive to TE and genomic characteristics, that may differ according to the species considered. Using Bos taurus population data as a case study to identify polymorphic LTR-retrotransposon insertions, we found low-frequency insertions particularly challenging to detect due to a high number of false positives. Increased sequencing coverage improved sensitivity but reduced precision. Our work underscores the importance of selecting appropriate tools and thresholds according to the specific research questions. | https://github.com/PacificBiosciences/pbsv, https://github.com/e-lerat/replicaTE | 1\n",
      "10 | A Targeted Reference Database for Improved Analysis of Environmental 16S rRNA Oxford Nanopore Sequencing Data | https://www.biorxiv.org/content/10.1101/2024.10.03.616456v2 | 10.1101/2024.10.03.616456 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.03.616456v2.full.pdf | The Oxford Nanopore Technologies (ONT) sequencing platform is compact and efficient, making it suitable for rapid biodiversity assessments in remote areas. Despite its long reads, ONT has a higher error rate compared to other platforms, necessitating high-quality reference databases for accurate taxonomic assignments. However, the absence of targeted databases for underexplored habitats, such as the seafloor, limits ONT9s broader applicability for exploratory analysis. To address this, we have developed a strategy to build environmentally-targeted databases to enhance 16S rRNA gene(16S) analysis using ONT. We start by using Illumina short-read data to create a database of full-length or near full-length 16S sequences from seafloor samples. Initially, amplicons are mapped to the SILVA database, with matches added to our database. Unmatched amplicons are reconstructed using METASEED and Barrnap methodologies with amplicon and metagenome data. Finally, if the previous strategies did not succeed, we included the short-read sequences in the database. This resulted in AQUAeD-DB, which contains 14 545 16S sequences clustered at 95% identity. Comparative database analysis and relative abundance correlation studies reveal that AQUAeD-DB provides consistent results for both Illumina and Nanopore read assignments (median correlation coefficient: 0.50), whereas a standard database showed a substantially weaker correlation. These findings also emphasis its potential to recognize both high and low-abundance taxa, which could be key indicators in environmental studies. This work highlights the necessity of targeted databases for environmental analysis, especially for ONT-based studies, and lays foundations for future extension of the database. | https://github.com/tseemann/barrnap | 1\n",
      "11 | Data-driven insights into the association between oxidative stress and calcium-regulating proteins in cardiovascular disease. | https://www.biorxiv.org/content/10.1101/2024.10.03.616559v1 | 10.1101/2024.10.03.616559 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.03.616559v1.full.pdf | A growing body of biomedical literature suggests a bidirectional regulatory relationship between cardiac calcium (Ca2+)-regulating proteins and reactive oxygen species (ROS), which is integral to the pathogenesis of various cardiac disorders via oxidative stress signaling. To address the challenge of finding hidden connections within the growing volume of biomedical research, we developed a data science pipeline for efficient data extraction, transformation, and loading. Employing the CaseOLAP (Context-Aware Semantic Analytic Processing) algorithm, our pipeline quantifies interactions between 128 human cardiomyocyte Ca2+-regulating proteins and eight cardiovascular disease (CVD) categories. Our machine learning analysis of CaseOLAP scores reveals that the molecular interfaces of Ca2+-regulating proteins uniquely associate with cardiac arrhythmias and diseases of the cardiac conduction system, distinguishing them from other CVDs. Additionally, a knowledge graph analysis identified 59 of the 128 Ca2+-regulating proteins as involved in OS-related cardiac diseases, with cardiomyopathy emerging as the predominant category. By leveraging a link prediction algorithm, our research illuminates interactions between Ca2+-regulating proteins, OS, and CVDs. The insights gained from our study provide a deeper understanding of the molecular interplay between cardiac ROS and Ca2+-regulating proteins in the context of CVDs. Such understanding is essential for the innovation and development of targeted therapeutic strategies. | None | 0\n",
      "12 | Bi-level Graph Learning Unveils Prognosis-Relevant Tumor Microenvironment Patterns in Breast Multiplexed Digital Pathology | https://www.biorxiv.org/content/10.1101/2024.04.22.590118v2 | 10.1101/2024.04.22.590118 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.04.22.590118v2.full.pdf | The tumor microenvironment is widely recognized for its central role in driving cancer progression and influencing prognostic outcomes. There have been increasing efforts dedicated to characterizing this complex and heterogeneous environment, including developing potential prognostic tools by leveraging modern deep learning methods. However, the identification of generalizable data-driven biomarkers has been limited, in part due to the inability to interpret the complex, black-box predictions made by these models. In this study, we introduce a data-driven yet interpretable approach for identifying patterns of cell organizations in the tumor microenvironment that are associated with patient prognoses. Our methodology relies on the construction of a bi-level graph model: (i) a cellular graph, which models the intricate tumor microenvironment, and (ii) a population graph that captures inter-patient similarities, given their respective cellular graphs, by means of a soft Weisfeiler-Lehman subtree kernel. This systematic integration of information across different scales enables us to identify patient subgroups exhibiting unique prognoses while unveiling tumor microenvironment patterns that characterize them. We demonstrate our approach in a cohort of breast cancer patients and show that the identified tumor microenvironment patterns result in a risk stratification system that provides new complementary information with respect to standard stratification systems. Our results, which are validated in two independent cohorts, allow for new insights into the prognostic implications of the breast tumor microenvironment. This methodology could be applied to other cancer types more generally, providing insights into the cellular patterns of organization associated with different outcomes. | https://github.com/Sulam-Group/BiGraph4TME.git | 1\n",
      "13 | Protein Language Model Fitness Is a Matter of Preference | https://www.biorxiv.org/content/10.1101/2024.10.03.616542v1 | 10.1101/2024.10.03.616542 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.03.616542v1.full.pdf | Leveraging billions of years of evolution, scientists have trained protein language models (pLMs) to understand the sequence and structure space of proteins aiding in the design of more functional proteins. Although they have shown ability to improve efficiency in engineering, it remains unclear if such models capture true biological patterns or artifacts of the training data. We aim to predict the circumstances in which pLMs can successfully perform zero-shot fitness estimation. Our work studies trends observed over hundreds of deep mutational scans across multiple different fitness objectives. We find that the likelihood, or abstractly, implicit preference of a certain protein sequence imbued during pretraining is predictive of fitness prediction capabilities. Both over-preferred and under-preferred wild type sequences harm performance. Using influence functions to causally understand how individual data points increase protein likelihoods, we find that there exists a power law tail due to sequence homology. Lastly, under-performance on low likelihood wild type proteins can be remedied by unsupervised finetuning. These findings that pLM zero-shot fitness estimation can be predicted by the likelihood of the engineered sequence can motivate and improve pLMs9 deployment in protein maturation campaigns. | None | 0\n",
      "14 | LIET Model: Capturing the kinetics of RNA polymerase from loading to termination | https://www.biorxiv.org/content/10.1101/2024.10.03.616401v1 | 10.1101/2024.10.03.616401 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.03.616401v1.full.pdf | Transcription by RNA polymerases is an exquisitely regulated process and the principle step of the central dogma. Transcription is the primary determinate of cell-state and most cellular perturbations (e.g. disease, drug treatment, or abiotic stressors) impact transcription by altering the activity of polymerases. Thus, the ability to detect and quantify detailed changes in polymerase activity would yield considerable insight into most cellular processes. Nascent run-on sequencing assays provide a direct readout of polymerase activity, but no tools exist that comprehensively model this activity at genes.  Thus, we focus on RNA polymerase II (RNAP2), which is responsible for transcribing all protein-coding genes and many ncRNAs. Here we present the first model to fully capture the complete process of gene transcription by RNAP2.  For an individual gene, this model parameterizes each distinct stage of RNAP2 transcription---Loading, Initiation, Elongation, and Termination, hence LIET---in a biologically interpretable way, and combines them in a Bayesian mixture model intended to be applied to nascent run-on sequencing data. Additionally, we provide a user-friendly, open-source software implementation of the model, which we use to demonstrate its performance. With our improved modeling of Loading and Initiation we demonstrate that these processes are characteristically different between sense (gene coding) and antisense (PROMPT) strands. Furthermore, our model is the first to robustly quantify the Termination phase of transcription. By applying the LIET model to 24 different human cell-types, our analysis indicates that the position of dissociation appears to be highly consistent across cell-types, indicative of a highly regulated process. Furthermore, by applying the LIET model to perturbation experiments, we demonstrate its ability to detect specific regulatory changes in pausing ratio (at 59end), strand-bias, and dissociation location (at 39end). This opens the door to differential assessment of transcription regulation at the level of not only individual genes but also individual transcription stages within a gene. | None | 0\n",
      "15 | PoreMeth2: decoding the evolution of methylome alterations with Nanopore sequencing | https://www.biorxiv.org/content/10.1101/2024.10.03.616449v1 | 10.1101/2024.10.03.616449 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.03.616449v1.full.pdf | In epigenetic analysis, identifying differentially methylated regions (DMRs) typically involves detecting groups of consecutive CpGs that show significant changes in their average methylation levels. However, the methylation state of a genomic region can also be characterized by a mixture of patterns (epialleles) with variable frequencies, and the relative proportions of such patterns can provide insights into its mechanisms of formation.\n",
      "Traditional methods based on bisulfite conversion and NGS, due to the read size (150 bp), allow epiallele frequency analysis only in high-CpG-density regions, limiting differential methylation studies to just 50% of the human methylome. Nanopore sequencing, with its long reads, enables the analysis of epiallele frequency across both high- and low-CpG-density regions.\n",
      "We introduce a novel computational approach, PoreMeth2, an R library that integrates epiallelic diversity and methylation frequency changes from Nanopore data to identify DMRs, assess their formation mechanisms, and annotate them to genic and regulatory elements. We applied PoreMeth2 to cancer and glial cell datasets, demonstrating its ability to distinguish epigenomic changes with a strong effect on gene expression from those with a weaker impact on transcriptional activity.\n",
      "PoreMeth2 is publicly available at https://github.com/Lab-CoMBINE/PoreMeth2. | https://github.com/Lab-CoMBINE/PoreMeth2 | 1\n",
      "16 | HLAIIPred: Cross-Attention Mechanism for Modeling the Interaction of HLA Class II Molecules with Peptides | https://www.biorxiv.org/content/10.1101/2024.10.01.616078v1 | 10.1101/2024.10.01.616078 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.616078v1.full.pdf | We introduce HLAIIPred, a deep learning model to predict peptides presented by class II human leukocyte antigens (HLAII) on the surface of antigen presenting cells. HLAIIPred is trained using a Transformer-based neural network and a dataset comprising of HLAII-presented peptides identified by mass spectrometry. In addition to predicting peptide presentation, the model can also provide important insights into peptide-HLAII interactions by identifying core peptide residues that form such interactions. We evaluate the performance of HLAIIPred on three different tasks, peptide presentation in monoallelic samples, immunogenicity prediction of therapeutic antibodies, and neoantigen prioritization for cancer immunotherapy. Additionally, we created a new dataset of biotherapeutics HLAII peptides presented by human dendritic cells. This data is used to develop screening strategies to predict the unwanted immunogenic segments of therapeutic antibodies by HLAII presentation models. HLAIIPred demonstrates superior or equivalent performance when compared to the latest models across all evaluated benchmark datasets. We achieve a 16% increase in prediction of presented peptides compared to the second-best model on a set of unseen peptides presented by less frequent alleles. The model also improves the area under the precision-recall curve by 3% for distinguishing between immunogenic and non-immunogenic antibodies. We show that HLAIIPred can identify epitopes in therapeutic antibodies and prioritize neoantigens with high accuracy. | None | 0\n",
      "17 | PUNCH2: explore the strategy for Intrinsically Disordered Protein predictor | https://www.biorxiv.org/content/10.1101/2024.10.03.616458v1 | 10.1101/2024.10.03.616458 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.03.616458v1.full.pdf | Intrinsically disordered proteins (IDPs) lack stable three-dimensional structures, which poses significant challenges for their computational prediction. This study introduces PUNCH2, a novel approach for predicting intrinsically disordered regions (IDRs) in proteins. We address key issues such as the scarcity of comprehensive IDR databases, effective feature extraction, and robust model architecture. By integrating sequences from experimental PDB datasets and fully disordered DisProt sequences for training, PUNCH2 achieves superior prediction accuracy. Various sequence embeddings, including One-Hot, MSA-based, and PLM-based methods, were evaluated, with ProtTrans-based embeddings showing the best performance. The optimal model architecture features multiple convolutional layers, enhancing predictive confidence. PUNCH2 and its lighter variant, PUNCH2-light, demonstrate high efficiency and accuracy in benchmarking against top predictors on the CAID2 dataset, offering promising tools for advancing IDP research and understanding. | https://github.com/deemeng/punch2, https://github.com/deemeng/punch2_light | 1\n",
      "18 | Real-Time Spatiotemporal Tracking of Infectious Outbreaks in Confined Environments with a Host-Pathogen Agent-Based System | https://www.biorxiv.org/content/10.1101/2024.10.01.616085v1 | 10.1101/2024.10.01.616085 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.616085v1.full.pdf | Deadly infection outbreaks in confined spaces, whether it is a COVID-19 outbreak on a cruise ship or at a veteran home, or measles and stomach flu outbreaks in schools, can be characterized by their rapid spread due to the abundance of common spaces, shared airways, and high population density. Preventing future infectious outbreaks and developing efficient mitigation protocols can benefit from advanced computational modeling approaches. Here, we developed an agent-based modeling approach to study the spatiotemporal dynamics of an infection outbreak in a confined environment caused by a specific pathogen and to determine effective infection containment protocols. The approach integrates the 3D geographic information system of a confined environment, the behavior of the hosts, key biological parameters about the pathogen obtained from the experimental data, and the general mechanics of host-pathogen and pathogen-fomite interactions. To assess our approach, we applied it to the historical data of infectious outbreak caused by norovirus, H1N1 influenza A, and SARS-CoV-2 viruses. As a result our model was able to accurately predict the number of infections per day, correctly identify the day when the CDC vessel sanitation protocol would be triggered,  single out key biological parameters affecting the infection spread, and propose important changes to existing containment protocols, specific for different pathogens. This research not only contributes to our understanding of infection spread and containment in cruise ships but also offers insights applicable to other similar confined settings, such as nursing homes, schools, and hospitals. By providing a robust framework for real-time outbreak modeling, this study proposes new, more effective containment protocols and enhances our preparedness for managing infectious diseases and emerging pathogens in confined environments. | https://github.com/KorkinLab/infection-dynamics | 1\n",
      "19 | Building a community-driven bioinformatics platform to facilitate Cannabis sativa multi-omics research | https://www.biorxiv.org/content/10.1101/2024.10.02.616368v1 | 10.1101/2024.10.02.616368 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.616368v1.full.pdf | Global changes in Cannabis legislation after decades of stringent regulation, and heightened demand for its industrial and medicinal applications have spurred recent genetic and genomics research. An international research community emerged and identified the need for a web portal to host Cannabis-specific datasets that seamlessly integrates multiple data sources and serves omics-type analyses, fostering information sharing.\n",
      "The Tripal platform was used to host public genome assemblies, gene annotations, QTL and genetic maps, gene and protein expression, metabolic profile and their sample attributes. SNPs were called using public resequencing datasets on three genomes. Additional applications, such as SNP-Seek and MapManJS, were embedded into Tripal. A multi-omics data integration web-service API, developed on top of existing Tripal modules, returns generic tables of sample, property, and values. Use-cases demonstrate the API9s utility for various -omics analyses, enabling researchers to perform multi-omics analyses efficiently. | https://github.com/TransDecoder/TransDecoder, https://github.com/shergreen/pyppca | 1\n",
      "20 | ChIPbinner: An R package for analyzing broad histone marks binned in uniform windows from ChIP-Seq or CUT&RUN/TAG data | https://www.biorxiv.org/content/10.1101/2024.10.02.616288v1 | 10.1101/2024.10.02.616288 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.616288v1.full.pdf | Background: The decreasing costs of sequencing, along with the growing understanding of epigenetic mechanisms driving diseases, have led to the increased application of chromatin immunoprecipitation (ChIP-Seq), Cleavage Under Targets & Release Using Nuclease (CUT&RUN) and Cleavage Under Targets and Tagmentation (CUT&TAG) sequencing - which are designed to map DNA or chromatin-binding proteins to their genome targets - in biomedical research. Existing software tools, namely peak-callers, are available for analyzing data from these technologies, although they often struggle with diffuse and broad signals, such as those associated with broad histone post-translational modifications (PTMs). \n",
      "Results: To address this limitation, we present ChIPbinner, an open-source R package tailored for reference-agnostic analysis of broad PTMs. Instead of relying on pre-identified enriched regions from peak-callers, ChIPbinner divides (bins) the genome into uniform windows. Thus, users are provided with an unbiased method to explore genome-wide differences between two samples using scatterplots, principal component analysis (PCA), and correlation plots. It also facilitates the identification and characterization of differential clusters of bins, allowing users to focus on specific genomic regions significantly affected by treatments or mutations. We demonstrated the effectiveness of this tool through a case study assessing H3K36me2 depletion following NSD1 knockout in head and neck squamous cell carcinoma, highlighting the advantages of ChIPbinner in detecting broad histone mark changes over existing software.\n",
      "Conclusions: Binned analysis provides a more holistic view of the genomic landscape, allowing researchers to uncover broader patterns and correlations that may be missed when solely focusing on individual peaks. ChIPbinner offers researchers a convenient tool to perform binned analysis. It improves on previously published software by providing a clustering approach that is independent of each bin9s differential enrichment status, while also offering additional features for downstream analysis of these differentially enriched bins. | https://github.com/padilr1/ChIPbinner, https://github.com/padilr1/ChIPbinner_database | 1\n",
      "21 | Mosaic integration of spatial multi-omics with SpaMosaic | https://www.biorxiv.org/content/10.1101/2024.10.02.616189v1 | 10.1101/2024.10.02.616189 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.616189v1.full.pdf | With the advent of spatial multi-omics, we can mosaic integrate such datasets with partially overlapping modalities to construct higher dimensional views of the source tissue. SpaMosaic is a spatial multi-omics mosaic integration tool that employs contrastive learning and graph neural networks to construct a modality-agnostic and batch-corrected latent space suited for analyses like spatial domain identification and imputing missing omes. Using simulated and experimentally acquired datasets, we benchmarked SpaMosaic against single-cell multi-omics mosaic integration methods. The experimental spatial omics data encompassed RNA and protein abundance, chromatin accessibility or histone modifications, acquired from brain, embryo, tonsil, and lymph node tissues. SpaMosaic achieved superior performance over existing methods in identifying known spatial domains while reducing noise and batch effects. We also integrated a set of five mouse brain datasets of RNA and different epigenomic modalities, and imputed the missing omes. We found the genes in the imputed omes enriched in the correct tissue specific biological processes, supporting the imputation accuracy. | https://github.com/PeterZZQ/scMoMaT/blob/main/demo_scmomat.ipynb, https://github.com/MarioniLab/StabMap/tree/main/vignettes, https://github.com/epurdom/cobolt/blob/master/docs/tutorial.ipynb, https://github.com/JinmiaoChenLab/SpaMosaic, https://github.com/spotify/annoy | 1\n",
      "22 | Fully Hyperbolic Neural Networks: A Novel Approach to Studying Aging Trajectories | https://www.biorxiv.org/content/10.1101/2024.10.01.616153v1 | 10.1101/2024.10.01.616153 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.616153v1.full.pdf | Characterizing age-related alterations in brain networks is crucial for understanding aging trajectories and identifying deviations indicative of neurodegenerative disorders, such as Alzheimer9s disease. In this study, we developed a Fully Hyperbolic Neural Network (FHNN) to embed functional brain connectivity graphs derived from magnetoencephalography (MEG) data into low dimensions on a Lorentz model of hyperbolic space. Using this model, we computed hyperbolic embeddings of the MEG brain networks of 587 individuals from the Cambridge Centre for Ageing and Neuroscience (Cam-CAN) dataset. Notably, we leveraged a unique metric - the radius of the node embeddings - which effectively captures the hierarchical organization of the brain, to characterize subtle hierarchical organizational changes in various brain subnetworks attributed to the aging process. Our findings revealed that a considerable number of subnetworks exhibited a reduction in hierarchy during aging, with some showing gradual changes and others undergoing rapid transformations in the elderly. Moreover, we demonstrated that hyperbolic features outperform traditional graph-theoretic measures in capturing age-related information in brain networks. Overall, our study represents the first evaluation of hyperbolic embeddings in MEG brain networks for studying aging trajectories, shedding light on critical regions undergoing significant age-related alterations in the large cohort of the Cam-CAN dataset. | None | 0\n",
      "23 | CREATE: cell-type-specific cis-regulatory elements identification via discrete embedding | https://www.biorxiv.org/content/10.1101/2024.10.02.616391v1 | 10.1101/2024.10.02.616391 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.616391v1.full.pdf | Identifying cis-regulatory elements (CREs) within non-coding genomic regions-such as enhancers, silencers, promoters, and insulators-is pivotal for elucidating the intricate gene regulatory mechanisms underlying complex biological traits. The current prevalent sequence-based methods often focus on singular CRE types, limiting insights into cell-type-specific biological implications. Here, we introduce CREATE, a multimodal deep learning model based on the Vector Quantized Variational AutoEncoder framework, designed to extract discrete CRE embeddings and classify multiple CRE classes using genomic sequences, chromatin accessibility, and chromatin interaction data. CREATE excels in accurate CRE identification and exhibits strong effectiveness and robustness. We showcase CREATE9s capability in generating comprehensive CRE-specific feature spectrum, offering quantitative and interpretable insights into CRE specificity. By enabling large-scale prediction of CREs in specific cell types, CREATE facilitates the recognition of disease- or phenotype-related biological variabilities of CREs, thereby expanding our understanding of gene regulation landscapes. | https://github.com/ncbi/SilencerEnhancerPredict, https://github.com/lyli1013/DeepICSH, https://github.com/cuixj19/CREATE, https://github.com/uci-cbcl/DanQ | 1\n",
      "24 | A Predictive Model for Compound-Protein Interactions Based on Concatenated Vectorization | https://www.biorxiv.org/content/10.1101/2024.10.02.616275v1 | 10.1101/2024.10.02.616275 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.616275v1.full.pdf | Background. Large data sets of compound activity lend themselves to building predictive models based on compound and target structure. The simplest representation of structure is via vectorisation. Compound fingerprint vectorisation has been successfully employed in predicting compound activity classes.\n",
      "Results. A vector representation of a protein-compound pair based on a concatenation of the compound fingerprint and the protein triplet vector has been used to train random forest and neural network models on multiple datasets of protein-compound interaction together with compound associated transcription and activity profiles. Results for compound-target predictability are comparable with more complex published methodologies. \n",
      "Conclusion. A simple intuitive representation of a protein-compound pair can be employed in a variety of machine learning models to gain a predictive handle on the activity of compounds for which there is no activity data. It is hoped that this transparent approach will prove sufficiently portable and simple to implement that drug discovery will be opened up to the wider research community. | None | 0\n",
      "25 | Unlearning Virus Knowledge Toward Safe and Responsible Mutation Effect Predictions | https://www.biorxiv.org/content/10.1101/2024.10.02.616274v1 | 10.1101/2024.10.02.616274 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.616274v1.full.pdf | Pre-trained deep protein models have become essential tools in fields such as biomedical research, enzyme engineering, and therapeutics due to their ability to predict and optimize protein properties effectively. However, the diverse and broad training data used to enhance the generalizability of these models may also inadvertently introduce ethical risks and pose biosafety concerns, such as the enhancement of harmful viral properties like transmissibility or drug resistance. To address this issue, we introduce a novel approach using knowledge unlearning to selectively remove virus-related knowledge while retaining other useful capabilities. We propose a learning scheme, PROEDIT, for editing a pre-trained protein language model toward safe and responsible mutation effect prediction. Extensive validation on open benchmarks demonstrates that PROEDIT significantly reduces the model9s ability to enhance the properties of virus mutants without compromising its performance on non-virus proteins. As the first thorough exploration of safety issues in deep learning solutions for protein engineering, this study provides a foundational step toward ethical and responsible AI in biology. | None | 0\n",
      "26 | A new compression strategy to reduce the size of nanopore sequencing data | https://www.biorxiv.org/content/10.1101/2024.10.02.616377v1 | 10.1101/2024.10.02.616377 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.616377v1.full.pdf | Nanopore sequencing is an increasingly central tool for genomics. Despite rapid advances in the field, large data volumes and computational bottlenecks continue to pose major challenges. Here we introduce ex-zd, a new data compression strategy that helps address the large size of raw signal data generated during nanopore experiments. Ex-zd encompasses both a lossless compression method, which modestly outperforms all current methods for nanopore signal data compression, and a 9lossy9 method, which can be used to achieve dramatic additional savings. The latter component works by reducing the number of bits used to encode signal data. We show that the three least significant bits in signal data generated on instruments from Oxford Nanopore Technologies (ONT) predominantly encode noise. Their removal reduces file sizes by half without impacting downstream analyses, including basecalling and detection of DNA methylation. Ex-zd compression saves hundreds of gigabytes on a single ONT sequencing experiment, thereby increasing the scalability, portability and accessibility of nanopore sequencing. | https://github.com/hasindu2008/slow5tools, https://github.com/nanoporetech/vbz_compression, https://github.com/hasindu2008/slow5lib, https://github.com/nanoporetech/pod5-file-format | 1\n",
      "27 | EvoSeq-ML: Advancing Data-Centric Machine Learning with Evolutionary-Informed Protein Sequence Representation and Generation | https://www.biorxiv.org/content/10.1101/2024.10.02.616302v1 | 10.1101/2024.10.02.616302 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.616302v1.full.pdf | In protein engineering, machine learning (ML) advancements have led to significant progress, including protein structure prediction (e.g., AlphaFold), sequence representation through language models, and novel protein generation. However, the impact of data curation on ML model performance is underexplored. As more sequence and structural data become available, a data-centric approach is increasingly favored over a model-centric method. A data-centric approach prioritizes high-quality, domain-specific data, ensuring ML tools are trained on datasets that accurately reflect biological complexity and diversity. This paper introduces a novel methodology that integrates ancestral sequence reconstruction (ASR) into ML models, enhancing data-centric strategies in the field. ASR uses computational techniques to infer ancient protein sequences from modern descendants, providing diverse, stable sequences with rich evolutionary information. While multiple sequence alignments (MSAs) are commonly used in protein engineering frameworks to incorporate evolutionary information, ASR offers deeper insights into protein evolution. Unlike MSAs, ASR captures mutation rates, phylogenic relationships, evolutionary trajectories, and specific ancestral sequences, giving access to novel protein sequences beyond what is available in public databases by natural selection. We employed two statistical methods for ASR: joint Bayesian inference and maximum likelihood. Bayesian approaches infer ancestral sequences by sampling from the entire posterior distribution, accounting for epistatic interactions between multiple amino acid positions to capture the nuances and uncertainties of ancestral sequences. In contrast, maximum likelihood methods estimate the most probable amino acids at individual positions in isolation. Both methods provide extensive ancestral data, enhancing ML model performance in protein sequence generation and fitness prediction tasks. Our results demonstrate that generative ML models training on either Bayesian or maximum likelihood approaches produce highly stable and diverse protein sequences. We also fine-tuned the evolutionary scale ESM protein language model with reconstructed ancestral data to obtain evolutionary-driven protein representations, and downstream stability prediction tasks for Endolysin and Lysozyme C families. For Lysozyme C, ancestral-based representations outperformed the baseline ESM in KNN classification and matched the established InterPro method. In Endolysin, our novel ASR-Dist method performed on par with or better than the baseline and other fine-tuning approaches across various classification metrics. ASR-Dist showed consistent performance in both simple and complex classification models, suggesting the effectiveness of this data-centric approach in enhancing protein representations. This work demonstrates how evolutionary data can improve ML-driven protein engineering, presenting a novel data-centric approach that expands our exploration of protein sequence space and enhances our ability to predict and design functional proteins. | https://github.com/WoldringLabMSU/EvoSeq-ML | 1\n",
      "28 | De novo discovery of conserved gene clusters in microbial genomes with Spacedust | https://www.biorxiv.org/content/10.1101/2024.10.02.616292v1 | 10.1101/2024.10.02.616292 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.616292v1.full.pdf | Metagenomics has revolutionized environmental and human-associated microbiome studies. However, the limited fraction of proteins with known biological process and molecular functions presents a major bottleneck. In prokaryotes and viruses, evolution favors keeping genes participating in the same biological processes co-localized as conserved gene clusters. Conversely, conservation of gene neighborhood indicates functional association. Spacedust is a tool for systematic, de novo discovery of conserved gene clusters. To find homologous protein matches it uses fast and sensitive structure comparison with Foldseek. Partially conserved clusters are detected using novel clustering and order conservation P-values. We demonstrate Spacedust9s sensitivity with an all-vs-all analysis of 1308 bacterial genomes, identifying 72843 conserved gene clusters containing 58% of the 4.2 million genes. It recovered recover 95% of antiviral defense system clusters annotated by a specialized tool. Spacedust9s high sensitivity and speed will facilitate the large-scale annotation of the huge numbers of sequenced bacterial, archaeal and viral genomes. | None | 0\n",
      "29 | BioLLMNet: Enhancing RNA-Interaction Prediction with a Specialized Cross-LLM Transformation Network | https://www.biorxiv.org/content/10.1101/2024.10.02.616044v1 | 10.1101/2024.10.02.616044 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.616044v1.full.pdf | Existing computational methods for the prediction of RNA related interactions often rely heavily on manually crafted features. Language model features for bio-sequences has gain significant popularity in proteomics and genomics. However, during interaction prediction, how language model features from different modalities should be combined to extract the most representative features is yet to be explored. We introduce BioLLMNet, a novel framework that introduces an effective combination approach for multi-modal bio-sequences. BioLLMNet provides a way to transform feature space of different molecule9s language model features and uses learnable gating mechanism to effectively fuse features. Rigorous evaluations show that BioLLMNet achieves state-of-the-art performance in RNA-protein, RNA-small molecule, and RNA-RNA interactions, outperforming existing methods in RNA-associated interaction prediction. | None | 0\n",
      "30 | PertEval-scFM: Benchmarking Single-Cell Foundation Models for Perturbation Effect Prediction | https://www.biorxiv.org/content/10.1101/2024.10.02.616248v1 | 10.1101/2024.10.02.616248 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.616248v1.full.pdf | In silicomodeling of transcriptional responses to perturbations is crucial for advancing our understanding of cellular processes and disease mechanisms. We present PertEval-scFM, a standardized framework designed to evaluate models for perturbation effect prediction. We apply PertEval-scFM to benchmark zero-shot single-cell foundation model (scFM) embeddings against simpler baseline models to assess whether these contextualized representations enhance perturbation effect prediction. Our results show that scFM embeddings do not provide consistent improvements over baseline models, especially under distribution shift. Additionally, all models struggle with predicting strong or atypical perturbation effects. Overall, this study provides a systematic evaluation of zero-shot scFM embeddings for perturbation effect prediction, highlighting the challenges of this task and revealing the limitations of current-generation scFMs. Our findings underscore the need for specialized models and high-quality datasets that capture a broader range of cellular states. Source code and documentation can be found at: <a>https://github.com/aaronwtr/PertEval</a>. | https://github.com/aaronwtr/PertEval | 1\n",
      "31 | scMODAL: A general deep learning framework for comprehensive single-cell multi-omics data alignment with feature links | https://www.biorxiv.org/content/10.1101/2024.10.01.616142v1 | 10.1101/2024.10.01.616142 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.616142v1.full.pdf | Recent advancements in single-cell technologies have enabled comprehensive characterization of cellular states through transcriptomic, epigenomic, and proteomic profiling at single-cell resolution. These technologies have significantly deepened our understanding of cell functions and disease mechanisms from various omics perspectives. As these technologies evolve rapidly and data resources expand, there is a growing need for computational methods that can integrate information from different modalities to facilitate joint analysis of single-cell multi-omics data. However, integrating single-cell omics datasets presents unique challenges due to varied feature correlations and technology-specific limitations. To address these challenges, we introduce scMODAL, a deep learning framework tailored for single-cell multi-omics data alignment using feature links. scMODAL integrates datasets with limited known positively correlated features, leveraging neural networks and generative adversarial networks to align cell embeddings and preserve feature topology. Our experiments demonstrate scMODAL9s effectiveness in removing unwanted variation, preserving biological information, and accurately identifying cell subpopulations across diverse datasets. scMODAL not only advances integration tasks but also supports downstream analyses such as feature imputation and feature relationship inference, offering a robust solution for advancing single-cell multi-omics research. | https://github.com/gefeiwang/scMODAL, https://github.com/lmweber/benchmark-data-Levine-32-dim | 1\n",
      "32 | The Impact of Clinical and Molecular Variant Properties on Calibration and Performance of Variant Effect Prediction Tools | https://www.biorxiv.org/content/10.1101/2024.10.02.614907v1 | 10.1101/2024.10.02.614907 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.614907v1.full.pdf | Background: Variant Effect Prediction (VEP) tools are essential for determining the potential pathogenicity of genetic variants, aiding clinical diagnostics and genetic counseling. However, their performance can vary depending on molecular and clinical contexts, complicating variant classification. Aim: This study aims to assess the performance variability of commonly used VEP tools under different conditions. Additionally, the study aims to recalibrate score thresholds to better reflect evidence of pathogenicity. Methods: ClinVar variants classified as pathogenic (P), likely pathogenic (LP), benign (B), or likely benign (LB) were analyzed using 25 VEP tools. Tools were evaluated based on discriminatory performance. Data were stratified by variant creation date, allele frequency, conservation level, mode of inheritance (MOI), and disease category. For each subset, Bayesian methods were employed to recalibrate score thresholds corresponding to the levels of evidence defined by the American College of Medical Genetics (ACMG). Results: The performance of VEP tools varied significantly across different subsets. Variants created after 2020 showed a mild yet significant decrease in the performance of certain VEP tools, particularly those trained on earlier datasets. VEP tools exhibited reduced accuracy for variants with higher allele frequencies, particularly those exceeding a frequency of 10-4, suggesting a need for recalibration when assessing more common variants. Tools demonstrated lower discriminatory performance for variants located in regions with high conservation, mostly due to a decrease in specificity. Variants affecting autosomal recessive (AR) and X-linked (XL) genes were more accurately classified compared to those affecting autosomal dominant (AD) genes by most tools. Differences in MOI and conservation levels within certain disease categories were shown to correlate with overall performance. Recalibration of prediction scores resulted in lower score thresholds in low conservation regions compared to high and uncovered subsets in which higher levels of evidence may be achieved. Conclusion: VEP tools exhibit context-dependent performance variability, necessitating score recalibration for accurate classification. | None | 0\n",
      "33 | Mitigation of multi-scale biases in cell-type deconvolution for spatially resolved transcriptomics using HarmoDecon | https://www.biorxiv.org/content/10.1101/2024.10.02.616209v1 | 10.1101/2024.10.02.616209 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.02.616209v1.full.pdf | The advent of spatially resolved transcriptomics (SRT) has revolutionized our understanding of tissue molecular microenvironments by enabling the study of gene expression in its spatial context. However, many SRT platforms lack single-cell resolution, necessitating cell-type deconvolution methods to estimate cell-type proportions in SRT spots. Despite advancements in existing tools, these methods have not addressed biases occurring at three scales: individual spots, entire tissue samples, and discrepancies between SRT and reference scRNA-seq datasets. These biases result in overbalanced cell-type proportions for each spot, mismatched cell-type fractions at the sample level, and data distribution shifts across platforms. To mitigate these biases, we introduce HarmoDecon, a novel semi-supervised deep learning model for spatial cell-type deconvolution. HarmoDecon leverages pseudo-spots derived from scRNA-seq data and employs Gaussian Mixture Graph Convolutional Networks to address the aforementioned issues. Through extensive simulations on multi-cell spots from STARmap and osmFISH, HarmoDecon outperformed 11 state-of-the-art methods. Additionally, when applied to legacy SRT platforms and 10x Visium datasets, HarmoDecon achieved the highest accuracy in spatial domain clustering and maintained strong correlations between cancer marker genes and cancer cells in human breast cancer samples. These results highlight the utility of HarmoDecon in advancing spatial transcriptomics analysis. | https://github.com/ericcombiolab/HarmoDecon/tree/main | 1\n",
      "34 | ProteusAI: An Open-Source and User-Friendly Platform for Machine Learning-Guided Protein Design and Engineering | https://www.biorxiv.org/content/10.1101/2024.10.01.616114v1 | 10.1101/2024.10.01.616114 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.616114v1.full.pdf | Protein design and engineering are crucial for advancements in biotechnology, medicine, and sustainability. Machine learning (ML) models are used to design or enhance protein properties such as stability, catalytic activity, and selectivity. However, many existing ML tools require specialized expertise or lack open-source availability, limiting broader use and further development. To address this, we developed ProteusAI, a user-friendly and open-source ML platform to streamline protein engineering and design tasks. ProteusAI offers modules to support researchers in various stages of the design-build-test-learn (DBTL) cycle, including protein discovery, structure-based design, zero-shot predictions, and ML-guided directed evolution (MLDE). Our benchmarking results demonstrate ProteusAI9s efficiency in improving proteins and enyzmes within a few DBTL-cycle iterations. ProteusAI democratizes access to ML-guided protein engineering and is freely available for academic and commercial use. Future work aims to expand and integrate novel methods in computational protein and enzyme design to further develop ProteusAI. | None | 0\n",
      "35 | The Specification Game: Rethinking the Evaluation of Drug Response Prediction for Precision Oncology | https://www.biorxiv.org/content/10.1101/2024.10.01.616046v1 | 10.1101/2024.10.01.616046 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.616046v1.full.pdf | Precision oncology plays a pivotal role in contemporary healthcare, aiming to optimize treatments for\n",
      "each patient based on their unique characteristics. This objective has spurred the emergence of various\n",
      "cancer cell line drug-response datasets, driven by the need to facilitate pre-clinical studies by exploring\n",
      "the impact of multi-omics data on drug response. Despite the proliferation of machine learning models\n",
      "for Drug Response Prediction (DRP), their validation remains critical to reliably assess their usefulness\n",
      "for drug discovery, precision oncology and their actual ability to generalize over the immense space of\n",
      "cancer cells and chemical compounds.\n",
      "This paper shows that the commonly used evaluation strategies for DRP methods learn solutions that\n",
      "optimize an unintended DRP score and fail to predict the proper drug-response activity (\"specification\n",
      "gaming\"). This problem hinders the advancement of the DRP field, and here we propose a new validation\n",
      "paradigm composed by three Aggregation Strategies (Global, Fixed-Drug, and Fixed-Cell Line) and\n",
      "three train-test Splitting Strategies to ensure a realistic assessment of the prediction performance. We\n",
      "also scrutinize the challenges associated with using IC50 as a prediction label, showing how its close\n",
      "correlation with the drug concentration ranges worsens the risk of misleading performance assessment.\n",
      "We thus propose also an alternative prediction label for DRP which is safer from this perspective. | None | 0\n",
      "36 | scChat: A Large Language Model-Powered Co-Pilot for Contextualized Single-Cell RNA Sequencing Analysis | https://www.biorxiv.org/content/10.1101/2024.10.01.616063v1 | 10.1101/2024.10.01.616063 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.616063v1.full.pdf | Data-driven methods, including unsupervised and supervised learning, have become essential tools for analyzing single-cell RNA sequencing (scRNA-seq) data. However, these approaches cannot incorporate research context into the analysis, limiting their ability to provide in-depth, contextual insights to researchers. We introduce scChat, a pioneering AI assistant designed for contextualized scRNA-seq analysis. Unlike existing tools that focus on standard tasks like cell annotation, scChat offers advanced capabilities, including research context-based analysis of experimental outcomes, detailed validation of research hypotheses, identification of potential explanations when hypotheses fail, and suggestions for next-step experimental design. Powered by a large language model, scChat integrates function calls for quantitative tasks, employs retrieval-augmented generation to reduce hallucination, and uses an LLM-powered search engine to seamlessly incorporate current literature into the analytical conversation. | None | 0\n",
      "37 | RAPDOR: Using Jensen-Shannon Distance for the computational analysis of complex proteomics datasets | https://www.biorxiv.org/content/10.1101/2024.09.30.615781v2 | 10.1101/2024.09.30.615781 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615781v2.full.pdf | The computational analysis of large proteomics datasets, such as those from gradient profiling or spatially resolved proteomics, is often as crucial as the experimental design. We present RAPDOR, a tool for intuitive analyzing and visualizing such datasets, based on the Jensen-Shannon distance and subsequent analysis of similarities between replicates, applied to three datasets. First, we examined the in-gradient distribution profiles of protein complexes with or without RNase treatment (GradR) to identify the set of RNA-binding proteins (RBPs) in the cyanobacterium Synechocystis sp. PCC 6803. RBPs play pivotal regulatory and structural roles; although numerous RBPs have been identified, the complete set is unknown for any species. RAPDOR identified 80 potential RBPs, including ribosomal proteins, likely RNA-modifying enzymes, and several proteins not previously associated with RNA binding. High-ranking putative RBPs, such as the universal stress protein Sll1388, or the translation inhibitor LrtA/RaiA, were predicted by RAPDOR but not the TriPepSVM algorithm, indicating uncharacterized RBP domains. These data are available online at https://synecho-rapdor.biologie.uni-freiburg.de, providing a comprehensive resource for RNase-sensitive protein complexes in cyanobacteria. We then show by reanalyzing existing datasets, that RAPDOR is effective in examining the intracellular redistribution of proteins under stress conditions. RAPDOR is a generic, non-parametric tool for the intuitive and versatile analysis of highly complex data sets such as the study of protein distributions using fractionation protocols. | None | 0\n",
      "38 | Interpretable models for scRNA-seq data embedding with multi-scale structure preservation | https://www.biorxiv.org/content/10.1101/2023.11.23.568428v3 | 10.1101/2023.11.23.568428 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.11.23.568428v3.full.pdf | The power to efficiently explore high-dimensional single-cell omics data is crucial for many biological applications. Dimensionality reduction techniques have therefore emerged as a basic building block of analytical workflows. By producing salient low-dimensional embeddings, they capture important structures in the data and are often used in visualisation, quality control, discovery and downstream analysis. However, current methods lack trustworthiness measures, resulting in the danger of misinterpretation. We tackle this with ViVAE: a deep learning model that achieves multi-scale preservation of structure and enhances interpretability through a set of novel, visual quality control techniques. In addition, we provide ViScore: a robust scoring framework that advances quantitative evaluation of embeddings with both unsupervised and supervised metrics that are scalable and fair. By applying our novel methods to a variety of datasets, we demonstrate the potential of our framework to spearhead the development of more trustworthy dimensionality reduction. | None | 0\n",
      "39 | ClassiCOL: LC-MS/MS analysis for ancient species Classification via Collagen peptide ambiguation | https://www.biorxiv.org/content/10.1101/2024.10.01.616034v2 | 10.1101/2024.10.01.616034 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.616034v2.full.pdf | LC-MS/MS extends on the MALDI-TOF ZooMS approach by providing fragmentation spectra for each peptide. However, ancient bone samples generate sparse datasets containing only a few collagen proteins, rendering target-decoy strategies unusable and increasing uncertainty in peptide annotation. ClassiCOL embraces and even extends this ambiguity using a novel isoBLAST approach. The exhaustive set of potential peptide candidates created in this way is then used to retain or reject different potential paths at each taxonomic branching point down to the taxonomic level attainable with the sample information, always allowing for potential mixtures in the process. As an end point, all considered ambiguity is graphically represented with a clear prioritization of the species in the sample. Using public as well as in-house data, we demonstrate the performance of this universal postprocessing approach on different instruments and explore the possibility of identifying genetic as well as sample mixtures. Diet reconstruction from 40,000 year old cave hyena coprolites illustrates the exciting potential of this approach. | https://github.com/EngelsI/ClassiCOL | 1\n",
      "40 | HairSplitter: haplotype assembly from long, noisy reads | https://www.biorxiv.org/content/10.1101/2024.02.13.580067v4 | 10.1101/2024.02.13.580067 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.02.13.580067v4.full.pdf | Long-read assemblers face challenges in discerning closely related viral or bacterial strains, often collapsing similar strains in a single sequence. This limitation has been hampering metagenome analysis, where diverse strains may harbor crucial functional distinctions.\n",
      "We introduce a novel software, HairSplitter, designed to retrieve strains from a strain-oblivious assembly and long reads. The method uses a custom variant calling process to operate with erroneous long reads and introduces a new read clustering algorithm to recover an a priori unknown number of strains. On noisy long reads, HairSplitter can recover more strains while being faster than state-of-the-art tools, both in the viral and the bacterial case. | https://github.com/rolandfaure/hairsplitter | 1\n",
      "41 | Investigation of cell metabolism dysfunction in autism spectrum disorders using genome-scale metabolic network analysis of neuronal development | https://www.biorxiv.org/content/10.1101/2024.10.01.616152v1 | 10.1101/2024.10.01.616152 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.616152v1.full.pdf | Autism Spectrum Disorders (ASDs) are a group of highly heterogeneous neurodevelopmental conditions characterized by impairments in social interactions and repetitive behaviors. In spite of extensive efforts, known genetic mutations can hardly explain the ASD cases, which signifies the highly heterogeneous nature of ASDs and calls for shifting to a more holistic approach for understanding the mechanism of these disorders. Several studies have reported that a wide range of metabolic markers is present in blood, urine and brain samples of ASD patients. In this study, using transcriptome data of neurons derived from ASD and control individuals, we reconstructed neuron-specific metabolic network models for two neurodevelopmental time points representing late neural progenitor stage and mature neurons. We simulate gene inactivation in these networks to investigate the altered metabolic fluxes and biological pathways in ASD metabolism. We further examine these networks using reporter metabolite analysis to identify those metabolites around which significant gene expression changes occur. Briefly, our model was successful in predicting previously reported metabolites and pathways implicated in ASD metabolism, including oxidative phosphorylation and fatty acid oxidation, and points to a new finding concerning alterations in the less studied pathways like keratan sulfate degradation pathway and N/O-glycosylation in ASD. Finally, by employing the robust metabolic transformation algorithm (rMTA) we set to find potential drug targets that can reverse the ASD metabolic state into a healthy one. Interestingly, our model found phosphatidic phosphatase enzyme, as well as phosphatidylethaolamine and phosphatidylserine transfer proteins as targets that could recover the healthy metabolic state. | None | 0\n",
      "42 | HERMES: Holographic Equivariant neuRal network model for Mutational Effect and Stability prediction | https://www.biorxiv.org/content/10.1101/2024.07.09.602403v2 | 10.1101/2024.07.09.602403 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.07.09.602403v2.full.pdf | Predicting the stability and fitness effects of amino-acid mutations in proteins is a cornerstone of biological discovery and engineering. Various experimental techniques have been developed to measure mutational effects, providing us with extensive datasets across a diverse range of proteins. By training on these data, machine learning approaches have advanced significantly in predicting mutational effects. Here, we introduce HERMES, a 3D rotationally equivariant structure-based neural network model for mutation effect prediction. Pre-trained to predict amino-acid propensities from their surrounding 3D structure atomic environments, HERMES can be efficiently fine-tuned to predict mutational effects, thanks to its symmetry-aware parameterization of the output space. Benchmarking against other models demonstrates that HERMES often outperforms or matches their performance in predicting mutation effects on stability, binding, and fitness, using either computationally or experimentally resolved protein structures. HERMES offers a versatile suit of tools for evaluating mutation effects and can be easily fine-tuned for specific predictive objectives using our open-source code. | https://github.com/StatPhysBio/hermes/tree/main | 1\n",
      "43 | MiRGraph: A hybrid deep learning approach to identify microRNA-target interactions by integrating heterogeneous regulatory network and genomic sequences | https://www.biorxiv.org/content/10.1101/2023.11.04.565620v3 | 10.1101/2023.11.04.565620 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.11.04.565620v3.full.pdf | MicroRNAs (miRNAs) mediates gene expression regulation by targeting specific messenger RNAs (mRNAs) in the cytoplasm. They can function as both tumor suppressors and oncogenes depending on the specific miRNA and its target genes. Detecting miRNA-target interactions (MTIs) is critical for unraveling the complex mechanisms of gene regulation and promising towards RNA therapy for cancer. There is currently a lack of MTIs prediction methods that simultaneously perform feature learning from heterogeneous gene regulatory network (GRN) and genomic sequences. To improve the prediction performance of MTIs, we present a novel transformer-based multi-view feature learning method -- MiRGraph, which consists of two main modules for learning the sequence-based and GRN-based feature embedding. For the former, we utilize the mature miRNA sequences and the complete 39UTR sequence of the target mRNAs to encode sequence features using a hybrid transformer and convolutional neural network (CNN) (TransCNN) architecture. For the latter, we utilize a heterogeneous graph transformer (HGT) module to extract the relational and structural information from the GRN consisting of miRNA-miRNA, gene-gene and miRNA-target interactions. The TransCNN and HGT modules can be learned end-to-end to predict experimentally validated MTIs from MiRTarBase. MiRGraph outperforms  existing methods in not only recapitulating the true MTIs but also in predicting strength of the MTIs based on the in-vitro measurements of miRNA transfections. In a case study on breast cancer, we identified plausible target genes of an oncomir. | None | 0\n",
      "44 | IgBlend: Unifying 3D Structures and Sequences in Antibody Language Models | https://www.biorxiv.org/content/10.1101/2024.10.01.615796v1 | 10.1101/2024.10.01.615796 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.615796v1.full.pdf | Large language models (LLMs) trained on antibody sequences have shown significant potential in the rapidly advancing field of machine learning-assisted antibody engineering and drug discovery. However, current state-of-the-art antibody LLMs often overlook structural information, which could enable the model to more effectively learn the functional properties of antibodies by providing richer, more informative data. In response to this limitation, we introduce IgBlend, which integrates both the 3D coordinates of backbone atoms (C-alpha, N, and C) and antibody sequences. Our model is trained on a diverse dataset containing over 4 million unique structures and more than 200 million unique sequences, including heavy and light chains as well as nanobodies. We rigorously evaluate IgBlend using established benchmarks such as sequence recovery, complementarity-determining region (CDR) editing and inverse folding and demonstrate that IgBlend consistently outperforms current state-of-the-art models across all benchmarks. Furthermore, experimental validation shows that the model9s log probabilities correlate well with measured binding affinities. | None | 0\n",
      "45 | Better confidence intervals in simulation-based inference | https://www.biorxiv.org/content/10.1101/2024.09.30.615940v1 | 10.1101/2024.09.30.615940 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615940v1.full.pdf | We describe and evaluate a method of statistical inference of model parameters, which revisits the idea of inferring a likelihood surface using simulation when the likelihood function cannot be evaluated. The method aims in particular to provide confidence intervals with controlled coverage, and its performance is assessed accordingly. It is based on a combining the random forest machine-learning method, and multivariate Gaussian mixture (MGM) models, in an effective inference workflow, here used to fit models with up to 15 variable parameters. Masked autoregressive flows, a deep learning technique, is also tested as an alternative do MGM models. The method is compared to that of approximate Bayesian computation (ABC) with random forests, with which it shares some technical features, on scenarios of inference of historical demography from population genetic data. These comparisons highlight the importance of an iterative workflow for exploring the parameter space efficiently. For equivalent simulation effort of the data-generating process, the new summary-likelihood method provides better control of coverage than ABC with random forests, and than generally reported for ABC methods. | None | 0\n",
      "46 | Miniature: Unsupervised glimpses into multiplexed tissue imaging datasets as thumbnails for data portals | https://www.biorxiv.org/content/10.1101/2024.10.01.615855v1 | 10.1101/2024.10.01.615855 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.615855v1.full.pdf | Multiplexed tissue imaging can illuminate complex spatial protein expression patterns in healthy and diseased specimens. Large-scale atlas programs such as Human Tumor Atlas Network and are relying heavily on highly-multiplexed approaches including CyCIF and CODEX to image up to 100 antigens. Such high dimensionality allows a deep understanding of cellular diversity and spatial structure, but can provide a challenge for image visualization and exploration. One challenge for data portals and visualization tools is the generation of an informative and pleasing image preview that captures the full heterogeneity of the image, rather than relying on a multi-channel overlay that may be restricted to 4-6 channels. We describe Miniature, a tool to automatically generate informative image thumbnails from multiplexed tissue images in an unsupervised and scalable manner. Miniature aims to aid researchers in understanding tissue heterogeneity and identifying potential pathological features without extensive manual intervention. Miniature uses a choice of unsupervised dimensionality reduction methods including uniform manifold embedding and projection (UMAP), t-distributed stochastic neighbor Embedding (t-SNE), and principal Component analysis (PCA) to reduce on-tissue pixels from a low-resolution, high dimensional image to two or three dimensions. Pixels are then color encoded by their coordinate in low dimensional space using a choice of color maps. We show that perceptually distinct regions in Miniature thumbnails reflect known pathological features seen in both the source multiplexed tissue image and H&E imaging of the same sample. We evaluate Miniature parameters for dimensionality reduction and pixel color encoding to recommend default configurations that maximize perceptual trustworthiness to both the low-dimensional embedding and high-dimensional image and provide high Mantel correlation between the perceived color difference (delta E 2000) and distance in high- and low-dimensional space. By simulating color vision deficiency, we show that Miniature thumbnails are accessible to all. We demonstrate that Miniature thumbnails are suitable for a wide range of multiplexed tissue imaging modalities and show their application in the Human Tumor Atlas Network Data Portal. | None | 0\n",
      "47 | PHLOWER - Single cell trajectory analysis using Decomposition of the Hodge Laplacian | https://www.biorxiv.org/content/10.1101/2024.10.01.613179v1 | 10.1101/2024.10.01.613179 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.613179v1.full.pdf | Cellular differentiation, the process by which a cell changes its chromatin and expression programs to acquire more specialized functions, is not only crucial in the development of multicellular organisms but also key during onset and progression of diseases.  Transcription factors, which are proteins binding to regulatory DNA regions (open chromatin regions), are key regulators of gene expression thereby orchestrating cellular differentiation processes. Dissecting these key regulatory events will help to develop protocols for cellular reprogramming or ex vivo differentiation in e.g. organoids and to understand disease-related differentiation processes for potential therapeutic interventions. In this context, a unique resource to understand the interplay between chromatin, regulatory signals (transcription factor binding), and expression changes during cellular differentiation, is multi-modal single-cell sequencing, which can measure both full expression programs and genome-wide open chromatin. However, the associated experimental protocols dissociate cells, making it impossible to track how an individual cell differentiates over time experimentally. | None | 0\n",
      "48 | Building pangenome graphs | https://www.biorxiv.org/content/10.1101/2023.04.05.535718v2 | 10.1101/2023.04.05.535718 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.04.05.535718v2.full.pdf | Pangenome graphs can represent all variation between multiple reference genomes, but current approaches to build them exclude complex sequences or are based upon a single reference. In response, we developed the PanGenome Graph Builder (PGGB), a pipeline for constructing pangenome graphs without bias or exclusion. PGGB uses all-to-all alignments to build a variation graph in which we can identify variation, measure conservation, detect recombination events, and infer phylogenetic relationships. | None | 0\n",
      "49 | Analyzing the Molecular Mechanism of Eucalyptol, Limonene and Pinene Enteric Capsules (QIENUO) in the Treatment of Pulmonary cystic fibrosis with Network Pharmacology and Verifying Molecular Docking | https://www.biorxiv.org/content/10.1101/2024.09.30.615978v1 | 10.1101/2024.09.30.615978 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615978v1.full.pdf | Objective: To explore the mechanism of Eucalyptol ,Limonene and Pinene Enteric Capsules（QIENUO） in the treatment of pulmonary cystic fibrosis (CPF), analyze the common targets of QIENUO and CPF, and verify the molecular docking of core proteins and small molecules. Methods: The main active compounds and their corresponding targets were obtained from PubChem, SwissTargetprediction, GeneCards, PharmMapper and TCMSP databases. Targets related to CPF were screened from GeneCards, OMIM, DisGeNET and TTD databases. The common \"QIENUO-CPF\" targets were analyzed by gene ontology (GO) and Kyoto Encyclopedia of Genes and Genomes (KEGG) through the website of Weishengxin. Protein-protein interaction (PPI) network and compound-target-pathway network were constructed by Cytoscape, and the network parameters were systematically analyzed. The interaction between core protein and monomer components was evaluated and verified by molecular docking method. Results: 228 active compounds target and 1354 CPF-related targets were screened out, and 92 common targets were analyzed by GO and KEGG. The results showed that the therapeutic effect of QIENUO on CPF was mainly through AMPK signaling pathway, cGMP-PKG signaling pathway and TGF-β signaling pathway. The results of molecular docking show that the binding energy of 9 of 15 pairs of ligand-receptor pairs is lower than-6 kjmol-1. Conclusion: QIENUO exhibits huge potential as a therapeutic agent for the treatment of pulmonary cystic fibrosis. The specific molecular mechanism and effective active components of QIENUO treat CPF were studied and demonstrated, which provided theoretical basis for better clinical application of QIENUO. | None | 0\n",
      "50 | Enhanced prediction of protein functional identity through the integration of sequence and structural features | https://www.biorxiv.org/content/10.1101/2024.09.30.615718v1 | 10.1101/2024.09.30.615718 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615718v1.full.pdf | Although over 300 million protein sequences are registered in a reference sequence database, only 0.2% have experimentally determined functions. This suggests that many valuable proteins, potentially catalyzing novel enzymatic reactions, remain undiscovered among the vast number of function-unknown proteins. In this study, we developed a method to predict whether two proteins catalyze the same enzymatic reaction by analyzing sequence and structural similarities, utilizing structural models predicted by AlphaFold2. We performed pocket detection and domain decomposition for each structural model. The similarity between protein pairs was assessed using features such as full-length sequence similarity, domain structural similarity, and pocket similarity. We developed several models using conventional machine learning algorithms and found that the LightGBM-based model outperformed the models. Our method also surpassed existing approaches, including those based solely on full-length sequence similarity and state-of-the-art deep learning models. Feature importance analysis revealed that domain sequence identity, calculated through structural alignment, had the greatest influence on the prediction. Therefore, our findings demonstrate that integrating sequence and structural information improves the accuracy of protein function prediction. | None | 0\n",
      "51 | Phage display enables machine learning discovery of cancer antigen specific TCRs | https://www.biorxiv.org/content/10.1101/2024.06.27.600973v2 | 10.1101/2024.06.27.600973 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.06.27.600973v2.full.pdf | T cells targeting epitopes in infectious diseases or cancer play a central role in spontaneous and therapy-induced immune responses. T-cell epitope recognition is mediated by the binding of the T-Cell Receptor (TCR) and TCRs recognizing clinically relevant epitopes are promising for T-cell based therapies. Starting from one of the few known TCRs targeting the cancer-testis antigen NY-ESO-1 157-165 epitope, we built large phage display libraries of TCRs with randomized Complementary Determining Region 3 of the β chain. The TCR libraries were panned against the NY-ESO-1 epitope, which enabled us to collect thousands of epitope-specific TCR sequences. We then trained a machine learning TCR-epitope interaction predictor with this data and could identify several epitope-specific TCRs directly from TCR repertoires. Cellular binding and functional assays revealed that the predicted TCRs displayed activity towards the NY-ESO-1 epitope and no detectable cross-reactivity with self-peptides. Overall, our work demonstrates how display technologies combined with machine learning models of TCR-epitope recognition can effectively leverage large TCR repertoires for TCR discovery. | None | 0\n",
      "52 | AptaDiff: de novo design and optimization of aptamers based on diffusion models | https://www.biorxiv.org/content/10.1101/2023.11.25.568693v2 | 10.1101/2023.11.25.568693 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.11.25.568693v2.full.pdf | Aptamers are single-stranded nucleic acid ligands, featuring high affinity and specificity to target molecules. Traditionally they are identified from large DNA/RNA libraries using in vitro methods, like Systematic Evolution of Ligands by Exponential Enrichment (SELEX). However, these libraries capture only a small fraction of theoretical sequence space, and various aptamer candidates are constrained by actual sequencing capabilities from the experiment. Addressing this, we proposed AptaDiff, the first in silico aptamer design and optimization method based on the diffusion model. Our Aptadiff can generate aptamers beyond the constraints of high-throughput sequencing data, leveraging motif-dependent latent embeddings from variational autoencoder, and can optimize aptamers by affinity-guided aptamer generation according to Bayesian optimization. Comparative evaluations revealed AptaDiff9s superiority over existing aptamer generation methods in terms of quality and fidelity across four high-throughput screening data targeting distinct proteins. Moreover, Surface Plasmon Resonance (SPR) experiments were conducted to validate the binding affinity of aptamers generated through Bayesian optimization for two target proteins. The results unveiled a significant boost of 87.9% and 60.2% in RU values, along with a 3.6-fold and 2.4-fold decrease in KD values for the respective target proteins. Notably, the optimized aptamers demonstrated superior binding affinity compared to top experimental candidates selected through SELEX, underscoring the promising outcomes of our AptaDiff in accelerating the discovery of superior aptamers. | https://github.com/wz-create/AptaDiff, http://github.com/SheffieldML/ | 1\n",
      "53 | Gut microbiota and type 2 diabetes associaction: A meta-analysis of 16S studies and their methodological challenges | https://www.biorxiv.org/content/10.1101/2024.10.01.615711v1 | 10.1101/2024.10.01.615711 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.10.01.615711v1.full.pdf | Diabetes mellitus is a prevalent chronic non-communicable disease, and recent studies have explored the link between gut microbiota and its development. Despite some evidence suggesting an association, the influence of gut microbiota on type 2 diabetes (DM2) remains unclear. A systematic search of PubMed (Janeiro 2016 to 10 December 2023) using the keywords: (16S and diabetes or DM2) and (gut microbiota and diabetes or DM2). The studies included compared gut microbiome diversity between diabetic and non-diabetic adults using 16S rRNA sequencing, excluding children, interventions, and type 1 diabetes. Alpha diversity indices and bacterial mean abundance were analyzed, with statistical assessments using a random-effects model and I2 for heterogeneity. Thirteen studies met the criteria, with the Shannon index being the most commonly used measure. Results showed significant heterogeneity (I2 > 75%) and no notable differences between diabetic and non-diabetic groups. Other indices, such as Chao1 and phylogenetic whole tree, similarly showed no consistent differences. Taxonomic analysis also failed to find phyla consistently correlated with DM2, with variability across studies. The relationship between gut microbiota and diabetes remains uncertain due to technical and biological factors that are often overlooked. The inconsistencies across studies highlight the low reproducibility common in microbiota research. | None | 0\n",
      "54 | scGNN+: Adapting ChatGPT for Seamless Tutorial and Code Optimization | https://www.biorxiv.org/content/10.1101/2024.09.30.615735v1 | 10.1101/2024.09.30.615735 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615735v1.full.pdf | Foundation models have transformed AI by leveraging large-scale data to efficiently perform diverse tasks, and their applications in bioinformatics are primarily focused on data-centric tasks like cell type annotation and gene expression analysis. However, their potential extends beyond data analysis, offering significant opportunities in software development and optimization, such as code refinement, tutorial generation, and advanced visualization. For example, models like OpenAI Codex can suggest optimized code snippets and generate well-documented, reproducible workflows, enhancing accessibility and reducing computational complexity. Despite these advantages, the use of foundation models for improving computational tool engineering in single-cell research remains underutilized. To address this gap, we developed scGNN+, a web-based platform that combines the power of graph neural networks with the capabilities of ChatGPT to enhance reproducibility, code optimization, and visualization. scGNN+ further simplifies the process for users by generating standardized, well-annotated code, making complex procedures more accessible to non-programmers. Additionally, ChatGPT integration allows users to create high-quality, customizable visualizations through natural language prompts, improving data interpretation and presentation. Ultimately, scGNN+ offers a user-friendly, reproducible, and optimized solution for single-cell research, leveraging the full potential of foundation models in bioinformatics. scGNN+ is publicly available at https://bmblx.bmi.osumc.edu/scgnn+. | None | 0\n",
      "55 | Isobaric labeling update in MaxQuant | https://www.biorxiv.org/content/10.1101/2024.09.30.615788v1 | 10.1101/2024.09.30.615788 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615788v1.full.pdf | We present an update of the MaxQuant software for isobaric labeling data and evaluate its performance on benchmark datasets. Impurity correction factors can be applied to labels mixing C- and N-type reporter ions, such as TMT Pro. Application to a single-cell multi-species mixture benchmark shows high accuracy of the impurity-corrected results. TMT data recorded with FAIMS separation can be analyzed directly in MaxQuant without splitting the raw data into separate files per FAIMS voltage. Weighted median normalization, is applied to several datasets, including large-scale human body atlas data. In the benchmark datasets the weighted median normalization either removes or strongly reduces the batch effects between different TMT plexes and results in clustering by biology. In datasets including a reference channel, we find that weighted median normalization performs as well or better when the reference channel is ignored and only the sample channel intensities are used, suggesting that the measurement of a reference channel is unnecessary when using weighted median normalization in MaxQuant. We demonstrate that MaxQuant including the weighted median normalization performs well on multi-notch MS3 data, as well as on phosphorylation data. MaxQuant is freely available for any purpose and can be downloaded from https://www.maxquant.org/. | https://github.com/cox-labs/CoxLab_Bug_Reporting, https://github.com/JurgenCox/mqtools7 | 1\n",
      "56 | Preserving Full Spectrum Information in Imaging Mass Spectrometry Data Reduction | https://www.biorxiv.org/content/10.1101/2024.09.30.614425v1 | 10.1101/2024.09.30.614425 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.614425v1.full.pdf | Imaging Mass Spectrometry (IMS) has become an important tool for molecular characterization of biological tissue. However, IMS experiments tend to yield large datasets, routinely recording over 200,000 ion intensity values per mass spectrum and more than 100,000 pixels, i.e., spectra, per dataset. Traditionally, IMS data size challenges have been addressed by feature selection or extraction, such as by peak picking and peak integration. Selective data reduction techniques such as peak picking only retain certain parts of a mass spectrum, and often these describe only medium-to-high-abundance species. Since lower-intensity peaks and, for example, near-isobar species are sometimes missed, selective methods can potentially bias downstream analysis towards a subset of species in the data rather than considering all species measured. Results: We present an alternative to selective data reduction of IMS data that achieves similar data size reduction while better conserving the ion intensity profiles across all recorded m/z-bins, thereby preserving full spectrum information. Our method utilizes a low-rank matrix completion model combined with a randomized sparse-format-aware algorithm to approximate IMS datasets. This representation offers reduced dimensionality and a data footprint comparable to peak picking, but also retains complete spectral profiles, enabling comprehensive analysis and compression. We demonstrate improved preservation of lower signal-to-noise-ratio signals and near-isobars, mitigation of selection bias, and reduced information loss compared to current state-of-the art data reduction methods in IMS. | https://github.com/vandeplaslab/full_profile | 1\n",
      "57 | BindCraft: one-shot design of functional protein binders | https://www.biorxiv.org/content/10.1101/2024.09.30.615802v1 | 10.1101/2024.09.30.615802 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615802v1.full.pdf | Protein-protein interactions (PPIs) are at the core of all key biological processes. However, the complexity of the structural features that determine PPIs makes their design challenging. We present BindCraft, an open-source and automated pipeline for de novo protein binder design with experimental success rates of 10-100%. BindCraft leverages the trained deep learning weights of AlphaFold2 to generate nanomolar binders without the need for high-throughput screening or experimental optimization, even in the absence of known binding sites. We successfully designed binders against a diverse set of challenging targets, including cell-surface receptors, common allergens, de novo designed proteins, and multi-domain nucleases, such as CRISPR-Cas9. We showcase their functional and therapeutic potential by demonstrating that designed binders can reduce IgE binding to birch allergen in patient-derived samples. This work represents a significant advancement towards a \"one design-one binder\" approach in computational design, with immense potential in therapeutics, diagnostics, and biotechnology. | https://github.com/martinpacesa/BindCraft/blob/main/notebooks/BindCraft.ipynb, https://github.com/martinpacesa/BindCraft | 1\n",
      "58 | UniGEN-DDI: Computing drug-drug interactions using a unified graph embedding network | https://www.biorxiv.org/content/10.1101/2024.09.30.615812v1 | 10.1101/2024.09.30.615812 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615812v1.full.pdf | Finding drug-drug interaction is crucial for patient safety and treatment efficacy. Two drugs may show a synergistic effect but may sometimes cause a severe health issue, including lethality. Wet lab studies are often performed to understand such interactions but are limited by cost and time. However, the biochemical data generated can be explored to compute unknown interactions. Here, we developed a computational model named UniGEN-DDI (Unified Graph Embedding Network for Drug-Drug Interaction) for the estimation of interactions between drugs. It is a simple unified network model containing the biochemical information of drug association developed using the compiled data from DrugBank 5.1.0. The feature learning of the drugs was carried out using a combination of GraphSAGE and Node2Vec algorithms, which were found efficient in extracting diverse features. The simple architecture of our model led to a significant reduction in computational time compared to the baselines, while maintaining a high prediction accuracy. The model performed well on the data, which was equally distributed between interacting and non-interacting drugs. As a more challenging evaluation, we performed non-overlapping splitting of the data based on the drug action on different parts of the body, and our model performed well in both interaction estimation and time efficiency. Our model successfully identified the 12 unknown drug interactions in DrugBank 5.1.0, which were updated in DrugBank 6.0. | None | 0\n",
      "59 | Divergence of the Individual repeats in the leucine-rich repeat domains of human Toll-like receptors explain their diversity and functional adaptations | https://www.biorxiv.org/content/10.1101/2024.09.30.615863v1 | 10.1101/2024.09.30.615863 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615863v1.full.pdf | The receptor domains of Toll-like receptors (TLRs) are characterized by a solenoid-like structure composed of tandem repeats of α/β units known as Leucine Rich Repeats (LRRs). LRR proteins form large paralogous families, with nearly 400 in the human genome alone, all sharing similar semi-regular solenoid-like structures. Despite this structural similarity, they exhibit remarkable diversity in binding specificity. For TLR receptors, this includes a range of pathogen-associated molecular patterns (PAMPs), while other LRR proteins bind an extensive array of ligands, including proteins, DNA, RNA, and small molecules. The LRR domains contain repeats that have similar, yet not identical, 3D structures and patterns of conserved residues. Through in-depth analysis of sequence and structural conservation in individual repeats of human TLRs, we demonstrate that even subtle variations between these repeats alter the local solenoid structure, leading to significant functional changes. Variations in repeat length and defining patterns result in local changes in curvature and the emergence of structural features such as loops, cavities, or specific interaction interfaces. Understanding how divergence in LRR repeats influences their functional roles can provide deeper insights into their binding mechanisms, including interactions with unknown ligands, both in humans and across the diverse phylogenetic spectrum of animals that rely on their TLR repertoires for immune activation. | None | 0\n",
      "60 | SAMURAI: Shallow Analysis of Copy nuMber alterations Using a Reproducible And Integrated bioinformatics pipeline | https://www.biorxiv.org/content/10.1101/2024.09.30.615766v1 | 10.1101/2024.09.30.615766 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615766v1.full.pdf | Shallow whole-genome sequencing (sWGS) offers a cost-effective approach to detect  copy number alterations (CNAs). However, there remains a gap for a standardized workflow specifically designed for sWGS analysis. To address this need, in this work we present SAMURAI  a bioinformatics pipeline specifically designed for analyzing CNAs from sWGS data in a standardized and reproducible manner. SAMURAI is built using established community standards, ensuring portability, scalability, and reproducibility. The pipeline features a modular design with independent blocks for data pre-processing, copy number analysis, and customized reporting. Users can select workflows tailored for either solid or liquid biopsy analysis (e.g., circulating tumor DNA), with specific tools integrated for each sample type. The final report generated by SAMURAI provides detailed results to facilitate data interpretation and potential downstream analyses.To demonstrate its robustness, SAMURAI was validated using simulated and real-world datasets. The pipeline achieved high concordance with ground truth data and maintained consistent performance across various scenarios. By promoting standardization and offering a versatile workflow, SAMURAI empowers researchers in diverse environments to reliably analyze CNAs from sWGS data. This, in turn, holds promise for advancements in precision medicine. | https://github.com/DIncalciLab/samurai, https://github.com/DincalciLab/samurai, https://github.com/DIncalciLab/SAMURAI_paper_scripts | 1\n",
      "61 | Capturing the Hierarchically Assorted Protein-protein Interaction Modules of Mammalian Cell | https://www.biorxiv.org/content/10.1101/2024.09.30.615776v1 | 10.1101/2024.09.30.615776 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615776v1.full.pdf | Proteins are organized into modules by both functions and physical interactions within compartments of an eukaryotic cell. The in vivo chemical crosslinking mass spectrometry (XL-MS) data collected from organelles, the whole cells and tissues are able to provide unique information about both protein-protein interaction (PPI) and the intensity of PPI. In the present study, we have retrieved 55,982 crosslinked peptides (XL-peptides) from the XL-MS databases, out of which 6,356 in vivo PPIs were identified. Introduction of the MONET software into analysis of 4,526 hetero PPIs revealed a total of 402 protein modules, including 15, 58 and 163 stable protein complex(s), condensate-forming protein module(s) and intrinsically disordered region (IDR)-containing protein module(s), respectively. The application of ChatGPT in analysis of these modules unexpectedly identified 4 vesicle-related modules. Together, these modules were assorted into 6 communities (module of modules) and 3 systems (module of communities) hierarchically. The bioinformatic analysis found that the three systems are corresponding to three compartments of eukaryotic cell: nuclei, mitochondria, endoplasmic reticulum (ER), respectively. This study presents a novel and comprehensive biomolecular modulome of a mammalian cell, which captures putative protein compositions of protein complexes, protein condensates and vesicles and provides a hierarchical protein organization and function within compartments of mammalian cell. | None | 0\n",
      "62 | Separating biological variance from noise by applying EM algorithm to modified General Linear Model | https://www.biorxiv.org/content/10.1101/2024.09.29.615661v1 | 10.1101/2024.09.29.615661 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.29.615661v1.full.pdf | Introduction: The General Linear Model (GLM) has been widely used in research, where error term has been treated as noise. However, compelling evidence suggests that in biological systems, the target variables may possess their innate variances. \n",
      "Methods: A modified GLM was proposed to explicitly model biological variance and non-biological noise. Employing the Expectation and Maximization (EM) scheme can distinguish biological variance from noise, termed EMSEV (EM for Separating Variances). The performance of EMSEV was evaluated by varying noise levels, dimensions of the design matrix, and covariance structures of the target variables.\n",
      "Results: The deviation between EMSEV outputs and the pre-defined distribution parameters increased with noise level. With a proper initial guess, when the noise magnitude and the variance of the target variables were similar, there were deviations of 3% and 10 to 16% in the estimated mean and covariance of the target variables, respectively, along with a 1.7% deviation in noise estimation.\n",
      "Conclusion: EMSEV appears promising for distinguishing signal variance from noise in biological systems. The potential applications and implications in biological science and statistical inference are discussed. | None | 0\n",
      "63 | Advancing Protein-DNA Binding Site Prediction: Integrating Sequence Models and Machine Learning Classifiers | https://www.biorxiv.org/content/10.1101/2023.08.23.554389v2 | 10.1101/2023.08.23.554389 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.08.23.554389v2.full.pdf | Predicting protein-DNA binding sites is a challenging computational problem that has led to the development of advanced algorithms and techniques in the field of bioinformatics. Identifying the specific residues where proteins bind to DNA is of paramount importance, as it enables the modeling of their interactions and facilitates downstream studies. Nevertheless, the development of accurate and efficient computational methods for this task remains a persistent challenge. Accurate prediction of protein-DNA binding sites has far-reaching implications for understanding molecular mechanisms, disease processes, drug discovery, and synthetic biology applications. It helps bridge the gap between genomics and functional biology, enabling researchers to uncover the intricacies of cellular processes and advance our knowledge of the biological world. The method used to predict DNA binding residues in this study is a potent combination of conventional bioinformatics tools, protein language models, and cutting-edge machine learning and deep learning classifiers. On a dataset of protein-DNA binding sites, our model is meticulously trained, and it is then rigorously examined using several experiments. As indicated by higher predictive behavior with AUC values on two benchmark datasets, the results show superior performance when compared to existing models. The suggested model has a strong capacity for generalization and shows specificity for DNA-binding sites. We further demonstrated the adaptability of our model as a universal framework for binding site prediction by training it on a variety of protein-ligand binding site datasets. In conclusion, our innovative approach for predicting protein-DNA binding residues holds great promise in advancing our understanding of molecular interactions, thus paving the way for several groundbreaking applications in the field of molecular biology and genetics. Our approach demonstrated efficacy and versatility underscore its potential for driving transformative discoveries in biomolecular research. | https://github.com/sarwanpasha/Binding-Site-Prediction | 1\n",
      "64 | Exploring the link between extended red blood cell parameters and platelet indices in voluntary blood donors | https://www.biorxiv.org/content/10.1101/2024.09.29.615743v1 | 10.1101/2024.09.29.615743 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.29.615743v1.full.pdf | Background: As regular blood donors are prone to iron deficiency, importance of extended red blood cell (eRBC) parameters in identifying donors with depleted iron stores was investigated.Thrombocytosis has been well documented in patients affected with IDA. Thus, significance of eRBC parameters in identifying iron deficiency associated thrombocytosis was also examined in this cohort. Methods: Blood samples were collected in EDTA tubes from consenting donors for analyses of routine haematological and eRBC parameters. Serum samples were isolated for estimation of iron parameters. Results: Iron deficient donors had significantly altered eRBC parameters. Among them, Ret-He with a cut-off of ≥32 pg had high AUC (0.822) and showed relatively high sensitivity & specificity in detecting iron deficiency. Combination of Ret-He with CCI increased sensitivity & specificity to 90.6% and 98.2%, in detection of donors affected with iron restricted erythropoiesis. This cohort had increased platelet counts, which showed significant association with Ret-He (β= -0.373), RBC-He (β= -0.384), CCI (β= 0.384), Hypo-He (β=0.494) and Micro-R (β= 0.299). Elevated platelet counts also showed significant correlations with these eRBC parameters, which was absent in iron replete donors. Conclusions: eRBC parameters are sensitive indicators of non-anaemic iron deficiency, which may be enhanced by combining them. Their significant association with elevated platelet counts in iron deficient donors, highlights their importance in reflecting iron deficiency associated thrombocytosis. | None | 0\n",
      "65 | PlasmidGPT: a generative framework for plasmid design and annotation | https://www.biorxiv.org/content/10.1101/2024.09.30.615762v1 | 10.1101/2024.09.30.615762 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615762v1.full.pdf | We introduce PlasmidGPT, a generative language model pretrained on 153k engineered plasmid sequences from Addgene. PlasmidGPT generates de novo sequences that share similar characteristics with engineered plasmids but show low sequence identity to the training data. We demonstrate its ability to generate plasmids in a controlled manner based on the input sequence or specific design constraint. Moreover, our model learns informative embeddings of both engineered and natural plasmids, allowing for efficient prediction of a wide range of sequence-related attributes. | https://github.com/lingxusb/PlasmidGPT | 1\n",
      "66 | Non-detection during excursions by citizen scientists modeled as a function of weather, season, list length, and individual preferences | https://www.biorxiv.org/content/10.1101/2024.09.30.615418v1 | 10.1101/2024.09.30.615418 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615418v1.full.pdf | INTRODUCTION: Citizen science is an increasingly valuable source of information about biodiversity. It is challenging to use this information for analysis of distribution and trends. The lack of a protocol leads to bias in observations and therefore data are not representative. The bias is a consequence of unequal detection probabilities, caused by different preferences and habits of citizen scientists. METHODS: We propose to incorporate characteristics of these excursions in analyses of data collected by citizen scientists to improve estimates of the probability that a species is not detected and reported, even though it does occur. By limiting these models to areas that are known to be occupied, detection can be modeled separately without considering variation in occupancy. We apply this idea to 150 common species in the Southwest Delta of The Netherlands, and illustrate the data selection, the modeling process and the results using four species. RESULTS: The strongest features to predict detection are the number of species during a visit (list length), earlier observations of the target species by the same observer, and the day of year. We compare three approaches to predict the total non-detection probability that takes all visits to an area into account. Predictions based on only the number of visits were outperformed by predictions that also take the list length into account. Our predictions based on all features combined consistently beat both other approaches, across all 10 species groups that were compared. DISCUSSION: We thus show that explicitly modelling the characteristics of all visits to an occupied area results in estimation of non-detection probabilities, while providing insight into the causes of detection and reporting bias. Furthermore, predictions of our model provide a basis for quantifying the sampling effort in each area, which is a promising first step to correct bias in citizen science data when aiming to map a species′ distribution. | None | 0\n",
      "67 | Unlocking the Potential of Low Quality Total RNA-seq Data: A Stepwise Mapping Approach for Improved Quantitative Analyses | https://www.biorxiv.org/content/10.1101/2024.09.30.615750v1 | 10.1101/2024.09.30.615750 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615750v1.full.pdf | High-throughput sequencing assays face persistent challenges when analyzing low-quality RNAs, often assessed by the RNA integrity number (RIN). Current preprocessing methods and pipelines designed for mRNA-seq presume high-quality RNAs, overlooking the nuanced complexities arising from degraded transcripts in low-quality samples. This study questions the applicability of standard analysis pipelines, especially when sequencing low-quality total RNAs, which are sometimes the sole recourse for specific inquiries. To address this, we conducted a comprehensive investigation using large sequencing reads obtained from blood biospecimens with varying RINs. Introducing a novel mapping approach, termed 9stepwise mapping9, our systematic comparative analyses propose an optimal practice for total RNA-seq data analysis. Contrary to conventional mapping procedures, the 9stepwise mapping9 approach unveils additional transcriptome information, crucial for stable differential expression analysis, even with total RNA-seq data from specimens with relatively low RINs. Our method proves particularly valuable when analyzing limited specimens with low RNA quality. | None | 0\n",
      "68 | STIX: Long-reads based Accurate Structural Variation Annotation at Population Scale | https://www.biorxiv.org/content/10.1101/2024.09.30.615931v1 | 10.1101/2024.09.30.615931 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.30.615931v1.full.pdf | The prioritization of Structural Variants (SV), which is needed to rank and identify potential pathogenic alleles, is still in its infancy. This is exemplified over gnomAD only being able to annotate 33.5% of GIAB SVs. To overcome this, we present the first long-read based annotation resource for both GRCh38 and CHM13-T2T reference using STIX. In contrast to previous methods, STIX indexes SV-informative long-reads themselves, can thus be easily extended and accurately annotate all SV types including insertions. STIX successfully annotated 95.9% of GIAB Tier1 SVs. STIX further improved cancer based SV prioritization by highlighting 3,563 SV from COSMIC being common in the population. We further showcase that mosaic SV can be independently gained and may be widely spread throughout the population. This highlights the need for accurate SV population frequency annotation to further facilitate the adoption of SV via long-read sequencing in medical research and clinical applications. | https://github.com/behzodcu/staffr, https://github.com/zhengxinchang/excord-lr, https://github.com/ryanlayer/stix | 1\n",
      "69 | A patient-centric knowledge graph approach to prioritize mutants for selective anti-cancer targeting | https://www.biorxiv.org/content/10.1101/2024.09.29.615658v1 | 10.1101/2024.09.29.615658 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.29.615658v1.full.pdf | Personalized oncology has revolutionized cancer treatment by targeting specific genetic aberrations in tumors. However, the identification of suitable targets for anti-cancer therapies remains a challenge. In this study, we introduce a knowledge graph approach to prioritize cancer mutations with clinical, functional, and structural significance as potential therapeutic targets. Focusing on the human kinome, we integrate protein-protein interaction and patient-centric mutation networks to construct a comprehensive network enriched with transcriptomic, structural, and drug response data, together covering five layers of information. Moreover, we make the constructed knowledge graph publicly available, along with a plethora of scripts to facilitate further annotation and expansion of the network. Interactive visualization resources are also provided, ensuring accessibility for researchers regardless of computational expertise and enabling detailed analysis by cancer type and individual layers of information. This comprehensive resource has the potential to identify relevant mutations for targeted therapeutic interventions, thereby advancing personalized oncology and improving patient outcomes. | None | 0\n",
      "70 | DeepPWM-BindingNet: Unleashing Binding Prediction with Combined Sequence and PWM Features | https://www.biorxiv.org/content/10.1101/2024.08.21.609040v2 | 10.1101/2024.08.21.609040 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.08.21.609040v2.full.pdf | A crucial challenge in molecular biology is the prediction of DNA-protein binding interactions, which has applications in the study of gene regulation and genome functionality. In this paper, we present a novel deep-learning framework to predict DNA-protein binding interactions with increased precision and interoperability. Our proposed framework DeepPWM-BindingNet leverages the rich information encoded in Position Weight Matrices (PWMs), which capture the sequence-specific binding preferences of proteins. These PWM-derived features are seamlessly integrated into a hybrid model of convolutional recurrent neural networks (CRNNs) that extracts hierarchical features from DNA sequences and protein structures. The sequential dependencies within the sequences are captured by recurrent layers. By incorporating PWM-derived features, the model9s interpretability is improved, enabling researchers to learn more about the underlying binding mechanisms. The model9s capacity to locate crucial binding sites is improved by the incorporation of an attention mechanism that highlights crucial regions. Experiments on diverse DNA-protein interaction datasets demonstrate the proposed approach improves the predictive performance. The proposed model holds significant potential in deciphering intricate DNA-protein interactions, ultimately advancing our comprehension of gene regulation mechanisms. | None | 0\n",
      "71 | A new framework for evaluating model out-of-distribution for the biochemical domain* | https://www.biorxiv.org/content/10.1101/2024.03.14.584508v2 | 10.1101/2024.03.14.584508 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.03.14.584508v2.full.pdf | Quantifying model generalization to out-of-distribution data has been a longstanding challenge in machine learning. Addressing this issue is crucial for leveraging machine learning in scientific discovery, where models must generalize to new molecules or materials. Current methods typically split data into train and test sets using various criteria, like temporal, sequence identity, scaffold, or random cross-validation; before evaluating model performance. However, with so many splitting criteria available, existing approaches offer limited guidance on selecting the most appropriate one, and they do not provide mechanisms for incorporating prior knowledge about the target deployment distribution(s). To tackle this problem, we have developed a novel metric, AU-GOOD, which quantifies expected model performance under conditions of increasing dissimilarity between train and test sets, while also accounting for prior knowledge about the target deployment distribution(s), when available. This metric is broadly applicable to biochemical entities, including proteins, small molecules, nucleic acids, or cells; as long as a relevant similarity function is defined for them. Recognizing the wide range of similarity functions used in biochemistry, we propose criteria to guide the selection of the most appropriate metric for partitioning. We also introduce a new partitioning algorithm that generates more challenging test sets, and we propose statistical methods for comparing models based on AU-GOOD. Finally, we demonstrate the insights that can be gained from this framework by applying it to two different use cases: developing predictors for pharmaceutical properties of small molecules, and using protein language models as embeddings to build biophysical property predictors. Source code is available at https://github.com/IBM/Hestia-OOD. | https://github.com/IBM/Hestia-OOD | 1\n",
      "72 | SMARTER-database: a tool to integrate SNP array datasets for sheep and goat breeds | https://www.biorxiv.org/content/10.1101/2024.09.01.610681v2 | 10.1101/2024.09.01.610681 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.01.610681v2.full.pdf | Underutilized sheep and goat breeds have the ability to adapt to challenging environments due to their genetic composition. Integrating publicly available genomic datasets with new data will facilitate genetic diversity analyses; however, this process is complicated by important data discrepancies, such as outdated assembly versions or different data formats. Here we present the SMARTER-database, a collection of tools and scripts to standardize genomic data and metadata mainly from SNP chips arrays on global small ruminant populations with a focus on reproducibility. SMARTER-database harmonizes genotypes for about 12,000 sheep and 6,000 goats to a uniform coding and assembly version. Users can access the genotype data via FTP and interact with the metadata through a web interface or programmatically using their custom scripts, enabling efficient filtering and selection of samples. These tools will empower researchers to focus on the crucial aspects of adaptation and contribute to livestock sustainability, leveraging the rich dataset provided by the SMARTER-database. | https://github.com/cnr-ibba/SMARTER-database, https://github.com/EBIvariation/vcf-validator?tab=readme-ov-file#assembly-checker,accessed:2024-04-16 | 1\n",
      "73 | Novel AI-powered computational method using tensor decomposition can discover the common optimal bin sizes when integrating multiple Hi-C datasets | https://www.biorxiv.org/content/10.1101/2024.09.29.615651v1 | 10.1101/2024.09.29.615651 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.29.615651v1.full.pdf | iscovering the optimal bin sizes (or resolutions) for the integration of multiple Hi-C datasets is difficult, since bin sizes must be common over multiple datasets whereas dependence of quality upon bin sizes can vary from dataset to dataset. Moreover, we are not supposed to seek common structures in the smaller bin sizes than optimal bin sizes if we cannot suppress the appearance of phase transition-like phenomena even after increasing the number of mapped short reads per bins, since it might mean that there are no common structures any more in finer resolutions and individual Hi-C datasets might have to be analyzed separately. Thus, quality assessments of individual datasets have the limited ability to determine the best bin size common for all datasets. In this study, we propose, the first-ever to our knowledge, a novel adoption of tensor decomposition (TD) based unsupervised feature extraction (FE) to choose the optimal bin sizes for the integration of multiple Hi-C datasets. The use of TD-based unsupervised FE can exhibit something like phase-transition phenomena, by which we can automatically estimate the possible smallest bin size (or the highest resolution) empirically without manually setting some threshold value by hand for the integration of multiple Hi-C datasets. | https://github.com/tagtag/TDbasedUFEHiC | 1\n",
      "74 | A systematic comparison of computational methods for expression forecasting | https://www.biorxiv.org/content/10.1101/2023.07.28.551039v2 | 10.1101/2023.07.28.551039 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.07.28.551039v2.full.pdf | Expression forecasting methods use machine learning models to predict how a cell will alter its transcriptome upon perturbation. Such methods are enticing because they promise to answer pressing questions in fields ranging from developmental genetics to cell fate engineering and because they are a fast, cheap, and accessible complement to the corresponding experiments. However, the absolute and relative accuracy of these methods is poorly characterized, limiting their informed use, their improvement, and the interpretation of their predictions. To address these issues, we created a benchmarking platform that combines a panel of 11 large-scale perturbation datasets with an expression forecasting software engine that encompasses or interfaces to a wide variety of methods. We used our platform to systematically assess methods, parameters, and sources of auxiliary data, finding that performance strongly depends on the choice of metric, and especially for simple metrics like mean squared error, it is uncommon for expression forecasting methods to out-perform simple baselines. Our platform will serve as a resource to improve methods and to identify contexts in which expression forecasting can succeed. | https://github.com/ekernf01/perturbation_benchmarking | 1\n",
      "75 | Quantifying Cancer Drug Targetability across Biochemical Pathways | https://www.biorxiv.org/content/10.1101/2024.09.27.615497v1 | 10.1101/2024.09.27.615497 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.27.615497v1.full.pdf | The evaluation of cancer drug treatments within the complex context of biological networks and pathways is challenging. Drugs can be \"promiscuous,\" targeting multiple genes or pathways, not just the ones of interest. This complexity makes it difficult to assess the full impact of a drug solely based on individual knowledgebases. We make the impact more comprehensible through the characterization of drug targetability in terms of a combination of biochemical pathway data from the Reactome Knowledgebase and drug-target interaction data from the Cancer Targetome. This leads to a graphical representation of drug targetability. We further improve comprehensibility by developing metrics to quantify the graph structures in order to evaluate drug target specificity and navigate the hierarchical nature of biological networks and identify candidate treatment options to increase specificity and minimize off-target effects. We have developed software that calculates these metrics for drugs in the Cancer Targetome. The software is available for research purposes only at https://github.com/sehagler/drug_targetability. | https://github.com/sehagler/drug_targetability | 1\n",
      "76 | Deep learning-based predictions of gene perturbation effects do not yet outperform simple linear methods | https://www.biorxiv.org/content/10.1101/2024.09.16.613342v3 | 10.1101/2024.09.16.613342 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.16.613342v3.full.pdf | Advanced deep-learning methods, such as transformer-based foundation models, promise to learn representations of biology that can be employed to predict in silico the outcome of unseen experiments, such as the effect of genetic perturbations on the transcriptomes of human cells. To see whether current models already reach this goal, we benchmarked two state-of-the-art foundation models and one popular graph-based deep learning framework against deliberately simplistic linear models in two important use cases: For combinatorial perturbations of two genes for which only data for the individual single perturbations have been seen, we find that a simple additive model outperformed the deep learning-based approaches. Also, for perturbations of genes that have not yet been seen, but which may be \"interpolated\" from biological similarity or network context, a simple linear model performed as good as the deep learning-based approaches. While the promise of deep neural networks for the representation of biological systems and prediction of experimental outcomes is plausible, our work highlights the need for critical benchmarking to direct research efforts that aim to bring transfer learning to biology. | None | 0\n",
      "77 | An Efficient Computational Method to Create Positive NIPT Samples with Autosomal Trisomy | https://www.biorxiv.org/content/10.1101/2023.11.24.568620v4 | 10.1101/2023.11.24.568620 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.11.24.568620v4.full.pdf | Noninvasive prenatal test (NIPT) has been widely used for screening trisomy on chromosomes 13 (T13), 18 (T18), and 21 (T21). However, the false negative rate of NIPT algorithms has not been thoroughly evaluated due to the lack of positive samples. In this study, we present an efficient computational approach to create positive samples with autosomal trisomy from negative samples. We applied the approach to establish a low coverage dataset of 1440 positive samples with T13, T18, and T21 aberrations for both mosaic and non-mosaic conditions. We examined the performance of WisecondorX and its improvement, called VINIPT, on both negative and positive datasets. Experiments showed that WisecondorX and VINIPT with a z-score threshold of 3.3 were able to detect all non-mosaic samples with T13, T18, and T21 aberrations (i.e., the sensitivity of 100%). Using a lower z-score threshold of 2.58 when analyzing mosaic samples, both WisecondorX and VINIPT have the overall sensitivity of 99.7% on detecting T13, T18, and T21 aberrations from mosaic samples. WisecondorX has the specificity of 98.5% for the non-mosaic analysis, but a considerably lower specificity of 95% for the mosaic analysis. VINIPT has a much better specificity than WisecondorX, i.e., 99.9% for the non-mosaic analysis, and 98.2% for the mosaic analysis.  The results suggest that VINIPT can play as a powerful NIPT tool for the low coverage data. | None | 0\n",
      "78 | Impact of Interval Censoring on Data Accuracy and Machine Learning Performance in Biological High-Throughput Screening | https://www.biorxiv.org/content/10.1101/2024.09.25.615059v2 | 10.1101/2024.09.25.615059 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.25.615059v2.full.pdf | High-throughput screening (HTS) combined with deep mutational scanning (DMS) and next-generation DNA sequencing (NGS) have great potential to accelerate discovery and optimization of biological therapeutics. Typical workflows involve generation of a mutagenized variant library, screening/selection of variants based on phenotypic fitness, and comprehensive analysis of binned variant populations by NGS. However, in such cases, the HTS data are subject to interval censoring, where each fitness value is calculated based on the assignment of variants to bins. Such censoring leads to increased uncertainty, which can impact data accuracy and, consequently, the performance of machine learning (ML) algorithms tasked with predicting sequence-fitness pairings. Here, we investigated the impact of interval censoring on data quality and ML performance in biological HTS experiments. We theoretically analyzed the impact of data censoring and propose a dimensionless number, theRatio of Discretization(RD), to assist in optimizing HTS parameters such as the bin width and the sampling size. This approach can be used to minimize errors in fitness prediction by ML and to improve the reliability of these methods. These findings are not limited to biological HTS techniques and can be applied to other systems where interval censoring is an advantageous measurement strategy. | None | 0\n",
      "79 | Profiling drug sensitivity of leukemic stem cells via bulk-to-single-cell deconvolution | https://www.biorxiv.org/content/10.1101/2023.05.10.540140v2 | 10.1101/2023.05.10.540140 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.05.10.540140v2.full.pdf | Ex-vivo drug sensitivity screening allows the prediction of cancer treatment effectiveness in a personalized fashion. However, it only provides a readout on mixtures of cells, potentially occulting important information on clinically relevant cell subtypes. To address this shortcoming, we developed a machine-learning framework to decompose drug sensitivity recorded at the bulk level into cell subtype-specific drug sensitivity. We first determined that our method could decipher the cellular composition of bulk samples with top-ranking accuracy across five cancer types compared to state-of-the-art bulk deconvolution methods. We emphasize its effectiveness in the realm of Acute Myeloid Leukemia, where it appears to offer the most precise estimation of leukemic stem cell fractions across three test datasets and three patient cohorts. We then optimized an algorithm capable of estimating cell subtype- and single-cell-specific drug sensitivity, which we evaluated by performing in-vitro drug studies and in-depth simulations. We then applied our deconvolution strategy to the beatAML cohort dataset, currently the most extensive database of ex-vivo drug screening data. We developed a drug sensitivity profile tailored to specific cell subtypes, focusing on four therapeutic compounds predicted to target leukemic stem cells: the previously known midostaurin and A-674563, as well as SNS-032 and foretinib, which have not been previously linked to leukemic stem cells. Our work provides an attractive new computational tool for drug development and precision medicine. | None | 0\n",
      "80 | R2DT: a comprehensive platform for visualising RNA secondary structure | https://www.biorxiv.org/content/10.1101/2024.09.29.611006v1 | 10.1101/2024.09.29.611006 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.29.611006v1.full.pdf | RNA secondary (2D) structure visualisation is an essential tool for understanding RNA function. R2DT is a software package designed to visualise RNA 2D structures in consistent, recognisable, and reproducible layouts. The latest release, R2DT 2.0, introduces multiple significant features, including the ability to display position-specific information, such as single nucleotide polymorphisms (SNPs) or SHAPE reactivities. It also offers a new template-free mode allowing visualisation of RNAs without pre-existing templates, alongside a constrained folding mode and support for animated visualisations. Users can interactively modify R2DT diagrams, either manually or using natural language prompts, to generate new templates or create publication-quality images. Additionally, R2DT features faster performance, an expanded template library, and a growing collection of compatible tools and utilities. Already integrated into multiple biological databases, R2DT has evolved into a comprehensive platform for RNA 2D visualisation, accessible at https://r2dt.bio. | https://github.com/DasLab/draw_rna, https://github.com/LDWLab/RNA2D-data-schema/, https://github.com/PDBeurope/pdb-rna-viewer, https://github.com/rnacentral/r2dt-web, https://github.com/LDWLab/RNA2D-data-schema, https://github.com/r2dt-bio/r2dt, https://github.com/r2dt-bio/r2dt-client | 1\n",
      "81 | Chromosome-level genome assembly and annotation of the crested gecko, Correlophus ciliatus, a lizard incapable of tail regeneration | https://www.biorxiv.org/content/10.1101/2024.09.28.615630v1 | 10.1101/2024.09.28.615630 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.28.615630v1.full.pdf | The vast majority of gecko species are capable of tail regeneration, but singular geckos of Correlophus, Uroplatus, and Nephrurus genera are unable to regrow lost tails. Of these non-regenerative geckos, the crested gecko (Correlophus ciliatus) is distinguished by ready availability, ease of care, high productivity, and hybridization potential. These features make C. ciliatus particularly suited as a model for studying the genetic, molecular, and cellular mechanisms underlying loss of tail regeneration capabilities. We report a contiguous genome of C. ciliatus with a total size of 1.65 Gb, a total of 152 scaffolds, L50 of 6, and N50 of 109 Mb. Repetitive content consists of 40.41% of the genome, and a total of 30,780 genes were annotated. Assembly of the crested gecko genome provides a valuable resource for future comparative genomic studies between non-regenerative and regenerative geckos and other squamate reptiles. | None | 0\n",
      "82 | Distinct gene regulatory networks govern hematopoietic and leukemia stem cells | https://www.biorxiv.org/content/10.1101/2024.09.28.614980v1 | 10.1101/2024.09.28.614980 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.28.614980v1.full.pdf | The underlying gene regulatory networks (GRN) that govern leukemia stem cells (LSC) in acute myeloid leukemia (AML) and hematopoietic stem cells (HSC) are not well understood. Here, we identified GRNs by integrating gene expression (GE) and chromatin accessibility data derived from functionally defined cell populations enriched for HSC and LSC. We analyzed n=32 LSC+ and n=32 LSC- cell fractions from n=22 AML patients, along with n=7 stem and n=10 progenitor enriched cell populations sorted from human umbilical cord blood (hUCB), producing a database of n≈17,000 transcription factor (TF) regulatory interactions for hUCB-HSPC and AML. We developed an iterative algorithm that associates the degree of chromatin openness with TF binding preferences, and the GE of candidate TF and target genes within 100kb upstream of transcription start sites. A putative regulatory structure was found to be enriched in HSC-enriched cell populations, comprising TF-target gene interactions between ETS1, EGR1, RUNX2, and ZNF683 oriented in a self-reinforcing configuration. A regulatory loop comprising FOXK1 and MEIS1, rather than the 4-factor HSC subnetwork, was detected in the LSC-specific GRN. The core HSC and LSC TF networks were extended using protein-protein interaction (PPI) data to determine connectivity with interacting genes whose expression strongly associated with LSC/HSC frequency estimates, producing a database of n=103,516 PPI target pathways. The effect of perturbing genes along the identified pathways on functional HSC and LSC frequency was predicted based on statistical regression analyses. To validate GRN predictions, we used pharmacologic and CRISPR targeting, in addition to re-examining published functional data associated with several network nodes that were predicted to impact stemness. Notably, we found that inhibition of CDK6 in AML samples markedly reduced LSC numbers as assessed in de novo serial xenotransplantation studies (fold change ≈ 10), as predicted by the LSC GRN model. Additionally, in-house CRISPR-based knockdown of ETS1 resulted in a significant decrease in HSC quiescence-associated microRNA-126 expression, and increased HSC frequency. Taken together, our models provide a comprehensive view of the underlying regulatory structures governing functional human HSC and LSC. This approach has translational potential as it can be used as a high-throughput in-silico screening tool for the systematic identification of gene targets for LSC elimination and HSC expansion. | None | 0\n",
      "83 | FragLlama: Next-fragment prediction for molecular design | https://www.biorxiv.org/content/10.1101/2024.09.28.615626v1 | 10.1101/2024.09.28.615626 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.28.615626v1.full.pdf | The emergence of ChatGPT has drawn significant attention to Large Language Models (LLMs) due to their impressive performance. While LLMs primarily focus on next token/word prediction, we apply this principle to molecular design by reframing the task as predicting the next token/fragment. We present FragLlama, a large language model trained for molecular design, featuring custom tokens that represent molecular fragments and functional groups. The model is for generating molecules given one or two fragments, for application scenarios like general hit-to-lead and lead optimization stage drug design, PROTAC linker design; mapping to commonly used drug design strategies like fragment growing and scaffold hopping. In the pre-training stage, we adapted the Llama 3 architecture to create FragLlama, training it to learn conditional probabilities of these fragment-level tokens. The subsequent alignment stage employed fine-tuning to guide the model towards generating molecules with desired properties. The effectiveness of FragLlama is demonstrated through its applications in designing molecular glue libraries, PROTAC linkers and EGFR binders. FragLlama demonstrates proficiency in reproducing expert-level designs while also exploring novel and promising chemical spaces, highlighting its potential to augment the capabilities of medicinal chemists in drug design. | None | 0\n",
      "84 | On Long-Term Species Coexistence in Five-Species Evolutionary Spatial Cyclic Games with Ablated and Non-Ablated Dominance Networks | https://www.biorxiv.org/content/10.1101/2024.09.27.615336v1 | 10.1101/2024.09.27.615336 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.27.615336v1.full.pdf | I present a replication and, to some extent, a refutation of key results published by Zhong, Zhang, Li, Dai, & Yang in their 2022 paper \"Species coexistence in spatial cyclic game of five species\" (Chaos, Solitons and Fractals, 156: 111806), where ecosystem species coexistence was explored via simulation studies of the evolutionary spatial cyclic game (ESCG) Rock-Paper-Scissors-Lizard-Spock (RPSLS) with certain predator-prey relationships removed from the game9s \"interaction structure\", i.e. with specific arcs ablated in the ESCG9s dominance network, and with the ESCG run for 100,000 Monte Carlo Steps (MCS) to identify its asymptotic behaviors. I replicate the results presented by Zhong et al. for interaction structures with one, two, three, and four arcs ablated from the dominance network. I then empirically demonstrate that the dynamics of the RPSLS ESCG have sufficiently long time constants that the true asymptotic outcomes can often only be identified after running the ablated ESCG for 10,000,000 MCS or longer, and that the true long-term outcomes can be markedly less diverse than those reported by Zhong et al. as asymptotic.  Finally I demonstrate that, when run for sufficiently many MCS, the original unablated RPSLS system exhibits essentially the same asymptotic outcomes as the ablated RPSLS systems, and in this sense the only causal effect of the ablations is to alter the time required for the system to converge to the long-term asymptotic states that the unablated system eventually settles to anyhow. | None | 0\n",
      "85 | Allo-Allo: Data-efficient prediction of allosteric sites | https://www.biorxiv.org/content/10.1101/2024.09.28.615583v1 | 10.1101/2024.09.28.615583 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.28.615583v1.full.pdf | Allostery, a fundamental structural mechanism where ligand binding at a protein site affects protein function at another site, plays a crucial role in key drug-target proteins like GPCRs. Unfortunately, existing methods for predicting allosteric sites have limited performance-- they are particularly constrained by scarce ground-truth experimental data. We introduce Allo-Allo, a data-efficient, sequence-based method that predicts allosteric sites by leveraging protein language models (PLMs). Honing in on ESM-2 attention heads that capture allosteric residue associations, Allo-Allo achieves a 67% higher AUPRC than state-of-the-art methods. Our innovative, data-efficient pipeline not only outperforms alternate, commonly-used PLM-based prediction architectures but also generalizes well. Notably, mutations in Allo-Allo-predicted sites show significant association with elevated disease risk scores from AlphaMissense, highlighting its translational potential. Beyond Allo-Allo9s biological and translational applicability, its architecture presents a powerful framework for other data-scarce problems in protein analysis. | None | 0\n",
      "86 | LightMed: A Light-weight and Robust FFT-Based Model for Adversarially Resilient Medical Image Segmentation | https://www.biorxiv.org/content/10.1101/2024.09.28.615584v1 | 10.1101/2024.09.28.615584 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.28.615584v1.full.pdf | Accurate and reliable medical image segmentation is essential for computer-aided diagnosis and formulating appropriate treatment plans. However, real-world challenges such as suboptimal image quality and computational resource constraints hinder the effective deployment of deep learning-based segmentation models. To address these issues, we propose LightMed, a novel efficient neural architecture based on Fast Fourier Transform (FFT). Different from prior works, our model directly learns on the frequency domain, harnessing its resilience to noise and uneven brightness, which common artifacts found in medical images. By focusing on low-frequency image components, we significantly reduce computational complexity while preserving essential image features. Our deep learning architecture extracts discriminative features directly from the Fourier domain, leading to improved segmentation accuracy and robustness compared to traditional spatial domain methods. Additionally, we propose a new benchmark incorporating various levels of Gaussian noise to assess susceptibility to noise attacks. The experimental results demonstrate that LightMed not only effectively eliminates noise and consistently achieves accurate image segmentation but also shows robust resistance to imperceptible adversarial attacks compared to other baseline models. Our new benchmark datasets and source code are publicly available at https://github.com/HySonLab/LightMed | None | 0\n",
      "87 | Improved robustness to gene tree incompleteness, estimation errors, and systematic homology errors with weighted TREE-QMC | https://www.biorxiv.org/content/10.1101/2024.09.27.615467v1 | 10.1101/2024.09.27.615467 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.27.615467v1.full.pdf | Summary methods are widely used to reconstruct species trees from gene trees while accounting for incomplete lineage sorting; however, it is increasingly recognized that their accuracy can be negatively impacted by incomplete and/or error-ridden  gene trees. To address the latter, Zhang and Mirarab (2022) leverage gene tree branch lengths and support values to weight quartets within the popular summary method ASTRAL. Although these quartet weighting schemes improved the robustness of ASTRAL to gene tree estimation error, implementing the weighting schemes presented computational challenges, resulting in the authors abandoning ASTRAL9s original search algorithm (i.e., computing an exact solution within a constrained search space) in favor of search heuristics (i.e., hill climbing with nearest neighbor interchange moves from a starting tree constructed via randomized taxon addition). Here, we show that these quartet weighting schemes can be leveraged within the Quartet Max Cut framework of Snir and Rao (2010), with only a small increase in time complexity compared to the unweighted algorithm, which behaves more like a constant factor in our simulation study. Moreover, our new algorithm, implemented within the TREE-QMC software, was highly competitive with weighted ASTRAL, even outperforming it in terms of species tree accuracy on some challenging model conditions, such as large numbers of taxa. In comparing unweighted and weighted summary methods on two avian data sets, we found that weighting quartets by gene tree branch lengths improves their robustness to \\textit{systematic} homology errors and is as effective as removing the impacted taxa from individual gene trees or removing the impacted gene trees entirely. Lastly, our study revealed that TREE-QMC is highly robust to high rates of missing data and is promising as a supertree method. TREE-QMC is written in C++ and is publicly available on Github: https://github.com/molloy-lab/TREE-QMC | https://github.com/molloy-lab/TREE-QMC, https://github.com/molloy-lab/tree-qmc-study/tree/main/han2024wtreeqmc | 1\n",
      "88 | MPCD Index for Hepatocellular Carcinoma Patients Based on Mitochondrial Function and Cell Death Patterns | https://www.biorxiv.org/content/10.1101/2024.09.27.615369v1 | 10.1101/2024.09.27.615369 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.27.615369v1.full.pdf | Hepatocellular carcinoma (HCC) is a highly heterogeneous cancer with a poor prognosis. During the development of cancer cells, mitochondria influence various cell death patterns by regulating metabolic pathways such as oxidative phosphorylation. However, the relationship between mitochondrial function and cell death patterns in HCC remains unclear. In this study, we used a comprehensive machine learning framework to construct a mitochondrial functional activity-associated programmed cell death index (MPCDI) based on scRNA-seq and RNA-seq data from TCGA, GEO, and ICGC datasets. The index signature was used to classify HCC patients, and studied the multi-omics features, immune microenvironment, and drug sensitivity of the subtypes. Finally, we constructed the MPCDI signature consisting of four genes (S100A9, FYN, LGALS3, and HMOX1), which was one of the independent risk factors for the prognosis of HCC patients. The HCC patients were divided into high- and low-MPCDI groups, and the immune status was different between the two groups. Patients with high MPCDI had higher TIDE scores and poorer responses to immunotherapy, suggesting that high-MPCDI patients might not be suitable for immunotherapy. By analyzing the drug sensitivity data of CTRP, GDSC, and PRISM databases, it was found that staurosporine has potential therapeutic significance for patients with high MPCDI. In summary, based on the characteristics of mitochondria function and PCD patterns, we used single-cell and transcriptome data to identify four genes and construct the MPCDI signature, which provided new perspectives and directions for the clinical diagnosis and personalized treatment of HCC patients. | https://github.com/wanglongxing2020/ML_code | 1\n",
      "89 | INSTINCT: Multi-sample integration of spatial chromatin accessibility sequencing data via stochastic domain translation | https://www.biorxiv.org/content/10.1101/2024.05.26.595944v2 | 10.1101/2024.05.26.595944 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.05.26.595944v2.full.pdf | Recent advances in spatial epigenomic techniques have given rise to spatial assay for transposase-accessible chromatin using sequencing (spATAC-seq) data, enabling the characterization of epigenomic heterogeneity and spatial information simultaneously. Integrative analysis of multiple spATAC-seq samples, for which no method has been developed, allows for effective identification and elimination of unwanted non-biological factors within the data, enabling comprehensive exploration of tissue structures and providing a holistic epigenomic landscape, thereby facilitating the discovery of biological implications and the study of regulatory processes. In this article, we present INSTINCT, a method for multi-sample INtegration of Spatial chromaTIN accessibility sequencing data via stochastiC domain Translation. INSTINCT can efficiently handle the high dimensionality of spATAC-seq data and eliminate the complex noise and batch effects of samples from different conditions through a stochastic domain translation procedure. We demonstrate the superiority and robustness of INSTINCT in integrating spATAC-seq data across multiple simulated scenarios and real datasets. Additionally, we highlight the advantages of INSTINCT in spatial domain identification, visualization, spot-type annotation, and various downstream analyses, including motif enrichment analysis, expression enrichment analysis, and partitioned heritability analysis. | https://github.com/Chen-Li-17/simCAS/tree/main/data, https://github.com/yyLIU12138/INSTINCT | 1\n",
      "90 | Identifying similar populations across independent single cell studies without data integration | https://www.biorxiv.org/content/10.1101/2024.09.27.615367v1 | 10.1101/2024.09.27.615367 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.27.615367v1.full.pdf | Supervised and unsupervised methods have emerged to address the complexity of single cell data analysis in the context of large pools of independent studies. Here, we present ClusterFoldSimilarity (CFS), a novel statistical method design to quantify the similarity between cell groups acroos any number of independent datasets, without the need for data correction or integration. By bypassing these processes, CFS avoids the introduction of artifacts and loss of information, offering a simple, efficient, and scalable solution. This method match groups of cells that exhibit conserved phenotypes across datasets, including different tissues and species, and in a multimodal scenario, including single-cell RNA-Seq, ATAC-Seq, single-cell proteomics, or, more broadly, data exhibiting differential abundance effects among groups of cells. Additionally, CFS performs feature selection, obtaining cross-dataset markers of the similar phenotypes observed, providing an inherent interpretability of relationships between cell populations. To showcase the effectiveness of our methodology we generated single-nuclei RNA-Seq data from the motor cortex and spinal cord of adult mice. By using CFS, we identified three distinct sub-populations of astrocytes conserved on both tissues. CFS includes various visualization methods for the interpretation of the similarity scores and similar cell populations. | https://github.com/OscarGVelasco/ClusterFoldSimilarity | 1\n",
      "91 | Plasmid conjugation drives within-patient plasmid diversity | https://www.biorxiv.org/content/10.1101/2024.09.27.615342v1 | 10.1101/2024.09.27.615342 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.27.615342v1.full.pdf | Plasmids are well known vehicles of antimicrobial resistance (AMR) genes dissemination. Through conjugation, plasmid-encoded AMR genes are spread among neighbouring bacteria, irrespective of their strain or even their species. This process is very concerning from a public health perspective, as plasmid-borne AMR gene outbreaks are often not confined to single species or strains and are therefore more difficult to be fully uncovered. At the moment, the impact of plasmid conjugation on within-patient plasmid diversity is not well understood. In this work we will tackle the role of conjugation on within-patient plasmid diversity using a dataset of carbapenemase-producing Enterobacterales (CPEs). The dataset of 256 sequences from 115 patients was sampled across England over 30 months. Each patient has more than one sequence, with at least one sequence carrying an OXA-48 gene, a well-known plasmid-borne carbapenemase-encoding gene. If more than one sequence carries the OXA-48 gene, they are carried on different bacterial hosts. Using a hybrid de novo-on-reference assembly pipeline, we were able to reconstruct the full OXA-48 plasmid from short read sequencing data for 232 of the 256 sequences. Of the 115 patients, 83 (72%) of patients had an identical OXA-48 plasmid in two or more sequences. Only 2 patients carried very different (>200 SNPs) alleles of the OXA-48 plasmid, probably from separate acquisitions. Our study shows that when more than one bacterial host carrying an OXA-48 plasmid is found in a patient, it is most likely that the same plasmid has been shared via conjugation. The event of separate acquisition of different plasmids in different bacterial hosts is highly unlikely in our dataset. | None | 0\n",
      "92 | gLinDA: Global Differential Abundance Analysis of Microbiomes | https://www.biorxiv.org/content/10.1101/2024.09.29.615668v1 | 10.1101/2024.09.29.615668 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.29.615668v1.full.pdf | Microbiome composition plays a significant role in various diseases, including cancer, obesity, inflammatory bowel disease, and mental health disorders. Understanding the differences in microbial abundance between patients is key to uncovering the microbiome9s impact on these conditions. Differential abundance analysis (DAA) can detect significant changes in microbiomes at the taxa level base. However, since individuals have unique microbial fingerprints that can potentially identify them, microbiome data must be treated as sensitive. We introduce gLinDA, a global differential abundance analysis software offering a swarm learning approach to analyzing distributed datasets while maintaining result accuracy without sharing patients9 raw data. | None | 0\n",
      "93 | stFormer: a foundation model for spatial transcriptomics | https://www.biorxiv.org/content/10.1101/2024.09.27.615337v1 | 10.1101/2024.09.27.615337 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.27.615337v1.full.pdf | Here we present stFormer, a foundation model which incorporates ligand genes within spatial niches into Transformer encoders of single-cell transcriptomics. We demonstrate that despite a moderate pretraining data size, the spatially informed gene representations generated by stFormer more consistently cluster the cells, more accurately encode hierarchy and membership within receptor-dependent gene networks, remarkably boost identification of ligand-receptor interaction pairs, and could simulate perturbation effects of ligand-receptor interactions on downstream targets. | None | 0\n",
      "94 | In Silico identification and modelling of FDA-approved drugs targeting T-type calcium channels | https://www.biorxiv.org/content/10.1101/2024.09.27.615366v1 | 10.1101/2024.09.27.615366 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.27.615366v1.full.pdf | Studies have shown that inhibition of the Cav3.1 T-type calcium channel can prevent or suppress neurological diseases, such as epileptic seizures and diabetic neuropathy. In this study we aimed to use in silico simulations to identify a U.S. Food and Drug Administration (FDA)-approved drug that can bind to the Cav3.1 T-type calcium channel. We used the automated docking suite GOLD v5.5 with the genetic algorithm to simulate molecular docking and predict the protein-ligand binding modes, and the ChemPLP empirical scoring function to estimate the binding affinities of 2,115 FDA-approved drugs to the human Cav3.1 channel. Drugs with high binding affinity and appropriate pharmacodynamic and pharmacokinetic properties were selected for molecular mechanics Poisson–Boltzmann surface area (MMPBSA) and molecular mechanics generalised Born surface area (MMGBSA) binding free energy calculations, GROMACS molecular dynamics (MD) simulations and Monte Carlo Cell (MCell) simulations. The docking results indicated that the FDA-approved drug montelukast has a high binding affinity to Cav3.1, and data from the literature suggested that montelukast has the appropriate drug-like properties to cross the human blood-brain barrier and reach synapses in the central nervous system. MMPBSA, MMGBSA and MD simulations showed the high stability of the montelukast-Cav3.1 complex. MCell simulations indicated that the blockage of Cav3.1 by montelukast reduced the number of synaptic vesicles being released from the pre-synaptic region to the synaptic cleft, which may reduce the probability and amplitude of postsynaptic potentials. | None | 0\n",
      "95 | Benchmarking copy number aberrations inference tools using single-cell multi-omics datasets | https://www.biorxiv.org/content/10.1101/2024.09.26.615284v1 | 10.1101/2024.09.26.615284 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.26.615284v1.full.pdf | Copy number aberrations (CNAs) are an important type of genomic variation which play a crucial role in the initiation and progression of cancer. With the explosion of single-cell RNA sequencing (scRNA-seq), several computational methods have been developed to infer CNAs from scRNA-seq studies. However, to date, no independent studies have comprehensively benchmarked their performance. Herein, we evaluated five state-of-the-art methods based on their performance in tumor vs normal cell classification, CNAs profile accuracy, tumor subclone inference and aneuploidy identification in non-malignant cells. Our results showed that Numbat outperformed others across most evaluation criteria, while CopyKAT excelled in scenarios when expression matrix alone was used as input. Additionally, we investigated how referencing settings, inclusion of tumor microenvironment cells, tumor type, and tumor purity impact the performance of these tools. This study provides a valuable guideline for researchers in selecting the appropriate methods for their datasets. | https://github.com/AntonioDeFalco/SCEVAN, https://github.com/bioliyezhang/Benchmark_scRNASeq_by_scMultiomics, https://github.com/akdess/CaSpER, https://github.com/broadinstitute/infercnv, https://github.com/navinlabcode/copykat | 1\n",
      "96 | EmbedSimScore: Advancing Protein Similarity Analysis with Structural and Contextual Embeddings | https://www.biorxiv.org/content/10.1101/2024.09.26.615264v1 | 10.1101/2024.09.26.615264 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.26.615264v1.full.pdf | Accurately computing protein similarity is challenging due to the intricate interplay between local substructures and the global structure within protein molecules. Traditional metrics like TM-score often focus on aligning the global structures of the proteins in a rather geometry-based algorithmic way, potentially overlooking critical local global relations and contextual comparisons. We introduce Embed-SimScore, a novel self-supervised method that generates structural and contextual embeddings by jointly considering both local substructures and global proteins9s structures. Utilizing contrastive language-structure pre training (CLSP) and structural contrastive learning, EmbedSimScore captures comprehensive features across different scales of protein structure. These embeddings provide a more precise and holistic means of computing protein similarities, resulting in the identification of intrinsic relations among proteins that traditional approaches overlook. | None | 0\n",
      "97 | Research on the Mechanism of Lphn1 Knockout in Inhibiting Colorectal Cancer | https://www.biorxiv.org/content/10.1101/2024.08.07.606975v5 | 10.1101/2024.08.07.606975 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.08.07.606975v5.full.pdf | Decades ago, colorectal cancer was rarely diagnosed. Today, it is the fourth leading cause of cancer-related deaths worldwide, with nearly 90,000 fatalities each year. By analyzing single-cell data from tumor-bearing colorectal cancer model mice with Lphn1 knockout and wild-type Lphn1, we identified five key target genes for anticancer therapy: Ulbp1, Klrk1, Ccl6, Tlr4, Cd48, Prdm5, Vstm2a, Ret, Oas2, Hdac11 and Ptchd4, along with their corresponding cell types. Additionally, we discovered tumor-inhibiting cell subpopulations, including Cd244a_T_cells_subcluster_1, Cd48_Cd244a_NK_cells_subcluster_2, and C3_Macrophages_subcluster_1, which are potential candidates for therapeutic intervention. We propose that cancer-associated fibroblasts (CAFs) serve as the primary antigen presenters for MHC class I, providing antigens to macrophages, NK cells, and T cells to combat colorectal cancer. From a cellular perspective, the knockout of Lphn1 activates the anti-colorectal cancer functions of subpopulations of macrophages, NK cells, and T cells. Macrophages enhance antitumor immune activity by engaging the Ulbp1-Klrk1 receptor pair to activate NK cells. Additionally, macrophages activate downstream functions of T cells against colorectal cancer through CD48 signaling. After the knockout of Lphn1, macrophages are recruited by autocrine Ccl6 and Ccl6 secReted by CAFs. They exhibit high expression of Tlr4 and have the potential to transition into M1-type macrophages due to changes in their cellular state. After the knockout of Lphn1, the CAFs were reduced by half. CAFs, which are part of the cell network communication associated with tumor cells, typically play an immunosuppressive role. A reduction by half indicates that the immunosuppression in the Lphn1 knockout group has significantly decreased. This suggests that the efficacy of various cancer immunotherapy drugs can be significantly enhanced. In specific cell types, the four colorectal cancer-resistant transcription factors Irf7, Nr2f1 (with uncertain function), Bclaf1, and Irf2 co-localize with Tlr4, Cd48, Prdm5, and Oas2, respectively, and have been analyzed by pyscenic to interact and functionally contribute to the resistance against colorectal cancer. The four differential metabolic pathways between the Lphn1 group and the luc group are Arginine and proline metabolism, Histidine metabolism, Phenylalanine metabolism, and Riboflavin metabolism, with Arginine and proline metabolism being more active in the Lphn1 group and Histidine metabolism being more active in the luc group. These findings could open new avenues for the treatment of colorectal cancer and contribute to the development of personalized medicine. | None | 0\n",
      "98 | Deep learning facilitates precise identification of disease-resistance genes in plants | https://www.biorxiv.org/content/10.1101/2024.09.26.615248v1 | 10.1101/2024.09.26.615248 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.26.615248v1.full.pdf | The identification of plant disease-resistance genes is essential for understanding the plant immune system and accelerating crop breeding with disease resistance. Therefore, there is a pressing need for a method capable of accurately and comprehensively annotating resistance genes at a genome-wide scale. In this study, we propose Evolutionary Scale Modeling for LRR (ESM-LRR), a novel approach based on the deep protein language model to accurately identify LRR domains which are substantially variable structures in disease-resistance proteins. ESM-LRR achieved its highest F1 score of 0.80 on a test set using the 90% identity as the matching threshold. Building upon ESM-LRR, we developed the Plant Disease-Resistance Gene Predictor (R-Predictor), a framework designed to simultaneously annotate 15 diverse domain topologies, extensively covering recently characterised resistance genes across the entire genome. R-Predictor integrates four modules, each employing superior methods that outperform existing methods (F1 score of 0.89 for RLKs and 0.88 for NLRs), demonstrating its high accuracy and practicality in annotating disease-resistance genes (R genes). R-Predictor were also applied to identify R genes in rice, tomato, and grapevine.  Furthermore, AlphaFold3 was employed to screen interactions between 1,116 protein pairs involving three NLRs identified exclusively by R-Predictor and 372 literature-reported plant-pathogen effectors, resulting in the identification of 15 putative NLR-effector complexes. Overall, this study presents a novel approach for advancing our understanding of plant immune mechanisms. | https://github.com/labstructbioinf/DeepCoil, https://github.com/phytolrr/predict-phytolrr, https://github.com/SwiftSeal/resistify, https://github.com/fteufel/signalp-6.0, https://github.com/slt666666/NLRtracker, https://github.com/zhouyflab/R-Predictor, https://github.com/ElofssonLab/scampi2, https://github.com/eliza-m/NLRexpress, https://github.com/zhenyaliu77/DeepLRR | 1\n",
      "99 | Mastering Artifact Correction in Neuroimaging Analysis: A Retrospective Approach | https://www.biorxiv.org/content/10.1101/2024.08.02.606374v2 | 10.1101/2024.08.02.606374 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.08.02.606374v2.full.pdf | The correction of artifacts in Magnetic Resonance Imaging (MRI) is increasingly relevant as voluntary and involuntary artifacts can hinder data acquisition. Reverting from corrupted to artifact-free images is a complex task. Deep Learning (DL) models have been employed to preserve data characteristics and to identify and correct those artifacts. We propose MOANA, a novel DL-based solution to correct artifacts in multi-contrast brain MRI scans. MOANA offers two models: the simulation and the correction models. The simulation model introduces perturbations similar to those occurring in an exam while preserving the original image as ground truth; this is required as publicly available datasets rarely have motion-corrupted images. It allows the addition of three types of artifacts with different degrees of severity. The DL-based correction model adds a fourth contrast to state-of-the-art solutions while improving the overall performance of the models. MOANA achieved the highest results in the FLAIR contrast, with a Structural Similarity Index Measure (SSIM) of 0.9803 and a Normalized Mutual Information (NMI) of 0.8030. With this, the MOANA model can correct large volumes of images in less time and adapt to different levels of artifact severity, allowing for better diagnosis | None | 0\n",
      "100 | Profiling ranked list enrichment scoring in sparse data elucidates algorithmic tradeoffs | https://www.biorxiv.org/content/10.1101/2024.06.03.597180v2 | 10.1101/2024.06.03.597180 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.06.03.597180v2.full.pdf | Gene Set Enrichment Analysis (GSEA) is a method for quantifying pathway and process activation in groups of samples, and its single sample version (ssGSEA) scores activation using mRNA abundance in a single sample. GSEA and ssGSEA were developed for bulk samples rather than individual cell technologies such as microarrays and bulk RNA-sequencing (RNA-seq) data. The growing use of single cell RNA-sequencing (scRNA-seq) raises the possibility of using ssGSEA to quantify pathway and process activation in individual cells. However, scRNA-seq data is much sparser than RNA-seq data. Here we show that ssGSEA as designed for bulk data is subject to some amount of score uncertainty and other technical issues when applied to individual cells from scRNA-seq data. We also show that a ssGSEA can be applied robustly to pseudobulk aggregate groups of a few hundred to a few thousand cells provided appropriate normalization is used. Finally, in comparing this approach to other ranked list enrichment methods, we find that the UCell method is most robust to sparsity. We have made the aggregate cell version of ssGSEA available as a Python package and GenePattern module and will also modularize UCell for use on GenePattern as well. | None | 0\n",
      "101 | SIMVI reveals intrinsic and spatial-induced states in spatial omics data | https://www.biorxiv.org/content/10.1101/2023.08.28.554970v2 | 10.1101/2023.08.28.554970 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.08.28.554970v2.full.pdf | <p>Spatial omics technologies enable the analysis of gene expression and interaction dynamics in relation to tissue structure and function. However, existing computational methods do not properly distinguish cellular intrinsic variability and intercellular interactions, thus failing to capture spatial regulations for further biological discoveries. Here, we present Spatial Interaction Modeling using Variational Inference (SIMVI), an annotation-free framework that disentangles cell intrinsic and spatial-induced latent variables for modeling gene expression in spatial omics data. We establish rigorous theoretical support for SIMVI in disentangling intrinsic and spatial-induced variations. By this disentanglement, SIMVI enables estimation of spatial effects (SE) at a single-cell resolution, and opens up various opportunities for downstream analyses. To demonstrate the power of SIMVI, we applied SIMVI to spatial omics data from diverse platforms and tissues (MERFISH human cortex, Slide-seqv2 mouse hippocampus, Slide-tags human tonsil, spatial multiome human melanoma, cohort-level CosMx melanoma). In all tested datasets, SIMVI reveal refined disentangled variations and accurate spatial effects compared with various alternative methods. Moreover, our applications of SIMVI in these datasets uncovered complex spatial regulations and dynamics of biological significance. In the human tonsil data, SIMVI characterizes the cyclical spatial dynamics of germinal center B cells during maturation. In the multiome melanoma data, SIMVI incorporates ATAC-seq measurements to discover potential tumor epigenetic reprogramming states. Application of SIMVI on our newly-collected cohort-level CosMx melanoma dataset uncovers space-outcome-dependent macrophage states and cellular communication machinery shifted across patient phenotypes.</p> | None | 0\n",
      "102 | Functional protein mining with conformal guarantees | https://www.biorxiv.org/content/10.1101/2024.06.27.601042v5 | 10.1101/2024.06.27.601042 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.06.27.601042v5.full.pdf | <p>Molecular structure prediction and homology detection provide a promising path to discovering new protein function and evolutionary relationships. However, current approaches lack statistical reliability assurances, limiting their practical utility for selecting proteins for further experimental and in-silico characterization. To address this challenge, we introduce a novel approach to protein search leveraging principles from conformal prediction, offering a framework that ensures statistical guarantees with user-specified risk and provides calibrated probabilities (rather than raw ML scores) for any protein search model. Our method (1) lets users select many biologically-relevant loss metrics (i.e. false discovery rate) and assigns reliable functional probabilities for annotating genes of unknown function; (2) achieves state-of-the-art performance in enzyme classification without training new models; and (3) robustly and rapidly pre-filters proteins for computationally intensive structural alignment algorithms. Our framework enhances the reliability of protein homology detection and enables the discovery of new proteins with likely desirable functional properties.</p> | None | 0\n",
      "103 | MetaLab Platform Enables Comprehensive DDA and DIA Metaproteomics Analysis | https://www.biorxiv.org/content/10.1101/2024.09.27.615406v1 | 10.1101/2024.09.27.615406 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.27.615406v1.full.pdf | <p>Metaproteomics studies the collective protein composition of complex microbial communities, providing insights into microbial roles in various environments. Despite its importance, metaproteomic data analysis is challenging due to the data9s large and heterogeneous nature. While Data-Independent Acquisition (DIA) mode enhances proteomics sensitivity, it traditionally requires Data-Dependent Acquisition (DDA) results to build the library for peptide identification. This paper introduces an updated version of MetaLab, a software solution that streamlines metaproteomic analysis by supporting both DDA and DIA modes across various mass spectrometry (MS) platforms, including Orbitrap and timsTOF. MetaLab9s key feature is its ability to perform DIA analysis without DDA results, allowing more experimental flexibility. It incorporates a deep learning strategy to train a neural network model, enhancing the accuracy and coverage of DIA results. Evaluations using diverse datasets demonstrate MetaLab9s robust performance in accuracy and sensitivity. Benchmarks from large-scale human gut microbiome studies show that MetaLab increases peptide identification by 2.7 times compared to conventional methods. MetaLab is a versatile tool that facilitates comprehensive and flexible metaproteomic data analysis, aiding researchers in exploring microbial communities9 functionality and dynamics.</p> | None | 0\n",
      "104 | Molecular Evolution and Functional Divergence of Heterodisulfide Reductase A (HdrA) | https://www.biorxiv.org/content/10.1101/2024.09.26.615249v1 | 10.1101/2024.09.26.615249 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.26.615249v1.full.pdf | <p>Heterodisulfide reductase is a key enzyme in methanogenesis pathway in archaea, which helps to recycle the coenzymes after methane is formed from C1 carbon sources. Homologous enzymes are observed in nonmethanogenic archaea and prokaryotes but their functional and evolutionary details are not fully explored. This report aims at understanding the distribution of one such homologous enzyme, heterodisulfide reductaseA; the domain architecture, overall structural contribution in biological function and evolutionary details in organisms other than methanogen. Exhaustive sequence analysis from both archaea and prokaryotic lineages reveals the presence of highly conserved homologues which probably have an altered oxidoreductase function. Phylogenetic analysis and coclustering showed the differential diversity of the homologues enzyme in metabolically diverse groups. The distribution of Heterodisulfide homologues in both archaea and prokaryotes shows differential expansion of this protein by multiple horizontal gene transfer events. Conserved domain analysis and structural modelling of different phylogenetic clades shows sequence and structural diversity at the Cterminal domain. Bimolecular network analysis highlights the functional association and interactions of the heterodisulfide homologues with diverse protein partners which participate in different metabolic function. HdrA involved in species specific metabolic function while the functional conservation is evident between close related species. Analysis of structural domain within HdrA  reveals that clade specific sequence which helps to differentiate structural domain arrangement and contributes to functional constrains. The presence of HdrA protein in methanogenic archaea and nonmethanogens suggests that the protein function involving redox reaction and electron transfer shared a common history.</p> | None | 0\n",
      "105 | Unified Sampling and Ranking for Protein Docking with DFMDock | https://www.biorxiv.org/content/10.1101/2024.09.27.615401v1 | 10.1101/2024.09.27.615401 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.27.615401v1.full.pdf | <p>Diffusion models have shown promise in addressing the protein docking problem. Traditionally, these models are used solely for sampling docked poses, with a separate confidence model for ranking. We introduce DFMDock (Denoising Force Matching Dock), a diffusion model that unifies sampling and ranking within a single framework. DFMDock features two output heads: one for predicting forces and the other for predicting energies. The forces are trained using a denoising force matching objective, while the energy gradients are trained to align with the forces. This design enables our model to sample using the predicted forces and rank poses using the predicted energies, thereby eliminating the need for an additional confidence model. Our approach outperforms the previous diffusion model for protein docking, DiffDock-PP, with a sampling success rate of 44% compared to its 8%, and a Top-1 ranking success rate of 16% compared to 0% on the Docking Benchmark 5.5 test set. In successful decoy cases, the DFMDock Energy forms a binding funnel similar to the physics-based Rosetta Energy, suggesting that DFMDock can capture the underlying energy landscape.</p> | None | 0\n",
      "106 | CAMDA 2023: finding patterns in urban microbiomes | https://www.biorxiv.org/content/10.1101/2024.09.26.614772v1 | 10.1101/2024.09.26.614772 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.26.614772v1.full.pdf | <p>The Critical Assessment of Massive Data Analysis (CAMDA) addresses the challenge of effectively utilizing Big Data in life science. Serving as both a conference and a catalyst for research groups, CAMDA annually presents challenges that foster innovative solutions. For the Forensics CAMDA 2023 challenge, we analyzed 365 metagenomic samples from 16 cities worldwide to characterize their origin. The forensic challenge was addressed from two perspectives: using the reduced abundance OTU tables and employing functional annotations. To identify the most informative Operational Taxonomic Units, we fit negative binomial models ultimately reducing variables to 294. After OTU selection, we implemented supervised models and conducted 5-fold cross-validation (CV) with a 4:1 training-to-validation ratio in each scenario. Support vector classification (SVC) achieved the highest F1 score (0.96) for the abundance tables, accurately classifying most cities, although New York City (NYC) posed a challenge. Via functional profiles with Mifaser at level 4, we achieved the best functional classification using the Neural Network (NN) model.\n",
      "\n",
      "Additionally, to gain insight into further associations between bacterial distribution with other covariates, we applied Dirichlet regression over <i>Escherichia</i>, <i>Enterobacter</i>, and <i>Klebsiella</i> bacteria abundances. We considered climatic and demographic variables of the cities, observing that population increase is indeed associated with a rise in the mean of <i>Escherichia</i> while decreasing temperature is linked to higher proportions of <i>Klebsiella</i>.\n",
      "\n",
      "For replicability of the scripts, a Docker container and a Conda environment are available at the repository: \n",
      "GitHub:github.com/ccm-bioinfo/cambda2023</p> | None | 0\n",
      "107 | ConoDL: A Deep Learning Framework for Rapid Generation and Prediction of Conotoxins | https://www.biorxiv.org/content/10.1101/2024.09.27.614001v1 | 10.1101/2024.09.27.614001 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.27.614001v1.full.pdf | <p>Conotoxins, being small disulfide-rich and bioactive peptides, manifest notable pharmacological potential and find extensive applications. However, the exploration of conotoxins9 vast molecular space using traditional methods is severely limited, necessitating the urgent need of developing novel approaches. Recently, deep learning (DL)-based methods have advanced to the molecular generation of proteins and peptides. Nevertheless, the limited data and the intricate structure of conotoxins constrain the application of deep learning models in the generation of conotoxins. We propose ConoDL, a framework for the generation and prediction of conotoxins, comprising the end-to-end conotoxin generation model (ConoGen) and the conotoxin prediction model (ConoPred). ConoGen employs transfer learning and a large language model (LLM) to tackle the challenges in conotoxin generation. Meanwhile, ConoPred filters artificial conotoxins generated by ConoGen, narrowing down the scope for subsequent research. A comprehensive evaluation of the peptide properties at both sequence and structure levels indicates that the artificial conotoxins generated by ConoDL exhibit a certain degree of similarity to natural conotoxins. Furthermore, ConoDL has generated artificial conotoxins with novel cysteine scaffolds. Therefore, ConoDL may uncover new cysteine scaffolds and conotoxin molecules, facilitating further exploration of the molecular space of conotoxins and the discovery of pharmacologically active variants. The code is available at https://github.com/xueww/ConoDL.The data and model are available at https://zenodo.org/records/10679280.</p> | None | 0\n",
      "108 | A qualitative review of Oxford Nanopore Sequencing datasets for RNA modifications | https://www.biorxiv.org/content/10.1101/2024.09.26.615132v1 | 10.1101/2024.09.26.615132 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.26.615132v1.full.pdf | <p>There are many oxford nanopore datasets available to study methylations. Methylations and modifications occur at nucleotides such as adenine (A), cytosine (C), guanine (G) and theanine (T) or uracil (U). Among all these provided datasets, some have the most common m6A methylation and others have m5C etc. using various real organism reference sequences such as human, mouse and artificial reference sequences which are prepared in the laboratory such as curlcake and IVT. These datasets are required to be organized by the methylation types to research ONT datasets. Here we provide a summary of the read qualities, base mapping success rates etc. for these methylation types and reference genomes. We have used minimap2 base mapping and longreadsum results. We find that methylated data have lower success rates than non-methylated data and mapping quality is lower for the real reference genomes such as human and mice. This could be because they contain more than 100,000 transcriptomes whereas artificial reference sequences contain only a few transcriptomes. Datasets which contain artificially created reference sequences have a higher quality than the others, thus they are recommended to be used for methylation or modification classification tasks in the future.</p> | None | 0\n",
      "109 | FishCODE: a web-based information platform for comprehensive omics data exploration in fish research | https://www.biorxiv.org/content/10.1101/2024.09.25.614839v1 | 10.1101/2024.09.25.614839 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.25.614839v1.full.pdf | <p>In terms of the utilization of omics data, the current fish database analysis functions are primarily relatively simple tools at the transcriptional level, aimed at obtaining the co-expression levels of specified genes or the data visualization of multiple genes, and do not enable users to perform comprehensive omics data analysis. Furthermore, the gene-level information currently provided by these multispecies fish genomics databases is incomplete, and there is a lack of a comprehensive portal that can offer multidimensional genetic information. To address these challenges, we collected extensive multi-omics information on 35 fishes and established the primary comprehensive multi-omics data information platform for fish, FishCODE (http://bioinfo.ihb.ac.cn/fishcode). We have collected experimental background of dataset which pertaining to the target fishes, selected a range of datasets that encompass a broad spectrum of research areas, and downloaded the corresponding raw omics data from public repositories such as the Sequence Read Archive (SRA). Through a unified pipeline analysis, FishCODE contains 11,216 samples from 540 sets of genomic, transcriptomic, and methylomic datasets. These data encompass transcript structure and expression, gene methylation levels, protein domains, protein subcellular localization, protein interactions, best matched protein (Swiss-Prot), associated SNP site information (47,111,018), orthologous genes, phylogenetic tree and GO/KEGG annotations. To facilitate comparison, we annotated the experimental background data sets of the FishCODE, FishGET, PhyloFish, FishSED and FishSCT databases using the Fish Experimental Condition Ontology. Currently, the FishCODE database omics dataset includes 146 unique experimental condition words, 654 cumulative experimental condition words, and 13 species with rich experimental background (more than 20 unique FECO words). These data are 3.5 times (42), 8.3 times (74), and 6.5 times (2) those of the second-ranked databases respectively. We generated word cloud maps for the experimental condition vocabularies of FishCODE and FishGET, illustrating the superior richness of FishCODE9s experimental background.</p> | https://github.com/hliihbcas/FishCODE | 1\n",
      "110 | Multi-View Integrative Approach For Imputing Short-Chain Fatty Acids and Identifying Key factors predicting Blood SCFA | https://www.biorxiv.org/content/10.1101/2024.09.25.614767v1 | 10.1101/2024.09.25.614767 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.25.614767v1.full.pdf | <p>Short-chain fatty acids (SCFAs) are the main metabolites produced by bacterial fermentation of dietary fiber within gastrointestinal tract. SCFAs produced by gut microbiotas (GMs) are absorbed by host, reach bloodstream, and are distributed to different organs, thus influencing host physiology. However, due to the limited budget or the poor sensitivity of instruments, most studies on GMs have incomplete blood SCFA data, limiting our understanding of the metabolic processes within the host. To address this gap, we developed an innovative multi-task multi-view integrative approach (M2AE, Multi-task Multi-View Attentive Encoders), to impute blood SCFA levels using gut metagenomic sequencing (MGS) data, while taking into account the intricate interplay among the gut microbiome, dietary features, and host characteristics, as well as the nuanced nature of SCFA dynamics within the body. Here, each view represents a distinct type of data input (i.e., gut microbiome compositions, dietary features, or host characteristics). Our method jointly explores both view-specific representations and cross-view correlations for effective predictions of SCFAs. We applied M2AE to two in-house datasets, which both include MGS and blood SCFAs profiles, host characteristics, and dietary features from 964 subjects and 171 subjects, respectively. Results from both of two datasets demonstrated that M2AE outperforms traditional regression-based and neural-network based approaches in imputing blood SCFAs. Furthermore, a series of gut bacterial species (e.g., Bacteroides thetaiotaomicron and Clostridium asparagiforme), host characteristics (e.g., race, gender), as well as dietary features (e.g., intake of fruits, pickles) were shown to contribute greatly to imputation of blood SCFAs. These findings demonstrated that GMs, dietary features and host characteristics might contribute to the complex biological processes involved in blood SCFA productions. These might pave the way for a deeper and more nuanced comprehension of how these factors impact human health.</p> | https://github.com/Wonderangela123/M2AE | 1\n",
      "111 | Unipept 6.0: Expanding Metaproteomics Analysis with Support for Missed Cleavages, Semi-Tryptic and Non-Tryptic Peptides | https://www.biorxiv.org/content/10.1101/2024.09.26.615136v1 | 10.1101/2024.09.26.615136 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.26.615136v1.full.pdf | <p>Unipept, a pioneering software tool in metaproteomics, has significantly advanced the analysis of complex ecosystems by facilitating both taxonomic and functional insights from environmental samples. From the onset, Unipept9s capabilities focused on tryptic peptides, utilizing the predictability and consistency of trypsin digestion to efficiently construct a protein reference database. However, the evolving landscape of proteomics and emerging fields like immunopeptidomics necessitate a more versatile approach that extends beyond the analysis of tryptic peptides. In this article, we present Unipept 6.0, featuring a novel protein reference database, powered by a Sparse Suffix Array index. This advancement enables the analysis of semi-tryptic peptides, peptides with missed cleavages, and non-tryptic peptides such as those from immunopeptidomics (e.g. MHC- and HLA-peptides). A benchmark study demonstrates that this version of Unipept9s performance matches or exceeds that of previous versions, while significantly expanding its application range.</p> | https://github.com/y-256/libdivsufsort, https://github.com/IlyaGrebnov/libsais | 1\n",
      "112 | Seed2LP: seed inference in metabolic networks for reverse ecology applications | https://www.biorxiv.org/content/10.1101/2024.09.26.615309v1 | 10.1101/2024.09.26.615309 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.26.615309v1.full.pdf | <p>A challenging problem in microbiology is to determine nutritional requirements of microorganisms and culture them, especially for the microbial dark matter detected solely with culture-independent methods. The latter foster an increasing amount of genomic sequences that can be explored with reverse ecology approaches to raise hypotheses on the corresponding populations. Building upon genome scale metabolic networks (GSMNs) obtained from genome annotations, metabolic models predict contextualised phenotypes using nutrient information. We developed the tool Seed2LP, addressing the inverse problem of predicting source nutrients, or seeds, from a GSMN and a metabolic objective. The originality of Seed2LP is its hybrid model, combining a scalable and discrete Boolean approximation of metabolic activity, with the numerically accurate flux balance analysis (FBA). Seed inference is highly customisable, with multiple search and solving modes, exploring the search space of external and internal metabolites combinations. Application to a benchmark of 107 curated GSMNs highlights the usefulness of a logic modelling method over a graph-based approach to predict seeds, and the relevance of hybrid solving to satisfy FBA constraints. Focusing on the dependency between metabolism and environment, Seed2LP is a computational support contributing to address the multifactorial challenge of culturing possibly uncultured microorganisms. Seed2LP is available on https://github.com/bioasp/seed2lp.</p> | https://github.com/potassco/clingo-lpx, https://github.com/bioasp/seed2lp | 1\n",
      "113 | Skeletons in the Forest: Using Entropy-Based Rogue Detection on Bayesian Phylogenetic Tree Distributions | https://www.biorxiv.org/content/10.1101/2024.09.25.615070v1 | 10.1101/2024.09.25.615070 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.25.615070v1.full.pdf | <p>In a phylogenetic analysis, rogue taxa and clades are those that, due to their uncertain placement, contribute disproportionally to the variance in a set of phylogenetics trees. They can obscure clear evolutionary relationships and complicate the interpretation of results. While existing rogue detection algorithms focus on improving the consensus tree for a set of trees, we introduce algorithms based on phylogenetic entropy to improve the analysis in a Bayesian framework. In particular, we introduce a tool that extracts a skeleton distribution of the posterior tree distribution that contains the core relationships by removing a minimal subset of rogues. Additionally, we enable detailed analyses of rogues at both the taxon and clade levels, including the visualisation of different rogue placements.</p> | https://github.com/aberer/RogueNaRok/blob/master/example/ | 1\n",
      "114 | Leveraging cell type-specificity for gene set analysis of single cell transcriptomics | https://www.biorxiv.org/content/10.1101/2024.09.25.615040v1 | 10.1101/2024.09.25.615040 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.25.615040v1.full.pdf | <p>Although single cell RNA-sequencing (scRNA-seq) provides unprecedented insights into the biology of complex tissues, analyzing such data on a gene-by-gene basis is challenging due to the large number of tested hypotheses and consequent low statistical power and difficult interpretation. These issues are magnified by the increased noise, significant sparsity and multi-modal distributions characteristic of single cell data. One promising approach for addressing these challenges is gene set testing, or pathway analysis. Unfortunately, statistical and biological differences between single cell and bulk transcriptomic data make it challenging to use existing gene set collections, which were developed for bulk tissue analysis, on scRNA-seq data. In this paper, we describe a procedure for customizing gene set collections originally created for bulk tissue analysis to reflect the structure of gene activity within specific cell types. Our approach leverages information about mean gene expression in the 81 human cell types profiled via scRNA-seq by the Human Protein Atlas (HPA) Single Cell Type Atlas. This HPA information is used to compute cell type-specific gene and gene set weights that can be used to filter or weight gene set collections. As demonstrated through the analysis of immune cell scRNA-seq data using gene sets from the Molecular Signatures Database (MSigDB), accounting for cell type-specificity can significantly improve gene set testing power and interpretability. An example vignette along with gene and gene set weights for the 81 HPA SCTA cell types and the MSigDB collections are available at https://hrfrost.host.dartmouth.edu/SCGeneSetOpt.</p> | None | 0\n",
      "115 | AnalyzAIRR: A user-friendly guided workflow for AIRR data analysis | https://www.biorxiv.org/content/10.1101/2024.09.25.614939v1 | 10.1101/2024.09.25.614939 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.25.614939v1.full.pdf | <p>The analysis of bulk adaptive immune receptor repertoires (AIRR) enables the understanding of immune responses in both normal and pathological conditions. However, the complexity of AIRR calls for advanced, specialized methods to extract meaningful biological insights. These sophisticated approaches often present challenges for researchers with limited bioinformatics expertise, hindering access to comprehensive immune system analysis. To address this challenge, we developed AnalyzAIRR, an AIRR-compliant R package enabling advanced bulk AIRR sequencing data. The tool integrates state-of-the-art statistical and visualization methods applicable at various levels of granularity. It offers a platform for general data exploration, filtering and manipulation, and in-depth cross-comparisons of AIRR datasets, aimed at answering specific biological questions. We illustrate AnalyzAIRR functionalities using of a published murine dataset of 18 T-cell receptor repertoires from three diferrent T cell subsets. We first detected and removed a major contaminant in a group of samples, before proceeding with to the comparative analysis. Subsequent cross-sample analysis revealed differences in repertoire diversity that aligned with the respective cell phenotypes, and in repertoire convergence among the studied subsets. AnalyzAIRR set of analytical metrics is integrated into a Shiny web application and complemented with a tutorial to help users in their analytical strategy, making it user-friendly for biologists with little or no background in bioinformatics.</p> | https://github.com/i3-unit/AnalyzAIRR | 1\n",
      "116 | Towards improving full-length ribosome density prediction by bridging sequence and graph-based representations | https://www.biorxiv.org/content/10.1101/2024.04.08.588507v3 | 10.1101/2024.04.08.588507 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.04.08.588507v3.full.pdf | <p>Translation elongation plays an important role in regulating protein concentrations in the cell, and dysregulation of this process has been linked to several human diseases. In this study, we use data from ribo-seq experiments to model ribosome densities, and in turn, predict the speed of translation. The proposed method, RiboGL, combines graph and recurrent neural networks to account for both graph and sequence-based features. The model takes a graph representing the secondary structure of the mRNA sequence as input, which incorporates both sequence and structural codon neighbors. In our experiments, RiboGL greatly outperforms the state-of-the-art RiboMIMO model for ribosome density prediction. We also conduct ablation studies to justify the design choices made in building the pipeline. Additionally, we use gradient-based interpretability to understand how the codon context and the structural neighbors affect the ribosome density at the A-site. By individually analyzing the genes in the dataset, we elucidate how structural neighbors could also potentially play a role in defining the ribosome density. Importantly, since these neighbors can be far away in the sequence, a recurrent model alone could not easily extract this information. This study lays the foundation for understanding how the mRNA secondary structure can be exploited for ribosome density prediction, and how in the future other graph modalities such as features from the nascent polypeptide can be used to further our understanding of translation in general.</p> | https://github.com/cgob/codonDT_snakemake, https://github.com/vam-sin/ribogl/ | 1\n",
      "117 | WIMOAD: Weighted Integration of Multi-Omics data for Alzheimer’s Disease (AD) Diagnosis | https://www.biorxiv.org/content/10.1101/2024.09.25.614862v1 | 10.1101/2024.09.25.614862 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.25.614862v1.full.pdf | <p>As the most common subtype of dementia, Alzheimer9s disease (AD) is characterized by a progressive decline in cognitive functions, especially in memory, thinking, and reasoning ability. Early diagnosis and interventions enable the implementation of measures to reduce or slow further regression of the disease, preventing individuals from severe brain function decline. The current framework of AD diagnosis depends on A/T/(N) biomarkers detection from cerebrospinal fluid or brain imaging data, which is invasive and expensive during the data acquisition process. Moreover, the pathophysiological changes of AD accumulate in amino acids, metabolism, neuroinflammation, etc., resulting in heterogeneity in newly registered patients. Recently, next generation sequencing (NGS) technologies have found to be a non-invasive, efficient and less-costly alternative on AD screening. However, most of existing studies rely on single omics only. To address these concerns, we introduce WIMOAD, a weighted integration of multi-omics data for AD diagnosis. WIMOAD synergistically leverages specialized classifiers for patients9 paired gene expression and methylation data for multi-stage classification. The resulting scores were then stacked with MLP-based meta-models for performance improvement. The prediction results of two distinct meta-models were integrated with optimized weights for the final decision-making of the model, providing higher performance than using single omics only. Remarkably, WIMOAD achieves significantly higher performance than using single omics alone in the classification tasks. The model9s overall performance also outperformed most existing approaches, highlighting its ability to effectively discern intricate patterns in multi-omics data and their correlations with clinical diagnosis results. In addition, WIMOAD also stands out as a biologically interpretable model by leveraging the SHapley Additive exPlanations (SHAP) to elucidate the contributions of each gene from each omics to the model output. We believe WIMOAD is a very promising tool for accurate AD diagnosis and effective biomarker discovery across different progression stages, which eventually will have consequential impacts on early treatment intervention and personalized therapy design on AD.</p> | https://github.com/wan-mlab/WIMOAD | 1\n",
      "118 | HuBMAPR: an R Client for the HuBMAP Data Portal | https://www.biorxiv.org/content/10.1101/2024.09.26.615227v1 | 10.1101/2024.09.26.615227 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.26.615227v1.full.pdf | <p>The Human BioMolecular Atlas Program (HuBMAP) constructs the worldwide available platform to research the human body at the cellular level. The HuBMAP Data Portal encompasses a wide range of data resources measured on emerging experimental technologies at a spatial resolution. To broaden access to the HuBMAP Data Portal, we introduce an R client called HuBMAPR available on Bioconductor. This gives an efficient and programmatic interface, enabling researchers to discover and retrieve HuBMAP data easier and faster.</p> | https://github.com/hubmapconsortium, https://github.com/christinehou11/HuBMAPR | 1\n",
      "119 | Lifestyle factors in the biomedical literature: An ontology and comprehensive resources for named entity recognition | https://www.biorxiv.org/content/10.1101/2024.06.13.598816v3 | 10.1101/2024.06.13.598816 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.06.13.598816v3.full.pdf | <p>Despite lifestyle factors (LSFs) being increasingly acknowledged in shaping individual health trajec-tories, particularly in chronic diseases, they have still not been systematically described in the bio-medical literature. This is in part because no named entity recognition (NER) system exists, which can comprehensively detect all types of LSFs in text. The task is challenging due to their inherent diversity, lack of a comprehensive LSF classification for dictionary-based NER, and lack of a cor-pus for deep learning-based NER. Results: We present a novel Lifestyle Factor Ontology (LSFO), which we used to develop a dic-tionary-based system for recognition and normalization of LSFs. Additionally, we introduce a manu-ally annotated corpus for LSFs (LSF200) suitable for training and evaluation of NER systems, and use it to train a transformer-based system. Evaluating the performance of both NER systems on the corpus revealed an F-score of 64% for the dictionary-based system and 76% for the transformer-based system. Large-scale application of these systems on PubMed abstracts and PMC Open Access articles identified over 300 million mentions of LSF in the biomedical literature. Availability: LSFO, the annotated LSF200 corpus, and the detected LSFs in PubMed and PMC-OA articles using both NER systems, are available under open licenses via the following GitHub repository: https://github.com/EsmaeilNourani/LSFO-expansion. This repository contains links to two associated GitHub repositories and a Zenodo project related to the study. LSFO is also availa-ble at BioPortal: https://bioportal.bioontology.org/ontologies/LSFO</p> | https://github.com/larsjuhljensen/tagger, https://github.com/spyysalo/bert-vocab-eval, https://github.com/EsmaeilNourani/LSFO-expansion | 1\n",
      "120 | LOCAS: Multi-label mRNA Localization with Supervised Contrastive Learning | https://www.biorxiv.org/content/10.1101/2024.09.24.614785v2 | 10.1101/2024.09.24.614785 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614785v2.full.pdf | <p>Traditional methods for mRNA subcellular localization often fail to account for multiple compartmentalization. Recent multi-label models have improved performance, but still face challenges in capturing complex localization patterns. We introduce LOCAS (Localization with Supervised Contrastive Learning), which integrates an RNA language model to generate initial embeddings, employs supervised contrastive learning (SCL) to identify distinct RNA clusters, and uses a multi-label classification head (ML-Decoder) with cross-attention for accurate predictions. Through extensive ablation studies and multi-label overlapping threshold tuning, LOCAS achieves state-of-the-art performance across all metrics, providing a robust solution for RNA localization tasks.</p> | None | 0\n",
      "121 | Cell-type-specific mapping of enhancers and target genes from single-cell multimodal data | https://www.biorxiv.org/content/10.1101/2024.09.24.614814v1 | 10.1101/2024.09.24.614814 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614814v1.full.pdf | <p>Mapping enhancers and target genes in disease-related cell types has provided critical insights into the functional mechanisms of genetic variants identified by genomewide association studies (GWAS). However, most existing analyses rely on bulk data or cultured cell lines, which may fail to identify cell-type-specific enhancers and target genes. Recently, single-cell multimodal data measuring both gene expression and chromatin accessibility within the same cells have enabled the inference of enhancer-gene pairs in a cell-type-specific and context-specific manner. However, this task is challenged by the data9s high sparsity, sequencing depth variation, and the computational burden of analyzing a large number of enhancer-gene pairs. To address these challenges, we propose scMultiMap, a statistical method that infers enhancer-gene association from sparse multimodal counts using a joint latent-variable model. It adjusts for technical confounding, permits fast moment-based estimation and provides analytically derived p-values. In systematic analyses of blood and brain data, scMultiMap shows appropriate type I error control, high statistical power with greater reproducibility across independent datasets and stronger consistency with orthogonal data modalities. Meanwhile, its computational cost is less than 1% of existing methods. When applied to single-cell multimodal data from postmortem brain samples from Alzheimer9s disease (AD) patients and controls, scMultiMap gave the highest heritability enrichment in microglia and revealed new insights into the regulatory mechanisms of AD GWAS variants in microglia.</p> | None | 0\n",
      "122 | Labels as a Feature: Network Homophily for Systematically Discovering human GPCR Drug-Target Interactions | https://www.biorxiv.org/content/10.1101/2024.03.29.586957v2 | 10.1101/2024.03.29.586957 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.03.29.586957v2.full.pdf | <p>Machine learning (ML) has revolutionized drug discovery by enabling the exploration of vast, uncharted chemical spaces essential for discovering novel patentable drugs. Despite the critical role of human G protein-coupled receptors (hGPCRs) in FDA approved drugs, exhaustive in-distribution drug-target interaction (DTI) testing across all pairs of hGPCRs and known drugs is rare due to significant economic and technical challenges. This often leaves off-target effects unexplored, which poses a considerable risk to drug safety. In contrast to the traditional focus on out-of-distribution (OOD) exploration (drug discovery), we introduce a neighbourhood-to-prediction model termed Chemical Space Neural Networks (CSNN) that leverages network homophily and training-free graph neural networks (GNNs) with Labels as Features (LaF). We show that CSNN′s ability to make accurate predictions strongly correlates with network homophily. Thus, LaFs strongly increase a ML model′s capacity to enhance in distribution prediction accuracy, which we show by integrating labeled data during inference. We validate these advancements in a high-throughput yeast biosensing system (3773 DTIs, 539 compounds, 7 hGPCRs) to discover novel DTIs for FDA approved drugs and to expand the general understanding of how to build reliable predictors to guide experimental verification.</p> | https://github.com/ashvardanian/SimSIMD | 1\n",
      "123 | Machine Learning-Enhanced Drug Discovery for BACE1: A Novel Approach to Alzheimer’s Therapeutics | https://www.biorxiv.org/content/10.1101/2024.09.24.614844v1 | 10.1101/2024.09.24.614844 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614844v1.full.pdf | <p>Alzheimer9s disease (AD) remains a significant global health challenge, with BACE1, a critical therapeutic target due to its role in the amyloidogenic pathway. This study presents an innovative approach to drug development by integrating machine learning (ML) techniques to identify the potential BACE1 inhibitors. Utilizing a comprehensive dataset of protein-ligand interactions, the current study employed a machine learning model to generate novel ligands with high binding affinity and specificity for the BACE1 active site. The model9s efficacy was validated through molecular docking studies, which demonstrated superior binding affinities compared to existing FDA-approved inhibitors such as Atabecestat and Lanabecestat. The findings reveal that the top candidate, MLC10, exhibits a unique mechanism of action by promoting controlled flexibility in BACE1, contrasting with the rigid conformations induced by traditional inhibitors. This dynamic modulation enhances the enzyme9s inhibition, suggesting a promising avenue for therapeutic intervention. Furthermore, solvent-accessible surface area analyses indicate that MLC10 facilitates a more favourable protein conformation, potentially altering catalytic behaviour through increased solvent interactions. This work underscores the transformative potential of machine learning in drug discovery, paving the way for the development of next-generation BACE1 inhibitors. By harnessing computational techniques, the present work aims to address the limitations of current therapies and contribute to more effective treatments for Alzheimer9s disease, ultimately improving patient outcomes and quality of life.</p> | None | 0\n",
      "124 | Generalizable Morphological Profiling of Cells by Interpretable Unsupervised Learning | https://www.biorxiv.org/content/10.1101/2024.09.24.614684v1 | 10.1101/2024.09.24.614684 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614684v1.full.pdf | <p>The intersection of advanced microscopy and machine learning is revolutionizing cell biology into a quantitative, datadriven science. While traditional morphological profiling of cells relies on laborintensive manual feature extraction susceptible to biases, deep learning offers promising alternatives but struggles with the interpretability of its black-box operation and dependency on extensive labeled data. We introduce MorphoGenie, an unsupervised deeplearning framework designed to address these challenges in singlecell morphological profiling. Enabling disentangled representation learning integrated with high-fidelity image reconstructions, MorphoGenie possesses a critical attribute to learn a compact, generalizable and interpretable latent space. This facilitates the extraction of biologically meaningful features without human annotation, additionally overcoming the \"curse of dimensionality\" inherent in manual methods. Unlike prior models, MorphoGenie introduces a systematic approach to mapping disentangled latent representations to fundamental hierarchical morphological attributes, ensuring both semantic and biological interpretability. Moreover, it adheres to the concept of combinatorial generalization, a core principle of human intelligence, which greatly enhances the model9s capacity to generalize across a broad spectrum of imaging modalities (e.g., quantitative phase imaging and fluorescence imaging) and experimental conditions (ranging from discrete cell type/state classification to continuous trajectory inference). The framework offers a new, generalized strategy for unbiased and comprehensive morphological profiling, potentially revealing insights into cellular behavior in health and disease that might be overlooked by expert visual examination.</p> | https://github.com/rashmisrm/MorphoGenie | 1\n",
      "125 | Local Mean Suppression Filter for Effective Background Identification in Fluorescence Images | https://www.biorxiv.org/content/10.1101/2024.09.25.614955v1 | 10.1101/2024.09.25.614955 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.25.614955v1.full.pdf | <p>We present an easy-to-use, nonlinear filter for effective background identification in fluorescence microscopy images with dense and low-contrast foreground. The pixel-wise filtering is based on comparison of the pixel intensity with the mean intensity of pixels in its local neighborhood. The pixel is given a background or foreground label depending on whether its intensity is less than or greater than the mean respectively. Multiple labels are generated for the same pixel by computing mean expression values by varying neighborhood size. These labels are accumulated to decide the final pixel label. We demonstrate that the performance of our filter favorably compares with state-of-the-art image processing, machine learning, and deep learning methods. We present three use cases that demonstrate its effectiveness, and also show how it can be used in multiplexed fluorescence imaging contexts and as a denoising step in image segmentation. A fast implementation of the filter is available in Python~3 on GitHub.</p> | https://github.com/uttamLab/LMSF | 1\n",
      "126 | FerroEnrich: An Interactive web tool for computing Ferroptosis index and gene enrichment | https://www.biorxiv.org/content/10.1101/2024.09.25.615075v1 | 10.1101/2024.09.25.615075 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.25.615075v1.full.pdf | <p>Ferroptosis is an iron-dependent form of controlled cell death and is characterized by the formation of lipid peroxides. Understanding gene expressions associated with ferroptosis is critical for determining its function in illnesses and potential therapeutic approaches. Despite its significance, no computational model is currently available to accurately quantify the ferroptosis incident. FerroEnrich is a sophisticated web-based tool built using R Shiny application to enumerate the occurrence of ferroptosis based on the relevant gene expressions. This tool available at https://ferroenrich.shinyapps.io/ferroenrich/ processes the input gene expression file to identify genes that are resistant or prone to ferroptosis, calculates ferroptosis index value with dynamic colored heatmap and gene network plot. This manuscript describes the design, operation and usability of FerroEnrich, including examples and a discussion of its potential impact on ferroptosis research. FerroEnrich is a vital tool for researchers, allowing them to explore and analyze complicated gene expression data related to ferroptosis.</p> | None | 0\n",
      "127 | An explainable graph neural network approach for effectively integrating multi-omics with prior knowledge to identify biomarkers from interacting biological domains | https://www.biorxiv.org/content/10.1101/2024.08.23.609465v2 | 10.1101/2024.08.23.609465 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.08.23.609465v2.full.pdf | <p>The rapid growth of multi-omics datasets, in addition to the wealth of existing biological prior knowledge, necessitates the development of effective methods for their integration. Such methods are essential for building predictive models and identifying disease-related molecular markers. We propose a framework for supervised integration of multi-omics data with biological priors represented as knowledge graphs. Our framework leverages graph neural networks (GNNs) to model the relationships among features from high-dimensional 9omics data and set transformers to integrate low-dimensional representations of 9omics features. Furthermore, our framework incorporates explainability methods to elucidate important biomarkers and extract interaction relationships between biological quantities of interest. We demonstrate the effectiveness of our approach by applying it to Alzheimer9s disease (AD) multi-omics data from the ROSMAP cohort, showing that the integration of transcriptomics and proteomics data with AD biological domain network priors improves the prediction accuracy of AD status and highlights robust AD biomarkers.</p> | None | 0\n",
      "128 | Continual integration of single-cell multimodal data with MIRACLE | https://www.biorxiv.org/content/10.1101/2024.09.24.613833v2 | 10.1101/2024.09.24.613833 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.613833v2.full.pdf | <p>Single-cell sequencing technologies have revolutionized our understanding of cellular heterogeneity and facilitated the construction of multi-omics cell atlases via data integration. However, updating these atlases with new data conventionally requires reintegration of all data and is computationally intensive, hindering timely updates and dynamic adjustments in biological and medical research. To address this challenge, we present Multimodal Integration with Continual Learning (MIRACLE), a novel online learning framework for the adaptive and efficient integration of single-cell multimodal data. MIRACLE employs dynamic architectures and data rehearsal strategies to support continual learning, allowing diverse data to be integrated while minimizing information loss over time. Our evaluations demonstrate that MIRACLE achieves accurate online integration with reduced computational requirements, effectively updating and expanding atlases with new cross-tissue and cross-modal data, and precisely identifying novel cell types and transferring labels across datasets. MIRACLE provides an efficient and flexible tool for single-cell community to integrate, share and explore biological knowledge from single-cell multimodal data.</p> | https://github.com/theislab/multigrate, https://github.com/YosefLab/, https://github.com/SONGDONGYUAN1994/scsampler, https://github.com/jaydu1/scVAEIT, https://github.com/welch-lab/liger, https://github.com/jlakkis/ | 1\n",
      "129 | Forecasting dominance of SARS-CoV-2 lineages by anomaly detection using deep AutoEncoders | https://www.biorxiv.org/content/10.1101/2023.10.24.563721v4 | 10.1101/2023.10.24.563721 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.10.24.563721v4.full.pdf | <p>The coronavirus disease of 2019 (COVID19) pandemic is characterized by sequential emergence of severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) variants, lineages, and sublineages, outcompeting previously circulating ones because of, among other factors, increased transmissibility and immune escape. We propose DeepAutoCoV, an unsupervised deep learning anomaly detection system to predict future dominant lineages (FDLs). We define FDLs as viral (sub)lineages that will constitute more than 10% of all the viral sequences added to the GISAID database on a given week. DeepAutoCoV is trained and validated by assembling global and country-specific data sets from over 16 million Spike protein sequences sampled over a period of about 4 years. DeepAutoCoV successfully flags FDLs at very low frequencies (0.01%,3%), with median lead times of 4-17 weeks, and predicts FDLs ~5 and ~25 times better than a baseline approach For example, the B.1.617.2 vaccine reference strain was flagged as FDL when its frequency was only 0.01%, more than a year before it was considered for an updated COVID-19 vaccine. Furthermore, DeepAutoCoV outputs interpretable results by pinpointing specific mutations potentially linked to increased fitness, and may provide significant insights for the optimization of public health pre-emptive intervention strategies.</p> | https://github.com/simoRancati/DeepAutoCoV | 1\n",
      "130 | Protein Sequence Modelling with Bayesian Flow Networks | https://www.biorxiv.org/content/10.1101/2024.09.24.614734v1 | 10.1101/2024.09.24.614734 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614734v1.full.pdf | <p>Exploring the vast and largely uncharted territory of amino acid sequences is crucial for understanding complex protein functions and the engineering of novel therapeutic proteins. Whilst generative machine learning has advanced protein sequence modelling, no existing approach is proficient for both unconditional and conditional generation. In this work, we propose that Bayesian Flow Networks (BFNs),  a recently introduced framework for generative modelling, can address these challenges. We present ProtBFN, a 650M parameter model trained on protein sequences curated from UniProtKB, which generates natural-like, diverse, structurally coherent, and novel protein sequences, significantly outperforming leading autoregressive and discrete diffusion models. Further, we fine-tune ProtBFN on heavy chains from the Observed Antibody Space (OAS) to obtain an antibody-specific model, AbBFN, which we use to evaluate zero-shot conditional generation capabilities. AbBFN is found to be competitive with, or better than, antibody-specific BERT-style models, when applied to predicting individual framework or complimentary determining regions (CDR).</p> | https://github.com/instadeepai/protein-sequence-bfn | 1\n",
      "131 | RanBALL: An Ensemble Random Projection Model for Identifying Subtypes of B-cell Acute Lymphoblastic Leukemia | https://www.biorxiv.org/content/10.1101/2024.09.24.614777v1 | 10.1101/2024.09.24.614777 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614777v1.full.pdf | <p>As the most common pediatric malignancy, B-cell acute lymphoblastic leukemia (B-ALL) has multiple distinct subtypes characterized by recurrent and sporadic somatic and germline genetic alterations. Identification of B-ALL subtypes can facilitate risk stratification and enable tailored therapeutic approaches. Existing methods for B-ALL subtyping primarily depend on immunophenotypic, cytogenetic and genomic analyses, which would be costly, complicated, and laborious in clinical practice applications. To overcome these challenges, we present RanBALL (an Ensemble Random Projection-Based Model for Identifying B-Cell Acute Lymphoblastic Leukemia Subtypes), an accurate and cost-effective model for B-ALL subtype identification based on transcriptomic profiling only. RanBALL leverages random projection (RP) to construct an ensemble of dimension-reduced multi-class support vector machine (SVM) classifiers for B-ALL subtyping. Results based on 10 times 5-fold cross validation tests for &gt;1700 B-ALL patients demonstrated that the proposed model achieved an accuracy of 93.35%, indicating promising prediction capabilities of RanBALL for B-ALL subtyping. The high accuracies of RanBALL suggested that our model could effectively capture underlying patterns of transcriptomic profiling for accurate B-ALL subtype identification. We believe RanBALL will facilitate the discovery of B-ALL subtype-specific marker genes and therapeutic targets, and eventually have consequential positive impacts on downstream risk stratification and tailored treatment design.</p> | https://github.com/wan-mlab/RanBALL | 1\n",
      "132 | scEMB: Learning context representation of genes based on large-scale single-cell transcriptomics | https://www.biorxiv.org/content/10.1101/2024.09.24.614685v1 | 10.1101/2024.09.24.614685 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614685v1.full.pdf | <p>Background: The rapid advancement of single-cell transcriptomic technologies has led to the curation of millions of cellular profiles, providing unprecedented insights into cellular heterogeneity across various tissues and developmental stages. This growing wealth of data presents an opportunity to uncover complex gene-gene relationships, yet also poses significant computational challenges. Results: We present scEMB, a transformer-based deep learning model developed to capture context-aware gene embeddings from large-scale single-cell transcriptomics data. Trained on over 30 million single-cell transcriptomes, scEMB utilizes an innovative binning strategy that integrates data across multiple platforms, effectively preserving both gene expression hierarchies and cell-type specificity. In downstream tasks such as batch integration, clustering, and cell type annotation, scEMB demonstrates superior performance compared to existing models like scGPT and Geneformer. Notably, scEMB excels in silico correlation analysis, accurately predicting gene perturbation effects in CRISPR-edited datasets and microglia state transition, identifying a few known Alzheimer9s disease (AD) risks genes in top gene list. Additionally, scEMB offers robust fine-tuning capabilities for domain-specific applications, making it a versatile tool for tackling diverse biological problems such as therapeutic target discovery and disease modeling. Conclusions: scEMB represents a powerful tool for extracting biologically meaningful insights from complex gene expression data. Its ability to model in silico perturbation effects and conduct correlation analyses in the embedding space highlights its potential to accelerate discoveries in precision medicine and therapeutic development.</p> | https://github.com/bowang-lab/scGPT | 1\n",
      "133 | Analytical Assessment of Metagenomic Workflows for Pathogen Detection with NIST RM 8376 and Two Sample Matrices | https://www.biorxiv.org/content/10.1101/2024.09.24.614717v1 | 10.1101/2024.09.24.614717 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614717v1.full.pdf | <p>We assessed the analytical performance of metagenomic workflows using NIST Reference Material 8376 DNA from bacterial pathogens spiked into two simulated clinical samples: cerebral spinal fluid (CSF) and stool. Sequencing and taxonomic classification were used to generate signals for each sample and taxa of interest, and to estimate the limit of detection (LOD), the response function, and linear dynamic range. We found that the LODs for taxa spiked into CSF ranged from approximately (0.1 to 0.3) copy/μL, with a linearity of 0.96 to 0.99. For stool, the LODs ranged from (10 to 221) copy/μL, with a linearity of 0.99 to 1.01. Further, discriminating different E. coli strains proved to be workflow-dependent, as only one classifier:database combination of the three tested showed the ability to differentiate the two pathogenic and  commensal strains. Surprisingly, when we compared the response functions of the same taxa in the two different sample types, we found those functions to be the same, despite large differences in LODs. This suggests that the agnostic-diagnostic theory for metagenomics may apply to different target organisms and different sample types. Using RMs, we were able to generate quantitative analytical performance metrics for each workflow and sample set, enabling relatively rapid workflow screening before employing clinical samples. This makes these RMs a useful tool that will generate data needed to support translation of metagenomics into regulated use.</p> | None | 0\n",
      "134 | BRDKRM: An Explainable Framework for Disease Modifying Drug Identification | https://www.biorxiv.org/content/10.1101/2024.09.24.614653v1 | 10.1101/2024.09.24.614653 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614653v1.full.pdf | <p>Background: Drug classification into disease-modifying (DM) and symptomatic (SYM) categories are crucial for clinical decision-making and therapeutic strategy development. Current methods often lack transparency and interpretability, limiting their utility in the biomedical domain. Method: We propose the Boundary Restricted Dynamic Key Route Mapping (BRDKRM) framework, which leverages the contextual overlap between disease and drug nodes in a heterogeneous graph. Our method aggregates genes from the top K shortest paths between diseases and\n",
      "known drugs, delineating disease neighborhood boundaries. We then classify unknown drug-disease treatments based on the average score of boundary genes in paths to unknown drugs. Results: BRDKRM achieved an accuracy of 86.78% in categorizing drug-disease treatments. We identified 530 genes playing roles in both modifying diseases and providing symptomatic relief. We demonstrated the method9s efficacy is demonstrated by using case studies on multiple sclerosis. Conclusion: BRDKRM offers an explainable approach to drug classification, bypassing extensive clinical trials. By providing biologically sound interpretations of drug classifications, our framework enhances understanding of therapeutic interventions and paves the way for more precise and efficient healthcare solutions.</p> | None | 0\n",
      "135 | PlantConnectome: knowledge graph encompassing >70,000 plant articles | https://www.biorxiv.org/content/10.1101/2023.07.11.548541v3 | 10.1101/2023.07.11.548541 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.07.11.548541v3.full.pdf | <p>One of the main quests of plant biology is understanding how genes and metabolites work together to form complex networks that drive plant growth, development, and responses to environmental stimuli. However, the ever-growing volume and diversity of scientific literature make it increasingly challenging to stay current with the latest advances in gene function studies. Here, we tackle the challenge by deploying the text-mining capacities of large language models to process over 71,000 plant biology abstracts. Our approach unveiled nearly 5 million functional relationships between a wide array of biological entities -genes, metabolites, tissues, and others - with a high accuracy of over 85%. We encapsulated these findings in PlantConnectome, a user-friendly database, and demonstrated its diverse utility by providing insights into gene regulatory networks, protein-protein interactions, and stress responses. We believe this innovative use of AI in the life sciences will allow plant scientists to keep up to date with the rapidly growing corpus of scientific literature. PlantConnectome is available at https://plant.connectome.tools/.</p> | None | 0\n",
      "136 | Evaluating the representational power of pre-trained DNA language models for regulatory genomics | https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2 | 10.1101/2024.02.29.582810 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2.full.pdf | <p>The emergence of genomic language models (gLMs) offers an unsupervised approach to learning a wide diversity of cis- regulatory patterns in the non-coding genome without requiring labels of functional activity generated by wet-lab experiments. Previous evaluations have shown that pre-trained gLMs can be leveraged to improve predictive performance across a broad range of regulatory genomics tasks, albeit using relatively simple benchmark datasets and baseline models. Since the gLMs in these studies were tested upon fine-tuning their weights for each downstream task, determining whether gLM representations embody a foundational understanding of cis-regulatory biology remains an open question. Here we evaluate the representational power of pre-trained gLMs to predict and interpret cell-type-specific functional genomics data that span DNA and RNA regulation. Our findings suggest that probing the representations of pre-trained gLMs do not offer substantial advantages over conventional machine learning approaches that use one-hot encoded sequences. This work highlights a major gap with current gLMs, raising potential issues in conventional pre-training strategies for the non-coding genome.</p> | None | 0\n",
      "137 | Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq | https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2 | 10.1101/2024.09.22.614322 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2.full.pdf | <p>Single-cell RNA sequencing (scRNA-seq) in multi-condition experiments enables the systematic assessment of treatment effects. Analyzing scRNA-seq data relies on linear dimensionality reduction (DR) methods like principal component analysis (PCA). These methods decompose high-dimensional gene expression profiles into tractable factor representations and prototypical gene expression patterns (components), facilitating the study of cell type variation. However, integrating study covariates within linear DR frameworks remains a challenging task. We present scPCA, a flexible DR framework that jointly models cellular heterogeneity and conditioning variables, allowing it to recover an integrated factor representation and reveal transcriptional changes across conditions and components of the decomposition. We show that scPCA extracts an interpretable latent representation by analyzing unstimulated and IFN-beta-treated PBMCs, and showcase that the model may be employed to effectively address batch effects. We examine age-related changes in rodent lung cell populations, uncovering a previously unreported surge in Ccl5 expression in T cells. We illustrate how scPCA may be employed to identify coordinated transcriptional changes across multiple time-points in depolarized visual cortex neurons. Finally, we show that scPCA elucidates transcriptional shifts in CRISPR-Cas9 chordin knockout zebrafish fish single-cell data despite large difference cell abundance across conditions. Since scPCA introduces a general approach to account for conditioning variables in high-dimensional data, it may also be applicable to datasets other than scRNA-seq.</p> | None | 0\n",
      "138 | varCADD: large sets of standing genetic variation enable genome-wide pathogenicity prediction | https://www.biorxiv.org/content/10.1101/2024.09.24.614666v1 | 10.1101/2024.09.24.614666 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614666v1.full.pdf | <p>Machine learning and artificial intelligence are increasingly being applied to identify phenotypically causal genetic variation. These data-driven methods require comprehensive training sets to deliver reliable results. However, large unbiased datasets for variant prioritization and effect predictions are rare as most of the available databases do not represent a broad ensemble of variant effects and are often biased towards coding genome, or even towards few well-studied genes. To overcome these issues, we propose several alternative training sets derived from subsets of human standing variation. Specifically, we use variants identified from whole-genome sequences of 71,156 individuals contained in gnomAD v3.0 and approximate the benign set with frequent and the deleterious set with rare standing variation. We apply the Combined Annotation Dependent Depletion framework (CADD) and train several alternative models using CADD v1.6. Using the NCBI ClinVar validation set, we demonstrate that the alternative models have state-of-the art accuracy, outperforming the widely used pathogenicity score CADD v1.6 in certain genomic regions. Being larger than conventional databases, including the evolutionary-derived training dataset of about 30 million variants in CADD, standing variation cover a broader range of genomic regions and rare instances of the applied annotations. For example, they cover more recent evolutionary changes common in gene regulatory regions, which are more challenging to assess with conventional tools. Finally, datasets derived from standing variation better represent allelic changes in the human genome and do not require extensive simulations and adaptations to annotations of the evolutionary-derived sequence alterations used for CADD training. We provide datasets as well as trained models to the community for further development and application.</p> | None | 0\n",
      "139 | Diversity and abundance of ring nucleases in type III CRISPR-Cas loci | https://www.biorxiv.org/content/10.1101/2024.09.24.614671v1 | 10.1101/2024.09.24.614671 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614671v1.full.pdf | <p>Most type III CRISPR-Cas systems facilitate immune responses against invading mobile genetic elements such as phages by generating cyclic oligoadenylates (cOAs). Downstream effectors activated by cOAs are typically non-specific proteins that induce damage to essential cellular components, thereby preventing phage epidemics. Due to these toxic effects, it is crucial that the production and concentration of cOAs remain under tight regulatory control during infection-free periods or when deactivating the immune response after clearing an infection. Type III CRISPR loci often encode enzymes known as ring nucleases (RNs) that bind and degrade specific cOAs, while some effectors are auto-deactivating. Despite the discovery of several classes of RNs, a comprehensive bioinformatic analysis of type III CRISPR-Cas loci in this context is lacking. Here, we examined 38,742 prokaryotic genomes to provide a global overview of type III CRISPR loci, focussing on the known and predicted RNs. The candidate RNs Csx16 and Csx20 are confirmed as active enzymes, joining Crn1-3. Distributions and patterns of co-occurrence of RNs and associated effectors are explored, allowing the conclusion that a sizeable majority of type III CRISPR systems regulate cOA levels by degrading the signalling molecules, which has implications for cell fate following viral infection.</p> | https://github.com/vihoikka/ring_nucleases | 1\n",
      "140 | Germinated brown rice research: bibliometric analysis of progress, hotspots and trends | https://www.biorxiv.org/content/10.1101/2024.09.24.614820v1 | 10.1101/2024.09.24.614820 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614820v1.full.pdf | <p>To gain a deeper understanding of global research trends and focal points in germinated brown rice, this article takes the relevant literature on germinated brown rice in the core database of Web of Science as the research object. By using bibliometric analysis, the literature on germinated brown rice published from 2003 to 2023 is deeply analyzed, and the global research progress, hotspots and future development trend of germinated brown rice are summarized. Since 2018, research on germinated brown rice has been rapidly developing, with a significant surge in interest since 2020. China leads in the number of publications, institutions, and core authors in this research area. The primary research topics include the nutritional value, physiological active components, and industrial applications of germinated brown rice. Current research frontiers involve identifying, evaluating, and enhancing bioactive substances in germinated brown rice for food applications. Research in this field remains active, and application scenarios are becoming increasingly diverse. Future studies may explore new directions in related equipment and ingredients.</p> | None | 0\n",
      "141 | p53motifDB: integration of genomic information and tumor suppressor p53 binding motifs | https://www.biorxiv.org/content/10.1101/2024.09.24.614594v1 | 10.1101/2024.09.24.614594 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614594v1.full.pdf | <p>The tumor suppressor gene TP53 encodes the DNA binding transcription factor p53 and is one of the most commonly mutated genes in human cancer. Tumor suppressor activity requires binding of p53 to its DNA response elements and subsequent transcriptional activation of a diverse set of target genes. Despite decades of close study, the logic underlying p53 interactions with its numerous potential genomic binding sites and target genes is not yet fully understood. Here, we present a database of DNA and chromatin-based information focused on putative p53 binding sites in the human genome to allow users to generate and test new hypotheses related to p53 activity in the genome. Users can query genomic locations based on experimentally observed p53 binding, regulatory element activity, genetic variation, evolutionary conservation, chromatin modification state, and chromatin structure. We present multiple use cases demonstrating the utility of this database for generating novel biological hypotheses, such as chromatin-based determinants of p53 binding and potential cell type-specific p53 activity. All database information is also available as a precompiled sqlite database for use in local analysis or as a Shiny web application.</p> | None | 0\n",
      "142 | RNA-DCGen: Dual Constrained RNA Sequence Generation with LLM-Attack | https://www.biorxiv.org/content/10.1101/2024.09.23.614570v1 | 10.1101/2024.09.23.614570 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.23.614570v1.full.pdf | <p>Designing RNA sequences with specific properties is critical for developing personalized medications and therapeutics. While recent diffusion and flow-matching-based generative models have made strides in conditional sequence design, they face two key limitations: specialization for fixed constraint types, such as tertiary structures, and lack of flexibility in imposing additional conditions beyond the primary property of interest. To address these challenges, we introduce RNA-DCGen, a generalized framework for RNA sequence generation that is adaptable to any structural or functional properties through straightforward finetuning with an RNA language model (RNA-LM). Additionally, RNA-DCGen can enforce conditions on the generated sequences by fixing specific conserved regions. On RNA generation conditioned on RNA distance maps, RNA-DCGen generates sequences with an average R2 score of 0.625 compared to random sequences that score only 0.118 over 250 generations as judged by a separate more capable RNA-LM. When conditioned on RNA secondary structures, RNA-DCGen achieves an average F1 score of 0.4 against a random baseline of 0.006.</p> | None | 0\n",
      "143 | Clustering-independent estimation of cell abundances in bulk tissues using single-cell RNA-seq data | https://www.biorxiv.org/content/10.1101/2023.02.06.527318v2 | 10.1101/2023.02.06.527318 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.02.06.527318v2.full.pdf | <p>Single-cell RNA-sequencing has transformed the study of biological tissues by enabling transcriptomic characterizations of their constituent cell states. Computational methods for gene expression deconvolution use this information to infer the cell composition of related tissues profiled at the bulk level. However, current deconvolution methods are restricted to discrete cell types and have limited power to make inferences about continuous cellular processes like cell differentiation or immune cell activation. We present ConDecon, a clustering-independent method for inferring the likelihood for each cell in a single-cell dataset to be present in a bulk tissue. ConDecon represents an improvement in phenotypic resolution and functionality with respect to regression-based methods. Using ConDecon, we discover the implication of neurodegenerative microglia inflammatory pathways in the mesenchymal transformation of pediatric ependymoma and characterize their spatial trajectories of activation. The generality of this approach enables the deconvolution of other data modalities such as bulk ATAC-seq data.</p> | https://github.com/amitfrish/scBio, https://github.com/favilaco/deconv_benchmark | 1\n",
      "144 | SeaMoon: Prediction of molecular motions based on language models | https://www.biorxiv.org/content/10.1101/2024.09.23.614585v1 | 10.1101/2024.09.23.614585 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.23.614585v1.full.pdf | <p>How protein move and deform determines their interactions with the environment and is thus of utmost importance for cellular functioning. Following the revolution in single protein 3D structure prediction, researchers have focused on repurposing or developing deep learning models for sampling alternative protein conformations. In this work, we explored whether continuous compact representations of protein motions could be predicted directly from protein sequences, without exploiting nor sampling protein structures. Our approach, called SeaMoon, leverages protein Language Model (pLM) embeddings as input to a lightweight (≈1M trainable parameters) convolutional neural network. SeaMoon achieves a success rate of up to 40% when assessed against ≈1~000 collections of experimental conformations exhibiting a wide range of motions. SeaMoon capture motions not accessible to the normal mode analysis, an unsupervised physics-based method relying solely on a protein structure9s 3D geometry, and generalises to proteins that do not have any detectable sequence similarity to the training set. SeaMoon is easily retrainable with novel or updated pLMs.</p> | None | 0\n",
      "145 | INSPIRE: interpretable, flexible and spatially-aware integration of multiple spatial transcriptomics datasets from diverse sources | https://www.biorxiv.org/content/10.1101/2024.09.23.614539v1 | 10.1101/2024.09.23.614539 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.23.614539v1.full.pdf | <p>Recent advances in spatial transcriptomics technologies have led to a growing number of diverse datasets, offering unprecedented opportunities to explore tissue organizations and functions within spatial contexts. However, it remains a significant challenge to effectively integrate and interpret these data, often originating from different samples, technologies, and developmental stages. In this paper, we present INSPIRE, a deep learning method for integrative analyses of multiple spatial transcriptomics datasets to address this challenge. With designs of graph neural networks and an adversarial learning mechanism, INSPIRE enables spatially informed and adaptable integration of data from varying sources. By incorporating non-negative matrix factorization, INSPIRE uncovers interpretable spatial factors with corresponding gene programs, revealing tissue architectures, cell type distributions and biological processes. We demonstrate the capabilities of INSPIRE by applying it to human cortex slices from different samples, mouse brain slices with complementary views, mouse hippocampus and embryo slices generated through different technologies, and spatiotemporal organogenesis atlases containing half a million spatial spots. INSPIRE shows superior performance in identifying detailed biological signals, effectively borrowing information across distinct profiling technologies, and elucidating dynamical changes during embryonic development. Furthermore, we utilize INSPIRE to build 3D models of tissues and whole organisms from multiple slices, demonstrating its power and versatility.</p> | https://github.com/jiazhao97/INSPIRE | 1\n",
      "146 | Clustering individuals using INMTD: a novel versatile multi-view embedding framework integrating omics and imaging data | https://www.biorxiv.org/content/10.1101/2024.09.23.614478v1 | 10.1101/2024.09.23.614478 | 2024-01-01 | Lien PDF non disponible | <p>Motivation: Combining omics and images, can lead to a more comprehensive clustering of individuals than classic single-view approaches. Among the various approaches for multi-view clustering, nonnegative matrix tri-factorization (NMTF) and nonnegative Tucker decomposition (NTD) are advantageous in learning low-rank embeddings with promising interpretability. Besides, there is a need to handle unwanted drivers of clusterings (i.e. confounders). Results: In this work, we introduce a novel multi-view clustering method based on NMTF and NTD, named INMTD, that integrates omics and 3D imaging data to derive unconfounded subgroups of individuals. In the application to real-life facial-genomic data, INMTD generated biologically relevant embeddings for individuals, genetics and facial morphology. By removing confounded embedding vectors, we derived an unconfounded clustering with better internal and external quality; the genetic and facial annotations of each derived subgroup highlighted distinctive characteristics. In conclusion, INMTD can effectively integrate omics data and 3D images for unconfounded clustering with biologically meaningful interpretation. Availability and implementation: https://github.com/ZuqiLi/INMTD</p> | None | 0\n",
      "147 | MedGraphNet : Leveraging Multi-Relational Graph Neural Networks and Text Knowledge for Biomedical Predictions | https://www.biorxiv.org/content/10.1101/2024.09.24.614782v1 | 10.1101/2024.09.24.614782 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.614782v1.full.pdf | <p>Genetic, molecular, and environmental factors influence diseases through complex interactions with genes, phenotypes, and drugs. Current methods often fail to integrate diverse multi-relational biological data meaningfully, limiting the discovery of novel risk genes and drugs. To address this, we present MedGraphNet, a multi-relational Graph Neural Network (GNN) model designed to infer relationships among drugs, genes, diseases, and phenotypes. MedGraphNet initializes nodes using informative embeddings from existing text knowledge, allowing for robust integration of various data types and improved generalizability. Our results demonstrate that MedGraphNet matches and often outperforms traditional single-relation approaches, particularly in scenarios with isolated or sparsely connected nodes. The model shows generalizability to external datasets, achieving high accuracy in identifying disease-gene associations and drug-phenotype relationships. Notably, MedGraphNet accurately inferred drug side effects without direct training on such data. Using Alzheimer9s disease as a case study, MedGraphNet successfully identified relevant phenotypes, genes, and drugs, corroborated by existing literature. These findings demonstrate the potential of integrating multi-relational data with text knowledge to enhance biomedical predictions and drug repurposing for diseases.</p> | https://github.com/vinash85/MedGraphNet | 1\n",
      "148 | BrainUnit: Integrating Physical Units into High-Performance AI-Driven Scientific Computing | https://www.biorxiv.org/content/10.1101/2024.09.20.614111v3 | 10.1101/2024.09.20.614111 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.20.614111v3.full.pdf | <p>Artificial intelligence (AI) is revolutionizing scientific research across various disciplines. The foundation of scientific research lies in rigorous scientific computing based on standardized physical units. However, current mainstream high-performance numerical computing libraries for AI generally lack native support for physical units, significantly impeding the integration of AI methodologies into scientific research. To fill this gap, we introduce BrainUnit, a unit system designed to seamlessly integrate physical units into AI libraries, with a focus on compatibility with JAX. BrainUnit offers a comprehensive library of over 2000 physical units and more than 300 unit-aware mathematical functions. It is fully compatible with JAX transformations, allowing for automatic differentiation, just-in-time compilation, vectorization, and parallelization while maintaining unit consistency. We demonstrate BrainUnit9s efficacy through several use cases in brain dynamics modeling, including detailed biophysical neuron simulations, multiscale brain network modeling, neuronal activity fitting, and cognitive task training. Our results show that BrainUnit enhances the accuracy, reliability, and interpretability of scientific computations across scales, from ion channels to whole-brain networks, without significantly impacting performance. By bridging the gap between abstract computational frameworks and physical units, BrainUnit represents a crucial step towards more robust and physically grounded AI-driven scientific computing.</p> | https://github.com/chaoming0625/brainunit-experiments, https://github.com/chaoming0625/ | 1\n",
      "149 | ArchMap: A web-based platform for reference-based analysis of single-cell datasets | https://www.biorxiv.org/content/10.1101/2024.09.19.613883v2 | 10.1101/2024.09.19.613883 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.19.613883v2.full.pdf | <p>Leveraging single cell reference atlases to analyse new data has brought about a paradigm shift in single cell data science akin to the first reference genome in genomics. However methods to perform this mapping require computational expertise as well as sometimes considerable compute power, and thus may exclude the researchers from this innovation who may benefit the most from it. ArchMap, a no-code query-to-reference mapping tool, removes this barrier by providing all-in-one automated mapping, cell type annotation, and collaborative features to analyse single-cell datasets from a wide range of integrated, often published, reference atlases and allows extension of atlases with the growing Human Cell Atlas and related efforts. This paves the way for a democratisation of reference mapping capabilities.</p> | None | 0\n",
      "150 | High-throughput bacterial aggregation analysis in droplets | https://www.biorxiv.org/content/10.1101/2024.09.24.613170v1 | 10.1101/2024.09.24.613170 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.613170v1.full.pdf | <p>The communal lifestyle of bacteria manifested as bacterial aggregation and potential biofilm formation is posing as a significant contributor to antimicrobial resistance. A key approach to adress the emerging pandemic is to implement novel techniques that enhance the precision and speed of analysis in various microbiological experiments. Droplet-based platforms coupled with image analysis techniques exhibit great potential in advancing the precision of investigating bacterial behaviour and increasing the speed of microbiological experiments. As most of the image analysis techniques require advanced programming skills or implement high-priced fees, open-source software interweaving user-friendly interface is in great demand. In this paper, we introduce two image analysis pipelines constructed in CellProfilerTM to quantify bacterial aggregation responding to a presence of microplastics as well as suboptimal concentrations of antibiotics in droplets.</p> | None | 0\n",
      "151 | GeoAB: Towards Realistic Antibody Design and Reliable Affinity Maturation | https://www.biorxiv.org/content/10.1101/2024.05.15.594274v2 | 10.1101/2024.05.15.594274 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.05.15.594274v2.full.pdf | <p>Increasing works for antibody design are emerging to generate sequences and structures in Complementarity Determining Regions (CDRs), but problems still exist. We focus on two of them:  (i) authenticity of the generated structure and (ii) rationality of the affinity maturation, and propose GeoAB as a solution. In specific, GeoAB-Designer generates CDR structures with realistic internal geometries, composed of a generative geometry initializer (Geo-Initializer) and a position refiner (Geo-Refiner); GeoAB-Optimizer achieves affinity maturation by accurately predicting both the mutation effects and structures of mutant antibodies with the same network architecture as Geo-Refiner. Experiments show that GeoAB achieves state-of-the-art performance in CDR co-design and mutation effect predictions, and fulfills the discussed tasks effectively.</p> | https://github.com/Edapinenut/GeoAB | 1\n",
      "152 | Fast and accurate local ancestry inference with Recomb-Mix | https://www.biorxiv.org/content/10.1101/2023.11.17.567650v3 | 10.1101/2023.11.17.567650 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.11.17.567650v3.full.pdf | <p>The availability of large genotyped cohorts brings new opportunities for revealing the high-resolution genetic structure of admixed populations via local ancestry inference (LAI), the process of identifying the ancestry of each segment of an individual haplotype. Though current methods achieve high accuracy in standard cases, LAI is still challenging when reference populations are more similar (e.g., intra-continental), when the number of reference populations is too numerous, or when the admixture events are deep in time, all of which are increasingly unavoidable in large biobanks. Here, we present a new LAI method, Recomb-Mix. Recomb-Mix integrates the elements of existing methods of the site-based Li and Stephens model and introduces a new graph collapsing trick to simplify counting paths with the same ancestry label readout. Through comprehensive benchmarking on various simulated datasets, we show that Recomb-Mix is more accurate than existing methods in diverse sets of scenarios while being competitive in terms of resource efficiency. We expect that Recomb-Mix will be a useful method for advancing genetics studies of admixed populations.</p> | None | 0\n",
      "153 | Pan-tissue Transcriptome Analysis Reveals Sex-dimorphic Human Aging | https://www.biorxiv.org/content/10.1101/2023.05.26.542373v2 | 10.1101/2023.05.26.542373 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.05.26.542373v2.full.pdf | <p>Complex diseases often exhibit sex-dimorphism in morbidity and prognosis, many of which are age-related. However, the underlying mechanisms of the sex-dimorphic aging remain foggy, with limited studies across multiple tissues. We systematically analyzed ~17,000 transcriptomes from 35 human tissues to quantitatively evaluate the individual and combined contributions of sex and age to transcriptomic variations. We discovered extensive sex-dimorphisms during aging with distinct patterns of change in gene expression and alternative splicing (AS). Intriguingly, the male-biased age-associated AS events have a stronger association with Alzheimer9s disease, and the females-biased events are often regulated by several sex-biased splicing factors that may be controlled by estrogen receptors.  Breakpoint analysis showed sex-dimorphic aging rates that are significantly associated with decline of sex hormones, with males having a larger and earlier transcriptome change.  Collectively, this study uncovered an essential role of sex during aging at the molecular and multi-tissue levels, providing insight into sex-dimorphic regulatory patterns.</p> | https://github.com/rnasys/Sex-dimorphic-aging | 1\n",
      "154 | An integrated single-cell atlas of the limb skeleton from development through adulthood | https://www.biorxiv.org/content/10.1101/2022.03.14.484345v3 | 10.1101/2022.03.14.484345 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2022.03.14.484345v3.full.pdf | <p>The recent growth of single-cell transcriptomics has turned single-cell RNA sequencing (scRNA-seq) into a near-routine experiment. Breakthroughs in improving scalability have led to the creation of organism-wide transcriptomic datasets, aiming to comprehensively profile the cell types and states within an organism throughout its lifecycle. To date, however, the skeleton remains a majorly underrepresented organ system in organism-wide atlases. Considering how the skeleton not only serves as the central framework of the vertebrate body but is also the home of the hematopoietic niche and a central player in major metabolic and homeostatic processes, this presents a major deficit in current reference atlas projects. To address this issue, we integrated ten separate scRNA-seq datasets containing limb skeletal cells and their developmental precursors, generating an atlas of 133 332 cells. This limb skeletal cell atlas describes cells across the mesenchymal lineage from the induction of the limb to the adult bone and encompasses 39 different cell states. Furthermore, expanding the repertoire of available time points and cell types within a single dataset allowed for more complete analyses of cell-cell communication or in silico perturbation studies. Taken together, we present a missing piece in the current atlas mapping efforts, which will be of value to researchers in the fields of skeletal biology, hematopoiesis, metabolism and regenerative medicine.</p> | https://github.com/TElabSBE/LimbSkeletalCellAtlas | 1\n",
      "155 | scParadise: Tunable highly accurate multi-task cell type annotation and surface protein abundance prediction | https://www.biorxiv.org/content/10.1101/2024.09.23.614509v1 | 10.1101/2024.09.23.614509 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.23.614509v1.full.pdf | <p>scRNA-seq is revolutionizing biomedical research by revealing tissue architecture, cellular composition, and functional interactions. However, accurate cell type annotation remains a challenge, particularly for rare cell types, with existing automated methods often falling short. Multimodal data, combining mRNA expression and protein markers, improves deep cellular analysis and make functional characterization of complex tissues more accurate. However, it is costly and complex to obtain. We present scParadise, a cutting-edge Python framework featuring three tools: scAdam for multi-level cell annotation, scEve for surface protein prediction, and scNoah for benchmarking. scAdam surpasses current methods in annotating rare cell types and ensures consistent results across diverse datasets, while scEve enhances clustering and cell type separation. With scNoah′s advanced metrics, scParadise offers a powerful, fast, and reliable solution for single-cell analysis, setting a new standard in scRNA-seq data processing.</p> | https://github.com/Chechekhins/scParadise/tree/main, https://github.com/satijalab/azimuth, https://github.com/Chechekhins/scParadise/tree/main/scripts_article | 1\n",
      "156 | GWA-X: An Extensible GPU Accelerated Package for Permutation Testing in Genome-Wide Association Studies | https://www.biorxiv.org/content/10.1101/2024.09.15.613119v2 | 10.1101/2024.09.15.613119 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.15.613119v2.full.pdf | <p>Genome-wide association studies (GWAS) aim to identify associations of genetic variants with a trait or disease. The scale of genomic datasets has increased to millions of genetic variants and hundreds of thousands of individuals, opening the possibilities for discoveries from GWAS. However, large-scale GWAS analyses are prone to high false positive rates because of the multiple testing problem. Permutation testing is the gold standard for maintaining false positive rates, yet it is impractical for large-scale GWAS because it requires vast computational resources.\n",
      "\n",
      "This paper presents GWA-X, a software package that can fully benefit from GPUs and accelerate permutation testing in GWAS. In contrast to previous methods, GWA-X employs a novel whole-genome regression method to batch the computations of many genetic markers. It achieved a two-order magnitude speed-up compared with the existing CPU-based and GPU-based methods and more than one-order magnitude speed-up compared with the current state-of-the-art GPU-based library. In addition, GWA-X provides an extensible framework for conducting permutation tests in GWAS.</p> | None | 0\n",
      "157 | dScaff – an automatic bioinformatics framework for scaffolding draft de novo assemblies based on reference genome data | https://www.biorxiv.org/content/10.1101/2024.09.23.614313v1 | 10.1101/2024.09.23.614313 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.23.614313v1.full.pdf | <p>Rapid evolution of long-read sequencing technologies requires accurate, fast and user-friendly genome assembly and scaffolding tools. In this article we present Digital Scaffolding (dScaff), a bioinformatics tool that performs scaffolding of <i>de novo</i> genome assemblies based on reference scaffolds or chromosomes. dScaff makes use of a series of bash and R scripts in order to create a minimal complete scaffold from a genome assembly, with future features to be implemented. Herein, we demonstrate the functionality of dScaff on a novel genome assembly pertaining to a recently sequenced local strain of <i>Drosophila suzukii</i>.</p> | https://github.com/DL-UB/SubSequencesExtractor | 1\n",
      "158 | Camera Paths, Modeling, and Image Processing Tools for ArtiaX | https://www.biorxiv.org/content/10.1101/2024.09.23.614454v1 | 10.1101/2024.09.23.614454 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.23.614454v1.full.pdf | <p>The enhancement of biomolecular image analysis and data interpretation is significantly improved through the application of advanced visualization techniques. Numerous visualization packages are currently available, spanning a broad spectrum of applications. Recently, we have extended the capabilities of UCSF ChimeraX to address the specific demands of cryo-electron tomography. Here, we introduce the evolution of our existing plugin, ArtiaX, designed to generate models that facilitate particle selection, define camera recording paths, and execute particle selection routines. In particular, diverse models can be generated and populated with putative particle positions and orientations. A specifically tailored coarse grained algorithm was developed to rectify overlaps, as encountered in template matching, employing a rapid and efficient approach. In addition, models can be used to drive the camera position, thereby simplifying the process of movie creation. The plugin incorporates fundamental image filtering options for the on-the-fly analysis of tomographic data. Collectively, this update of ArtiaX comprehensively encompasses essential tools for the analysis and visualization of electron tomograms. It retains its hallmark attributes of speed, reliability, and user-friendliness, fostering seamless human-machine interaction.</p> | None | 0\n",
      "159 | Jaeger: an accurate and fast deep-learning tool to detect bacteriophage sequences | https://www.biorxiv.org/content/10.1101/2024.09.24.612722v1 | 10.1101/2024.09.24.612722 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.24.612722v1.full.pdf | <p>Abstract \n",
      "Viruses are integral to every biome on Earth, yet we still need a more comprehensive picture of their identity and global distribution. Global metagenomics sequencing efforts revealed the genomic content of tens of thousands of environmental samples, however identifying the viral sequences in these datasets remains challenging due to their vast genomic diversity. Here, we address identifying bacteriophage sequences in sequencing data. In a recent benchmarking paper, we observed that existing deep-learning tools show a high true positive rate, but often produce many false positives when confronted with divergent sequences. To tackle this challenge, we introduce Jaeger, a novel deep-learning method designed specifically for identifying bacteriophage genome fragments. Extensive benchmarking on the IMG/VR database and real-world metagenomes reveals Jaeger9s consistent performance across various scenarios. Applying Jaeger to over 16,000 metagenomic assemblies from the MGnify database yielded over five million putative phage contigs at an estimated 2-27% false discovery rate. On average, Jaeger is around 20 times faster than the other state-of-the-art methods, highlighting its efficacy in bacteriophage identification within global metagenomes. Jaeger is available at https://github.com/MGXlab/Jaeger.</p> | https://github.com/MGXlab/Jaeger | 1\n",
      "160 | STANCE: a unified statistical model to detect cell-type-specific spatially variable genes in spatial transcriptomics | https://www.biorxiv.org/content/10.1101/2024.09.22.614385v1 | 10.1101/2024.09.22.614385 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.22.614385v1.full.pdf | <p>A significant challenge in analyzing spatial transcriptomics data is the effective and efficient detection of spatially variable genes (SVGs), whose expression exhibits non-random spatial patterns in tissues. Many SVGs show spatial variation in expression that is highly correlated with cell type categories or compositions, leading to the concept of cell type-specific spatially variable genes (ctSVGs). Existing statistical methods for detecting ctSVGs treat cell type-specific spatial effects as fixed effects when modeling, resulting in a critical issue: the testing results are not invariant to the rotation of spatial coordinates. Additionally, an SVG may display random spatial patterns within a cell type, and a ctSVG may exhibit random spatial patterns from a general perspective, indicating that an SVG does not necessarily have to be a ctSVG, and vice versa. This poses challenges in real analysis when detecting SVGs or ctSVGs. To address these problems, we propose STANCE, a unified statistical model developed to detect both SVG and ctSVGs in spatial transcriptomics. By integrating gene expression, spatial location, and cell type composition through a linear mixed-effect model, STANCE enables the identification of both SVGs and ctSVGs in an initial stage, followed by a second stage test dedicated to ctSVG detection. Its design ensures robustness in complex scenarios and the results are spatial rotation invariant. We demonstrated the performance of STANCE through comprehensive simulations and analyses of three public datasets. The downstream analyses based on ctSVGs detected by STANCE suggest promising future applications of the model in spatial transcriptomics and various areas of genome biology. A software implementation of STANCE is available at https://github.com/Cui-STT-Lab/STANCE.</p> | https://github.com/Cui-STT-Lab/STANCE | 1\n",
      "161 | OPLS-based Multiclass Classification and Data-Driven Inter-Class Relationship Discovery | https://www.biorxiv.org/content/10.1101/2024.09.23.614438v1 | 10.1101/2024.09.23.614438 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.23.614438v1.full.pdf | <p>Multiclass datasets and large-scale studies are increasingly common in omics sciences, drug discovery, and clinical research due to advancements in analytical platforms. Efficiently handling these datasets and discerning subtle differences across multiple classes remains a significant challenge. \n",
      "In metabolomics, two-class OPLS-DA (Orthogonal Projection to Latent Structures Discriminant Analysis) models are widely used due to their strong discrimination capabilities and ability to provide interpretable information on class differences. However, these models face challenges in multiclass settings. A common solution is to transform the multiclass comparison into multiple two-class comparisons, which, while more effective than a global multiclass OPLS-DA model, unfortunately results in a manual, time-consuming model-building process with complicated interpretation.\n",
      "Here, we introduce an extension of OPLS-DA for data-driven multiclass classification: Orthogonal Partial Least Squares-Hierarchical Discriminant Analysis (OPLS-HDA). OPLS-HDA integrates Hierarchical Cluster Analysis (HCA) with the OPLS-DA framework to create a decision tree, addressing multiclass classification challenges and providing intuitive visualization of inter-class relationships. To avoid overfitting and ensure reliable predictions, we use cross-validation during model building. Benchmark results show that OPLS-HDA performs competitively across diverse datasets compared to eight established methods. \n",
      "This method represents a significant advancement, offering a powerful tool to dissect complex multiclass datasets. With its versatility, interpretability, and ease of use, OPLS-HDA is an efficient approach to multiclass data analysis applicable across various fields.</p> | None | 0\n",
      "162 | AI-Augmented R-Group Exploration in Medicinal Chemistry | https://www.biorxiv.org/content/10.1101/2024.09.23.614417v1 | 10.1101/2024.09.23.614417 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.23.614417v1.full.pdf | <p>Efficient R-group exploration in the vast chemical space, enabled by increasingly available building blocks or generative AI, remains an open challenge. Here, we developed an enhanced Free-Wilson QSAR model embedding R-groups by atom-centric pharmacophoric features. Regioisomers of R-groups can be distinguished by explicitly accounting for the atomic positions. Good predictivity is observed consistently across 12 public datasets. Integrated into an open-source program, we showcase its application in performing classic Free-Wilson analysis as well as R-group exploration in uncharted chemical space.</p> | https://github.com/WIMNZhao/AIR | 1\n",
      "163 | Beyond Static Brain Atlases: AI-Powered Open Databasing and Dynamic Mining of Brain-Wide Neuron Morphometry | https://www.biorxiv.org/content/10.1101/2024.09.22.614319v1 | 10.1101/2024.09.22.614319 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.22.614319v1.full.pdf | <p>We introduce NeuroXiv (neuroxiv.org), a large-scale, AI-powered database that provides detailed 3D morphologies of individual neurons mapped to a standard brain atlas, designed to support a wide array of dynamic, interactive neuroscience applications. NeuroXiv offers a comprehensive collection of 175,149 atlas-oriented reconstructed morphologies of individual neurons derived from more than 518 mouse brains, classified into 292 distinct types and mapped into the Common Coordinate Framework Version 3 (CCFv3). Different from conventional static brain atlases that are often limited to data-browsing, NeuroXiv allows interactive analyses as well as uploading and databasing custom neuron morphologies, which are mapped to the brain atlas for objective comparisons. Powered by a cutting-edge AI engine (AIPOM), NeuroXiv enables dynamic, user-specific analysis and data mining. We specifically developed a mixture-of-experts algorithm to harness the capabilities of multiple large language models. We also developed a client program to achieve more than 10 times better performance compared to a typical server-side setup. We demonstrate NeuroXiv9s scalability, efficiency, flexibility, openness, and robustness through various applications.</p> | https://github.com/Vaa3D/vaa3d_tools/tree/master/hackathon/shengdian/NeuroMorphoLib, https://github.com/Vaa3D, https://github.com/SEU-ALLEN-codebase/NeuroXiv, https://github.com/Vaa3D/vaa3d_tools/tree/master/hackathon/mBrainAligner | 1\n",
      "164 | AlphaMut: a deep reinforcement learning model to suggest helix-disrupting mutations | https://www.biorxiv.org/content/10.1101/2024.09.21.614241v1 | 10.1101/2024.09.21.614241 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.21.614241v1.full.pdf | <p>Helices are important secondary structural motifs within proteins and are pivotal in numerous physiological processes. While amino acids (AA) such as alanine and leucine are known to promote helix formation, proline and glycine disfavor it. Helical structure formation, however, also depends on its environment, and hence, prior prediction of a mutational effect on a helical structure is difficult. Here, we employ a reinforcement learning algorithm to develop a predictive model for helix-disrupting mutations. We start with a toy model consisting of helices with only 30 AA and train different models. Our results show that only a few mutations lead to a drastic disruption of the target helix. We further extend our approach to helices in proteins and validate the results using rigorous free energy calculations. Our strategy identifies amino acids crucial for maintaining structural integrity and predicts key mutations that could alter protein function. Through our work, we present a new use case for reinforcement learning in protein structure disruption.</p> | None | 0\n",
      "165 | Accurate non-invasive quantification of astaxanthin content using hyperspectral images and machine learning | https://www.biorxiv.org/content/10.1101/2024.09.23.614444v1 | 10.1101/2024.09.23.614444 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.23.614444v1.full.pdf | <p>Commercial cultivation of the microalgae Haematococcus pluvialis to produce natural astaxanthin has gained significant traction due to the high antioxidant capacity of this pigment and its application in foods, feed, cosmetics and nutraceuticals. However, monitoring of astaxanthin content in cultures remains challenging and relies on invasive, time consuming and expensive approaches. In this study, we employed reflectance hyperspectral imaging (HSI) of H. pluvialis suspensions within the visible spectrum, combined with a 1-dimensional convolutional neural network (CNN) to predict the astaxanthin content (ug mg-1) as quantified by high-performance liquid chromatography (HPLC). This approach had low average prediction error (5.9%) across a gradient of astaxanthin contents and was only unreliable at very low contents (&lt;0.6 micrograms mg-1). In addition, our machine learning model outperformed single or dual wavelength linear regression models even when the spectral data was obtained with a spectrophotometer coupled with an integrating sphere. Overall, this study proposes the use of HSI in combination with a CNN for precise non-invasive quantification of astaxanthin in cell suspensions.</p> | None | 0\n",
      "166 | Building a literature knowledge base towards transparent biomedical AI | https://www.biorxiv.org/content/10.1101/2024.09.22.614323v1 | 10.1101/2024.09.22.614323 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.22.614323v1.full.pdf | <p>Knowledge graphs have recently emerged as a powerful data structure to organize biomedical knowledge with explicit representation of nodes and edges. The knowledge representation is in a machine-learning ready format and supports explainable AI models. However, PubMed, the largest and richest biomedical knowledge repository, exists as free text, limiting its utility for advanced machine learning tasks. To address the limitation, we present LiteralGraph, a computational framework that rigorously extracts biomedical terms and relationships from PubMed literature. Using this framework, we established Genomic Literature Knowledge Base (GLKB), a knowledge graph that consolidates 263,714,413 biomedical terms, 14,634,427 biomedical relationships, and 10,667,370 genomic events from 33 million PubMed abstracts and nine well-established biomedical repositories. The database is coupled with RESTful APIs and a user-friendly web interface that make it accessible to researchers for various usages, including machine learning using the semantic knowledge in PubMed articles, reducing hallucination of large language models (LLM), and helping experimental scientists explore their data using vast PubMed evidence.</p> | None | 0\n",
      "167 | AsaruSim: a single-cell and spatial RNA-Seq Nanopore long-reads simulation workflow | https://www.biorxiv.org/content/10.1101/2024.09.20.613625v1 | 10.1101/2024.09.20.613625 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.20.613625v1.full.pdf | <p>Motivation: The combination of long-read sequencing technologies like Oxford Nanopore with single-cell RNA sequencing (scRNAseq) assays enables the detailed exploration of transcriptomic complexity, including isoform detection and quantification, by capturing full-length cDNAs. However, challenges remain, including the lack of advanced simulation tools that can effectively mimic the unique complexities of scRNAseq long-read datasets. Such tools are essential for the evaluation and optimization of isoform detection methods dedicated to single-cell long readstudies. Results: We developed AsaruSim, a workflow that simulates synthetic single-cell long-read Nanopore datasets, closely mimicking real experimental data. AsaruSim employs a multi-step process that includes the creation of a synthetic UMI count matrix, generation of perfect reads, optional PCR amplification, introduction of sequencing errors, and comprehensive quality control reporting. Applied to a dataset of human peripheral blood mononuclear cells (PBMCs), AsaruSim accurately reproduced experimental read characteristics. Availability and implementation: The source code and full documentation are available at: https://github.com/GenomiqueENS/AsaruSim. Data availability: The 1,090 Human PBMCs count matrix and cell type annotation files are accessible on zenodo under DOI: 10.5281/zenodo.12731408.</p> | https://github.com/GenomiqueENS/AsaruSim, https://github.com/youyupei/SLSim | 1\n",
      "168 | Identification and study of Prolyl Oligopeptidases and related sequences in bacterial lineages | https://www.biorxiv.org/content/10.1101/2024.09.22.614393v1 | 10.1101/2024.09.22.614393 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.22.614393v1.full.pdf | <p>Proteases are enzymes that break down proteins, and serine proteases are an important subset of these enzymes. Prolyl oligopeptidase (POP) is a family of serine proteases (S9 family) that has the ability to cleave peptide bonds involving proline residues and it is unique for its ability to cleave various small oligopeptides shorter than 30 amino acids. The S9 family from the MEROPS database, is classified into four subfamilies based on active site motifs. These S9 subfamilies assume a crucial position owing to their diverse biological roles and potential therapeutic applications in various diseases. In this study, we have examined ~32000 completely annotated bacterial genomes from the NCBI RefSeq Assembly database to identify annotated S9 family proteins. This results in the discovery of ~53,000 bacterial S9 family proteins (referred to as POP homologues). These sequences are classified into distinct subfamilies through various machine-learning approaches and comprehensive analysis of their distribution across various phyla and species and domain architecture analysis are also conducted. Distinct subclusters and class-specific motifs of POPs were identified, suggesting differences in substrate specificity in POP homologues. This study can enable future research of these gene families that are involved in many important biological processes.</p> | None | 0\n",
      "169 | Improving the accuracy of pose prediction by incorporating symmetry-related molecules | https://www.biorxiv.org/content/10.1101/2024.09.21.614298v1 | 10.1101/2024.09.21.614298 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.21.614298v1.full.pdf | <p>Accurate prediction of biologically relevant binding poses is crucial for the success of computer-aided drug development. In this study, we describe a general strategy to enhance the precision of pose prediction in molecular docking by incorporating symmetry-related molecules (SRMs). Our objective was to demonstrate the significant impact of SRMs on the accuracy of pose prediction. To achieve this, we evaluated our method on high-quality protein-ligand complex structures, focusing on the presence and absence of SRMs during molecular docking studies. We have extracted the co-crystal ligands from the selected crystal structure and were redocked in presence and absence of SRM to assess their influence. Additionally, we calculated the free energy of the docked poses using the Molecular Mechanics Generalized Born Surface Area (MM-GBSA) method, comparing the results in the presence and absence of SRMs. The findings revealed that redocking performed in the presence of SRMs significantly improved the prediction of biologically significant/crystallographically relevant poses. Consequently, our proposed strategy offers a robust method for enhancing pose prediction in current molecular docking programs, potentially leading to more effective and reliable drug development processes.</p> | None | 0\n",
      "170 | Identification and Diagnostic Potential of Pyroptosis-Related Genes in Endometriosis: A Novel Bioinformatics Analysis | https://www.biorxiv.org/content/10.1101/2024.09.23.614461v1 | 10.1101/2024.09.23.614461 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.23.614461v1.full.pdf | <p>Objective: This study aimed to identify and analyze potential signatures of pyroptosis-related genes in EMs. \n",
      "Methods: Transcriptomic datasets related to endometriosis were retrieved from the GEO databases (GSE7305, GSE7307, and GSE11691). Differential gene expression analysis was performed to identify pyroptosis-related differentially expressed genes (PRDEGs) by intersecting DEGs with a curated list of PRGs. Various bioinformatics tools were employed to explore the biological functions and pathways associated with PRDEGs. \n",
      "Results: We identified 26 PRDEGs from combined datasets and constructed an EMs diagnostic model using LASSO regression based on pyroptosis scores. The model included 5 DEGs: KIF13B, BAG6, MYO5A, HEATR, and AK055981. Additionally, 21 Key Module Genes (KMGs) were identified, leading to the classification of 3 distinct EMs subtypes. These subtypes were analyzed for immune cell infiltration, revealing a complex immune landscape in EMs.\n",
      "Conclusions: This study reveals pyroptosis9 crucial role in EMs and offers a novel diagnostic model based on pyroptosis-related genes. Modulating pyroptosis may provide a new therapeutic approach for managing EMs.</p> | None | 0\n",
      "171 | Identification of potential inhibitors against Inosine 5’-Monophosphate Dehydrogenase of Cryptosporidium parvum through an integrated in silico approach | https://www.biorxiv.org/content/10.1101/2024.09.22.614371v1 | 10.1101/2024.09.22.614371 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.22.614371v1.full.pdf | <p>The protozoan parasite Cryptosporidium, found in many vertebrate species, including humans, is the source of the global infection known as cryptosporidiosis, which manifests as acute gastroenteritis, abdominal pain, and diarrhea. Although infections in certain individuals have been linked to other species, Cryptosporidium parvum is the main cause of illnesses in humans. Lactate Dehydrogenase, Inosine 59-Monophosphate Dehydrogenase (IMPDH), and several other targets have been identified by the genome sequencing of C. parvum. Bioactive phytochemicals derived from nature have enormous potential as anti-cryptosporidiosis agents. The study aimed to identify new anti-cryptosporidial agents that work against the IMPDH of the parasite by using integrated in silico approaches. In this study, a total of 24 bioactive phytochemicals were screened virtually through molecular docking and ADMET (Absorption, Distribution, Metabolism, Excretion, and Toxicity) analyses. Four lead compounds were identified, including Brevelin A (-8.9 kcal/mol), Vernodalin (-8.7 kcal/mol), Luteolin (-8.6 kcal/mol), and Pectolinarigenin (-8.1 kcal/mol), against the IMPDH protein (PDB ID: 4IXH) from the parasite. All the lead compounds were found to possess favorable pharmacokinetic and pharmacodynamic properties. The toxicity analysis showed satisfactory results with no major side effects. All of the selected compounds showed no violation of Lipinski9s rules of five, indicating the possibility of oral bioavailability as potential drug candidates. Target class prediction unveiled enzymes in most of the cases, and some experimental and investigational drugs were found to be structurally similar to the lead compounds. With significant biochemical interactions, all of the targeted phytochemical compounds have demonstrated excellent pharmacokinetics and good bioavailabilities. The findings strongly recommend in vitro experimental studies to aid in the development of novel therapeutics against Cryptosporidium parvum.</p> | None | 0\n",
      "172 | Designing of thermostable proteins with a desired melting temperature | https://www.biorxiv.org/content/10.1101/2024.09.21.614294v1 | 10.1101/2024.09.21.614294 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.21.614294v1.full.pdf | <p>The stability of proteins at higher temperatures is crucial for its functionality that is measured by their melting temperature (Tm).  The Tm is the temperature at which 50% of the protein loses its native structure and activity. Existing methods for predicting Tm have two major limitations: first, they are often trained on redundant proteins, and second, they do not allow users to design proteins with the desired Tm. To address these limitations, we developed a regression method for predicting the Tm value of proteins using 17,312 non-redundant proteins, where no two proteins are more than 40% similar. We used 80% of the data for training and testing; remaining 20% of the data for validation. Initially, we developed a machine learning model using standard features from protein sequences. Our best model, developed using Shannon entropy for all residues, achieved the highest Pearson correlation of 0.80 with an R2 of 0.63 between the predicted and actual Tm of proteins on the validation dataset. Next, we fine-tuned large language models (e.g., ProtBert, ProtGPT2, ProtT5) on our training dataset and generated embeddings. These embeddings have been used for developing machine learning models. Our best model, developed using ProtBert embeddings, achieved a maximum correlation of 0.89 with an R2 of 0.80 on the validation dataset. Finally, we developed an ensemble method that combines standard protein features and embeddings. One of the aims of the study is to assist the scientific community in the design of targeted melting temperatures. We created a user-friendly web server and a python package for predicting and designing thermostable proteins. Our standalone software can be used to screen thermostable proteins in genomes and metagenomes. We demonstrated the application of PPTstab in identifying thermostable proteins in different organisms from their genomes, the model and data is available at: https://webs.iiitd.edu.in/raghava/pptstab.</p> | None | 0\n",
      "173 | Exploring midgut expression dynamics: longitudinal transcriptomic analysis of adult female Amblyomma americanum midgut and comparative insights with other hard tick species | https://www.biorxiv.org/content/10.1101/2024.09.20.614175v1 | 10.1101/2024.09.20.614175 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.20.614175v1.full.pdf | <p>Background: Female ticks remain attached to their host for multiple days to complete a blood meal. This prolonged feeding period is accompanied by a significant increase in the tick size and body weight, paralleled by noteworthy changes to the tick midgut. While the midgut is recognized for its established role in blood storage and processing, its importance extends to playing a crucial role in the acquisition, survival, and proliferation of pathogens. Despite this, our overall understanding of tick midgut biology is limited. \n",
      "Results: We conducted a comprehensive longitudinal transcriptome analysis of the midgut in adult female A. americanum ticks across various feeding stages, including unfed, slow-feeding, and rapid-feeding phases. Our analysis revealed 15,599 putative DNA coding sequences (CDS) classified within 26 functional groups. Dimensional and differential expression analysis highlighted the dynamic transcriptional changes in the tick midgut as feeding progresses, particularly during the initial period of feeding and the transition from the slow-feeding to the rapid-feeding phase. Additionally, we performed an orthology analysis comparing our dataset with midgut transcriptomes from other hard ticks, such as Ixodes scapularis and Rhipicephalus microplus. This comparison allowed us to identify transcripts commonly expressed during different feeding phases across these three species. \n",
      "Conclusion: Our findings provide a detailed temporal resolution of numerous metabolic pathways in A. americanum, emphasizing the dynamic transcriptional changes occurring in the tick midgut throughout the feeding process. Furthermore, we identified conserved transcripts across three different tick species that exhibit similar expression patterns. This knowledge has significant implications for future research aimed at deciphering the physiological pathways relevant within the tick midgut. It also offers potential avenues for developing control methods that target multiple tick species.</p> | https://github.com/FelixKrueger/TrimGalore | 1\n",
      "174 | Decoding Protein Dynamics: ProFlex as a Linguistic Bridge in Normal Mode Analysis | https://www.biorxiv.org/content/10.1101/2024.09.21.614246v1 | 10.1101/2024.09.21.614246 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.21.614246v1.full.pdf | <p>Artificial intelligence has revolutionized structural bioinformatics, with AlphaFold being arguably the most impactful development to date. The structural atlases generated by these methods present significant opportunities for unraveling biological mysteries, but also pose challenges in leveraging such massive datasets effectively. In this work, we explore the dynamic landscape of hundreds of thousands of AlphaFold-predicted structures using normal mode analysis. The resulting data is used to define an alphabet summarizing relative protein flexibility, termed ProFlex. We believe that refining and further applying ProFlex-like approaches offers novel opportunities for understanding protein function and enhancing other methods.</p> | https://github.com/DamianJM/proFlex | 1\n",
      "175 | Metabolic modeling identifies determinants of thermal growth responses in Arabidopsis thaliana | https://www.biorxiv.org/content/10.1101/2024.09.20.614037v1 | 10.1101/2024.09.20.614037 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.20.614037v1.full.pdf | <p>Temperature is a critical environmental factor affecting nearly all plant processes, including growth, development, and yield. Yet, despite decades of research, we lack the ability to predict plant performance at different temperatures, limiting the development of climate-resilient crops. Further, there is a pressing need to bridge the gap between the prediction of physiological and molecular traits to improve our understanding and manipulation of plant temperature responses. Here, we developed the first enzyme-constrained model of <i>Arabidopsis thaliana</i>′s metabolism, facilitating predictions of growth-related phenotypes at different temperatures. We showed that the model can be employed for <i>in silico</i> identification of genes that affect plant growth at suboptimal growth temperature. Using mutant lines, we validated the genes predicted to affect plant growth, demonstrating the potential of metabolic modeling in accurately predicting plant thermal responses. The temperature-dependent enzyme-constrained metabolic model provides a template that can be used for developing sophisticated strategies to engineer climate-resilient crops.</p> | https://github.com/pwendering/ArabidopsisCoreModel, https://github.com/pwendering/topt-predict | 1\n",
      "176 | Learning-Augmented Sketching Offers Improved Performance for Privacy Preserving and Secure GWAS | https://www.biorxiv.org/content/10.1101/2024.09.19.613975v1 | 10.1101/2024.09.19.613975 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.19.613975v1.full.pdf | <p>The introduction of trusted execution environments (TEEs), such as secure enclaves provided by the Intel SGX technology has enabled secure and privacy-preserving computation on the cloud. The stringent resource limitations, such as memory constraints, required by some TEEs necessitates the development of computational approaches with reduced memory usage, such as sketching. One example is the SkSES method for GWAS on a cohort of case and control samples from multiple institutions, which identifies the most significant SNPs in a privacy-preserving manner without disclosing sensitive genotype information to other institutions or the cloud service provider. Here we show how to improve the performance of SkSES on large datasets by augmenting it with a learning-augmented approach. Specifically, we show how individual institutions can perform smaller scale GWAS on their own datasets and identify two sets of variants according to certain criteria, which are then used to guide the sketching process to more accurately identify significant variants over the collective dataset. The new method achieves up to 40% accuracy gain compared to the original SkSES method under the same memory constraints on datasets we tested on. The code is available at https://github.com/alreadydone/sgx-genome-variants-search.</p> | https://github.com/odelaneau/shapeit4/tree/master/ | 1\n",
      "177 | Bacopa monnieri phytochemicals as promising BACE1 inhibitors for Alzheimer’s Disease Therapy | https://www.biorxiv.org/content/10.1101/2024.09.19.613992v1 | 10.1101/2024.09.19.613992 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.19.613992v1.full.pdf | <p>Alzheimers disease (AD) remains a formidable challenge, necessitating the discovery of effective therapeutic agents targeting β-site amyloid precursor protein cleaving enzyme 1 (BACE1). This study investigates the inhibitory potential of phytochemicals derived from Bacopa monnieri, a plant renowned for its cognitive-enhancing properties, in comparison to established synthetic inhibitors such as Atabecestat, Lanabecestat, and Verubecestat. Utilizing molecular docking and advanced computational simulations, we demonstrate that Bacopaside I exhibits superior binding affinity and a unique interaction profile with BACE1, suggesting a more nuanced inhibitory mechanism. Our findings highlight the promising role of Bacopa monnieri phytochemicals as viable alternatives to synthetic drugs, emphasizing their potential to overcome limitations faced in clinical settings. Furthermore, the development of the SIMANA (https://simana.streamlit.app/) platform enhances the visualization and analysis of\n",
      "protein-ligand interactions, facilitating a deeper understanding of the dynamics involved. This research not only underscores the therapeutic promise of natural compounds in AD treatment but also advocates for a paradigm shift towards integrating traditional medicinal knowledge into contemporary drug discovery efforts.</p> | None | 0\n",
      "178 | Liberality is More Explainable than PCA of Transcriptome for Vertebrate Embryo Development | https://www.biorxiv.org/content/10.1101/2024.09.19.613970v1 | 10.1101/2024.09.19.613970 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.19.613970v1.full.pdf | <p>Dimensional reduction approaches of genome scale data are widely used. Liberality is one of the approaches which is a quantitative index of cellular differentiation and dedifferentiation. Here, we analyzed a trend of liberality of time course transcriptome data on vertebrate embryo development. Historically annotated embryo developmental stages matched changes in the trend of liberality. Analyzing liberality of various biological phenomena would be beneficial.</p> | None | 0\n",
      "179 | Differential quantification of alternative splicing events on spliced pangenome graphs | https://www.biorxiv.org/content/10.1101/2023.11.06.565751v2 | 10.1101/2023.11.06.565751 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.11.06.565751v2.full.pdf | <p>Pangenomes are becoming a powerful framework to perform many bioinformatics analyses taking into account the genetic variability of a population, thus reducing the bias introduced by a single reference genome. With the wider diffusion of pangenomes, integrating genetic variability with transcriptome diversity is becoming a natural extension that demands specific methods for its exploration. In this work, we extend the notion of spliced pangenomes to that of annotated spliced pangenomes; this allows us to introduce a formal definition of Alternative Splicing (AS) events on a graph structure. To investigate the usage of graph pangenomes for the quantification of AS events across conditions, we developed pantas, the first pangenomic method for the detection and differential analysis of AS events from short RNA-Seq reads. A comparison with state-of-the-art linear reference-based approaches proves that pantas achieves competitive accuracy, making spliced pangenomes effective for conducting AS events quantification and opening future directions for the analysis of population-based transcriptomes. pantas is open-source and freely available at https://github.com/algolab/pantas</p> | https://github.com/algolab/pantas?tab=readme-ov-file#experiments | 1\n",
      "180 | Pangenome-Informed Language Models for Privacy-Preserving Synthetic Genome Sequence Generation | https://www.biorxiv.org/content/10.1101/2024.09.18.612131v2 | 10.1101/2024.09.18.612131 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.18.612131v2.full.pdf | <p>The public availability of genome datasets, such as The Human Genome Project (HGP), The 1000 Genomes Project, The Cancer Genome Atlas, and the International HapMap Project, has significantly advanced scientific research and medical understanding. Here our goal is to share such genomic information for downstream analysis while protecting the privacy of individuals through Differential Privacy (DP). We introduce synthetic DNA data generation based on pangenomes in combination with Pretrained-Language Models (PTLMs). We introduce two novel tokenization schemes based on pangenome graphs to enhance the modeling of DNA. We evaluated these tokenization methods, and compared them with classical single nucleotide and <i>k</i>-mer tokenizations. We find <i>k</i>-mer tokenization schemes, indicating that our tokenization schemes boost the model9s performance consistency with long effective context length (covering longer sequences with the same number of tokens). Additionally, we propose a method to utilize the pangenome graph and make that comply with DP privacy standards. We assess the performance of DP training on the quality of generated sequences with discussion of the trade-offs between privacy and model accuracy. The source code for our work will be published under a free and open source license soon.</p> | None | 0\n",
      "181 | Accurate Multiple Sequence Alignment of Ultramassive Genome Sets | https://www.biorxiv.org/content/10.1101/2024.09.22.613454v1 | 10.1101/2024.09.22.613454 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.22.613454v1.full.pdf | <p>With ever increasing sequencing efficiency, there is a pressing need to tackle a presently intractable task of accurate multiple sequence alignment (MSA) for ultramassive genome sets of millions and beyond. Additionally, efficient graph and probabilistic representations for downstream analysis are in dire lack. In light of these challenges, we develop a set of linearly scalable algorithms, including that for constructing directed acyclic graphs, for training profile hidden Markov models and for conducting alignment on such graphs. The astounding power of these algorithms is demonstrated by both significantly improved accuracy and tremendous acceleration of SAR-CoV-2 MSA by three observed to five projected orders of magnitude for genome set sizes ranging from 40,000 to more than 4 million when compared with widely utilized MAFFT. Future application to other viral species and extension to more complex genomes will prove this algorithm set as a cornerstone for the coming era of ultramassive genome sets.</p> | None | 0\n",
      "182 | Machine Learning Driven Simulations of SARS-CoV-2 Fitness Landscape | https://www.biorxiv.org/content/10.1101/2024.09.20.614179v1 | 10.1101/2024.09.20.614179 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.20.614179v1.full.pdf | <p>SARS-CoV-2 infection is mediated by interactions between the receptor binding domain (RBD) of viral spike proteins and host cell angiotensin converting enzyme 2 (ACE2) receptors. Mutations in the spike protein are the primary cause for neutralizing antibody escape leading to breakthrough infections. We characterize the fitness landscape underpinning future variants of concern by combining supervised machine learning and Markov Chain Monte Carlo. Leveraging deep mutational scanning (DMS) data characterizing the binding affinity between RBD mutants to the ACE2 receptor, we predict variants of concern not seen in the training data and sample statistics of the fitness landscape. These simulations provide insight into the relationship between RBD sequence elements and offer a new perspective on utilizing DMS to predict emerging viral strains.</p> | None | 0\n",
      "183 | Improved permutation tests enable detection of dependence between nonstationary time series despite limited numbers of replicates | https://www.biorxiv.org/content/10.1101/2023.03.13.531689v3 | 10.1101/2023.03.13.531689 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.03.13.531689v3.full.pdf | <p>In disciplines from ecology to neuroscience, researchers analyze correlations between pairs of nonstationary time series to infer relationships among variables. This often involves a statistical test to determine whether an observed correlation is stronger than expected under the null hypothesis of independence. Testing for dependence between nonstationary time series with only one experimental replicate is exceedingly challenging. However, with many replicates, a nonparametric trial-swapping permutation test can be employed, comparing within-replicate correlations to between-replicate correlations. Although largely assumption-free, this test is severely limited by the number of replicates because its minimum achievable p-value is 1/n! where n is the number of replicates. This curtails its applicability to many biomedical studies, where n is frequently as low as 3, which would render significance thresholds like 0.05 unattainable. To address this, we propose modified permutation tests that can report lower p-values of 2/n<sup>n</sup> or 1/n<sup>n</sup> when there is strong evidence of dependence. We prove that the tests guarantee a false positive rate at or below the significance level, as long as replicates come from independent and identical experiments. We demonstrate this approach by confirming the observation that groups of zebrafish swim faster when directionally aligned, using an existing dataset with 3 biological replicates.</p> | None | 0\n",
      "184 | Integrative modeling in the age of machine learning: a summary of HADDOCK strategies in CAPRI rounds 47-55 | https://www.biorxiv.org/content/10.1101/2024.09.16.613212v2 | 10.1101/2024.09.16.613212 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.16.613212v2.full.pdf | <p>The HADDOCK team participated in CAPRI rounds 47-55 as both server, manual predictor, and scorers. Throughout these CAPRI rounds, we used a plethora of computational strategies to predict the structure of protein complexes. Of the 10 targets comprising 24 interfaces, we achieved acceptable or better models for 3 targets in the human category and 1 in the server category. Our performance in the scoring challenge was slightly better, with our simple scoring protocol being the only one capable of identifying an acceptable model for Target 234. This result highlights the robustness of the simple, fully physics-based HADDOCK scoring function, especially when applied to highly flexible antibody-antigen complexes. Inspired by the significant advances in machine learning for structural biology and the dramatic improvement in our success rates after the public release of Alphafold2, we identify the integration of classical approaches like HADDOCK with AI-driven structure prediction methods as a key strategy for improving the accuracy of model generation and scoring.</p> | None | 0\n",
      "185 | METASPACE-ML: Context-specific metabolite annotation for imaging mass spectrometry using machine learning | https://www.biorxiv.org/content/10.1101/2023.05.29.542736v3 | 10.1101/2023.05.29.542736 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.05.29.542736v3.full.pdf | <p>Imaging mass spectrometry is a powerful technology enabling spatial metabolomics, yet metabolites can be assigned only to a fraction of the data generated. METASPACE-ML is a machine learning-based approach addressing this challenge which incorporates new scores and computationally-efficient False Discovery Rate estimation. For training and evaluation, we use a comprehensive set of 1,710 datasets from 159 researchers from 47 labs encompassing both animal and plant-based datasets representing multiple spatial metabolomics contexts derived from the METASPACE knowledge base. Here we show that, METASPACE-ML outperforms its rule-based predecessor, exhibiting higher precision, increased throughput, and enhanced capability in identifying low-intensity and biologically-relevant metabolites.</p> | https://github.com/DinosaurInSpace/core_metabolome, https://github.com/metaspace2020/metaspace, https://github.com/Bisho2122/METASPACE-ML_reproducibility | 1\n",
      "186 | Epiregulon: Inference of single-cell transcription factor activity to predict drug response and drivers of cell state | https://www.biorxiv.org/content/10.1101/2023.11.27.568955v3 | 10.1101/2023.11.27.568955 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.11.27.568955v3.full.pdf | <p>Transcription factors (TFs) and transcriptional coregulators represent an emerging class of therapeutic targets in oncology. Gene regulatory networks (GRNs) can be used to evaluate pharmacological agents targeting these factors and to identify drivers of disease and drug resistance. However, GRN methods that rely solely on gene expression often fail to account for post-transcriptional modulation of TF function. We present Epiregulon, a method that constructs GRNs from single-cell ATAC-seq and RNA-seq data for accurate prediction of TF activity. This is achieved by considering the co-occurrence of TF expression and chromatin accessibility at TF binding sites in each cell. We leverage ChIP-seq data to extend inference to transcriptional coregulators lacking defined motifs or TF harboring neomorphic mutations. Epiregulon accurately predicted the effects of AR inhibition across various drug modalities including an AR antagonist and an AR degrader, delineated the mechanisms of a SMARCA4 degrader by identifying context-dependent interaction partners and prioritized known and novel drivers of lineage reprogramming and tumorigenesis. By mapping gene regulation across various cellular contexts, Epiregulon can accelerate the discovery of therapeutics targeting transcriptional regulators.</p> | https://github.com/xiaosaiyao/epiregulon.manuscript, https://github.com/xiaosaiyao/epiregulon, https://github.com/xiaosaiyao/scMultiome, https://github.com/xiaosaiyao/epiregulon.diffnetwork.simulation, https://github.com/xiaosaiyao/epiregulon.archr | 1\n",
      "187 | Prediction and functional interpretation of inter-chromosomal genome architecture from DNA sequence with TwinC | https://www.biorxiv.org/content/10.1101/2024.09.16.613355v2 | 10.1101/2024.09.16.613355 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.16.613355v2.full.pdf | <p>Three-dimensional nuclear DNA architecture comprises well-studied intra-chromosomal (cis) folding and less characterized inter-chromosomal (trans) interfaces. Current predictive models of 3D genome folding can effectively infer pairwise cis-chromatin interactions from the primary DNA sequence but generally ignore trans contacts. There is an unmet need for robust models of trans-genome organization that provide insights into their underlying principles and functional relevance. We present TwinC, an interpretable convolutional neural network model that reliably predicts trans contacts measurable through genome-wide chromatin conformation capture (Hi-C). TwinC uses a paired sequence design from replicate Hi-C experiments to learn single base pair relevance in trans interactions across two stretches of DNA. The method achieves high predictive accuracy (AUROC=0.80) on a cross-chromosomal test set from Hi-C experiments in heart tissue. Mechanistically, the neural network learns the importance of compartments, chromatin accessibility, clustered transcription factor binding and G-quadruplexes in forming trans contacts. In summary, TwinC models and interprets trans genome architecture, shedding light on this poorly understood aspect of gene regulation.</p> | https://github.com/Noble-Lab/twinc | 1\n",
      "188 | Simplifying bioinformatics data analysis through conversation | https://www.biorxiv.org/content/10.1101/2023.10.29.564479v2 | 10.1101/2023.10.29.564479 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.10.29.564479v2.full.pdf | <p>The rapid advancements in high-throughput sequencing technologies have produced a wealth of omics data, facilitating significant biological insights but presenting immense computational challenges. Traditional bioinformatics tools require substantial programming expertise, limiting accessibility for experimental researchers. Despite efforts to develop user-friendly platforms, the complexity of these tools continues to hinder efficient biological data analysis. In this paper, we introduce BioMANIA--an AI-driven, natural language-oriented bioinformatics pipeline that addresses these challenges by enabling the automatic and codeless execution of biological analyses. BioMANIA leverages large language models (LLMs) to interpret user instructions and execute sophisticated bioinformatics workflows, integrating API knowledge from existing Python tools. By streamlining the analysis process, BioMANIA simplifies complex omics data exploration and accelerates bioinformatics research. Compared to relying on general-purpose LLMs to conduct analysis from scratch, BioMANIA, informed by domain-specific biological tools, helps mitigate hallucinations and significantly reduces the likelihood of confusion and errors. Through comprehensive benchmarking and application to diverse biological data, ranging from single-cell omics to electronic health records, we demonstrate BioMANIA9s ability to lower technical barriers, enabling more accurate and comprehensive biological discoveries.</p> | https://github.com/batmen-lab/BioMANIA | 1\n",
      "189 | PhyloFunc: Phylogeny-informed Functional Distance as a New Ecological Metric for Metaproteomic Data Analysis | https://www.biorxiv.org/content/10.1101/2024.05.28.596184v2 | 10.1101/2024.05.28.596184 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.05.28.596184v2.full.pdf | <p>Background: Beta-diversity is a fundamental ecological metric for exploring dissimilarities between microbial communities. On the functional dimension, metaproteomics data can be used to quantify beta-diversity to understand how microbial community functional profiles vary under different environmental conditions. Conventional approaches to metaproteomic functional beta diversity often treat protein functions as independent features, ignoring the evolutionary relationships among microbial taxa from which different proteins originate. A more informative functional distance metric that incorporates evolutionary relatedness is needed to better understand microbiome functional dissimilarities. Results: Here, we introduce PhyloFunc, a novel functional beta-diversity metric that incorporates microbiome phylogeny to inform on metaproteomic functional distance. Leveraging the phylogenetic framework of weighted UniFrac distance, PhyloFunc innovatively utilizes branch lengths to weigh between-sample functional distances for each taxon, rather than differences in taxonomic abundance as in weighted UniFrac. Proof-of-concept using a simulated toy dataset and a real dataset from mouse inoculated with a synthetic gut microbiome and fed different diets show that PhyloFunc successfully captured functional compensatory effects between phylogenetically related taxa. We further tested a third dataset of complex human gut microbiomes treated with five different drugs to compare PhyloFunc9s performance with other traditional distance methods. PCoA and machine learning-based classification algorithms revealed higher sensitivity of PhyloFunc in microbiome responses to paracetamol. We offer PhyloFunc as an open-source Python package (https://pydigger.com/pypi/PhyloFunc) that efficiently calculates functional beta-diversity distances between sample pairs and generates distance matrices for all samples. Conclusions: Unlike traditional approaches that consider metaproteomics features as independent and unrelated, PhyloFunc acknowledges the role of phylogenetic context in shaping the functional landscape in metaproteomes. In particular, we report that PhyloFunc accounts for the functional compensatory effect of taxonomically related species. It is effective, ecologically significant, and has better sensitivity, as evidenced by the particular applications we presented.</p> | https://github.com/lumanottawa/PhyloFunc/tree/main/0_PhyloFunk_package_tutorial | 1\n",
      "190 | Possibilities of multiple disease progression pathways in female Alzheimer’s disease patients inferred from transcriptome and epigenome data of the dorsolateral prefrontal cortex | https://www.biorxiv.org/content/10.1101/2024.03.24.586514v2 | 10.1101/2024.03.24.586514 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.03.24.586514v2.full.pdf | <p>Late-onset Alzheimer9s disease (AD) is a typical type of dementia for which therapeutic strategies have not yet been established. Based on an analysis of publicly available data of women patients, this study found that the pathology of AD was suggested to be nonuniform; however, several different typical forms of cognitive deterioration were observed. First, cluster analysis was performed on individuals diagnosed with \"No Cognitive Impairment (NCI)\", \"Mild Cognitive Impairment (MCI)\", and \"Alzheimer9s Disease (AD)\" stages in clinical trials using gene expression and epigenome data. NCI was found to contain at least two substages, whereas MCI and AD were found to contain four and six substages, respectively. An inference of adjacency networks among substages, evaluated via partition-based graph abstraction using the gene expression profiles of individuals, suggested the possibilities of multiple typical disease progression pathways from NCI through different MCI substages and then to different AD substages. These results will aid future research on the pathological features and mechanisms of AD and induce changes in the strategic bases of conventional AD prevention and treatment.</p> | None | 0\n",
      "191 | mosna reveals different types of cellular interactions predictive of response to immunotherapies and survival in cancer | https://www.biorxiv.org/content/10.1101/2023.03.16.532947v2 | 10.1101/2023.03.16.532947 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.03.16.532947v2.full.pdf | <p>Spatially resolved omics enable the discovery of tissue organization of biological or clinical importance. Despite the existence of several methods, performing a rational analysis including multiple algorithms while integrating different conditions such as clinical data is still not trivial. To make such investigations more accessible, we developed mosna, a Python package to analyze spatial omics data with respect to clinical or biological data and to gain insight on cell interaction patterns or tissue architecture of biological relevance. mosna is compatible with all spatial omics methods, it leverages tysserand to build accurate spatial networks, and is compatible with Squidpy. It proposes an analysis pipeline, in which increasingly complex features computed at each step can be explored in integration with clinical data, either with easy-to-use descriptive statistics and data visualization, or by seamlessly training machine learning models and identifying variables with the most predictive power. mosna can take as input any dataset produced by spatial omics methods, including sub-cellular resolved transcriptomics (MERFISH, seqFISH) and proteomics (CODEX, MIBI-TOF, low-plex immuno-fluorescence), as well as spot-based spatial transcriptomics (10x Visium). Integration with experimental metadata or clinical data is adapted to binary conditions, such as biological treatments or response status of patients, and to survival data. We demonstrate the proposed analysis pipeline on two spatially resolved proteomic datasets containing either binary response to immunotherapy or survival data. mosna identifies features describing cellular composition and spatial distribution that can provide biological insight regarding factors that affect response to immunotherapies or survival.</p> | None | 0\n",
      "192 | Hidden GPCR structural transitions addressed by multiple walker supervised molecular dynamics (mwSuMD) | https://www.biorxiv.org/content/10.1101/2022.10.26.513870v4 | 10.1101/2022.10.26.513870 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2022.10.26.513870v4.full.pdf | <p>The structural basis for the pharmacology of G protein-coupled receptors (GPCRs), the most abundant membrane proteins and the target of about 35% of approved drugs, is still a matter of intense study. What makes GPCRs challenging to study is the inherent flexibility and the metastable nature of interaction with extra- and intracellular partners that drive their effects. Here, we present a molecular dynamics (MD) adaptive sampling algorithm, namely multiple walker supervised molecular dynamics (mwSuMD), to address complex structural transitions involving GPCRs without energy input. We first report the binding and unbinding of the vasopressin peptide from its receptor V2. Successively, we present the complete transition of the glucagon-like peptide-1 receptor (GLP-1R) from inactive to active, agonist and Gs-bound state, and the GDP release from Gs. To our knowledge, this is the first time the whole sequence of events leading from an inactive GPCR to the GDP release is simulated without any energy bias. We demonstrate that mwSuMD can address complex binding processes intrinsically linked to protein dynamics out of reach of classic MD.</p> | https://github.com/pipitoludovico/mwSuMD, https://github.com/pipitoludovico/mwSuMD_multiEngine | 1\n",
      "193 | Microsatellite Density Landscapes Illustrate Short Tandem Repeats Aggregation in The Complete Reference Human Genome | https://www.biorxiv.org/content/10.1101/2022.04.16.487617v3 | 10.1101/2022.04.16.487617 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2022.04.16.487617v3.full.pdf | <p>Background: Microsatellites are increasingly realized to have biological significance in human genome and health in past decades, the assembled complete reference sequence of human genome T2T-CHM13 brought great help for a comprehensive study of short tandem repeats in the human genome. Results: Microsatellites density landscapes of all 24 chromosomes were built here for the first complete reference sequence of human genome T2T-CHM13. These landscapes showed that short tandem repeats (STRs) are prone to aggregate characteristically to form a large number of STRs density peaks. We classified 8,823 High Microsatellites Density Peaks (HMDPs), 35,257 Middle Microsatellites Density Peaks (MMDPs) and 199, 649 Low Microsatellites Density Peaks (LMDPs) on the 24 chromosomes; and also classified the motif types of every microsatellites density peak. These STRs density aggregation peaks are mainly composing of a single motif, and AT is the most dominant motif, followed by AATGG and CCATT motifs. And 514 genomic regions were characterized by microsatellite density feature in the full T2T-CHM13 genome. Conclusions: These landscape maps exhibited that microsatellites aggregate in many genomic positions to form a large number of microsatellite density peaks with composing of mainly single motif type in the complete reference genome, indicating that the local microsatellites density varies enormously along the every chromosome of T2T-CHM13.</p> | https://github.com/zhongyangtan/DCM.git, https://github.com/zhongyangtan/GHC.git, https://github.com/zhongyangtan/MDPS.git | 1\n",
      "194 | Characterising tandem repeat complexities across long-read sequencing platforms with TREAT and otter | https://www.biorxiv.org/content/10.1101/2024.03.15.585288v2 | 10.1101/2024.03.15.585288 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.03.15.585288v2.full.pdf | <p>Tandem repeats (TR) play important roles in genomic variation and disease risk in humans. Long-read sequencing allows for the accurate characterisation of TRs, however, the underlying bioinformatics perspectives remain challenging. We present otter and TREAT: otter is a fast targeted local assembler, cross-compatible across different sequencing platforms. It is integrated in TREAT, an end-to-end workflow for TR characterisation, visualisation and analysis across multiple genomes. In a comparison with existing tools based on long-read sequencing data from both Oxford Nanopore Technology (ONT, Simplex and Duplex) and PacBio (Sequel 2 and Revio), otter and TREAT achieved state-of-the-art genotyping and motif characterisation accuracy. Applied to clinically relevant TRs, TREAT/otter significantly identified individuals with pathogenic TR expansions. When applied to a case-control setting, we significantly replicated previously reported associations of TRs with Alzheimer9s Disease, including those near or within APOC1 (p=2.63x10-9), SPI1 (p=6.5x10-3) and ABCA7 (p=0.04) genes. We finally used TREAT/otter to systematically evaluate potential biases when genotyping TRs using diverse ONT and PacBio long-read sequencing datasets. We showed that, in rare cases (0.06%), long-read sequencing suffers from coverage drops in TRs, including the disease-associated TRs in ABCA7 and RFC1 genes. Such coverage drops can lead to TR mis-genotyping, hampering the accurate characterisation of TR alleles. Taken together, our tools can accurately genotype TR across different sequencing technologies and with minimal requirements, allowing end-to-end analysis and comparisons of TR in human genomes, with broad applications in research and clinical fields.</p> | https://github.com/human-pangenomics/HPP_Year1_Data_Freeze_v1.0?tab=readme-ov-, https://github.com/holstegelab/treat, https://github.com/PacificBiosciences/trgt/tree/main/repeats, https://github.com/lmdu/pytrf, https://github.com/holstegelab/otter, https://github.com/marbl/CHM13 | 1\n",
      "195 | Efficient Generation of Protein Pockets with PocketGen | https://www.biorxiv.org/content/10.1101/2024.02.25.581968v4 | 10.1101/2024.02.25.581968 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.02.25.581968v4.full.pdf | <p>Designing small-molecule-binding proteins, such as enzymes and biosensors, is crucial in protein biology and bioengineering. Generating high-fidelity protein pockets areas where proteins interact with ligand molecules, is challenging due to complex interactions between ligand molecules and proteins, flexibility of ligand molecules and amino acid side chains, and complex sequence-structure dependencies. Here, we introduce PocketGen, a deep generative method for generating the residue sequence and the full-atom structure within the protein pocket region that leverages sequence structure consistency. PocketGen consists of a bilevel graph transformer for structural encoding and a sequence refinement module that uses a protein language model (pLM) for sequence prediction. The bilevel graph transformer captures interactions at multiple granularities (atom-level and residue/ligand-level) and aspects (intra-protein and protein-ligand) with bilevel attention mechanisms. For sequence refinement, a structural adapter using cross-attention is integrated into a pLM to ensure structure-sequence consistency. During training, only the adapter is fine-tuned, while the other layers of the pLM remain unchanged. Experiments show that PocketGen can efficiently generate protein pockets with higher binding affinity and validity than state-of-the-art methods. PocketGen is ten times faster than physics-based methods and achieves a 95% success rate (percentage of generated pockets with higher binding affinity than reference pockets) with over 64% amino acid recovery rate.</p> | https://github.com/RosettaCommons/RFdiffusion, https://github.com/zaixizhang/FAIR, https://github.com/THUNLP-MT/dyMEAN, https://github.com/Hoecker-Lab/pocketoptimizer | 1\n",
      "196 | A general substitution matrix for structural phylogenetics | https://www.biorxiv.org/content/10.1101/2024.09.19.613819v3 | 10.1101/2024.09.19.613819 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.09.19.613819v3.full.pdf | <p>Sequence-based maximum likelihood (ML) phylogenetics is a widely used method for inferring evolutionary relationships, which has illuminated the evolutionary histories of proteins and the organisms that harbour them. But modern implementations with sophisticated models of sequence evolution struggle to resolve deep evolutionary relationships, which can be obscured by excessive sequence divergence and substitution saturation.  Structural phylogenetics has emerged as a promising alternative, because protein structure evolves much more slowly than protein sequences. Recent developments protein in structure prediction using AI have made it possible to predict protein structures for entire protein families, and then to translate these structures into a sequence representation - the 3Di structural alphabet - that can in theory be directly fed into existing sequence based phylogenetic software. To unlock the full potential of this idea, however, requires the inference of a general substitution matrix for structural phylogenetics, which has so far been missing. Here we infer this matrix from large datasets of protein structures and show that it results in a better fit to empirical datasets that previous approaches. We then use this matrix to re-visit the question of the root of the tree of life. Using structural phylogenies of universal paralogs, we provide the first unambiguous evidence for a root between and archaea and bacteria. Finally, we discuss some practical and conceptual limitations of structural phylogenetics. Our 3Di substitution matrix provides a starting point for revisiting many deep phylogenetic problems that have so far been extremely difficult to solve.</p> | https://github.com/mheinzinger/ProstT5, https://github.com/steineggerlab/foldseek | 1\n",
      "197 | PAPET: a collection of performant algorithms to identify 5-methyl cytosine from PacBio SequelII data | https://www.biorxiv.org/content/10.1101/2023.03.17.533149v2 | 10.1101/2023.03.17.533149 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2023.03.17.533149v2.full.pdf | <p>CpGs methylation is an important feature for the regulation of gene expression in vertebreate genomes. In this paper, we present the PAcBio Predicting Epigenetics Toolkit (PAPET) algorithms. PAPET is a collection of general algorithms and tools to train predictive models and predict epigenetics from SequelII data. This set of tools is worth for the PacBio user community to keep up with the fast evolving pace of PacBio sequencing technology. We apply this framework to predict CpG methylation from SequelII data and demonstrate that the classifiers obtained compare equally with their best in class counterparts. PAPET is implemented in C++ to ensure resource efficiency and an easy scalability to large datasets. Moreover, PAPET is fully multi-threaded. The source code is available at https://github.com/ngs-ai-org/papet.</p> | https://github.com/PacificBiosciences/pbmm2, https://github.com/PacificBiosciences/pb-CpG-tools | 1\n",
      "198 | eMOSAIC: Multi-modal Out-of-distribution Uncertainty Quantification Streamlines Large-scale Polypharmacology | https://www.biorxiv.org/content/10.1101/2024.01.05.574359v2 | 10.1101/2024.01.05.574359 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.01.05.574359v2.full.pdf | <p>Polypharmacology has emerged as a new paradigm to discover novel therapeutics for unmet medical needs. Accurate, reliable and scalable predictions of  protein-ligand binding affinity across multiple proteins are essential for polypharmacology. Machine learning is a promising tool for multi-target binding affinity  predictions, often formulated as a multi-modal regression problem. Despite  considerable efforts, three challenges remain: out-of-distribution (OOD) generalizations for compounds with new chemical scaffolds, uncertainty quantification  of OOD predictions, and scalability to billions of compounds, which structure-based methods fail to achieve. To address aforementioned challenges, we propose a new model-agnostic anomaly detection-based uncertainty quantification  method, embedding Mahalanobis Outlier Scoring and Anomaly Identification  via Clustering (eMOSAIC). eMOSAIC uniquely quantifies distribution similarities or differences between the multi-modal representation of known cases and  that of a new unseen one. We apply eMOSAIC to a multi-modal deep neural network model for multi-target ligand binding affinity predictions, leveraging  a pre-trained strucrture-informed large protein language model. We extensively validate eMOSAIC in OOD settings, showing that it significantly outperforms state-of-the-art sequence-based deep learning and structure-based protein-ligand docking (PLD) methods by a large margin as well as existing uncertainty quantification methods. This finding highlights eMOSAIC9s potential for real-world polypharmacology and other applications.</p> | None | 0\n",
      "199 | Discovering nuclear localization signal universe through a novel deep learning model with interpretable attention units | https://www.biorxiv.org/content/10.1101/2024.08.10.606103v2 | 10.1101/2024.08.10.606103 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.08.10.606103v2.full.pdf | <p>Nuclear localization signals (NLSs) are essential peptide fragments within proteins that play a decisive role in guiding proteins into the cell nucleus. Determining the existence and precise locations of NLSs experimentally is time-consuming and complicated, resulting in a scarcity of experimentally validated NLS fragments. Consequently, annotated NLS datasets are relatively limited, presenting challenges for data-driven approaches. In this study, we propose an innovative interpretable approach, NLSExplorer, which leverages large-scale protein language models to capture crucial biological information with a novel attention-based deep network for NLS identification. By enhancing the knowledge retrieved from protein language models with a novel attention to key area module, NLSExplorer achieves superior predictive performance compared to existing methods on two NLS benchmark datasets. Additionally, NLSExplorer is able to detect various kinds of segments highly correlated with nuclear transport, such as nuclear export signals. We employ NLSExplorer to investigate potential NLSs and other domains that are important for nuclear transport in nucleus-localized proteins in the Swiss-Prot database. Furthermore, the comprehensive pattern analysis for all these segments uncovers a potential NLS space and internal relationship of important nuclear transport segments for 416 species. This study not only introduces a powerful tool for predicting and exploring NLS space, but also offers a versatile network that is powerful for detecting characteristic domains and motifs of NLSs.</p> | None | 0\n",
      "200 | Gene Function Revealed at the Moment of Stochastic Gene Silencing | https://www.biorxiv.org/content/10.1101/2024.07.16.603770v2 | 10.1101/2024.07.16.603770 | 2024-01-01 | https://www.biorxiv.org/content/10.1101/2024.07.16.603770v2.full.pdf | <p>Gene expression is a dynamic and stochastic process characterized by transcriptional bursting followed by periods of silence. Single-cell RNA sequencing (scRNA-seq) is a powerful tool to measure transcriptional bursting and silencing at the individual cell level. In this study, we introduce the single-cell Stochastic Gene Silencing (scSGS) method, which leverages the natural variability in single-cell gene expression to decipher gene function. For a target gene g under investigation, scSGS classifies cells into transcriptionally active (g+) and silenced (g-) samples. It then compares these cell samples to identify differentially expressed genes, referred to as SGS-responsive genes, which are used to infer the function of the target gene g. Analysis of real data demonstrates that scSGS can reveal regulatory relationships up- and downstream of target genes, circumventing the survivorship bias that often affects gene knockout and perturbation studies. scSGS thus offers an efficient approach for gene function prediction, with significant potential to reduce the use of genetically modified animals in gene function research.</p> | https://github.com/Xenon8778/scSGS | 1\n"
     ]
    }
   ],
   "source": [
    "import sqlite3\n",
    "\n",
    "# Connexion à la base de données SQLite\n",
    "database = sqlite3.connect('bioinformatics_articles.db')\n",
    "c = database.cursor()\n",
    "\n",
    "# Fonction pour afficher les attributs de la table articles\n",
    "def display_articles():\n",
    "    # Exécuter une requête pour récupérer toutes les colonnes et enregistrements de la table articles\n",
    "    c.execute(\"SELECT * FROM articles\")\n",
    "    \n",
    "    # Récupérer les noms de colonnes\n",
    "    column_names = [description[0] for description in c.description]\n",
    "    \n",
    "    # Afficher les noms des colonnes\n",
    "    print(f\"{' | '.join(column_names)}\")\n",
    "    print(\"-\" * (len(column_names) * 4))  # Ligne de séparation\n",
    "    \n",
    "    # Récupérer et afficher tous les enregistrements\n",
    "    for row in c.fetchall():\n",
    "        print(' | '.join(map(str, row)))\n",
    "\n",
    "# Appeler la fonction pour afficher les articles\n",
    "display_articles()\n",
    "\n",
    "# Fermer la connexion à la base de données\n",
    "database.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# suite du projet  :\n",
    "\n",
    "3.\tArchivage via Software Heritage:\n",
    "\n",
    "Utiliser l’API de Software Heritage pour vérifier si le dépôt logiciel est déjà archivé dans leur base de données. Si le dépôt n’est pas encore archivé, soumettre automatiquement une demande d’archivage via l’API.\n",
    "\n",
    "--->Script 3 : lire la BDD, récupérer les URL, vérifier que le dépôt est archivé dans Software Heritage\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
