{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Installation de bibliotheques \n",
    "\n",
    "**pip install beautifulsoup4**\n",
    "\n",
    "**pip install requests**\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import sqlite3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Scraper les articles bioRxiv liés à la bioinformatique "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = 'https://www.biorxiv.org'\n",
    "\n",
    "# requête récupére la page bioinfo\n",
    "response = requests.get(url + '/collection/bioinformatics')\n",
    "soup = BeautifulSoup(response.content, 'html.parser') #convertit le html et analyse \n",
    "                                                       #.content ->renvoie code htm sous forme de bytes  \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<sqlite3.Cursor at 0x7facb8a11f10>"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# creer base de données SQLite \"bioinformatics_articles\" \n",
    "database = sqlite3.connect('bioinformatics_articles.db')\n",
    "c = database.cursor() #ici le curseur 'c' permettra d'executer/parcourir les requetes SQL sur database\n",
    "\n",
    "# create table 'articles' avec col id,title et le lien \n",
    "c.execute('''CREATE TABLE IF NOT EXISTS articles\n",
    "             (id INTEGER PRIMARY KEY AUTOINCREMENT, title TEXT, link TEXT)''')\n",
    "#id c'est un identifiant unique généré automatiquement "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Requête pour récupérer la page bioinfo\n",
    "response = requests.get(f'{url}/collection/bioinformatics')\n",
    "soup = BeautifulSoup(response.content, 'html.parser')\n",
    "\n",
    "# Trouver les liens vers les articles via beautifulsoup\n",
    "articles = soup.find_all('a', class_='highwire-cite-linked-title')#find liens hypertextes avec classe CSS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Parcourir les articles et enregistrer leurs titres et liens "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Titre: scEMB: Learning context representation of genes based on large-scale single-cell transcriptomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614685v1\n",
      "\n",
      "Titre: Analytical Assessment of Metagenomic Workflows for Pathogen Detection with NIST RM 8376 and Two Sample Matrices\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614717v1\n",
      "\n",
      "Titre: BRDKRM: An Explainable Framework for Disease Modifying Drug Identification\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614653v1\n",
      "\n",
      "Titre: PlantConnectome: knowledge graph encompassing >70,000 plant articles\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.07.11.548541v3\n",
      "\n",
      "Titre: Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2\n",
      "\n",
      "Titre: Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2\n",
      "\n",
      "Titre: varCADD: large sets of standing genetic variation enable genome-wide pathogenicity prediction\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614666v1\n",
      "\n",
      "Titre: Diversity and abundance of ring nucleases in type III CRISPR-Cas loci\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614671v1\n",
      "\n",
      "Titre: Germinated brown rice research: bibliometric analysis of progress, hotspots and trends\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614820v1\n",
      "\n",
      "Titre: p53motifDB: integration of genomic information and tumor suppressor p53 binding motifs\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614594v1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Parcourir tous les articles et enregistrer leurs titres et liens\n",
    "for article in articles:\n",
    "    title = article.text.strip() # extraire le titre dans la balise <a>, accessible via .text.\n",
    "    link = url + article['href'] #extraire le lien , href=destination du lien de mon article\n",
    "    \n",
    "    # Afficher le titre et le lien de chaque article\n",
    "    print(f'Titre: {title}\\nLien: {link}\\n')\n",
    "    \n",
    "    # Insérer dans la base de données SQLite\n",
    "    c.execute(\"INSERT INTO articles (title, link) VALUES (?, ?)\", (title, link))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sauvegarder les modifications dans la base de données\n",
    "conn.commit("
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Affichage de mes articles dans la base de données"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "=== Liste des articles enregistrés ===\n",
      "\n",
      "ID: 1\n",
      "Titre: Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2\n",
      "--------------------------------------------------\n",
      "ID: 2\n",
      "Titre: Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2\n",
      "--------------------------------------------------\n",
      "ID: 3\n",
      "Titre: Clustering-independent estimation of cell abundances in bulk tissues using single-cell RNA-seq data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.02.06.527318v2\n",
      "--------------------------------------------------\n",
      "ID: 4\n",
      "Titre: Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2\n",
      "--------------------------------------------------\n",
      "ID: 5\n",
      "Titre: Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2\n",
      "--------------------------------------------------\n",
      "ID: 6\n",
      "Titre: Clustering-independent estimation of cell abundances in bulk tissues using single-cell RNA-seq data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.02.06.527318v2\n",
      "--------------------------------------------------\n",
      "ID: 7\n",
      "Titre: Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2\n",
      "--------------------------------------------------\n",
      "ID: 8\n",
      "Titre: Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2\n",
      "--------------------------------------------------\n",
      "ID: 9\n",
      "Titre: Clustering-independent estimation of cell abundances in bulk tissues using single-cell RNA-seq data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.02.06.527318v2\n",
      "--------------------------------------------------\n",
      "ID: 10\n",
      "Titre: Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2\n",
      "--------------------------------------------------\n",
      "ID: 11\n",
      "Titre: Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2\n",
      "--------------------------------------------------\n",
      "ID: 12\n",
      "Titre: Clustering-independent estimation of cell abundances in bulk tissues using single-cell RNA-seq data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.02.06.527318v2\n",
      "--------------------------------------------------\n",
      "ID: 13\n",
      "Titre: BRDKRM: An Explainable Framework for Disease Modifying Drug Identification\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614653v1\n",
      "--------------------------------------------------\n",
      "ID: 14\n",
      "Titre: Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2\n",
      "--------------------------------------------------\n",
      "ID: 15\n",
      "Titre: Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2\n",
      "--------------------------------------------------\n",
      "ID: 16\n",
      "Titre: RNA-DCGen: Dual Constrained RNA Sequence Generation with LLM-Attack\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614570v1\n",
      "--------------------------------------------------\n",
      "ID: 17\n",
      "Titre: Clustering-independent estimation of cell abundances in bulk tissues using single-cell RNA-seq data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.02.06.527318v2\n",
      "--------------------------------------------------\n",
      "ID: 18\n",
      "Titre: BRDKRM: An Explainable Framework for Disease Modifying Drug Identification\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614653v1\n",
      "--------------------------------------------------\n",
      "ID: 19\n",
      "Titre: Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2\n",
      "--------------------------------------------------\n",
      "ID: 20\n",
      "Titre: Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2\n",
      "--------------------------------------------------\n",
      "ID: 21\n",
      "Titre: RNA-DCGen: Dual Constrained RNA Sequence Generation with LLM-Attack\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614570v1\n",
      "--------------------------------------------------\n",
      "ID: 22\n",
      "Titre: Clustering-independent estimation of cell abundances in bulk tissues using single-cell RNA-seq data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.02.06.527318v2\n",
      "--------------------------------------------------\n",
      "ID: 23\n",
      "Titre: scEMB: Learning context representation of genes based on large-scale single-cell transcriptomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614685v1\n",
      "--------------------------------------------------\n",
      "ID: 24\n",
      "Titre: Analytical Assessment of Metagenomic Workflows for Pathogen Detection with NIST RM 8376 and Two Sample Matrices\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614717v1\n",
      "--------------------------------------------------\n",
      "ID: 25\n",
      "Titre: BRDKRM: An Explainable Framework for Disease Modifying Drug Identification\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614653v1\n",
      "--------------------------------------------------\n",
      "ID: 26\n",
      "Titre: PlantConnectome: knowledge graph encompassing >70,000 plant articles\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.07.11.548541v3\n",
      "--------------------------------------------------\n",
      "ID: 27\n",
      "Titre: Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2\n",
      "--------------------------------------------------\n",
      "ID: 28\n",
      "Titre: Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2\n",
      "--------------------------------------------------\n",
      "ID: 29\n",
      "Titre: varCADD: large sets of standing genetic variation enable genome-wide pathogenicity prediction\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614666v1\n",
      "--------------------------------------------------\n",
      "ID: 30\n",
      "Titre: Diversity and abundance of ring nucleases in type III CRISPR-Cas loci\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614671v1\n",
      "--------------------------------------------------\n",
      "ID: 31\n",
      "Titre: Germinated brown rice research: bibliometric analysis of progress, hotspots and trends\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614820v1\n",
      "--------------------------------------------------\n",
      "ID: 32\n",
      "Titre: p53motifDB: integration of genomic information and tumor suppressor p53 binding motifs\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614594v1\n",
      "--------------------------------------------------\n",
      "ID: 33\n",
      "Titre: scEMB: Learning context representation of genes based on large-scale single-cell transcriptomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614685v1\n",
      "--------------------------------------------------\n",
      "ID: 34\n",
      "Titre: Analytical Assessment of Metagenomic Workflows for Pathogen Detection with NIST RM 8376 and Two Sample Matrices\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614717v1\n",
      "--------------------------------------------------\n",
      "ID: 35\n",
      "Titre: BRDKRM: An Explainable Framework for Disease Modifying Drug Identification\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614653v1\n",
      "--------------------------------------------------\n",
      "ID: 36\n",
      "Titre: PlantConnectome: knowledge graph encompassing >70,000 plant articles\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.07.11.548541v3\n",
      "--------------------------------------------------\n",
      "ID: 37\n",
      "Titre: Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2\n",
      "--------------------------------------------------\n",
      "ID: 38\n",
      "Titre: Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2\n",
      "--------------------------------------------------\n",
      "ID: 39\n",
      "Titre: varCADD: large sets of standing genetic variation enable genome-wide pathogenicity prediction\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614666v1\n",
      "--------------------------------------------------\n",
      "ID: 40\n",
      "Titre: Diversity and abundance of ring nucleases in type III CRISPR-Cas loci\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614671v1\n",
      "--------------------------------------------------\n",
      "ID: 41\n",
      "Titre: Germinated brown rice research: bibliometric analysis of progress, hotspots and trends\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614820v1\n",
      "--------------------------------------------------\n",
      "ID: 42\n",
      "Titre: p53motifDB: integration of genomic information and tumor suppressor p53 binding motifs\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614594v1\n",
      "--------------------------------------------------\n",
      "Les titres et les liens des articles ont été enregistrés dans la base de données SQLite.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# affiche articles stockés dans database\n",
    "c.execute(\"SELECT * FROM articles\")\n",
    "\n",
    "rows = c.fetchall() #récupere tous les results de la requete d'avant =tuple(id,title,link) de chaque article\n",
    "\n",
    "print(\"=== Liste des articles enregistrés ===\\n\")\n",
    "\n",
    "for row in rows:\n",
    "    print(f\"ID: {row[0]}\")\n",
    "    print(f\"Titre: {row[1]}\")\n",
    "    print(f\"Lien: {row[2]}\")\n",
    "    print(\"-\" * 50)  # separateur visuel entre chaque article\n",
    "\n",
    "# Fermer la connexion à la base de données\n",
    "conn.close()\n",
    "\n",
    "print(\"Les titres et les liens des articles ont été enregistrés dans la base de données SQLite.\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "l1-python",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
