{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2 style=\"color: #004C47;\">Installation des bibliothèques nécessaires </h2>\n",
    "<div style=\"border: 2px solid #01796F; background-color: #E0F2F1; color: #01796F; padding: 15px; border-radius: 5px; font-size: 16px;\">\n",
    "    <strong>\n",
    "    \n",
    "**pip install beautifulsoup4**\n",
    "\n",
    "**pip install requests**\n",
    "\n",
    "**pip install PyPDF2**\n",
    "\n",
    "**pip install pdfplumber**   </div>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Imports internes\n",
    "from io import BytesIO\n",
    "import logging\n",
    "import os\n",
    "import PyPDF2\n",
    "import re\n",
    "import sqlite3\n",
    "import time\n",
    "from concurrent.futures import ThreadPoolExecutor\n",
    "from datetime import datetime\n",
    "\n",
    "\n",
    "# Imports externes\n",
    "import pdfplumber\n",
    "import requests\n",
    "from bs4 import BeautifulSoup"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2 style=\"color: #004C47;\">Chargement et rechargement du module tools</h2> "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "import tools"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2 style=\"color: #004C47;\">1. Scraper les articles bioRxiv liés à la bioinformatique</h2>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div style=\"border: 2px solid #01796F; background-color: #E0F2F1; color: #01796F; padding: 15px; border-radius: 5px; font-size: 16px;\">\n",
    "    <strong>Objectif:</strong> Ce script permet d'archiver automatiquement les dépôts logiciels non encore archivés dans la base Software Heritage.Récupérer les articles publiés sur bioRxiv dans le domaine de la bioinformatique, en collectant les informations essentielles pour une analyse et un traitement ultérieurs.\n",
    "\n",
    "Informations collectées :\n",
    "\n",
    "-Titre de l’article : Titre complet décrivant le sujet de recherche.\n",
    "\n",
    "-Résumé (abstract) : Vue d’ensemble concise du contenu scientifique.\n",
    "\n",
    "-Auteurs : Liste des auteurs, ordonnée telle qu’elle apparaît sur bioRxiv.\n",
    "\n",
    "-Date de publication : Date de mise en ligne pour le suivi des recherches.\n",
    "\n",
    "-DOI : Identifiant unique facilitant la référence et l’accès.\n",
    "\n",
    "-URL de l’article : Lien direct vers la page de l’article sur bioRxiv.\n",
    "\n",
    "-Mots-clés : Mots-clés associés pour décrire le domaine ou sujet de recherche.\n",
    "\n",
    "-URLs des dépôts logiciels : Liens vers les dépôts de code (GitHub, GitLab) associés aux ressources logicielles.\n",
    "\n",
    " \n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import tools\n",
    "\n",
    "def main():\n",
    "    \"\"\"\n",
    "    Scrape les articles bioRxiv dans le domaine de la bioinformatique\n",
    "    et enregistre les informations essentielles dans la base de données.\n",
    "    \"\"\"\n",
    "    tools.scrape_articles(total_pages=10)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2 style=\"color: #004C47;\">2. Extraction des URLs des Dépôts Logiciels</h2>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div style=\"border: 2px solid #01796F; background-color: #E0F2F1; color: #01796F; padding: 15px; border-radius: 5px; font-size: 16px;\">\n",
    "    <strong>Objectif : </strong> Extraire automatiquement les liens vers les dépôts logiciels (GitHub, GitLab) mentionnés dans les articles bioRxiv liés à la bioinformatique. Ce script parcourt les PDF et les résumés (abstracts) des articles pour identifier et collecter ces URLs de dépôt, qui seront ensuite utilisés pour archivage ou analyse.\n",
    "\n",
    " </div>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import tools\n",
    "import logging\n",
    "\n",
    "logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')\n",
    "\n",
    "def main():\n",
    "    \"\"\"\n",
    "    Traite les articles pour extraire les liens des dépôts logiciels\n",
    "    (GitHub, GitLab) et les associe aux articles dans la base de données.\n",
    "    \"\"\"\n",
    "    tools.extract_repository_links()\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2 style=\"color: #004C47;\">3. Archivage via Software Heritage</h2>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div style=\"border: 2px solid #01796F; background-color: #E0F2F1; color: #01796F; padding: 15px; border-radius: 5px; font-size: 16px;\">\n",
    "    <strong>Objectif : </strong>  Assurer l’archivage à long terme des dépôts logiciels associés aux articles via l’API de Software Heritage.\n",
    "\n",
    "-Vérification : Pour chaque dépôt, le script vérifie s’il est déjà archivé.\n",
    "\n",
    "-Soumission automatique : Si non archivé, une demande d’archivage est soumise automatiquement.\n",
    "\n",
    "-Mise à jour : La base de données est mise à jour avec les liens et dates d’archivage.\n",
    "\n",
    " </div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import logging\n",
    "from tools import archive_repositories\n",
    "\n",
    "logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')\n",
    "\n",
    "def main():\n",
    "    \"\"\"\n",
    "    Archive les dépôts présents dans la base de données.\n",
    "    \"\"\"\n",
    "    archive_repositories(db_name=\"bioinformatics_article.db\")\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2 style=\"color: #004C47;\">4. Re-vérification de l’archivage pour les dépôts soumis</h2>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div style=\"border: 2px solid #01796F; background-color: #E0F2F1; color: #01796F; padding: 15px; border-radius: 5px; font-size: 16px;\">\n",
    "    <strong>Objectif : </strong>  Vérifier l’état d’archivage des dépôts soumis pour archivage auprès de Software Heritage.\n",
    "\n",
    "-Vérification périodique : Ce script revisite les dépôts soumis (ceux en attente d’archivage) pour voir si l’archivage a été complété.\n",
    "\n",
    "-Mise à jour : Si un dépôt est archivé avec succès, le script enregistre le lien et la date d’archivage dans la base de données.\n",
    "</div>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import tools\n",
    "import logging\n",
    "\n",
    "logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')\n",
    "\n",
    "def main():\n",
    "    \"\"\"\n",
    "    Re-vérifie les dépôts marqués comme en cours d'archivage dans Software Heritage.\n",
    "    Met à jour la base de données avec les résultats.\n",
    "    \"\"\"\n",
    "    tools.recheck_archived_repositories()\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2 style=\"color: #004C47;\">Affichage des tables et colonnes de la base de données </h2>\n",
    "\n",
    "<div style=\"border: 2px solid #01796F; background-color: #E0F2F1; color: #01796F; padding: 15px; border-radius: 5px; font-size: 16px;\">\n",
    "    <strong>Objectif : </strong> Visualiser la structure de la base de données en restant sur le fichier jupyter pour confirmer la présence des tables attendues et des colonnes associées.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Contenu de la table 'articles':\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>title</th>\n",
       "      <th>link</th>\n",
       "      <th>doi</th>\n",
       "      <th>date</th>\n",
       "      <th>pdf_link</th>\n",
       "      <th>abstract</th>\n",
       "      <th>is_article_processed</th>\n",
       "      <th>contains_valid_repo_link</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>23</td>\n",
       "      <td>BC-Design: A Biochemistry-Aware Framework for ...</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>10.1101/2024.10.28.620755</td>\n",
       "      <td>2024-01-01</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>Inverse protein folding, which aims to design ...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>24</td>\n",
       "      <td>Parallel maximal common subgraphs with labels ...</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.0...</td>\n",
       "      <td>10.1101/2024.05.10.593525</td>\n",
       "      <td>2024-01-01</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.0...</td>\n",
       "      <td>Advances in graph algorithmics have allowed in...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>25</td>\n",
       "      <td>GenomicSign: A computational method to discove...</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>10.1101/2024.11.05.622192</td>\n",
       "      <td>2024-01-01</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>Molecular diagnostics for the rapid identifica...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26</td>\n",
       "      <td>Vcfexpress: flexible, rapid user-expressions t...</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>10.1101/2024.11.05.622129</td>\n",
       "      <td>2024-01-01</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>Motivation: Variant Call Format (VCF) files ar...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>27</td>\n",
       "      <td>pLM-DBPs: Enhanced DNA-Binding Protein Predict...</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>10.1101/2024.10.04.616755</td>\n",
       "      <td>2024-01-01</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>DNA-binding proteins (DBPs) play critical role...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>122</td>\n",
       "      <td>Robust fine-mapping in the presence of linkage...</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>10.1101/2024.10.29.620968</td>\n",
       "      <td>2024-01-01</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>Fine-mapping methods based on summary statisti...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>123</td>\n",
       "      <td>Microbial populations hardly ever grow logisti...</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.0...</td>\n",
       "      <td>10.1101/2024.09.02.610774</td>\n",
       "      <td>2024-01-01</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.0...</td>\n",
       "      <td>We investigate the growth dynamics of microbia...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>124</td>\n",
       "      <td>AlphaFold predictions on whole genomes at a gl...</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>10.1101/2024.11.16.623929</td>\n",
       "      <td>2024-01-01</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>AlphaFold has opened a new era in structural b...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>125</td>\n",
       "      <td>Unipept in 2024: Expanding metaproteomics anal...</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.0...</td>\n",
       "      <td>10.1101/2024.09.26.615136</td>\n",
       "      <td>2024-01-01</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.0...</td>\n",
       "      <td>Unipept, a pioneering software tool in metapro...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>126</td>\n",
       "      <td>PSTP: Decoding Latent Sequence Grammar for Pro...</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>10.1101/2024.10.29.620820</td>\n",
       "      <td>2024-01-01</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/2024.1...</td>\n",
       "      <td>Phase separation (PS) is essential in various ...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>104 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      id                                              title  \\\n",
       "0     23  BC-Design: A Biochemistry-Aware Framework for ...   \n",
       "1     24  Parallel maximal common subgraphs with labels ...   \n",
       "2     25  GenomicSign: A computational method to discove...   \n",
       "3     26  Vcfexpress: flexible, rapid user-expressions t...   \n",
       "4     27  pLM-DBPs: Enhanced DNA-Binding Protein Predict...   \n",
       "..   ...                                                ...   \n",
       "99   122  Robust fine-mapping in the presence of linkage...   \n",
       "100  123  Microbial populations hardly ever grow logisti...   \n",
       "101  124  AlphaFold predictions on whole genomes at a gl...   \n",
       "102  125  Unipept in 2024: Expanding metaproteomics anal...   \n",
       "103  126  PSTP: Decoding Latent Sequence Grammar for Pro...   \n",
       "\n",
       "                                                  link  \\\n",
       "0    https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "1    https://www.biorxiv.org/content/10.1101/2024.0...   \n",
       "2    https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "3    https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "4    https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "..                                                 ...   \n",
       "99   https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "100  https://www.biorxiv.org/content/10.1101/2024.0...   \n",
       "101  https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "102  https://www.biorxiv.org/content/10.1101/2024.0...   \n",
       "103  https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "\n",
       "                           doi        date  \\\n",
       "0    10.1101/2024.10.28.620755  2024-01-01   \n",
       "1    10.1101/2024.05.10.593525  2024-01-01   \n",
       "2    10.1101/2024.11.05.622192  2024-01-01   \n",
       "3    10.1101/2024.11.05.622129  2024-01-01   \n",
       "4    10.1101/2024.10.04.616755  2024-01-01   \n",
       "..                         ...         ...   \n",
       "99   10.1101/2024.10.29.620968  2024-01-01   \n",
       "100  10.1101/2024.09.02.610774  2024-01-01   \n",
       "101  10.1101/2024.11.16.623929  2024-01-01   \n",
       "102  10.1101/2024.09.26.615136  2024-01-01   \n",
       "103  10.1101/2024.10.29.620820  2024-01-01   \n",
       "\n",
       "                                              pdf_link  \\\n",
       "0    https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "1    https://www.biorxiv.org/content/10.1101/2024.0...   \n",
       "2    https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "3    https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "4    https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "..                                                 ...   \n",
       "99   https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "100  https://www.biorxiv.org/content/10.1101/2024.0...   \n",
       "101  https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "102  https://www.biorxiv.org/content/10.1101/2024.0...   \n",
       "103  https://www.biorxiv.org/content/10.1101/2024.1...   \n",
       "\n",
       "                                              abstract  is_article_processed  \\\n",
       "0    Inverse protein folding, which aims to design ...                     1   \n",
       "1    Advances in graph algorithmics have allowed in...                     1   \n",
       "2    Molecular diagnostics for the rapid identifica...                     1   \n",
       "3    Motivation: Variant Call Format (VCF) files ar...                     1   \n",
       "4    DNA-binding proteins (DBPs) play critical role...                     1   \n",
       "..                                                 ...                   ...   \n",
       "99   Fine-mapping methods based on summary statisti...                     1   \n",
       "100  We investigate the growth dynamics of microbia...                     1   \n",
       "101  AlphaFold has opened a new era in structural b...                     1   \n",
       "102  Unipept, a pioneering software tool in metapro...                     1   \n",
       "103  Phase separation (PS) is essential in various ...                     1   \n",
       "\n",
       "     contains_valid_repo_link  \n",
       "0                           0  \n",
       "1                           0  \n",
       "2                           0  \n",
       "3                           1  \n",
       "4                           0  \n",
       "..                        ...  \n",
       "99                          1  \n",
       "100                         0  \n",
       "101                         0  \n",
       "102                         1  \n",
       "103                         1  \n",
       "\n",
       "[104 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Contenu de la table 'code_repositories':\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>code_repo_id</th>\n",
       "      <th>code_repo_url</th>\n",
       "      <th>is_archived_in_swh</th>\n",
       "      <th>swh_archive_link</th>\n",
       "      <th>swh_date_last_archive</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>https://github.com/brentp/vcfexpress</td>\n",
       "      <td>1</td>\n",
       "      <td>https://archive.softwareheritage.org/browse/or...</td>\n",
       "      <td>2024-11-14T12:33:17.227000+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>https://github.com/satijalab/seurat</td>\n",
       "      <td>1</td>\n",
       "      <td>https://archive.softwareheritage.org/browse/or...</td>\n",
       "      <td>2024-11-14T12:33:15.176000+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>https://github.com/RenanSimoesBR</td>\n",
       "      <td>1</td>\n",
       "      <td>https://archive.softwareheritage.org/browse/or...</td>\n",
       "      <td>2024-11-14T12:33:15.696000+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>https://github.com/QuackenbushLab/tcga-data-nf</td>\n",
       "      <td>1</td>\n",
       "      <td>https://archive.softwareheritage.org/browse/or...</td>\n",
       "      <td>2024-11-14T12:33:16.465000+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>https://github.com/QuackenbushLab/NetworkDataC...</td>\n",
       "      <td>1</td>\n",
       "      <td>https://archive.softwareheritage.org/browse/or...</td>\n",
       "      <td>2024-11-14T12:33:17.982000+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>75</td>\n",
       "      <td>https://github.com/Morvan98/PSTP</td>\n",
       "      <td>1</td>\n",
       "      <td>https://archive.softwareheritage.org/browse/or...</td>\n",
       "      <td>2024-11-28T16:44:30.306000+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>76</td>\n",
       "      <td>https://github.com/KULL-Centre/_2023_Tesei_IDRome</td>\n",
       "      <td>1</td>\n",
       "      <td>https://archive.softwareheritage.org/browse/or...</td>\n",
       "      <td>2024-11-28T16:44:06.585000+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>77</td>\n",
       "      <td>https://github.com/zhwm/RSparsePro_</td>\n",
       "      <td>1</td>\n",
       "      <td>https://archive.softwareheritage.org/browse/or...</td>\n",
       "      <td>2024-11-28T16:44:12.515000+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>78</td>\n",
       "      <td>https://github.com/zhwm/RSparsePro_LD_</td>\n",
       "      <td>1</td>\n",
       "      <td>https://archive.softwareheritage.org/browse/or...</td>\n",
       "      <td>2024-11-28T16:44:07.492000+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>79</td>\n",
       "      <td>https://github.com/zhwm/RSparsePro_LD</td>\n",
       "      <td>1</td>\n",
       "      <td>https://archive.softwareheritage.org/browse/or...</td>\n",
       "      <td>2024-11-28T16:44:20.316000+00:00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    code_repo_id                                      code_repo_url  \\\n",
       "0              1               https://github.com/brentp/vcfexpress   \n",
       "1              2                https://github.com/satijalab/seurat   \n",
       "2              3                   https://github.com/RenanSimoesBR   \n",
       "3              4     https://github.com/QuackenbushLab/tcga-data-nf   \n",
       "4              5  https://github.com/QuackenbushLab/NetworkDataC...   \n",
       "..           ...                                                ...   \n",
       "74            75                   https://github.com/Morvan98/PSTP   \n",
       "75            76  https://github.com/KULL-Centre/_2023_Tesei_IDRome   \n",
       "76            77                https://github.com/zhwm/RSparsePro_   \n",
       "77            78             https://github.com/zhwm/RSparsePro_LD_   \n",
       "78            79              https://github.com/zhwm/RSparsePro_LD   \n",
       "\n",
       "    is_archived_in_swh                                   swh_archive_link  \\\n",
       "0                    1  https://archive.softwareheritage.org/browse/or...   \n",
       "1                    1  https://archive.softwareheritage.org/browse/or...   \n",
       "2                    1  https://archive.softwareheritage.org/browse/or...   \n",
       "3                    1  https://archive.softwareheritage.org/browse/or...   \n",
       "4                    1  https://archive.softwareheritage.org/browse/or...   \n",
       "..                 ...                                                ...   \n",
       "74                   1  https://archive.softwareheritage.org/browse/or...   \n",
       "75                   1  https://archive.softwareheritage.org/browse/or...   \n",
       "76                   1  https://archive.softwareheritage.org/browse/or...   \n",
       "77                   1  https://archive.softwareheritage.org/browse/or...   \n",
       "78                   1  https://archive.softwareheritage.org/browse/or...   \n",
       "\n",
       "               swh_date_last_archive  \n",
       "0   2024-11-14T12:33:17.227000+00:00  \n",
       "1   2024-11-14T12:33:15.176000+00:00  \n",
       "2   2024-11-14T12:33:15.696000+00:00  \n",
       "3   2024-11-14T12:33:16.465000+00:00  \n",
       "4   2024-11-14T12:33:17.982000+00:00  \n",
       "..                               ...  \n",
       "74  2024-11-28T16:44:30.306000+00:00  \n",
       "75  2024-11-28T16:44:06.585000+00:00  \n",
       "76  2024-11-28T16:44:12.515000+00:00  \n",
       "77  2024-11-28T16:44:07.492000+00:00  \n",
       "78  2024-11-28T16:44:20.316000+00:00  \n",
       "\n",
       "[79 rows x 5 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Contenu de la table 'articles_code_repositories':\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>code_repo_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>28</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>28</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>33</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>33</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>126</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>126</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>122</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>122</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>122</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>81 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    article_id  code_repo_id\n",
       "0           26             1\n",
       "1           28             2\n",
       "2           28             3\n",
       "3           33             4\n",
       "4           33             5\n",
       "..         ...           ...\n",
       "76         126            75\n",
       "77         126            76\n",
       "78         122            77\n",
       "79         122            78\n",
       "80         122            79\n",
       "\n",
       "[81 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import sqlite3\n",
    "import pandas as pd\n",
    "\n",
    "def show_tables(db_name=\"bioinformatics_article.db\"):\n",
    "    \"\"\"\n",
    "    Affiche le contenu des tables articles, code_repositories et articles_code_repositories\n",
    "    dans la base de données SQLite avec les noms des colonnes.\n",
    "    \"\"\"\n",
    "    # Connexion à la base de données\n",
    "    conn = sqlite3.connect(db_name)\n",
    "\n",
    "    # Fonction pour afficher le contenu d'une table sous forme de DataFrame\n",
    "    def show_table_content(table_name):\n",
    "        print(f\"\\nContenu de la table '{table_name}':\")\n",
    "        query = f\"SELECT * FROM {table_name}\"\n",
    "        df = pd.read_sql_query(query, conn)\n",
    "        display(df)  # Affiche le DataFrame dans le notebook\n",
    "\n",
    "    # Afficher le contenu des tables\n",
    "    show_table_content(\"articles\")\n",
    "    show_table_content(\"code_repositories\")\n",
    "    show_table_content(\"articles_code_repositories\")\n",
    "\n",
    "    # Fermer la connexion\n",
    "    conn.close()\n",
    "\n",
    "# Exécuter la fonction\n",
    "show_tables()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
