{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Installation de bibliotheques \n",
    "\n",
    "**pip install beautifulsoup4**\n",
    "\n",
    "**pip install requests**\n",
    "\n",
    "**pip install PyPDF2**\n",
    "\n",
    "**pip install pdfplumber**\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import sqlite3\n",
    "import time\n",
    "from datetime import datetime\n",
    "\n",
    "import pdfplumber\n",
    "import re\n",
    "import PyPDF2\n",
    "from io import BytesIO\n",
    "import os\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Scraper les articles bioRxiv liés à la bioinformatique "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = 'https://www.biorxiv.org'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<sqlite3.Cursor at 0x7fd4109d3f80>"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# creer base de données SQLite \"bioinformatics_articles\" \n",
    "database = sqlite3.connect('bioinformatics_articles.db')\n",
    "c = database.cursor() #ici le curseur 'c' permettra d'executer/parcourir les requetes SQL sur database\n",
    "\n",
    "\n",
    "c.execute(\"DROP TABLE IF EXISTS articles\")\n",
    "\n",
    "# create table 'articles' avec col id,title et le lien \n",
    "c.execute('''CREATE TABLE IF NOT EXISTS articles\n",
    "             (id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "             title TEXT, \n",
    "             link TEXT,\n",
    "             doi TEXT UNIQUE,\n",
    "             date TEXT,\n",
    "             pdf_link TEXT,\n",
    "             abstract TEXT)''')\n",
    "#id c'est un identifiant unique généré automatiquement "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Parcourir les articles  de la collection Bioinformatics contenant 2153 pages d'articles et enregistrer leurs informations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scraping page 1 sur 10\n",
      "Nombre total d'articles trouvés sur la page 1 : 10\n",
      "Article ajouté : PHLOWER - Single cell trajectory analysis using Decomposition of the Hodge Laplacian\n",
      "Article ajouté : Building pangenome graphs\n",
      "Article ajouté : Analyzing the Molecular Mechanism of Eucalyptol, Limonene and Pinene Enteric Capsules （QIENUO） in the Treatment of Pulmonary cystic fibrosis with Network Pharmacology and Verifying Molecular Docking\n",
      "Article ajouté : Enhanced prediction of protein functional identity through the integration of sequence and structural features\n",
      "Article ajouté : Phage display enables machine learning discovery of cancer antigen specific TCRs\n",
      "Article ajouté : AptaDiff: de novo design and optimization of aptamers based on diffusion models\n",
      "Article ajouté : Gut microbiota and type 2 diabetes associations: A meta-analysis of 16s studies and their methodological challenges\n",
      "Article ajouté : scGNN+: Adapting ChatGPT for Seamless Tutorial and Code Optimization\n",
      "Article ajouté : Isobaric labeling update in MaxQuant\n",
      "Article ajouté : Preserving Full Spectrum Information in Imaging Mass Spectrometry Data Reduction\n",
      "Scraping page 2 sur 10\n",
      "Nombre total d'articles trouvés sur la page 2 : 10\n",
      "Article ajouté : BindCraft: one-shot design of functional protein binders\n",
      "Article ajouté : UniGEN-DDI: Computing drug-drug interactions using a unified graph embedding network\n",
      "Article ajouté : Divergence of the Individual repeats in the leucine-rich repeat domains of human Toll-like receptors explain their diversity and functional adaptations.\n",
      "Article ajouté : SAMURAI: Shallow Analysis of Copy nuMber alterations Using a Reproducible And Integrated bioinformatics pipeline\n",
      "Article ajouté : Capturing the Hierarchically Assorted Protein-protein Interaction Modules of Mammalian Cell\n",
      "Article ajouté : Separating biological variance from noise by applying EM algorithm to modified General Linear Model\n",
      "Article ajouté : Advancing Protein-DNA Binding Site Prediction: Integrating Sequence Models and Machine Learning Classifiers\n",
      "Article ajouté : Exploring the link between extended red blood cell parameters and platelet indices in voluntary blood donors\n",
      "Article ajouté : PlasmidGPT: a generative framework for plasmid design and annotation\n",
      "Article ajouté : Non-detection during excursions by citizen scientists modeled as a function of weather, season, list length, and individual preferences\n",
      "Scraping page 3 sur 10\n",
      "Nombre total d'articles trouvés sur la page 3 : 10\n",
      "Article ajouté : Unlocking the Potential of Low Quality Total RNA-seq Data: A Stepwise Mapping Approach for Improved Quantitative Analyses\n",
      "Article ajouté : STIX: Long-reads based Accurate Structural Variation Annotation at Population Scale\n",
      "Article ajouté : A patient-centric knowledge graph approach to prioritize mutants for selective anti-cancer targeting\n",
      "Article ajouté : DeepPWM-BindingNet: Unleashing Binding Prediction with Combined Sequence and PWM Features\n",
      "Article ajouté : A new framework for evaluating model out-of-distribution for the biochemical domain\n",
      "Article ajouté : SMARTER-database: a tool to integrate SNP array datasets for sheep and goat breeds\n",
      "Article ajouté : Novel AI-powered computational method using tensor decomposition can discover the common optimal bin sizes when integrating multiple Hi-C datasets\n",
      "Article ajouté : A systematic comparison of computational methods for expression forecasting\n",
      "Article ajouté : Quantifying Cancer Drug Targetability across Biochemical Pathways\n",
      "Article ajouté : Deep learning-based predictions of gene perturbation effects do not yet outperform simple linear methods\n",
      "Scraping page 4 sur 10\n",
      "Nombre total d'articles trouvés sur la page 4 : 10\n",
      "Article ajouté : An Efficient Computational Method to Create Positive NIPT Samples with Autosomal Trisomy\n",
      "Article ajouté : Impact of Interval Censoring on Data Accuracy and Machine Learning Performance in Biological High-Throughput Screening\n",
      "Article ajouté : Profiling drug sensitivity of leukemic stem cells via bulk-to-single-cell deconvolution\n",
      "Article ajouté : R2DT: a comprehensive platform for visualising RNA secondary structure\n",
      "Article ajouté : Chromosome-level genome assembly and annotation of the crested gecko, Correlophus ciliatus, a lizard incapable of tail regeneration\n",
      "Article ajouté : Distinct gene regulatory networks govern hematopoietic and leukemia stem cells\n",
      "Article ajouté : FragLlama: Next-fragment prediction for molecular design\n",
      "Article ajouté : On Long-Term Species Coexistence in Five-Species Evolutionary Spatial Cyclic Games with Ablated and Non-Ablated Dominance Networks\n",
      "Article ajouté : Allo-Allo: Data-efficient prediction of allosteric sites\n",
      "Article ajouté : LightMed: A Light-weight and Robust FFT-Based Model for Adversarially Resilient Medical Image Segmentation\n",
      "Scraping page 5 sur 10\n",
      "Nombre total d'articles trouvés sur la page 5 : 10\n",
      "Article ajouté : Improved robustness to gene tree incompleteness, estimation errors, and systematic homology errors with weighted TREE-QMC\n",
      "Article ajouté : MPCD Index for Hepatocellular Carcinoma Patients Based on Mitochondrial Function and Cell Death Patterns\n",
      "Article ajouté : INSTINCT: Multi-sample integration of spatial chromatin accessibility sequencing data via stochastic domain translation\n",
      "Article ajouté : Identifying similar populations across independent single cell studies without data integration\n",
      "Article ajouté : Plasmid conjugation drives within-patient plasmid diversity\n",
      "Article ajouté : gLinDA: global differential abundance analysis of microbiomes\n",
      "Article ajouté : stFormer: a foundation model for spatial transcriptomics\n",
      "Article ajouté : In Silico identification and modelling of FDA-approved drugs targeting T-type calcium channels\n",
      "Article ajouté : Benchmarking copy number aberrations inference tools using single-cell multi-omics datasets\n",
      "Article ajouté : EmbedSimScore: Advancing Protein Similarity Analysis with Structural and Contextual Embeddings\n",
      "Scraping page 6 sur 10\n",
      "Nombre total d'articles trouvés sur la page 6 : 10\n",
      "Article ajouté : Research on the Mechanism of Lphn1 Knockout in Inhibiting Colorectal Cancer\n",
      "Article ajouté : Deep learning facilitates precise identification of disease-resistance genes in plants\n",
      "Article ajouté : Mastering Artifact Correction in Neuroimaging Analysis: A Retrospective Approach\n",
      "Article ajouté : Profiling ranked list enrichment scoring in sparse data elucidates algorithmic tradeoffs\n",
      "Article ajouté : SIMVI reveals intrinsic and spatial-induced states in spatial omics data\n",
      "Article ajouté : Functional protein mining with conformal guarantees\n",
      "Article ajouté : MetaLab Platform Enables Comprehensive DDA and DIA Metaproteomics Analysis\n",
      "Article ajouté : Molecular Evolution and Functional Divergence of Heterodisulfide Reductase A (HdrA)\n",
      "Article ajouté : Unified Sampling and Ranking for Protein Docking with DFMDock\n",
      "Article ajouté : CAMDA 2023: finding patterns in urban microbiomes\n",
      "Scraping page 7 sur 10\n",
      "Nombre total d'articles trouvés sur la page 7 : 10\n",
      "Article ajouté : ConoDL: A Deep Learning Framework for Rapid Generation and Prediction of Conotoxins\n",
      "Article ajouté : A qualitative review of Oxford Nanopore Sequencing datasets for RNA modifications\n",
      "Article ajouté : FishCODE: a web-based information platform for comprehensive omics data exploration in fish research\n",
      "Article ajouté : Multi-View Integrative Approach For Imputing Short-Chain Fatty Acids and Identifying Key factors predicting Blood SCFA\n",
      "Article ajouté : Unipept 6.0: Expanding Metaproteomics Analysis with Support for Missed Cleavages, Semi-Tryptic and Non-Tryptic Peptides\n",
      "Article ajouté : Seed2LP: seed inference in metabolic networks for reverse ecology applications\n",
      "Article ajouté : Skeletons in the Forest: Using Entropy-Based Rogue Detection on Bayesian Phylogenetic Tree Distributions\n",
      "Article ajouté : Leveraging cell type-specificity for gene set analysis of single cell transcriptomics\n",
      "Article ajouté : AnalyzAIRR: A user-friendly guided workflow for AIRR data analysis\n",
      "Article ajouté : Towards improving full-length ribosome density prediction by bridging sequence and graph-based representations\n",
      "Scraping page 8 sur 10\n",
      "Nombre total d'articles trouvés sur la page 8 : 10\n",
      "Article ajouté : WIMOAD: Weighted Integration of Multi-Omics data for Alzheimer’s Disease (AD) Diagnosis\n",
      "Article ajouté : HuBMAPR: an R Client for the HuBMAP Data Portal\n",
      "Article ajouté : Lifestyle factors in the biomedical literature: An ontology and comprehensive resources for named entity recognition\n",
      "Article ajouté : LOCAS: Multi-label mRNA Localization with Supervised Contrastive Learning\n",
      "Article ajouté : Cell-type-specific mapping of enhancers and target genes from single-cell multimodal data\n",
      "Article ajouté : Labels as a Feature: Network Homophily for Systematically Discovering human GPCR Drug-Target Interactions\n",
      "Article ajouté : Machine Learning-Enhanced Drug Discovery for BACE1: A Novel Approach to Alzheimer’s Therapeutics\n",
      "Article ajouté : Particular sequence characteristics induce bias in the detection of polymorphic transposable element insertions\n",
      "Article ajouté : Generalizable Morphological Profiling of Cells by Interpretable Unsupervised Learning\n",
      "Article ajouté : Local Mean Suppression Filter for Effective Background Identification in Fluorescence Images\n",
      "Scraping page 9 sur 10\n",
      "Nombre total d'articles trouvés sur la page 9 : 10\n",
      "Article ajouté : FerroEnrich: An Interactive web tool for computing Ferroptosis index and gene enrichment\n",
      "Article ajouté : An explainable graph neural network approach for effectively integrating multi-omics with prior knowledge to identify biomarkers from interacting biological domains\n",
      "Article ajouté : Continual integration of single-cell multimodal data with MIRACLE\n",
      "Article ajouté : Forecasting dominance of SARS-CoV-2 lineages by anomaly detection using deep AutoEncoders\n",
      "Article ajouté : Protein Sequence Modelling with Bayesian Flow Networks\n",
      "Article ajouté : RanBALL: An Ensemble Random Projection Model for Identifying Subtypes of B-cell Acute Lymphoblastic Leukemia\n",
      "Article ajouté : scEMB: Learning context representation of genes based on large-scale single-cell transcriptomics\n",
      "Article ajouté : Analytical Assessment of Metagenomic Workflows for Pathogen Detection with NIST RM 8376 and Two Sample Matrices\n",
      "Article ajouté : BRDKRM: An Explainable Framework for Disease Modifying Drug Identification\n",
      "Article ajouté : PlantConnectome: knowledge graph encompassing >70,000 plant articles\n",
      "Scraping page 10 sur 10\n",
      "Nombre total d'articles trouvés sur la page 10 : 10\n",
      "Article ajouté : Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Article ajouté : Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Article ajouté : varCADD: large sets of standing genetic variation enable genome-wide pathogenicity prediction\n",
      "Article ajouté : Diversity and abundance of ring nucleases in type III CRISPR-Cas loci\n",
      "Article ajouté : Germinated brown rice research: bibliometric analysis of progress, hotspots and trends\n",
      "Article ajouté : p53motifDB: integration of genomic information and tumor suppressor p53 binding motifs\n",
      "Article ajouté : RNA-DCGen: Dual Constrained RNA Sequence Generation with LLM-Attack\n",
      "Article ajouté : Clustering-independent estimation of cell abundances in bulk tissues using single-cell RNA-seq data\n",
      "Article ajouté : SeaMoon: Prediction of molecular motions based on language models\n",
      "Article ajouté : INSPIRE: interpretable, flexible and spatially-aware integration of multiple spatial transcriptomics datasets from diverse sources\n"
     ]
    }
   ],
   "source": [
    "\n",
    "## Nombre de pages à scraper (exemple : 10 pages à la fois)\n",
    "from pickle import TRUE\n",
    "\n",
    "\n",
    "total_pages = 10\n",
    "\n",
    "# boucle qui parcour chaque page de la collection Bioinformatics\n",
    "for page in range(1, total_pages + 1):\n",
    "    print(f\"Scraping page {page} sur {total_pages}\")\n",
    "\n",
    "    # requete qui recup la page bioinfo actuelle\n",
    "    response = requests.get(f'{url}/collection/bioinformatics?page={page}') #? mean numero de la {page}\n",
    "    soup = BeautifulSoup(response.content, 'html.parser')\n",
    "\n",
    "    # Trouver les liens vers les articles sur cette page\n",
    "    articles = soup.find_all('a', class_='highwire-cite-linked-title')  # find liens hypertextes avec classe CSS\n",
    "\n",
    "    print(f'Nombre total d\\'articles trouvés sur la page {page} : {len(articles)}')\n",
    "\n",
    "    # Parcourir tous les articles de la page et enregistrer leurs titres et liens\n",
    "    for article in articles:\n",
    "        title = article.text.strip()  # extraire le titre dans la balise <a>, accessible via .text.\n",
    "        link = url + article['href']  # extraire le lien , href=destination du lien de mon article\n",
    "\n",
    "        # Requête pour accéder à la page de l'article\n",
    "        article_response = requests.get(link)\n",
    "        article_soup = BeautifulSoup(article_response.content, 'html.parser')\n",
    "\n",
    "        # extraire le DOI (dans une balise meta ou dans l'URL)\n",
    "        doi_tag = article_soup.find('meta', {'name': 'citation_doi'})  # balise <meta> qui a l’attribut name=\"citation_doi\".\n",
    "        doi = doi_tag['content'] if doi_tag else 'DOI non disponible'  # si balise existe -> attribut content contient la valeur doi\n",
    "\n",
    "        # extraire la date de publication (dans une balise meta ou time)et la formater au format ISO 8601\n",
    "        date_tag = article_soup.find('meta', {'name': 'citation_publication_date'})\n",
    "        date = date_tag['content'] if date_tag else 'Date non disponible'\n",
    "        # Remplacer les barres obliques par des tirets avant la conversion\n",
    "        if date != 'Date non disponible':\n",
    "            date = date.replace('/', '-')  # Remplacer '/' par '-'\n",
    "            date = datetime.strptime(date, '%Y-%m-%d').date().isoformat()\n",
    "\n",
    "        # extraire le lien vers le fichier PDF\n",
    "        pdf_links = article_soup.find_all('a')\n",
    "        pdf_link = None\n",
    "        for pdf in pdf_links:\n",
    "            if pdf and 'PDF' in pdf.text:\n",
    "                potential_pdf_link = url + pdf['href']\n",
    "                if potential_pdf_link.endswith('.pdf'):  # Vérifier si le lien se termine par .pdf\n",
    "                    pdf_link = potential_pdf_link\n",
    "                    break  # Quitter la boucle une fois le lien trouvé\n",
    "\n",
    "        # En cas de lien invalide, assigner une valeur par défaut\n",
    "        if not pdf_link:\n",
    "            pdf_link = 'Lien PDF non disponible'\n",
    "\n",
    "        \n",
    "        \n",
    "         # Extraire l'abstract\n",
    "        abstract_tag = article_soup.find('meta', {'name': 'citation_abstract'})\n",
    "        if abstract_tag:\n",
    "            abstract_html = abstract_tag['content']\n",
    "            clean_abstract = BeautifulSoup(abstract_html, \"html.parser\").get_text(strip=True)\n",
    "        else:\n",
    "            clean_abstract = 'Abstract non disponible'\n",
    "\n",
    "\n",
    "        # tout inserer dans la base de données SQLite\n",
    "        # Vérifier si l'article existe déjà dans la base de données (vérification sur DOI et date)\n",
    "        c.execute(\"SELECT * FROM articles WHERE doi = ? AND date = ?\", (doi, date))\n",
    "        result = c.fetchone() # recupère la 1ere ligne du result donc si mm doi/date trouvé retourne cette ligne en tuple sinon NONE\n",
    "\n",
    "        if result:\n",
    "            print(f\"L'article avec DOI {doi} existe déjà, ignoré.\")\n",
    "        else:\n",
    "            # Insérer l'article dans la base de données\n",
    "            c.execute(\"INSERT INTO articles (title, link, doi, date, pdf_link, abstract) VALUES (?, ?, ?, ?, ?, ?)\", \n",
    "                      (title, link, doi, date, pdf_link, clean_abstract))\n",
    "            print(f\"Article ajouté : {title}\")\n",
    "\n",
    "        # Afficher les informations extraites\n",
    "        #print(f'Titre: {title}\\nLien: {link}\\nDOI: {doi}\\nDate de publication: {date}\\nLien PDF: {pdf_link}\\nRésumé:{clean_abstract}\\n')\n",
    "        # pause d'une seconde pour éviter de surcharger le serveur\n",
    "        time.sleep(1)\n",
    "        \n",
    "    # Sauvegarder les modifications dans la base de données après chaque page\n",
    "    database.commit()\n",
    "   \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Affichage de mes articles dans la base de données (A supprimer c'est juste pr que je vois bien l'ajout)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "=== Liste des articles enregistrés ===\n",
      "\n",
      "ID: 1\n",
      "Titre: PHLOWER - Single cell trajectory analysis using Decomposition of the Hodge Laplacian\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.10.01.613179v1\n",
      "DOI: 10.1101/2024.10.01.613179\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.10.01.613179v1.full.pdf\n",
      "Résumé: Cellular differentiation, the process by which a cell changes its chromatin and expression programs to acquire more specialized functions, is not only crucial in the development of multicellular organisms but also key during onset and progression of diseases.  Transcription factors, which are proteins binding to regulatory DNA regions (open chromatin regions), are key regulators of gene expression thereby orchestrating cellular differentiation processes. Dissecting these key regulatory events will help to develop protocols for cellular reprogramming or ex vivo differentiation in e.g. organoids and to understand disease-related differentiation processes for potential therapeutic interventions. In this context, a unique resource to understand the interplay between chromatin, regulatory signals (transcription factor binding), and expression changes during cellular differentiation, is multi-modal single-cell sequencing, which can measure both full expression programs and genome-wide open chromatin. However, the associated experimental protocols dissociate cells, making it impossible to track how an individual cell differentiates over time experimentally.\n",
      "--------------------------------------------------\n",
      "ID: 2\n",
      "Titre: Building pangenome graphs\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.04.05.535718v2\n",
      "DOI: 10.1101/2023.04.05.535718\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.04.05.535718v2.full.pdf\n",
      "Résumé: Pangenome graphs can represent all variation between multiple reference genomes, but current approaches to build them exclude complex sequences or are based upon a single reference. In response, we developed the PanGenome Graph Builder (PGGB), a pipeline for constructing pangenome graphs without bias or exclusion. PGGB uses all-to-all alignments to build a variation graph in which we can identify variation, measure conservation, detect recombination events, and infer phylogenetic relationships.\n",
      "--------------------------------------------------\n",
      "ID: 3\n",
      "Titre: Analyzing the Molecular Mechanism of Eucalyptol, Limonene and Pinene Enteric Capsules （QIENUO） in the Treatment of Pulmonary cystic fibrosis with Network Pharmacology and Verifying Molecular Docking\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615978v1\n",
      "DOI: 10.1101/2024.09.30.615978\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615978v1.full.pdf\n",
      "Résumé: Objective: To explore the mechanism of Eucalyptol ,Limonene and Pinene Enteric Capsules（QIENUO） in the treatment of pulmonary cystic fibrosis (CPF), analyze the common targets of QIENUO and CPF, and verify the molecular docking of core proteins and small molecules. Methods: The main active compounds and their corresponding targets were obtained from PubChem, SwissTargetprediction, GeneCards, PharmMapper and TCMSP databases. Targets related to CPF were screened from GeneCards, OMIM, DisGeNET and TTD databases. The common \"QIENUO-CPF\" targets were analyzed by gene ontology (GO) and Kyoto Encyclopedia of Genes and Genomes (KEGG) through the website of Weishengxin. Protein-protein interaction (PPI) network and compound-target-pathway network were constructed by Cytoscape, and the network parameters were systematically analyzed. The interaction between core protein and monomer components was evaluated and verified by molecular docking method. Results: 228 active compounds target and 1354 CPF-related targets were screened out, and 92 common targets were analyzed by GO and KEGG. The results showed that the therapeutic effect of QIENUO on CPF was mainly through AMPK signaling pathway, cGMP-PKG signaling pathway and TGF-β signaling pathway. The results of molecular docking show that the binding energy of 9 of 15 pairs of ligand-receptor pairs is lower than-6 kjmol-1. Conclusion: QIENUO exhibits huge potential as a therapeutic agent for the treatment of pulmonary cystic fibrosis. The specific molecular mechanism and effective active components of QIENUO treat CPF were studied and demonstrated, which provided theoretical basis for better clinical application of QIENUO.\n",
      "--------------------------------------------------\n",
      "ID: 4\n",
      "Titre: Enhanced prediction of protein functional identity through the integration of sequence and structural features\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615718v1\n",
      "DOI: 10.1101/2024.09.30.615718\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615718v1.full.pdf\n",
      "Résumé: Although over 300 million protein sequences are registered in a reference sequence database, only 0.2% have experimentally determined functions. This suggests that many valuable proteins, potentially catalyzing novel enzymatic reactions, remain undiscovered among the vast number of function-unknown proteins. In this study, we developed a method to predict whether two proteins catalyze the same enzymatic reaction by analyzing sequence and structural similarities, utilizing structural models predicted by AlphaFold2. We performed pocket detection and domain decomposition for each structural model. The similarity between protein pairs was assessed using features such as full-length sequence similarity, domain structural similarity, and pocket similarity. We developed several models using conventional machine learning algorithms and found that the LightGBM-based model outperformed the models. Our method also surpassed existing approaches, including those based solely on full-length sequence similarity and state-of-the-art deep learning models. Feature importance analysis revealed that domain sequence identity, calculated through structural alignment, had the greatest influence on the prediction. Therefore, our findings demonstrate that integrating sequence and structural information improves the accuracy of protein function prediction.\n",
      "--------------------------------------------------\n",
      "ID: 5\n",
      "Titre: Phage display enables machine learning discovery of cancer antigen specific TCRs\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.06.27.600973v2\n",
      "DOI: 10.1101/2024.06.27.600973\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.06.27.600973v2.full.pdf\n",
      "Résumé: T cells targeting epitopes in infectious diseases or cancer play a central role in spontaneous and therapy-induced immune responses. T-cell epitope recognition is mediated by the binding of the T-Cell Receptor (TCR) and TCRs recognizing clinically relevant epitopes are promising for T-cell based therapies. Starting from one of the few known TCRs targeting the cancer-testis antigen NY-ESO-1 157-165 epitope, we built large phage display libraries of TCRs with randomized Complementary Determining Region 3 of the β chain. The TCR libraries were panned against the NY-ESO-1 epitope, which enabled us to collect thousands of epitope-specific TCR sequences. We then trained a machine learning TCR-epitope interaction predictor with this data and could identify several epitope-specific TCRs directly from TCR repertoires. Cellular binding and functional assays revealed that the predicted TCRs displayed activity towards the NY-ESO-1 epitope and no detectable cross-reactivity with self-peptides. Overall, our work demonstrates how display technologies combined with machine learning models of TCR-epitope recognition can effectively leverage large TCR repertoires for TCR discovery.\n",
      "--------------------------------------------------\n",
      "ID: 6\n",
      "Titre: AptaDiff: de novo design and optimization of aptamers based on diffusion models\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.11.25.568693v2\n",
      "DOI: 10.1101/2023.11.25.568693\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.11.25.568693v2.full.pdf\n",
      "Résumé: Aptamers are single-stranded nucleic acid ligands, featuring high affinity and specificity to target molecules. Traditionally they are identified from large DNA/RNA libraries using in vitro methods, like Systematic Evolution of Ligands by Exponential Enrichment (SELEX). However, these libraries capture only a small fraction of theoretical sequence space, and various aptamer candidates are constrained by actual sequencing capabilities from the experiment. Addressing this, we proposed AptaDiff, the first in silico aptamer design and optimization method based on the diffusion model. Our Aptadiff can generate aptamers beyond the constraints of high-throughput sequencing data, leveraging motif-dependent latent embeddings from variational autoencoder, and can optimize aptamers by affinity-guided aptamer generation according to Bayesian optimization. Comparative evaluations revealed AptaDiff9s superiority over existing aptamer generation methods in terms of quality and fidelity across four high-throughput screening data targeting distinct proteins. Moreover, Surface Plasmon Resonance (SPR) experiments were conducted to validate the binding affinity of aptamers generated through Bayesian optimization for two target proteins. The results unveiled a significant boost of 87.9% and 60.2% in RU values, along with a 3.6-fold and 2.4-fold decrease in KD values for the respective target proteins. Notably, the optimized aptamers demonstrated superior binding affinity compared to top experimental candidates selected through SELEX, underscoring the promising outcomes of our AptaDiff in accelerating the discovery of superior aptamers.\n",
      "--------------------------------------------------\n",
      "ID: 7\n",
      "Titre: Gut microbiota and type 2 diabetes associations: A meta-analysis of 16s studies and their methodological challenges\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.10.01.615711v1\n",
      "DOI: 10.1101/2024.10.01.615711\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.10.01.615711v1.full.pdf\n",
      "Résumé: Diabetes mellitus is a prevalent chronic non-communicable disease, and recent studies have explored the link between gut microbiota and its development. Despite some evidence suggesting an association, the influence of gut microbiota on type 2 diabetes (DM2) remains unclear. A systematic search of PubMed (Janeiro 2016 to 10 December 2023) using the keywords: (16S and diabetes or DM2) and (gut microbiota and diabetes or DM2). The studies included compared gut microbiome diversity between diabetic and non-diabetic adults using 16S rRNA sequencing, excluding children, interventions, and type 1 diabetes. Alpha diversity indices and bacterial mean abundance were analyzed, with statistical assessments using a random-effects model and I2 for heterogeneity. Thirteen studies met the criteria, with the Shannon index being the most commonly used measure. Results showed significant heterogeneity (I2 > 75%) and no notable differences between diabetic and non-diabetic groups. Other indices, such as Chao1 and phylogenetic whole tree, similarly showed no consistent differences. Taxonomic analysis also failed to find phyla consistently correlated with DM2, with variability across studies. The relationship between gut microbiota and diabetes remains uncertain due to technical and biological factors that are often overlooked. The inconsistencies across studies highlight the low reproducibility common in microbiota research.\n",
      "--------------------------------------------------\n",
      "ID: 8\n",
      "Titre: scGNN+: Adapting ChatGPT for Seamless Tutorial and Code Optimization\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615735v1\n",
      "DOI: 10.1101/2024.09.30.615735\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615735v1.full.pdf\n",
      "Résumé: Foundation models have transformed AI by leveraging large-scale data to efficiently perform diverse tasks, and their applications in bioinformatics are primarily focused on data-centric tasks like cell type annotation and gene expression analysis. However, their potential extends beyond data analysis, offering significant opportunities in software development and optimization, such as code refinement, tutorial generation, and advanced visualization. For example, models like OpenAI Codex can suggest optimized code snippets and generate well-documented, reproducible workflows, enhancing accessibility and reducing computational complexity. Despite these advantages, the use of foundation models for improving computational tool engineering in single-cell research remains underutilized. To address this gap, we developed scGNN+, a web-based platform that combines the power of graph neural networks with the capabilities of ChatGPT to enhance reproducibility, code optimization, and visualization. scGNN+ further simplifies the process for users by generating standardized, well-annotated code, making complex procedures more accessible to non-programmers. Additionally, ChatGPT integration allows users to create high-quality, customizable visualizations through natural language prompts, improving data interpretation and presentation. Ultimately, scGNN+ offers a user-friendly, reproducible, and optimized solution for single-cell research, leveraging the full potential of foundation models in bioinformatics. scGNN+ is publicly available at https://bmblx.bmi.osumc.edu/scgnn+.\n",
      "--------------------------------------------------\n",
      "ID: 9\n",
      "Titre: Isobaric labeling update in MaxQuant\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615788v1\n",
      "DOI: 10.1101/2024.09.30.615788\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615788v1.full.pdf\n",
      "Résumé: We present an update of the MaxQuant software for isobaric labeling data and evaluate its performance on benchmark datasets. Impurity correction factors can be applied to labels mixing C- and N-type reporter ions, such as TMT Pro. Application to a single-cell multi-species mixture benchmark shows high accuracy of the impurity-corrected results. TMT data recorded with FAIMS separation can be analyzed directly in MaxQuant without splitting the raw data into separate files per FAIMS voltage. Weighted median normalization, is applied to several datasets, including large-scale human body atlas data. In the benchmark datasets the weighted median normalization either removes or strongly reduces the batch effects between different TMT plexes and results in clustering by biology. In datasets including a reference channel, we find that weighted median normalization performs as well or better when the reference channel is ignored and only the sample channel intensities are used, suggesting that the measurement of a reference channel is unnecessary when using weighted median normalization in MaxQuant. We demonstrate that MaxQuant including the weighted median normalization performs well on multi-notch MS3 data, as well as on phosphorylation data. MaxQuant is freely available for any purpose and can be downloaded from https://www.maxquant.org/.\n",
      "--------------------------------------------------\n",
      "ID: 10\n",
      "Titre: Preserving Full Spectrum Information in Imaging Mass Spectrometry Data Reduction\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.614425v1\n",
      "DOI: 10.1101/2024.09.30.614425\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.614425v1.full.pdf\n",
      "Résumé: Imaging Mass Spectrometry (IMS) has become an important tool for molecular characterization of biological tissue. However, IMS experiments tend to yield large datasets, routinely recording over 200,000 ion intensity values per mass spectrum and more than 100,000 pixels, i.e., spectra, per dataset. Traditionally, IMS data size challenges have been addressed by feature selection or extraction, such as by peak picking and peak integration. Selective data reduction techniques such as peak picking only retain certain parts of a mass spectrum, and often these describe only medium-to-high-abundance species. Since lower-intensity peaks and, for example, near-isobar species are sometimes missed, selective methods can potentially bias downstream analysis towards a subset of species in the data rather than considering all species measured. Results: We present an alternative to selective data reduction of IMS data that achieves similar data size reduction while better conserving the ion intensity profiles across all recorded m/z-bins, thereby preserving full spectrum information. Our method utilizes a low-rank matrix completion model combined with a randomized sparse-format-aware algorithm to approximate IMS datasets. This representation offers reduced dimensionality and a data footprint comparable to peak picking, but also retains complete spectral profiles, enabling comprehensive analysis and compression. We demonstrate improved preservation of lower signal-to-noise-ratio signals and near-isobars, mitigation of selection bias, and reduced information loss compared to current state-of-the art data reduction methods in IMS.\n",
      "--------------------------------------------------\n",
      "ID: 11\n",
      "Titre: BindCraft: one-shot design of functional protein binders\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615802v1\n",
      "DOI: 10.1101/2024.09.30.615802\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615802v1.full.pdf\n",
      "Résumé: Protein-protein interactions (PPIs) are at the core of all key biological processes. However, the complexity of the structural features that determine PPIs makes their design challenging. We present BindCraft, an open-source and automated pipeline for de novo protein binder design with experimental success rates of 10-100%. BindCraft leverages the trained deep learning weights of AlphaFold2 to generate nanomolar binders without the need for high-throughput screening or experimental optimization, even in the absence of known binding sites. We successfully designed binders against a diverse set of challenging targets, including cell-surface receptors, common allergens, de novo designed proteins, and multi-domain nucleases, such as CRISPR-Cas9. We showcase their functional and therapeutic potential by demonstrating that designed binders can reduce IgE binding to birch allergen in patient-derived samples. This work represents a significant advancement towards a \"one design-one binder\" approach in computational design, with immense potential in therapeutics, diagnostics, and biotechnology.\n",
      "--------------------------------------------------\n",
      "ID: 12\n",
      "Titre: UniGEN-DDI: Computing drug-drug interactions using a unified graph embedding network\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615812v1\n",
      "DOI: 10.1101/2024.09.30.615812\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615812v1.full.pdf\n",
      "Résumé: Finding drug-drug interaction is crucial for patient safety and treatment efficacy. Two drugs may show a synergistic effect but may sometimes cause a severe health issue, including lethality. Wet lab studies are often performed to understand such interactions but are limited by cost and time. However, the biochemical data generated can be explored to compute unknown interactions. Here, we developed a computational model named UniGEN-DDI (Unified Graph Embedding Network for Drug-Drug Interaction) for the estimation of interactions between drugs. It is a simple unified network model containing the biochemical information of drug association developed using the compiled data from DrugBank 5.1.0. The feature learning of the drugs was carried out using a combination of GraphSAGE and Node2Vec algorithms, which were found efficient in extracting diverse features. The simple architecture of our model led to a significant reduction in computational time compared to the baselines, while maintaining a high prediction accuracy. The model performed well on the data, which was equally distributed between interacting and non-interacting drugs. As a more challenging evaluation, we performed non-overlapping splitting of the data based on the drug action on different parts of the body, and our model performed well in both interaction estimation and time efficiency. Our model successfully identified the 12 unknown drug interactions in DrugBank 5.1.0, which were updated in DrugBank 6.0.\n",
      "--------------------------------------------------\n",
      "ID: 13\n",
      "Titre: Divergence of the Individual repeats in the leucine-rich repeat domains of human Toll-like receptors explain their diversity and functional adaptations.\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615863v1\n",
      "DOI: 10.1101/2024.09.30.615863\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615863v1.full.pdf\n",
      "Résumé: The receptor domains of Toll-like receptors (TLRs) are characterized by a solenoid-like structure composed of tandem repeats of α/β units known as Leucine Rich Repeats (LRRs). LRR proteins form large paralogous families, with nearly 400 in the human genome alone, all sharing similar semi-regular solenoid-like structures. Despite this structural similarity, they exhibit remarkable diversity in binding specificity. For TLR receptors, this includes a range of pathogen-associated molecular patterns (PAMPs), while other LRR proteins bind an extensive array of ligands, including proteins, DNA, RNA, and small molecules. The LRR domains contain repeats that have similar, yet not identical, 3D structures and patterns of conserved residues. Through in-depth analysis of sequence and structural conservation in individual repeats of human TLRs, we demonstrate that even subtle variations between these repeats alter the local solenoid structure, leading to significant functional changes. Variations in repeat length and defining patterns result in local changes in curvature and the emergence of structural features such as loops, cavities, or specific interaction interfaces. Understanding how divergence in LRR repeats influences their functional roles can provide deeper insights into their binding mechanisms, including interactions with unknown ligands, both in humans and across the diverse phylogenetic spectrum of animals that rely on their TLR repertoires for immune activation.\n",
      "--------------------------------------------------\n",
      "ID: 14\n",
      "Titre: SAMURAI: Shallow Analysis of Copy nuMber alterations Using a Reproducible And Integrated bioinformatics pipeline\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615766v1\n",
      "DOI: 10.1101/2024.09.30.615766\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615766v1.full.pdf\n",
      "Résumé: Shallow whole-genome sequencing (sWGS) offers a cost-effective approach to detect  copy number alterations (CNAs). However, there remains a gap for a standardized workflow specifically designed for sWGS analysis. To address this need, in this work we present SAMURAI  a bioinformatics pipeline specifically designed for analyzing CNAs from sWGS data in a standardized and reproducible manner. SAMURAI is built using established community standards, ensuring portability, scalability, and reproducibility. The pipeline features a modular design with independent blocks for data pre-processing, copy number analysis, and customized reporting. Users can select workflows tailored for either solid or liquid biopsy analysis (e.g., circulating tumor DNA), with specific tools integrated for each sample type. The final report generated by SAMURAI provides detailed results to facilitate data interpretation and potential downstream analyses.To demonstrate its robustness, SAMURAI was validated using simulated and real-world datasets. The pipeline achieved high concordance with ground truth data and maintained consistent performance across various scenarios. By promoting standardization and offering a versatile workflow, SAMURAI empowers researchers in diverse environments to reliably analyze CNAs from sWGS data. This, in turn, holds promise for advancements in precision medicine.\n",
      "--------------------------------------------------\n",
      "ID: 15\n",
      "Titre: Capturing the Hierarchically Assorted Protein-protein Interaction Modules of Mammalian Cell\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615776v1\n",
      "DOI: 10.1101/2024.09.30.615776\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615776v1.full.pdf\n",
      "Résumé: Proteins are organized into modules by both functions and physical interactions within compartments of an eukaryotic cell. The in vivo chemical crosslinking mass spectrometry (XL-MS) data collected from organelles, the whole cells and tissues are able to provide unique information about both protein-protein interaction (PPI) and the intensity of PPI. In the present study, we have retrieved 55,982 crosslinked peptides (XL-peptides) from the XL-MS databases, out of which 6,356 in vivo PPIs were identified. Introduction of the MONET software into analysis of 4,526 hetero PPIs revealed a total of 402 protein modules, including 15, 58 and 163 stable protein complex(s), condensate-forming protein module(s) and intrinsically disordered region (IDR)-containing protein module(s), respectively. The application of ChatGPT in analysis of these modules unexpectedly identified 4 vesicle-related modules. Together, these modules were assorted into 6 communities (module of modules) and 3 systems (module of communities) hierarchically. The bioinformatic analysis found that the three systems are corresponding to three compartments of eukaryotic cell: nuclei, mitochondria, endoplasmic reticulum (ER), respectively. This study presents a novel and comprehensive biomolecular modulome of a mammalian cell, which captures putative protein compositions of protein complexes, protein condensates and vesicles and provides a hierarchical protein organization and function within compartments of mammalian cell.\n",
      "--------------------------------------------------\n",
      "ID: 16\n",
      "Titre: Separating biological variance from noise by applying EM algorithm to modified General Linear Model\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615661v1\n",
      "DOI: 10.1101/2024.09.29.615661\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615661v1.full.pdf\n",
      "Résumé: Introduction: The General Linear Model (GLM) has been widely used in research, where error term has been treated as noise. However, compelling evidence suggests that in biological systems, the target variables may possess their innate variances. \n",
      "Methods: A modified GLM was proposed to explicitly model biological variance and non-biological noise. Employing the Expectation and Maximization (EM) scheme can distinguish biological variance from noise, termed EMSEV (EM for Separating Variances). The performance of EMSEV was evaluated by varying noise levels, dimensions of the design matrix, and covariance structures of the target variables.\n",
      "Results: The deviation between EMSEV outputs and the pre-defined distribution parameters increased with noise level. With a proper initial guess, when the noise magnitude and the variance of the target variables were similar, there were deviations of 3% and 10 to 16% in the estimated mean and covariance of the target variables, respectively, along with a 1.7% deviation in noise estimation.\n",
      "Conclusion: EMSEV appears promising for distinguishing signal variance from noise in biological systems. The potential applications and implications in biological science and statistical inference are discussed.\n",
      "--------------------------------------------------\n",
      "ID: 17\n",
      "Titre: Advancing Protein-DNA Binding Site Prediction: Integrating Sequence Models and Machine Learning Classifiers\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.08.23.554389v2\n",
      "DOI: 10.1101/2023.08.23.554389\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.08.23.554389v2.full.pdf\n",
      "Résumé: Predicting protein-DNA binding sites is a challenging computational problem that has led to the development of advanced algorithms and techniques in the field of bioinformatics. Identifying the specific residues where proteins bind to DNA is of paramount importance, as it enables the modeling of their interactions and facilitates downstream studies. Nevertheless, the development of accurate and efficient computational methods for this task remains a persistent challenge. Accurate prediction of protein-DNA binding sites has far-reaching implications for understanding molecular mechanisms, disease processes, drug discovery, and synthetic biology applications. It helps bridge the gap between genomics and functional biology, enabling researchers to uncover the intricacies of cellular processes and advance our knowledge of the biological world. The method used to predict DNA binding residues in this study is a potent combination of conventional bioinformatics tools, protein language models, and cutting-edge machine learning and deep learning classifiers. On a dataset of protein-DNA binding sites, our model is meticulously trained, and it is then rigorously examined using several experiments. As indicated by higher predictive behavior with AUC values on two benchmark datasets, the results show superior performance when compared to existing models. The suggested model has a strong capacity for generalization and shows specificity for DNA-binding sites. We further demonstrated the adaptability of our model as a universal framework for binding site prediction by training it on a variety of protein-ligand binding site datasets. In conclusion, our innovative approach for predicting protein-DNA binding residues holds great promise in advancing our understanding of molecular interactions, thus paving the way for several groundbreaking applications in the field of molecular biology and genetics. Our approach demonstrated efficacy and versatility underscore its potential for driving transformative discoveries in biomolecular research.\n",
      "--------------------------------------------------\n",
      "ID: 18\n",
      "Titre: Exploring the link between extended red blood cell parameters and platelet indices in voluntary blood donors\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615743v1\n",
      "DOI: 10.1101/2024.09.29.615743\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615743v1.full.pdf\n",
      "Résumé: Background: As regular blood donors are prone to iron deficiency, importance of extended red blood cell (eRBC) parameters in identifying donors with depleted iron stores was investigated.Thrombocytosis has been well documented in patients affected with IDA. Thus, significance of eRBC parameters in identifying iron deficiency associated thrombocytosis was also examined in this cohort. Methods: Blood samples were collected in EDTA tubes from consenting donors for analyses of routine haematological and eRBC parameters. Serum samples were isolated for estimation of iron parameters. Results: Iron deficient donors had significantly altered eRBC parameters. Among them, Ret-He with a cut-off of ≥32 pg had high AUC (0.822) and showed relatively high sensitivity & specificity in detecting iron deficiency. Combination of Ret-He with CCI increased sensitivity & specificity to 90.6% and 98.2%, in detection of donors affected with iron restricted erythropoiesis. This cohort had increased platelet counts, which showed significant association with Ret-He (β= -0.373), RBC-He (β= -0.384), CCI (β= 0.384), Hypo-He (β=0.494) and Micro-R (β= 0.299). Elevated platelet counts also showed significant correlations with these eRBC parameters, which was absent in iron replete donors. Conclusions: eRBC parameters are sensitive indicators of non-anaemic iron deficiency, which may be enhanced by combining them. Their significant association with elevated platelet counts in iron deficient donors, highlights their importance in reflecting iron deficiency associated thrombocytosis.\n",
      "--------------------------------------------------\n",
      "ID: 19\n",
      "Titre: PlasmidGPT: a generative framework for plasmid design and annotation\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615762v1\n",
      "DOI: 10.1101/2024.09.30.615762\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615762v1.full.pdf\n",
      "Résumé: We introduce PlasmidGPT, a generative language model pretrained on 153k engineered plasmid sequences from Addgene. PlasmidGPT generates de novo sequences that share similar characteristics with engineered plasmids but show low sequence identity to the training data. We demonstrate its ability to generate plasmids in a controlled manner based on the input sequence or specific design constraint. Moreover, our model learns informative embeddings of both engineered and natural plasmids, allowing for efficient prediction of a wide range of sequence-related attributes.\n",
      "--------------------------------------------------\n",
      "ID: 20\n",
      "Titre: Non-detection during excursions by citizen scientists modeled as a function of weather, season, list length, and individual preferences\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615418v1\n",
      "DOI: 10.1101/2024.09.30.615418\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615418v1.full.pdf\n",
      "Résumé: INTRODUCTION: Citizen science is an increasingly valuable source of information about biodiversity. It is challenging to use this information for analysis of distribution and trends. The lack of a protocol leads to bias in observations and therefore data are not representative. The bias is a consequence of unequal detection probabilities, caused by different preferences and habits of citizen scientists. METHODS: We propose to incorporate characteristics of these excursions in analyses of data collected by citizen scientists to improve estimates of the probability that a species is not detected and reported, even though it does occur. By limiting these models to areas that are known to be occupied, detection can be modeled separately without considering variation in occupancy. We apply this idea to 150 common species in the Southwest Delta of The Netherlands, and illustrate the data selection, the modeling process and the results using four species. RESULTS: The strongest features to predict detection are the number of species during a visit (list length), earlier observations of the target species by the same observer, and the day of year. We compare three approaches to predict the total non-detection probability that takes all visits to an area into account. Predictions based on only the number of visits were outperformed by predictions that also take the list length into account. Our predictions based on all features combined consistently beat both other approaches, across all 10 species groups that were compared. DISCUSSION: We thus show that explicitly modelling the characteristics of all visits to an occupied area results in estimation of non-detection probabilities, while providing insight into the causes of detection and reporting bias. Furthermore, predictions of our model provide a basis for quantifying the sampling effort in each area, which is a promising first step to correct bias in citizen science data when aiming to map a species′ distribution.\n",
      "--------------------------------------------------\n",
      "ID: 21\n",
      "Titre: Unlocking the Potential of Low Quality Total RNA-seq Data: A Stepwise Mapping Approach for Improved Quantitative Analyses\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615750v1\n",
      "DOI: 10.1101/2024.09.30.615750\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615750v1.full.pdf\n",
      "Résumé: High-throughput sequencing assays face persistent challenges when analyzing low-quality RNAs, often assessed by the RNA integrity number (RIN). Current preprocessing methods and pipelines designed for mRNA-seq presume high-quality RNAs, overlooking the nuanced complexities arising from degraded transcripts in low-quality samples. This study questions the applicability of standard analysis pipelines, especially when sequencing low-quality total RNAs, which are sometimes the sole recourse for specific inquiries. To address this, we conducted a comprehensive investigation using large sequencing reads obtained from blood biospecimens with varying RINs. Introducing a novel mapping approach, termed 9stepwise mapping9, our systematic comparative analyses propose an optimal practice for total RNA-seq data analysis. Contrary to conventional mapping procedures, the 9stepwise mapping9 approach unveils additional transcriptome information, crucial for stable differential expression analysis, even with total RNA-seq data from specimens with relatively low RINs. Our method proves particularly valuable when analyzing limited specimens with low RNA quality.\n",
      "--------------------------------------------------\n",
      "ID: 22\n",
      "Titre: STIX: Long-reads based Accurate Structural Variation Annotation at Population Scale\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615931v1\n",
      "DOI: 10.1101/2024.09.30.615931\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615931v1.full.pdf\n",
      "Résumé: The prioritization of Structural Variants (SV), which is needed to rank and identify potential pathogenic alleles, is still in its infancy. This is exemplified over gnomAD only being able to annotate 33.5% of GIAB SVs. To overcome this, we present the first long-read based annotation resource for both GRCh38 and CHM13-T2T reference using STIX. In contrast to previous methods, STIX indexes SV-informative long-reads themselves, can thus be easily extended and accurately annotate all SV types including insertions. STIX successfully annotated 95.9% of GIAB Tier1 SVs. STIX further improved cancer based SV prioritization by highlighting 3,563 SV from COSMIC being common in the population. We further showcase that mosaic SV can be independently gained and may be widely spread throughout the population. This highlights the need for accurate SV population frequency annotation to further facilitate the adoption of SV via long-read sequencing in medical research and clinical applications.\n",
      "--------------------------------------------------\n",
      "ID: 23\n",
      "Titre: A patient-centric knowledge graph approach to prioritize mutants for selective anti-cancer targeting\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615658v1\n",
      "DOI: 10.1101/2024.09.29.615658\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615658v1.full.pdf\n",
      "Résumé: Personalized oncology has revolutionized cancer treatment by targeting specific genetic aberrations in tumors. However, the identification of suitable targets for anti-cancer therapies remains a challenge. In this study, we introduce a knowledge graph approach to prioritize cancer mutations with clinical, functional, and structural significance as potential therapeutic targets. Focusing on the human kinome, we integrate protein-protein interaction and patient-centric mutation networks to construct a comprehensive network enriched with transcriptomic, structural, and drug response data, together covering five layers of information. Moreover, we make the constructed knowledge graph publicly available, along with a plethora of scripts to facilitate further annotation and expansion of the network. Interactive visualization resources are also provided, ensuring accessibility for researchers regardless of computational expertise and enabling detailed analysis by cancer type and individual layers of information. This comprehensive resource has the potential to identify relevant mutations for targeted therapeutic interventions, thereby advancing personalized oncology and improving patient outcomes.\n",
      "--------------------------------------------------\n",
      "ID: 24\n",
      "Titre: DeepPWM-BindingNet: Unleashing Binding Prediction with Combined Sequence and PWM Features\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.08.21.609040v2\n",
      "DOI: 10.1101/2024.08.21.609040\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.08.21.609040v2.full.pdf\n",
      "Résumé: A crucial challenge in molecular biology is the prediction of DNA-protein binding interactions, which has applications in the study of gene regulation and genome functionality. In this paper, we present a novel deep-learning framework to predict DNA-protein binding interactions with increased precision and interoperability. Our proposed framework DeepPWM-BindingNet leverages the rich information encoded in Position Weight Matrices (PWMs), which capture the sequence-specific binding preferences of proteins. These PWM-derived features are seamlessly integrated into a hybrid model of convolutional recurrent neural networks (CRNNs) that extracts hierarchical features from DNA sequences and protein structures. The sequential dependencies within the sequences are captured by recurrent layers. By incorporating PWM-derived features, the model9s interpretability is improved, enabling researchers to learn more about the underlying binding mechanisms. The model9s capacity to locate crucial binding sites is improved by the incorporation of an attention mechanism that highlights crucial regions. Experiments on diverse DNA-protein interaction datasets demonstrate the proposed approach improves the predictive performance. The proposed model holds significant potential in deciphering intricate DNA-protein interactions, ultimately advancing our comprehension of gene regulation mechanisms.\n",
      "--------------------------------------------------\n",
      "ID: 25\n",
      "Titre: A new framework for evaluating model out-of-distribution for the biochemical domain\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.03.14.584508v2\n",
      "DOI: 10.1101/2024.03.14.584508\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.03.14.584508v2.full.pdf\n",
      "Résumé: Quantifying model generalization to out-of-distribution data has been a longstanding challenge in machine learning. Addressing this issue is crucial for leveraging machine learning in scientific discovery, where models must generalize to new molecules or materials. Current methods typically split data into train and test sets using various criteria, like temporal, sequence identity, scaffold, or random cross-validation; before evaluating model performance. However, with so many splitting criteria available, existing approaches offer limited guidance on selecting the most appropriate one, and they do not provide mechanisms for incorporating prior knowledge about the target deployment distribution(s). To tackle this problem, we have developed a novel metric, AU-GOOD, which quantifies expected model performance under conditions of increasing dissimilarity between train and test sets, while also accounting for prior knowledge about the target deployment distribution(s), when available. This metric is broadly applicable to biochemical entities, including proteins, small molecules, nucleic acids, or cells; as long as a relevant similarity function is defined for them. Recognizing the wide range of similarity functions used in biochemistry, we propose criteria to guide the selection of the most appropriate metric for partitioning. We also introduce a new partitioning algorithm that generates more challenging test sets, and we propose statistical methods for comparing models based on AU-GOOD. Finally, we demonstrate the insights that can be gained from this framework by applying it to two different use cases: developing predictors for pharmaceutical properties of small molecules, and using protein language models as embeddings to build biophysical property predictors. Source code is available at https://github.com/IBM/Hestia-OOD.\n",
      "--------------------------------------------------\n",
      "ID: 26\n",
      "Titre: SMARTER-database: a tool to integrate SNP array datasets for sheep and goat breeds\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.01.610681v2\n",
      "DOI: 10.1101/2024.09.01.610681\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.01.610681v2.full.pdf\n",
      "Résumé: Underutilized sheep and goat breeds have the ability to adapt to challenging environments due to their genetic composition. Integrating publicly available genomic datasets with new data will facilitate genetic diversity analyses; however, this process is complicated by important data discrepancies, such as outdated assembly versions or different data formats. Here we present the SMARTER-database, a collection of tools and scripts to standardize genomic data and metadata mainly from SNP chips arrays on global small ruminant populations with a focus on reproducibility. SMARTER-database harmonizes genotypes for about 12,000 sheep and 6,000 goats to a uniform coding and assembly version. Users can access the genotype data via FTP and interact with the metadata through a web interface or programmatically using their custom scripts, enabling efficient filtering and selection of samples. These tools will empower researchers to focus on the crucial aspects of adaptation and contribute to livestock sustainability, leveraging the rich dataset provided by the SMARTER-database.\n",
      "--------------------------------------------------\n",
      "ID: 27\n",
      "Titre: Novel AI-powered computational method using tensor decomposition can discover the common optimal bin sizes when integrating multiple Hi-C datasets\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615651v1\n",
      "DOI: 10.1101/2024.09.29.615651\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615651v1.full.pdf\n",
      "Résumé: iscovering the optimal bin sizes (or resolutions) for the integration of multiple Hi-C datasets is difficult, since bin sizes must be common over multiple datasets whereas dependence of quality upon bin sizes can vary from dataset to dataset. Moreover, we are not supposed to seek common structures in the smaller bin sizes than optimal bin sizes if we cannot suppress the appearance of phase transition-like phenomena even after increasing the number of mapped short reads per bins, since it might mean that there are no common structures any more in finer resolutions and individual Hi-C datasets might have to be analyzed separately. Thus, quality assessments of individual datasets have the limited ability to determine the best bin size common for all datasets. In this study, we propose, the first-ever to our knowledge, a novel adoption of tensor decomposition (TD) based unsupervised feature extraction (FE) to choose the optimal bin sizes for the integration of multiple Hi-C datasets. The use of TD-based unsupervised FE can exhibit something like phase-transition phenomena, by which we can automatically estimate the possible smallest bin size (or the highest resolution) empirically without manually setting some threshold value by hand for the integration of multiple Hi-C datasets.\n",
      "--------------------------------------------------\n",
      "ID: 28\n",
      "Titre: A systematic comparison of computational methods for expression forecasting\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.07.28.551039v2\n",
      "DOI: 10.1101/2023.07.28.551039\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.07.28.551039v2.full.pdf\n",
      "Résumé: Expression forecasting methods use machine learning models to predict how a cell will alter its transcriptome upon perturbation. Such methods are enticing because they promise to answer pressing questions in fields ranging from developmental genetics to cell fate engineering and because they are a fast, cheap, and accessible complement to the corresponding experiments. However, the absolute and relative accuracy of these methods is poorly characterized, limiting their informed use, their improvement, and the interpretation of their predictions. To address these issues, we created a benchmarking platform that combines a panel of 11 large-scale perturbation datasets with an expression forecasting software engine that encompasses or interfaces to a wide variety of methods. We used our platform to systematically assess methods, parameters, and sources of auxiliary data, finding that performance strongly depends on the choice of metric, and especially for simple metrics like mean squared error, it is uncommon for expression forecasting methods to out-perform simple baselines. Our platform will serve as a resource to improve methods and to identify contexts in which expression forecasting can succeed.\n",
      "--------------------------------------------------\n",
      "ID: 29\n",
      "Titre: Quantifying Cancer Drug Targetability across Biochemical Pathways\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.615497v1\n",
      "DOI: 10.1101/2024.09.27.615497\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.615497v1.full.pdf\n",
      "Résumé: The evaluation of cancer drug treatments within the complex context of biological networks and pathways is challenging. Drugs can be \"promiscuous,\" targeting multiple genes or pathways, not just the ones of interest. This complexity makes it difficult to assess the full impact of a drug solely based on individual knowledgebases. We make the impact more comprehensible through the characterization of drug targetability in terms of a combination of biochemical pathway data from the Reactome Knowledgebase and drug-target interaction data from the Cancer Targetome. This leads to a graphical representation of drug targetability. We further improve comprehensibility by developing metrics to quantify the graph structures in order to evaluate drug target specificity and navigate the hierarchical nature of biological networks and identify candidate treatment options to increase specificity and minimize off-target effects. We have developed software that calculates these metrics for drugs in the Cancer Targetome. The software is available for research purposes only at https://github.com/sehagler/drug_targetability.\n",
      "--------------------------------------------------\n",
      "ID: 30\n",
      "Titre: Deep learning-based predictions of gene perturbation effects do not yet outperform simple linear methods\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.16.613342v3\n",
      "DOI: 10.1101/2024.09.16.613342\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.16.613342v3.full.pdf\n",
      "Résumé: Advanced deep-learning methods, such as transformer-based foundation models, promise to learn representations of biology that can be employed to predict in silico the outcome of unseen experiments, such as the effect of genetic perturbations on the transcriptomes of human cells. To see whether current models already reach this goal, we benchmarked two state-of-the-art foundation models and one popular graph-based deep learning framework against deliberately simplistic linear models in two important use cases: For combinatorial perturbations of two genes for which only data for the individual single perturbations have been seen, we find that a simple additive model outperformed the deep learning-based approaches. Also, for perturbations of genes that have not yet been seen, but which may be \"interpolated\" from biological similarity or network context, a simple linear model performed as good as the deep learning-based approaches. While the promise of deep neural networks for the representation of biological systems and prediction of experimental outcomes is plausible, our work highlights the need for critical benchmarking to direct research efforts that aim to bring transfer learning to biology.\n",
      "--------------------------------------------------\n",
      "ID: 31\n",
      "Titre: An Efficient Computational Method to Create Positive NIPT Samples with Autosomal Trisomy\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.11.24.568620v4\n",
      "DOI: 10.1101/2023.11.24.568620\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.11.24.568620v4.full.pdf\n",
      "Résumé: Noninvasive prenatal test (NIPT) has been widely used for screening trisomy on chromosomes 13 (T13), 18 (T18), and 21 (T21). However, the false negative rate of NIPT algorithms has not been thoroughly evaluated due to the lack of positive samples. In this study, we present an efficient computational approach to create positive samples with autosomal trisomy from negative samples. We applied the approach to establish a low coverage dataset of 1440 positive samples with T13, T18, and T21 aberrations for both mosaic and non-mosaic conditions. We examined the performance of WisecondorX and its improvement, called VINIPT, on both negative and positive datasets. Experiments showed that WisecondorX and VINIPT with a z-score threshold of 3.3 were able to detect all non-mosaic samples with T13, T18, and T21 aberrations (i.e., the sensitivity of 100%). Using a lower z-score threshold of 2.58 when analyzing mosaic samples, both WisecondorX and VINIPT have the overall sensitivity of 99.7% on detecting T13, T18, and T21 aberrations from mosaic samples. WisecondorX has the specificity of 98.5% for the non-mosaic analysis, but a considerably lower specificity of 95% for the mosaic analysis. VINIPT has a much better specificity than WisecondorX, i.e., 99.9% for the non-mosaic analysis, and 98.2% for the mosaic analysis.  The results suggest that VINIPT can play as a powerful NIPT tool for the low coverage data.\n",
      "--------------------------------------------------\n",
      "ID: 32\n",
      "Titre: Impact of Interval Censoring on Data Accuracy and Machine Learning Performance in Biological High-Throughput Screening\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.25.615059v2\n",
      "DOI: 10.1101/2024.09.25.615059\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.25.615059v2.full.pdf\n",
      "Résumé: High-throughput screening (HTS) combined with deep mutational scanning (DMS) and next-generation DNA sequencing (NGS) have great potential to accelerate discovery and optimization of biological therapeutics. Typical workflows involve generation of a mutagenized variant library, screening/selection of variants based on phenotypic fitness, and comprehensive analysis of binned variant populations by NGS. However, in such cases, the HTS data are subject to interval censoring, where each fitness value is calculated based on the assignment of variants to bins. Such censoring leads to increased uncertainty, which can impact data accuracy and, consequently, the performance of machine learning (ML) algorithms tasked with predicting sequence-fitness pairings. Here, we investigated the impact of interval censoring on data quality and ML performance in biological HTS experiments. We theoretically analyzed the impact of data censoring and propose a dimensionless number, theRatio of Discretization(RD), to assist in optimizing HTS parameters such as the bin width and the sampling size. This approach can be used to minimize errors in fitness prediction by ML and to improve the reliability of these methods. These findings are not limited to biological HTS techniques and can be applied to other systems where interval censoring is an advantageous measurement strategy.\n",
      "--------------------------------------------------\n",
      "ID: 33\n",
      "Titre: Profiling drug sensitivity of leukemic stem cells via bulk-to-single-cell deconvolution\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.05.10.540140v2\n",
      "DOI: 10.1101/2023.05.10.540140\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.05.10.540140v2.full.pdf\n",
      "Résumé: Ex-vivo drug sensitivity screening allows the prediction of cancer treatment effectiveness in a personalized fashion. However, it only provides a readout on mixtures of cells, potentially occulting important information on clinically relevant cell subtypes. To address this shortcoming, we developed a machine-learning framework to decompose drug sensitivity recorded at the bulk level into cell subtype-specific drug sensitivity. We first determined that our method could decipher the cellular composition of bulk samples with top-ranking accuracy across five cancer types compared to state-of-the-art bulk deconvolution methods. We emphasize its effectiveness in the realm of Acute Myeloid Leukemia, where it appears to offer the most precise estimation of leukemic stem cell fractions across three test datasets and three patient cohorts. We then optimized an algorithm capable of estimating cell subtype- and single-cell-specific drug sensitivity, which we evaluated by performing in-vitro drug studies and in-depth simulations. We then applied our deconvolution strategy to the beatAML cohort dataset, currently the most extensive database of ex-vivo drug screening data. We developed a drug sensitivity profile tailored to specific cell subtypes, focusing on four therapeutic compounds predicted to target leukemic stem cells: the previously known midostaurin and A-674563, as well as SNS-032 and foretinib, which have not been previously linked to leukemic stem cells. Our work provides an attractive new computational tool for drug development and precision medicine.\n",
      "--------------------------------------------------\n",
      "ID: 34\n",
      "Titre: R2DT: a comprehensive platform for visualising RNA secondary structure\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.611006v1\n",
      "DOI: 10.1101/2024.09.29.611006\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.611006v1.full.pdf\n",
      "Résumé: RNA secondary (2D) structure visualisation is an essential tool for understanding RNA function. R2DT is a software package designed to visualise RNA 2D structures in consistent, recognisable, and reproducible layouts. The latest release, R2DT 2.0, introduces multiple significant features, including the ability to display position-specific information, such as single nucleotide polymorphisms (SNPs) or SHAPE reactivities. It also offers a new template-free mode allowing visualisation of RNAs without pre-existing templates, alongside a constrained folding mode and support for animated visualisations. Users can interactively modify R2DT diagrams, either manually or using natural language prompts, to generate new templates or create publication-quality images. Additionally, R2DT features faster performance, an expanded template library, and a growing collection of compatible tools and utilities. Already integrated into multiple biological databases, R2DT has evolved into a comprehensive platform for RNA 2D visualisation, accessible at https://r2dt.bio.\n",
      "--------------------------------------------------\n",
      "ID: 35\n",
      "Titre: Chromosome-level genome assembly and annotation of the crested gecko, Correlophus ciliatus, a lizard incapable of tail regeneration\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.28.615630v1\n",
      "DOI: 10.1101/2024.09.28.615630\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.28.615630v1.full.pdf\n",
      "Résumé: The vast majority of gecko species are capable of tail regeneration, but singular geckos of Correlophus, Uroplatus, and Nephrurus genera are unable to regrow lost tails. Of these non-regenerative geckos, the crested gecko (Correlophus ciliatus) is distinguished by ready availability, ease of care, high productivity, and hybridization potential. These features make C. ciliatus particularly suited as a model for studying the genetic, molecular, and cellular mechanisms underlying loss of tail regeneration capabilities. We report a contiguous genome of C. ciliatus with a total size of 1.65 Gb, a total of 152 scaffolds, L50 of 6, and N50 of 109 Mb. Repetitive content consists of 40.41% of the genome, and a total of 30,780 genes were annotated. Assembly of the crested gecko genome provides a valuable resource for future comparative genomic studies between non-regenerative and regenerative geckos and other squamate reptiles.\n",
      "--------------------------------------------------\n",
      "ID: 36\n",
      "Titre: Distinct gene regulatory networks govern hematopoietic and leukemia stem cells\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.28.614980v1\n",
      "DOI: 10.1101/2024.09.28.614980\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.28.614980v1.full.pdf\n",
      "Résumé: The underlying gene regulatory networks (GRN) that govern leukemia stem cells (LSC) in acute myeloid leukemia (AML) and hematopoietic stem cells (HSC) are not well understood. Here, we identified GRNs by integrating gene expression (GE) and chromatin accessibility data derived from functionally defined cell populations enriched for HSC and LSC. We analyzed n=32 LSC+ and n=32 LSC- cell fractions from n=22 AML patients, along with n=7 stem and n=10 progenitor enriched cell populations sorted from human umbilical cord blood (hUCB), producing a database of n≈17,000 transcription factor (TF) regulatory interactions for hUCB-HSPC and AML. We developed an iterative algorithm that associates the degree of chromatin openness with TF binding preferences, and the GE of candidate TF and target genes within 100kb upstream of transcription start sites. A putative regulatory structure was found to be enriched in HSC-enriched cell populations, comprising TF-target gene interactions between ETS1, EGR1, RUNX2, and ZNF683 oriented in a self-reinforcing configuration. A regulatory loop comprising FOXK1 and MEIS1, rather than the 4-factor HSC subnetwork, was detected in the LSC-specific GRN. The core HSC and LSC TF networks were extended using protein-protein interaction (PPI) data to determine connectivity with interacting genes whose expression strongly associated with LSC/HSC frequency estimates, producing a database of n=103,516 PPI target pathways. The effect of perturbing genes along the identified pathways on functional HSC and LSC frequency was predicted based on statistical regression analyses. To validate GRN predictions, we used pharmacologic and CRISPR targeting, in addition to re-examining published functional data associated with several network nodes that were predicted to impact stemness. Notably, we found that inhibition of CDK6 in AML samples markedly reduced LSC numbers as assessed in de novo serial xenotransplantation studies (fold change ≈ 10), as predicted by the LSC GRN model. Additionally, in-house CRISPR-based knockdown of ETS1 resulted in a significant decrease in HSC quiescence-associated microRNA-126 expression, and increased HSC frequency. Taken together, our models provide a comprehensive view of the underlying regulatory structures governing functional human HSC and LSC. This approach has translational potential as it can be used as a high-throughput in-silico screening tool for the systematic identification of gene targets for LSC elimination and HSC expansion.\n",
      "--------------------------------------------------\n",
      "ID: 37\n",
      "Titre: FragLlama: Next-fragment prediction for molecular design\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.28.615626v1\n",
      "DOI: 10.1101/2024.09.28.615626\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.28.615626v1.full.pdf\n",
      "Résumé: The emergence of ChatGPT has drawn significant attention to Large Language Models (LLMs) due to their impressive performance. While LLMs primarily focus on next token/word prediction, we apply this principle to molecular design by reframing the task as predicting the next token/fragment. We present FragLlama, a large language model trained for molecular design, featuring custom tokens that represent molecular fragments and functional groups. The model is for generating molecules given one or two fragments, for application scenarios like general hit-to-lead and lead optimization stage drug design, PROTAC linker design; mapping to commonly used drug design strategies like fragment growing and scaffold hopping. In the pre-training stage, we adapted the Llama 3 architecture to create FragLlama, training it to learn conditional probabilities of these fragment-level tokens. The subsequent alignment stage employed fine-tuning to guide the model towards generating molecules with desired properties. The effectiveness of FragLlama is demonstrated through its applications in designing molecular glue libraries, PROTAC linkers and EGFR binders. FragLlama demonstrates proficiency in reproducing expert-level designs while also exploring novel and promising chemical spaces, highlighting its potential to augment the capabilities of medicinal chemists in drug design.\n",
      "--------------------------------------------------\n",
      "ID: 38\n",
      "Titre: On Long-Term Species Coexistence in Five-Species Evolutionary Spatial Cyclic Games with Ablated and Non-Ablated Dominance Networks\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.615336v1\n",
      "DOI: 10.1101/2024.09.27.615336\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.615336v1.full.pdf\n",
      "Résumé: I present a replication and, to some extent, a refutation of key results published by Zhong, Zhang, Li, Dai, & Yang in their 2022 paper \"Species coexistence in spatial cyclic game of five species\" (Chaos, Solitons and Fractals, 156: 111806), where ecosystem species coexistence was explored via simulation studies of the evolutionary spatial cyclic game (ESCG) Rock-Paper-Scissors-Lizard-Spock (RPSLS) with certain predator-prey relationships removed from the game9s \"interaction structure\", i.e. with specific arcs ablated in the ESCG9s dominance network, and with the ESCG run for 100,000 Monte Carlo Steps (MCS) to identify its asymptotic behaviors. I replicate the results presented by Zhong et al. for interaction structures with one, two, three, and four arcs ablated from the dominance network. I then empirically demonstrate that the dynamics of the RPSLS ESCG have sufficiently long time constants that the true asymptotic outcomes can often only be identified after running the ablated ESCG for 10,000,000 MCS or longer, and that the true long-term outcomes can be markedly less diverse than those reported by Zhong et al. as asymptotic.  Finally I demonstrate that, when run for sufficiently many MCS, the original unablated RPSLS system exhibits essentially the same asymptotic outcomes as the ablated RPSLS systems, and in this sense the only causal effect of the ablations is to alter the time required for the system to converge to the long-term asymptotic states that the unablated system eventually settles to anyhow.\n",
      "--------------------------------------------------\n",
      "ID: 39\n",
      "Titre: Allo-Allo: Data-efficient prediction of allosteric sites\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.28.615583v1\n",
      "DOI: 10.1101/2024.09.28.615583\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.28.615583v1.full.pdf\n",
      "Résumé: Allostery, a fundamental structural mechanism where ligand binding at a protein site affects protein function at another site, plays a crucial role in key drug-target proteins like GPCRs. Unfortunately, existing methods for predicting allosteric sites have limited performance-- they are particularly constrained by scarce ground-truth experimental data. We introduce Allo-Allo, a data-efficient, sequence-based method that predicts allosteric sites by leveraging protein language models (PLMs). Honing in on ESM-2 attention heads that capture allosteric residue associations, Allo-Allo achieves a 67% higher AUPRC than state-of-the-art methods. Our innovative, data-efficient pipeline not only outperforms alternate, commonly-used PLM-based prediction architectures but also generalizes well. Notably, mutations in Allo-Allo-predicted sites show significant association with elevated disease risk scores from AlphaMissense, highlighting its translational potential. Beyond Allo-Allo9s biological and translational applicability, its architecture presents a powerful framework for other data-scarce problems in protein analysis.\n",
      "--------------------------------------------------\n",
      "ID: 40\n",
      "Titre: LightMed: A Light-weight and Robust FFT-Based Model for Adversarially Resilient Medical Image Segmentation\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.28.615584v1\n",
      "DOI: 10.1101/2024.09.28.615584\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.28.615584v1.full.pdf\n",
      "Résumé: Accurate and reliable medical image segmentation is essential for computer-aided diagnosis and formulating appropriate treatment plans. However, real-world challenges such as suboptimal image quality and computational resource constraints hinder the effective deployment of deep learning-based segmentation models. To address these issues, we propose LightMed, a novel efficient neural architecture based on Fast Fourier Transform (FFT). Different from prior works, our model directly learns on the frequency domain, harnessing its resilience to noise and uneven brightness, which common artifacts found in medical images. By focusing on low-frequency image components, we significantly reduce computational complexity while preserving essential image features. Our deep learning architecture extracts discriminative features directly from the Fourier domain, leading to improved segmentation accuracy and robustness compared to traditional spatial domain methods. Additionally, we propose a new benchmark incorporating various levels of Gaussian noise to assess susceptibility to noise attacks. The experimental results demonstrate that LightMed not only effectively eliminates noise and consistently achieves accurate image segmentation but also shows robust resistance to imperceptible adversarial attacks compared to other baseline models. Our new benchmark datasets and source code are publicly available at https://github.com/HySonLab/LightMed\n",
      "--------------------------------------------------\n",
      "ID: 41\n",
      "Titre: Improved robustness to gene tree incompleteness, estimation errors, and systematic homology errors with weighted TREE-QMC\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.615467v1\n",
      "DOI: 10.1101/2024.09.27.615467\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.615467v1.full.pdf\n",
      "Résumé: Summary methods are widely used to reconstruct species trees from gene trees while accounting for incomplete lineage sorting; however, it is increasingly recognized that their accuracy can be negatively impacted by incomplete and/or error-ridden  gene trees. To address the latter, Zhang and Mirarab (2022) leverage gene tree branch lengths and support values to weight quartets within the popular summary method ASTRAL. Although these quartet weighting schemes improved the robustness of ASTRAL to gene tree estimation error, implementing the weighting schemes presented computational challenges, resulting in the authors abandoning ASTRAL9s original search algorithm (i.e., computing an exact solution within a constrained search space) in favor of search heuristics (i.e., hill climbing with nearest neighbor interchange moves from a starting tree constructed via randomized taxon addition). Here, we show that these quartet weighting schemes can be leveraged within the Quartet Max Cut framework of Snir and Rao (2010), with only a small increase in time complexity compared to the unweighted algorithm, which behaves more like a constant factor in our simulation study. Moreover, our new algorithm, implemented within the TREE-QMC software, was highly competitive with weighted ASTRAL, even outperforming it in terms of species tree accuracy on some challenging model conditions, such as large numbers of taxa. In comparing unweighted and weighted summary methods on two avian data sets, we found that weighting quartets by gene tree branch lengths improves their robustness to \\textit{systematic} homology errors and is as effective as removing the impacted taxa from individual gene trees or removing the impacted gene trees entirely. Lastly, our study revealed that TREE-QMC is highly robust to high rates of missing data and is promising as a supertree method. TREE-QMC is written in C++ and is publicly available on Github: https://github.com/molloy-lab/TREE-QMC\n",
      "--------------------------------------------------\n",
      "ID: 42\n",
      "Titre: MPCD Index for Hepatocellular Carcinoma Patients Based on Mitochondrial Function and Cell Death Patterns\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.615369v1\n",
      "DOI: 10.1101/2024.09.27.615369\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.615369v1.full.pdf\n",
      "Résumé: Hepatocellular carcinoma (HCC) is a highly heterogeneous cancer with a poor prognosis. During the development of cancer cells, mitochondria influence various cell death patterns by regulating metabolic pathways such as oxidative phosphorylation. However, the relationship between mitochondrial function and cell death patterns in HCC remains unclear. In this study, we used a comprehensive machine learning framework to construct a mitochondrial functional activity-associated programmed cell death index (MPCDI) based on scRNA-seq and RNA-seq data from TCGA, GEO, and ICGC datasets. The index signature was used to classify HCC patients, and studied the multi-omics features, immune microenvironment, and drug sensitivity of the subtypes. Finally, we constructed the MPCDI signature consisting of four genes (S100A9, FYN, LGALS3, and HMOX1), which was one of the independent risk factors for the prognosis of HCC patients. The HCC patients were divided into high- and low-MPCDI groups, and the immune status was different between the two groups. Patients with high MPCDI had higher TIDE scores and poorer responses to immunotherapy, suggesting that high-MPCDI patients might not be suitable for immunotherapy. By analyzing the drug sensitivity data of CTRP, GDSC, and PRISM databases, it was found that staurosporine has potential therapeutic significance for patients with high MPCDI. In summary, based on the characteristics of mitochondria function and PCD patterns, we used single-cell and transcriptome data to identify four genes and construct the MPCDI signature, which provided new perspectives and directions for the clinical diagnosis and personalized treatment of HCC patients.\n",
      "--------------------------------------------------\n",
      "ID: 43\n",
      "Titre: INSTINCT: Multi-sample integration of spatial chromatin accessibility sequencing data via stochastic domain translation\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.05.26.595944v2\n",
      "DOI: 10.1101/2024.05.26.595944\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.05.26.595944v2.full.pdf\n",
      "Résumé: Recent advances in spatial epigenomic techniques have given rise to spatial assay for transposase-accessible chromatin using sequencing (spATAC-seq) data, enabling the characterization of epigenomic heterogeneity and spatial information simultaneously. Integrative analysis of multiple spATAC-seq samples, for which no method has been developed, allows for effective identification and elimination of unwanted non-biological factors within the data, enabling comprehensive exploration of tissue structures and providing a holistic epigenomic landscape, thereby facilitating the discovery of biological implications and the study of regulatory processes. In this article, we present INSTINCT, a method for multi-sample INtegration of Spatial chromaTIN accessibility sequencing data via stochastiC domain Translation. INSTINCT can efficiently handle the high dimensionality of spATAC-seq data and eliminate the complex noise and batch effects of samples from different conditions through a stochastic domain translation procedure. We demonstrate the superiority and robustness of INSTINCT in integrating spATAC-seq data across multiple simulated scenarios and real datasets. Additionally, we highlight the advantages of INSTINCT in spatial domain identification, visualization, spot-type annotation, and various downstream analyses, including motif enrichment analysis, expression enrichment analysis, and partitioned heritability analysis.\n",
      "--------------------------------------------------\n",
      "ID: 44\n",
      "Titre: Identifying similar populations across independent single cell studies without data integration\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.615367v1\n",
      "DOI: 10.1101/2024.09.27.615367\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.615367v1.full.pdf\n",
      "Résumé: Supervised and unsupervised methods have emerged to address the complexity of single cell data analysis in the context of large pools of independent studies. Here, we present ClusterFoldSimilarity (CFS), a novel statistical method design to quantify the similarity between cell groups acroos any number of independent datasets, without the need for data correction or integration. By bypassing these processes, CFS avoids the introduction of artifacts and loss of information, offering a simple, efficient, and scalable solution. This method match groups of cells that exhibit conserved phenotypes across datasets, including different tissues and species, and in a multimodal scenario, including single-cell RNA-Seq, ATAC-Seq, single-cell proteomics, or, more broadly, data exhibiting differential abundance effects among groups of cells. Additionally, CFS performs feature selection, obtaining cross-dataset markers of the similar phenotypes observed, providing an inherent interpretability of relationships between cell populations. To showcase the effectiveness of our methodology we generated single-nuclei RNA-Seq data from the motor cortex and spinal cord of adult mice. By using CFS, we identified three distinct sub-populations of astrocytes conserved on both tissues. CFS includes various visualization methods for the interpretation of the similarity scores and similar cell populations.\n",
      "--------------------------------------------------\n",
      "ID: 45\n",
      "Titre: Plasmid conjugation drives within-patient plasmid diversity\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.615342v1\n",
      "DOI: 10.1101/2024.09.27.615342\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.615342v1.full.pdf\n",
      "Résumé: Plasmids are well known vehicles of antimicrobial resistance (AMR) genes dissemination. Through conjugation, plasmid-encoded AMR genes are spread among neighbouring bacteria, irrespective of their strain or even their species. This process is very concerning from a public health perspective, as plasmid-borne AMR gene outbreaks are often not confined to single species or strains and are therefore more difficult to be fully uncovered. At the moment, the impact of plasmid conjugation on within-patient plasmid diversity is not well understood. In this work we will tackle the role of conjugation on within-patient plasmid diversity using a dataset of carbapenemase-producing Enterobacterales (CPEs). The dataset of 256 sequences from 115 patients was sampled across England over 30 months. Each patient has more than one sequence, with at least one sequence carrying an OXA-48 gene, a well-known plasmid-borne carbapenemase-encoding gene. If more than one sequence carries the OXA-48 gene, they are carried on different bacterial hosts. Using a hybrid de novo-on-reference assembly pipeline, we were able to reconstruct the full OXA-48 plasmid from short read sequencing data for 232 of the 256 sequences. Of the 115 patients, 83 (72%) of patients had an identical OXA-48 plasmid in two or more sequences. Only 2 patients carried very different (>200 SNPs) alleles of the OXA-48 plasmid, probably from separate acquisitions. Our study shows that when more than one bacterial host carrying an OXA-48 plasmid is found in a patient, it is most likely that the same plasmid has been shared via conjugation. The event of separate acquisition of different plasmids in different bacterial hosts is highly unlikely in our dataset.\n",
      "--------------------------------------------------\n",
      "ID: 46\n",
      "Titre: gLinDA: global differential abundance analysis of microbiomes\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615668v1\n",
      "DOI: 10.1101/2024.09.29.615668\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615668v1.full.pdf\n",
      "Résumé: Microbiome composition plays a significant role in various diseases, including cancer, obesity, inflammatory bowel disease, and mental health disorders. Understanding the differences in microbial abundance between patients is key to uncovering the microbiome9s impact on these conditions. Differential abundance analysis (DAA) can detect significant changes in microbiomes at the taxa level base. However, since individuals have unique microbial fingerprints that can potentially identify them, microbiome data must be treated as sensitive. We introduce gLinDA, a global differential abundance analysis software offering a swarm learning approach to analyzing distributed datasets while maintaining result accuracy without sharing patients9 raw data.\n",
      "--------------------------------------------------\n",
      "ID: 47\n",
      "Titre: stFormer: a foundation model for spatial transcriptomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.615337v1\n",
      "DOI: 10.1101/2024.09.27.615337\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.615337v1.full.pdf\n",
      "Résumé: Here we present stFormer, a foundation model which incorporates ligand genes within spatial niches into Transformer encoders of single-cell transcriptomics. We demonstrate that despite a moderate pretraining data size, the spatially informed gene representations generated by stFormer more consistently cluster the cells, more accurately encode hierarchy and membership within receptor-dependent gene networks, remarkably boost identification of ligand-receptor interaction pairs, and could simulate perturbation effects of ligand-receptor interactions on downstream targets.\n",
      "--------------------------------------------------\n",
      "ID: 48\n",
      "Titre: In Silico identification and modelling of FDA-approved drugs targeting T-type calcium channels\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.615366v1\n",
      "DOI: 10.1101/2024.09.27.615366\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.615366v1.full.pdf\n",
      "Résumé: Studies have shown that inhibition of the Cav3.1 T-type calcium channel can prevent or suppress neurological diseases, such as epileptic seizures and diabetic neuropathy. In this study we aimed to use in silico simulations to identify a U.S. Food and Drug Administration (FDA)-approved drug that can bind to the Cav3.1 T-type calcium channel. We used the automated docking suite GOLD v5.5 with the genetic algorithm to simulate molecular docking and predict the protein-ligand binding modes, and the ChemPLP empirical scoring function to estimate the binding affinities of 2,115 FDA-approved drugs to the human Cav3.1 channel. Drugs with high binding affinity and appropriate pharmacodynamic and pharmacokinetic properties were selected for molecular mechanics Poisson–Boltzmann surface area (MMPBSA) and molecular mechanics generalised Born surface area (MMGBSA) binding free energy calculations, GROMACS molecular dynamics (MD) simulations and Monte Carlo Cell (MCell) simulations. The docking results indicated that the FDA-approved drug montelukast has a high binding affinity to Cav3.1, and data from the literature suggested that montelukast has the appropriate drug-like properties to cross the human blood-brain barrier and reach synapses in the central nervous system. MMPBSA, MMGBSA and MD simulations showed the high stability of the montelukast-Cav3.1 complex. MCell simulations indicated that the blockage of Cav3.1 by montelukast reduced the number of synaptic vesicles being released from the pre-synaptic region to the synaptic cleft, which may reduce the probability and amplitude of postsynaptic potentials.\n",
      "--------------------------------------------------\n",
      "ID: 49\n",
      "Titre: Benchmarking copy number aberrations inference tools using single-cell multi-omics datasets\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.26.615284v1\n",
      "DOI: 10.1101/2024.09.26.615284\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.26.615284v1.full.pdf\n",
      "Résumé: Copy number aberrations (CNAs) are an important type of genomic variation which play a crucial role in the initiation and progression of cancer. With the explosion of single-cell RNA sequencing (scRNA-seq), several computational methods have been developed to infer CNAs from scRNA-seq studies. However, to date, no independent studies have comprehensively benchmarked their performance. Herein, we evaluated five state-of-the-art methods based on their performance in tumor vs normal cell classification, CNAs profile accuracy, tumor subclone inference and aneuploidy identification in non-malignant cells. Our results showed that Numbat outperformed others across most evaluation criteria, while CopyKAT excelled in scenarios when expression matrix alone was used as input. Additionally, we investigated how referencing settings, inclusion of tumor microenvironment cells, tumor type, and tumor purity impact the performance of these tools. This study provides a valuable guideline for researchers in selecting the appropriate methods for their datasets.\n",
      "--------------------------------------------------\n",
      "ID: 50\n",
      "Titre: EmbedSimScore: Advancing Protein Similarity Analysis with Structural and Contextual Embeddings\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.26.615264v1\n",
      "DOI: 10.1101/2024.09.26.615264\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.26.615264v1.full.pdf\n",
      "Résumé: Accurately computing protein similarity is challenging due to the intricate interplay between local substructures and the global structure within protein molecules. Traditional metrics like TM-score often focus on aligning the global structures of the proteins in a rather geometry-based algorithmic way, potentially overlooking critical local global relations and contextual comparisons. We introduce Embed-SimScore, a novel self-supervised method that generates structural and contextual embeddings by jointly considering both local substructures and global proteins9s structures. Utilizing contrastive language-structure pre training (CLSP) and structural contrastive learning, EmbedSimScore captures comprehensive features across different scales of protein structure. These embeddings provide a more precise and holistic means of computing protein similarities, resulting in the identification of intrinsic relations among proteins that traditional approaches overlook.\n",
      "--------------------------------------------------\n",
      "ID: 51\n",
      "Titre: Research on the Mechanism of Lphn1 Knockout in Inhibiting Colorectal Cancer\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.08.07.606975v5\n",
      "DOI: 10.1101/2024.08.07.606975\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.08.07.606975v5.full.pdf\n",
      "Résumé: Decades ago, colorectal cancer was rarely diagnosed. Today, it is the fourth leading cause of cancer-related deaths worldwide, with nearly 90,000 fatalities each year. By analyzing single-cell data from tumor-bearing colorectal cancer model mice with Lphn1 knockout and wild-type Lphn1, we identified five key target genes for anticancer therapy: Ulbp1, Klrk1, Ccl6, Tlr4, Cd48, Prdm5, Vstm2a, Ret, Oas2, Hdac11 and Ptchd4, along with their corresponding cell types. Additionally, we discovered tumor-inhibiting cell subpopulations, including Cd244a_T_cells_subcluster_1, Cd48_Cd244a_NK_cells_subcluster_2, and C3_Macrophages_subcluster_1, which are potential candidates for therapeutic intervention. We propose that cancer-associated fibroblasts (CAFs) serve as the primary antigen presenters for MHC class I, providing antigens to macrophages, NK cells, and T cells to combat colorectal cancer. From a cellular perspective, the knockout of Lphn1 activates the anti-colorectal cancer functions of subpopulations of macrophages, NK cells, and T cells. Macrophages enhance antitumor immune activity by engaging the Ulbp1-Klrk1 receptor pair to activate NK cells. Additionally, macrophages activate downstream functions of T cells against colorectal cancer through CD48 signaling. After the knockout of Lphn1, macrophages are recruited by autocrine Ccl6 and Ccl6 secReted by CAFs. They exhibit high expression of Tlr4 and have the potential to transition into M1-type macrophages due to changes in their cellular state. After the knockout of Lphn1, the CAFs were reduced by half. CAFs, which are part of the cell network communication associated with tumor cells, typically play an immunosuppressive role. A reduction by half indicates that the immunosuppression in the Lphn1 knockout group has significantly decreased. This suggests that the efficacy of various cancer immunotherapy drugs can be significantly enhanced. In specific cell types, the four colorectal cancer-resistant transcription factors Irf7, Nr2f1 (with uncertain function), Bclaf1, and Irf2 co-localize with Tlr4, Cd48, Prdm5, and Oas2, respectively, and have been analyzed by pyscenic to interact and functionally contribute to the resistance against colorectal cancer. The four differential metabolic pathways between the Lphn1 group and the luc group are Arginine and proline metabolism, Histidine metabolism, Phenylalanine metabolism, and Riboflavin metabolism, with Arginine and proline metabolism being more active in the Lphn1 group and Histidine metabolism being more active in the luc group. These findings could open new avenues for the treatment of colorectal cancer and contribute to the development of personalized medicine.\n",
      "--------------------------------------------------\n",
      "ID: 52\n",
      "Titre: Deep learning facilitates precise identification of disease-resistance genes in plants\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.26.615248v1\n",
      "DOI: 10.1101/2024.09.26.615248\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.26.615248v1.full.pdf\n",
      "Résumé: The identification of plant disease-resistance genes is essential for understanding the plant immune system and accelerating crop breeding with disease resistance. Therefore, there is a pressing need for a method capable of accurately and comprehensively annotating resistance genes at a genome-wide scale. In this study, we propose Evolutionary Scale Modeling for LRR (ESM-LRR), a novel approach based on the deep protein language model to accurately identify LRR domains which are substantially variable structures in disease-resistance proteins. ESM-LRR achieved its highest F1 score of 0.80 on a test set using the 90% identity as the matching threshold. Building upon ESM-LRR, we developed the Plant Disease-Resistance Gene Predictor (R-Predictor), a framework designed to simultaneously annotate 15 diverse domain topologies, extensively covering recently characterised resistance genes across the entire genome. R-Predictor integrates four modules, each employing superior methods that outperform existing methods (F1 score of 0.89 for RLKs and 0.88 for NLRs), demonstrating its high accuracy and practicality in annotating disease-resistance genes (R genes). R-Predictor were also applied to identify R genes in rice, tomato, and grapevine.  Furthermore, AlphaFold3 was employed to screen interactions between 1,116 protein pairs involving three NLRs identified exclusively by R-Predictor and 372 literature-reported plant-pathogen effectors, resulting in the identification of 15 putative NLR-effector complexes. Overall, this study presents a novel approach for advancing our understanding of plant immune mechanisms.\n",
      "--------------------------------------------------\n",
      "ID: 53\n",
      "Titre: Mastering Artifact Correction in Neuroimaging Analysis: A Retrospective Approach\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.08.02.606374v2\n",
      "DOI: 10.1101/2024.08.02.606374\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.08.02.606374v2.full.pdf\n",
      "Résumé: The correction of artifacts in Magnetic Resonance Imaging (MRI) is increasingly relevant as voluntary and involuntary artifacts can hinder data acquisition. Reverting from corrupted to artifact-free images is a complex task. Deep Learning (DL) models have been employed to preserve data characteristics and to identify and correct those artifacts. We propose MOANA, a novel DL-based solution to correct artifacts in multi-contrast brain MRI scans. MOANA offers two models: the simulation and the correction models. The simulation model introduces perturbations similar to those occurring in an exam while preserving the original image as ground truth; this is required as publicly available datasets rarely have motion-corrupted images. It allows the addition of three types of artifacts with different degrees of severity. The DL-based correction model adds a fourth contrast to state-of-the-art solutions while improving the overall performance of the models. MOANA achieved the highest results in the FLAIR contrast, with a Structural Similarity Index Measure (SSIM) of 0.9803 and a Normalized Mutual Information (NMI) of 0.8030. With this, the MOANA model can correct large volumes of images in less time and adapt to different levels of artifact severity, allowing for better diagnosis\n",
      "--------------------------------------------------\n",
      "ID: 54\n",
      "Titre: Profiling ranked list enrichment scoring in sparse data elucidates algorithmic tradeoffs\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.06.03.597180v2\n",
      "DOI: 10.1101/2024.06.03.597180\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.06.03.597180v2.full.pdf\n",
      "Résumé: Gene Set Enrichment Analysis (GSEA) is a method for quantifying pathway and process activation in groups of samples, and its single sample version (ssGSEA) scores activation using mRNA abundance in a single sample. GSEA and ssGSEA were developed for bulk samples rather than individual cell technologies such as microarrays and bulk RNA-sequencing (RNA-seq) data. The growing use of single cell RNA-sequencing (scRNA-seq) raises the possibility of using ssGSEA to quantify pathway and process activation in individual cells. However, scRNA-seq data is much sparser than RNA-seq data. Here we show that ssGSEA as designed for bulk data is subject to some amount of score uncertainty and other technical issues when applied to individual cells from scRNA-seq data. We also show that a ssGSEA can be applied robustly to pseudobulk aggregate groups of a few hundred to a few thousand cells provided appropriate normalization is used. Finally, in comparing this approach to other ranked list enrichment methods, we find that the UCell method is most robust to sparsity. We have made the aggregate cell version of ssGSEA available as a Python package and GenePattern module and will also modularize UCell for use on GenePattern as well.\n",
      "--------------------------------------------------\n",
      "ID: 55\n",
      "Titre: SIMVI reveals intrinsic and spatial-induced states in spatial omics data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.08.28.554970v2\n",
      "DOI: 10.1101/2023.08.28.554970\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.08.28.554970v2.full.pdf\n",
      "Résumé: Spatial omics technologies enable the analysis of gene expression and interaction dynamics in relation to tissue structure and function. However, existing computational methods do not properly distinguish cellular intrinsic variability and intercellular interactions, thus failing to capture spatial regulations for further biological discoveries. Here, we present Spatial Interaction Modeling using Variational Inference (SIMVI), an annotation-free framework that disentangles cell intrinsic and spatial-induced latent variables for modeling gene expression in spatial omics data. We establish rigorous theoretical support for SIMVI in disentangling intrinsic and spatial-induced variations. By this disentanglement, SIMVI enables estimation of spatial effects (SE) at a single-cell resolution, and opens up various opportunities for downstream analyses. To demonstrate the power of SIMVI, we applied SIMVI to spatial omics data from diverse platforms and tissues (MERFISH human cortex, Slide-seqv2 mouse hippocampus, Slide-tags human tonsil, spatial multiome human melanoma, cohort-level CosMx melanoma). In all tested datasets, SIMVI reveal refined disentangled variations and accurate spatial effects compared with various alternative methods. Moreover, our applications of SIMVI in these datasets uncovered complex spatial regulations and dynamics of biological significance. In the human tonsil data, SIMVI characterizes the cyclical spatial dynamics of germinal center B cells during maturation. In the multiome melanoma data, SIMVI incorporates ATAC-seq measurements to discover potential tumor epigenetic reprogramming states. Application of SIMVI on our newly-collected cohort-level CosMx melanoma dataset uncovers space-outcome-dependent macrophage states and cellular communication machinery shifted across patient phenotypes.\n",
      "--------------------------------------------------\n",
      "ID: 56\n",
      "Titre: Functional protein mining with conformal guarantees\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.06.27.601042v5\n",
      "DOI: 10.1101/2024.06.27.601042\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.06.27.601042v5.full.pdf\n",
      "Résumé: Molecular structure prediction and homology detection provide a promising path to discovering new protein function and evolutionary relationships. However, current approaches lack statistical reliability assurances, limiting their practical utility for selecting proteins for further experimental and in-silico characterization. To address this challenge, we introduce a novel approach to protein search leveraging principles from conformal prediction, offering a framework that ensures statistical guarantees with user-specified risk and provides calibrated probabilities (rather than raw ML scores) for any protein search model. Our method (1) lets users select many biologically-relevant loss metrics (i.e. false discovery rate) and assigns reliable functional probabilities for annotating genes of unknown function; (2) achieves state-of-the-art performance in enzyme classification without training new models; and (3) robustly and rapidly pre-filters proteins for computationally intensive structural alignment algorithms. Our framework enhances the reliability of protein homology detection and enables the discovery of new proteins with likely desirable functional properties.\n",
      "--------------------------------------------------\n",
      "ID: 57\n",
      "Titre: MetaLab Platform Enables Comprehensive DDA and DIA Metaproteomics Analysis\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.615406v1\n",
      "DOI: 10.1101/2024.09.27.615406\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.615406v1.full.pdf\n",
      "Résumé: Metaproteomics studies the collective protein composition of complex microbial communities, providing insights into microbial roles in various environments. Despite its importance, metaproteomic data analysis is challenging due to the data9s large and heterogeneous nature. While Data-Independent Acquisition (DIA) mode enhances proteomics sensitivity, it traditionally requires Data-Dependent Acquisition (DDA) results to build the library for peptide identification. This paper introduces an updated version of MetaLab, a software solution that streamlines metaproteomic analysis by supporting both DDA and DIA modes across various mass spectrometry (MS) platforms, including Orbitrap and timsTOF. MetaLab9s key feature is its ability to perform DIA analysis without DDA results, allowing more experimental flexibility. It incorporates a deep learning strategy to train a neural network model, enhancing the accuracy and coverage of DIA results. Evaluations using diverse datasets demonstrate MetaLab9s robust performance in accuracy and sensitivity. Benchmarks from large-scale human gut microbiome studies show that MetaLab increases peptide identification by 2.7 times compared to conventional methods. MetaLab is a versatile tool that facilitates comprehensive and flexible metaproteomic data analysis, aiding researchers in exploring microbial communities9 functionality and dynamics.\n",
      "--------------------------------------------------\n",
      "ID: 58\n",
      "Titre: Molecular Evolution and Functional Divergence of Heterodisulfide Reductase A (HdrA)\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.26.615249v1\n",
      "DOI: 10.1101/2024.09.26.615249\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.26.615249v1.full.pdf\n",
      "Résumé: Heterodisulfide reductase is a key enzyme in methanogenesis pathway in archaea, which helps to recycle the coenzymes after methane is formed from C1 carbon sources. Homologous enzymes are observed in nonmethanogenic archaea and prokaryotes but their functional and evolutionary details are not fully explored. This report aims at understanding the distribution of one such homologous enzyme, heterodisulfide reductaseA; the domain architecture, overall structural contribution in biological function and evolutionary details in organisms other than methanogen. Exhaustive sequence analysis from both archaea and prokaryotic lineages reveals the presence of highly conserved homologues which probably have an altered oxidoreductase function. Phylogenetic analysis and coclustering showed the differential diversity of the homologues enzyme in metabolically diverse groups. The distribution of Heterodisulfide homologues in both archaea and prokaryotes shows differential expansion of this protein by multiple horizontal gene transfer events. Conserved domain analysis and structural modelling of different phylogenetic clades shows sequence and structural diversity at the Cterminal domain. Bimolecular network analysis highlights the functional association and interactions of the heterodisulfide homologues with diverse protein partners which participate in different metabolic function. HdrA involved in species specific metabolic function while the functional conservation is evident between close related species. Analysis of structural domain within HdrA  reveals that clade specific sequence which helps to differentiate structural domain arrangement and contributes to functional constrains. The presence of HdrA protein in methanogenic archaea and nonmethanogens suggests that the protein function involving redox reaction and electron transfer shared a common history.\n",
      "--------------------------------------------------\n",
      "ID: 59\n",
      "Titre: Unified Sampling and Ranking for Protein Docking with DFMDock\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.615401v1\n",
      "DOI: 10.1101/2024.09.27.615401\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.615401v1.full.pdf\n",
      "Résumé: Diffusion models have shown promise in addressing the protein docking problem. Traditionally, these models are used solely for sampling docked poses, with a separate confidence model for ranking. We introduce DFMDock (Denoising Force Matching Dock), a diffusion model that unifies sampling and ranking within a single framework. DFMDock features two output heads: one for predicting forces and the other for predicting energies. The forces are trained using a denoising force matching objective, while the energy gradients are trained to align with the forces. This design enables our model to sample using the predicted forces and rank poses using the predicted energies, thereby eliminating the need for an additional confidence model. Our approach outperforms the previous diffusion model for protein docking, DiffDock-PP, with a sampling success rate of 44% compared to its 8%, and a Top-1 ranking success rate of 16% compared to 0% on the Docking Benchmark 5.5 test set. In successful decoy cases, the DFMDock Energy forms a binding funnel similar to the physics-based Rosetta Energy, suggesting that DFMDock can capture the underlying energy landscape.\n",
      "--------------------------------------------------\n",
      "ID: 60\n",
      "Titre: CAMDA 2023: finding patterns in urban microbiomes\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.26.614772v1\n",
      "DOI: 10.1101/2024.09.26.614772\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.26.614772v1.full.pdf\n",
      "Résumé: The Critical Assessment of Massive Data Analysis (CAMDA) addresses the challenge of effectively utilizing Big Data in life science. Serving as both a conference and a catalyst for research groups, CAMDA annually presents challenges that foster innovative solutions. For the Forensics CAMDA 2023 challenge, we analyzed 365 metagenomic samples from 16 cities worldwide to characterize their origin. The forensic challenge was addressed from two perspectives: using the reduced abundance OTU tables and employing functional annotations. To identify the most informative Operational Taxonomic Units, we fit negative binomial models ultimately reducing variables to 294. After OTU selection, we implemented supervised models and conducted 5-fold cross-validation (CV) with a 4:1 training-to-validation ratio in each scenario. Support vector classification (SVC) achieved the highest F1 score (0.96) for the abundance tables, accurately classifying most cities, although New York City (NYC) posed a challenge. Via functional profiles with Mifaser at level 4, we achieved the best functional classification using the Neural Network (NN) model.\n",
      "\n",
      "Additionally, to gain insight into further associations between bacterial distribution with other covariates, we applied Dirichlet regression overEscherichia,Enterobacter, andKlebsiellabacteria abundances. We considered climatic and demographic variables of the cities, observing that population increase is indeed associated with a rise in the mean ofEscherichiawhile decreasing temperature is linked to higher proportions ofKlebsiella.\n",
      "\n",
      "For replicability of the scripts, a Docker container and a Conda environment are available at the repository: \n",
      "GitHub:github.com/ccm-bioinfo/cambda2023\n",
      "--------------------------------------------------\n",
      "ID: 61\n",
      "Titre: ConoDL: A Deep Learning Framework for Rapid Generation and Prediction of Conotoxins\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.614001v1\n",
      "DOI: 10.1101/2024.09.27.614001\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.614001v1.full.pdf\n",
      "Résumé: Conotoxins, being small disulfide-rich and bioactive peptides, manifest notable pharmacological potential and find extensive applications. However, the exploration of conotoxins9 vast molecular space using traditional methods is severely limited, necessitating the urgent need of developing novel approaches. Recently, deep learning (DL)-based methods have advanced to the molecular generation of proteins and peptides. Nevertheless, the limited data and the intricate structure of conotoxins constrain the application of deep learning models in the generation of conotoxins. We propose ConoDL, a framework for the generation and prediction of conotoxins, comprising the end-to-end conotoxin generation model (ConoGen) and the conotoxin prediction model (ConoPred). ConoGen employs transfer learning and a large language model (LLM) to tackle the challenges in conotoxin generation. Meanwhile, ConoPred filters artificial conotoxins generated by ConoGen, narrowing down the scope for subsequent research. A comprehensive evaluation of the peptide properties at both sequence and structure levels indicates that the artificial conotoxins generated by ConoDL exhibit a certain degree of similarity to natural conotoxins. Furthermore, ConoDL has generated artificial conotoxins with novel cysteine scaffolds. Therefore, ConoDL may uncover new cysteine scaffolds and conotoxin molecules, facilitating further exploration of the molecular space of conotoxins and the discovery of pharmacologically active variants. The code is available at https://github.com/xueww/ConoDL.The data and model are available at https://zenodo.org/records/10679280.\n",
      "--------------------------------------------------\n",
      "ID: 62\n",
      "Titre: A qualitative review of Oxford Nanopore Sequencing datasets for RNA modifications\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.26.615132v1\n",
      "DOI: 10.1101/2024.09.26.615132\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.26.615132v1.full.pdf\n",
      "Résumé: There are many oxford nanopore datasets available to study methylations. Methylations and modifications occur at nucleotides such as adenine (A), cytosine (C), guanine (G) and theanine (T) or uracil (U). Among all these provided datasets, some have the most common m6A methylation and others have m5C etc. using various real organism reference sequences such as human, mouse and artificial reference sequences which are prepared in the laboratory such as curlcake and IVT. These datasets are required to be organized by the methylation types to research ONT datasets. Here we provide a summary of the read qualities, base mapping success rates etc. for these methylation types and reference genomes. We have used minimap2 base mapping and longreadsum results. We find that methylated data have lower success rates than non-methylated data and mapping quality is lower for the real reference genomes such as human and mice. This could be because they contain more than 100,000 transcriptomes whereas artificial reference sequences contain only a few transcriptomes. Datasets which contain artificially created reference sequences have a higher quality than the others, thus they are recommended to be used for methylation or modification classification tasks in the future.\n",
      "--------------------------------------------------\n",
      "ID: 63\n",
      "Titre: FishCODE: a web-based information platform for comprehensive omics data exploration in fish research\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.25.614839v1\n",
      "DOI: 10.1101/2024.09.25.614839\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.25.614839v1.full.pdf\n",
      "Résumé: In terms of the utilization of omics data, the current fish database analysis functions are primarily relatively simple tools at the transcriptional level, aimed at obtaining the co-expression levels of specified genes or the data visualization of multiple genes, and do not enable users to perform comprehensive omics data analysis. Furthermore, the gene-level information currently provided by these multispecies fish genomics databases is incomplete, and there is a lack of a comprehensive portal that can offer multidimensional genetic information. To address these challenges, we collected extensive multi-omics information on 35 fishes and established the primary comprehensive multi-omics data information platform for fish, FishCODE (http://bioinfo.ihb.ac.cn/fishcode). We have collected experimental background of dataset which pertaining to the target fishes, selected a range of datasets that encompass a broad spectrum of research areas, and downloaded the corresponding raw omics data from public repositories such as the Sequence Read Archive (SRA). Through a unified pipeline analysis, FishCODE contains 11,216 samples from 540 sets of genomic, transcriptomic, and methylomic datasets. These data encompass transcript structure and expression, gene methylation levels, protein domains, protein subcellular localization, protein interactions, best matched protein (Swiss-Prot), associated SNP site information (47,111,018), orthologous genes, phylogenetic tree and GO/KEGG annotations. To facilitate comparison, we annotated the experimental background data sets of the FishCODE, FishGET, PhyloFish, FishSED and FishSCT databases using the Fish Experimental Condition Ontology. Currently, the FishCODE database omics dataset includes 146 unique experimental condition words, 654 cumulative experimental condition words, and 13 species with rich experimental background (more than 20 unique FECO words). These data are 3.5 times (42), 8.3 times (74), and 6.5 times (2) those of the second-ranked databases respectively. We generated word cloud maps for the experimental condition vocabularies of FishCODE and FishGET, illustrating the superior richness of FishCODE9s experimental background.\n",
      "--------------------------------------------------\n",
      "ID: 64\n",
      "Titre: Multi-View Integrative Approach For Imputing Short-Chain Fatty Acids and Identifying Key factors predicting Blood SCFA\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.25.614767v1\n",
      "DOI: 10.1101/2024.09.25.614767\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.25.614767v1.full.pdf\n",
      "Résumé: Short-chain fatty acids (SCFAs) are the main metabolites produced by bacterial fermentation of dietary fiber within gastrointestinal tract. SCFAs produced by gut microbiotas (GMs) are absorbed by host, reach bloodstream, and are distributed to different organs, thus influencing host physiology. However, due to the limited budget or the poor sensitivity of instruments, most studies on GMs have incomplete blood SCFA data, limiting our understanding of the metabolic processes within the host. To address this gap, we developed an innovative multi-task multi-view integrative approach (M2AE, Multi-task Multi-View Attentive Encoders), to impute blood SCFA levels using gut metagenomic sequencing (MGS) data, while taking into account the intricate interplay among the gut microbiome, dietary features, and host characteristics, as well as the nuanced nature of SCFA dynamics within the body. Here, each view represents a distinct type of data input (i.e., gut microbiome compositions, dietary features, or host characteristics). Our method jointly explores both view-specific representations and cross-view correlations for effective predictions of SCFAs. We applied M2AE to two in-house datasets, which both include MGS and blood SCFAs profiles, host characteristics, and dietary features from 964 subjects and 171 subjects, respectively. Results from both of two datasets demonstrated that M2AE outperforms traditional regression-based and neural-network based approaches in imputing blood SCFAs. Furthermore, a series of gut bacterial species (e.g., Bacteroides thetaiotaomicron and Clostridium asparagiforme), host characteristics (e.g., race, gender), as well as dietary features (e.g., intake of fruits, pickles) were shown to contribute greatly to imputation of blood SCFAs. These findings demonstrated that GMs, dietary features and host characteristics might contribute to the complex biological processes involved in blood SCFA productions. These might pave the way for a deeper and more nuanced comprehension of how these factors impact human health.\n",
      "--------------------------------------------------\n",
      "ID: 65\n",
      "Titre: Unipept 6.0: Expanding Metaproteomics Analysis with Support for Missed Cleavages, Semi-Tryptic and Non-Tryptic Peptides\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.26.615136v1\n",
      "DOI: 10.1101/2024.09.26.615136\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.26.615136v1.full.pdf\n",
      "Résumé: Unipept, a pioneering software tool in metaproteomics, has significantly advanced the analysis of complex ecosystems by facilitating both taxonomic and functional insights from environmental samples. From the onset, Unipept9s capabilities focused on tryptic peptides, utilizing the predictability and consistency of trypsin digestion to efficiently construct a protein reference database. However, the evolving landscape of proteomics and emerging fields like immunopeptidomics necessitate a more versatile approach that extends beyond the analysis of tryptic peptides. In this article, we present Unipept 6.0, featuring a novel protein reference database, powered by a Sparse Suffix Array index. This advancement enables the analysis of semi-tryptic peptides, peptides with missed cleavages, and non-tryptic peptides such as those from immunopeptidomics (e.g. MHC- and HLA-peptides). A benchmark study demonstrates that this version of Unipept9s performance matches or exceeds that of previous versions, while significantly expanding its application range.\n",
      "--------------------------------------------------\n",
      "ID: 66\n",
      "Titre: Seed2LP: seed inference in metabolic networks for reverse ecology applications\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.26.615309v1\n",
      "DOI: 10.1101/2024.09.26.615309\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.26.615309v1.full.pdf\n",
      "Résumé: A challenging problem in microbiology is to determine nutritional requirements of microorganisms and culture them, especially for the microbial dark matter detected solely with culture-independent methods. The latter foster an increasing amount of genomic sequences that can be explored with reverse ecology approaches to raise hypotheses on the corresponding populations. Building upon genome scale metabolic networks (GSMNs) obtained from genome annotations, metabolic models predict contextualised phenotypes using nutrient information. We developed the tool Seed2LP, addressing the inverse problem of predicting source nutrients, or seeds, from a GSMN and a metabolic objective. The originality of Seed2LP is its hybrid model, combining a scalable and discrete Boolean approximation of metabolic activity, with the numerically accurate flux balance analysis (FBA). Seed inference is highly customisable, with multiple search and solving modes, exploring the search space of external and internal metabolites combinations. Application to a benchmark of 107 curated GSMNs highlights the usefulness of a logic modelling method over a graph-based approach to predict seeds, and the relevance of hybrid solving to satisfy FBA constraints. Focusing on the dependency between metabolism and environment, Seed2LP is a computational support contributing to address the multifactorial challenge of culturing possibly uncultured microorganisms. Seed2LP is available on https://github.com/bioasp/seed2lp.\n",
      "--------------------------------------------------\n",
      "ID: 67\n",
      "Titre: Skeletons in the Forest: Using Entropy-Based Rogue Detection on Bayesian Phylogenetic Tree Distributions\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.25.615070v1\n",
      "DOI: 10.1101/2024.09.25.615070\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.25.615070v1.full.pdf\n",
      "Résumé: In a phylogenetic analysis, rogue taxa and clades are those that, due to their uncertain placement, contribute disproportionally to the variance in a set of phylogenetics trees. They can obscure clear evolutionary relationships and complicate the interpretation of results. While existing rogue detection algorithms focus on improving the consensus tree for a set of trees, we introduce algorithms based on phylogenetic entropy to improve the analysis in a Bayesian framework. In particular, we introduce a tool that extracts a skeleton distribution of the posterior tree distribution that contains the core relationships by removing a minimal subset of rogues. Additionally, we enable detailed analyses of rogues at both the taxon and clade levels, including the visualisation of different rogue placements.\n",
      "--------------------------------------------------\n",
      "ID: 68\n",
      "Titre: Leveraging cell type-specificity for gene set analysis of single cell transcriptomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.25.615040v1\n",
      "DOI: 10.1101/2024.09.25.615040\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.25.615040v1.full.pdf\n",
      "Résumé: Although single cell RNA-sequencing (scRNA-seq) provides unprecedented insights into the biology of complex tissues, analyzing such data on a gene-by-gene basis is challenging due to the large number of tested hypotheses and consequent low statistical power and difficult interpretation. These issues are magnified by the increased noise, significant sparsity and multi-modal distributions characteristic of single cell data. One promising approach for addressing these challenges is gene set testing, or pathway analysis. Unfortunately, statistical and biological differences between single cell and bulk transcriptomic data make it challenging to use existing gene set collections, which were developed for bulk tissue analysis, on scRNA-seq data. In this paper, we describe a procedure for customizing gene set collections originally created for bulk tissue analysis to reflect the structure of gene activity within specific cell types. Our approach leverages information about mean gene expression in the 81 human cell types profiled via scRNA-seq by the Human Protein Atlas (HPA) Single Cell Type Atlas. This HPA information is used to compute cell type-specific gene and gene set weights that can be used to filter or weight gene set collections. As demonstrated through the analysis of immune cell scRNA-seq data using gene sets from the Molecular Signatures Database (MSigDB), accounting for cell type-specificity can significantly improve gene set testing power and interpretability. An example vignette along with gene and gene set weights for the 81 HPA SCTA cell types and the MSigDB collections are available at https://hrfrost.host.dartmouth.edu/SCGeneSetOpt.\n",
      "--------------------------------------------------\n",
      "ID: 69\n",
      "Titre: AnalyzAIRR: A user-friendly guided workflow for AIRR data analysis\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.25.614939v1\n",
      "DOI: 10.1101/2024.09.25.614939\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.25.614939v1.full.pdf\n",
      "Résumé: The analysis of bulk adaptive immune receptor repertoires (AIRR) enables the understanding of immune responses in both normal and pathological conditions. However, the complexity of AIRR calls for advanced, specialized methods to extract meaningful biological insights. These sophisticated approaches often present challenges for researchers with limited bioinformatics expertise, hindering access to comprehensive immune system analysis. To address this challenge, we developed AnalyzAIRR, an AIRR-compliant R package enabling advanced bulk AIRR sequencing data. The tool integrates state-of-the-art statistical and visualization methods applicable at various levels of granularity. It offers a platform for general data exploration, filtering and manipulation, and in-depth cross-comparisons of AIRR datasets, aimed at answering specific biological questions. We illustrate AnalyzAIRR functionalities using of a published murine dataset of 18 T-cell receptor repertoires from three diferrent T cell subsets. We first detected and removed a major contaminant in a group of samples, before proceeding with to the comparative analysis. Subsequent cross-sample analysis revealed differences in repertoire diversity that aligned with the respective cell phenotypes, and in repertoire convergence among the studied subsets. AnalyzAIRR set of analytical metrics is integrated into a Shiny web application and complemented with a tutorial to help users in their analytical strategy, making it user-friendly for biologists with little or no background in bioinformatics.\n",
      "--------------------------------------------------\n",
      "ID: 70\n",
      "Titre: Towards improving full-length ribosome density prediction by bridging sequence and graph-based representations\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.04.08.588507v3\n",
      "DOI: 10.1101/2024.04.08.588507\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.04.08.588507v3.full.pdf\n",
      "Résumé: Translation elongation plays an important role in regulating protein concentrations in the cell, and dysregulation of this process has been linked to several human diseases. In this study, we use data from ribo-seq experiments to model ribosome densities, and in turn, predict the speed of translation. The proposed method, RiboGL, combines graph and recurrent neural networks to account for both graph and sequence-based features. The model takes a graph representing the secondary structure of the mRNA sequence as input, which incorporates both sequence and structural codon neighbors. In our experiments, RiboGL greatly outperforms the state-of-the-art RiboMIMO model for ribosome density prediction. We also conduct ablation studies to justify the design choices made in building the pipeline. Additionally, we use gradient-based interpretability to understand how the codon context and the structural neighbors affect the ribosome density at the A-site. By individually analyzing the genes in the dataset, we elucidate how structural neighbors could also potentially play a role in defining the ribosome density. Importantly, since these neighbors can be far away in the sequence, a recurrent model alone could not easily extract this information. This study lays the foundation for understanding how the mRNA secondary structure can be exploited for ribosome density prediction, and how in the future other graph modalities such as features from the nascent polypeptide can be used to further our understanding of translation in general.\n",
      "--------------------------------------------------\n",
      "ID: 71\n",
      "Titre: WIMOAD: Weighted Integration of Multi-Omics data for Alzheimer’s Disease (AD) Diagnosis\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.25.614862v1\n",
      "DOI: 10.1101/2024.09.25.614862\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.25.614862v1.full.pdf\n",
      "Résumé: As the most common subtype of dementia, Alzheimer9s disease (AD) is characterized by a progressive decline in cognitive functions, especially in memory, thinking, and reasoning ability. Early diagnosis and interventions enable the implementation of measures to reduce or slow further regression of the disease, preventing individuals from severe brain function decline. The current framework of AD diagnosis depends on A/T/(N) biomarkers detection from cerebrospinal fluid or brain imaging data, which is invasive and expensive during the data acquisition process. Moreover, the pathophysiological changes of AD accumulate in amino acids, metabolism, neuroinflammation, etc., resulting in heterogeneity in newly registered patients. Recently, next generation sequencing (NGS) technologies have found to be a non-invasive, efficient and less-costly alternative on AD screening. However, most of existing studies rely on single omics only. To address these concerns, we introduce WIMOAD, a weighted integration of multi-omics data for AD diagnosis. WIMOAD synergistically leverages specialized classifiers for patients9 paired gene expression and methylation data for multi-stage classification. The resulting scores were then stacked with MLP-based meta-models for performance improvement. The prediction results of two distinct meta-models were integrated with optimized weights for the final decision-making of the model, providing higher performance than using single omics only. Remarkably, WIMOAD achieves significantly higher performance than using single omics alone in the classification tasks. The model9s overall performance also outperformed most existing approaches, highlighting its ability to effectively discern intricate patterns in multi-omics data and their correlations with clinical diagnosis results. In addition, WIMOAD also stands out as a biologically interpretable model by leveraging the SHapley Additive exPlanations (SHAP) to elucidate the contributions of each gene from each omics to the model output. We believe WIMOAD is a very promising tool for accurate AD diagnosis and effective biomarker discovery across different progression stages, which eventually will have consequential impacts on early treatment intervention and personalized therapy design on AD.\n",
      "--------------------------------------------------\n",
      "ID: 72\n",
      "Titre: HuBMAPR: an R Client for the HuBMAP Data Portal\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.26.615227v1\n",
      "DOI: 10.1101/2024.09.26.615227\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.26.615227v1.full.pdf\n",
      "Résumé: The Human BioMolecular Atlas Program (HuBMAP) constructs the worldwide available platform to research the human body at the cellular level. The HuBMAP Data Portal encompasses a wide range of data resources measured on emerging experimental technologies at a spatial resolution. To broaden access to the HuBMAP Data Portal, we introduce an R client called HuBMAPR available on Bioconductor. This gives an efficient and programmatic interface, enabling researchers to discover and retrieve HuBMAP data easier and faster.\n",
      "--------------------------------------------------\n",
      "ID: 73\n",
      "Titre: Lifestyle factors in the biomedical literature: An ontology and comprehensive resources for named entity recognition\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.06.13.598816v3\n",
      "DOI: 10.1101/2024.06.13.598816\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.06.13.598816v3.full.pdf\n",
      "Résumé: Despite lifestyle factors (LSFs) being increasingly acknowledged in shaping individual health trajec-tories, particularly in chronic diseases, they have still not been systematically described in the bio-medical literature. This is in part because no named entity recognition (NER) system exists, which can comprehensively detect all types of LSFs in text. The task is challenging due to their inherent diversity, lack of a comprehensive LSF classification for dictionary-based NER, and lack of a cor-pus for deep learning-based NER. Results: We present a novel Lifestyle Factor Ontology (LSFO), which we used to develop a dic-tionary-based system for recognition and normalization of LSFs. Additionally, we introduce a manu-ally annotated corpus for LSFs (LSF200) suitable for training and evaluation of NER systems, and use it to train a transformer-based system. Evaluating the performance of both NER systems on the corpus revealed an F-score of 64% for the dictionary-based system and 76% for the transformer-based system. Large-scale application of these systems on PubMed abstracts and PMC Open Access articles identified over 300 million mentions of LSF in the biomedical literature. Availability: LSFO, the annotated LSF200 corpus, and the detected LSFs in PubMed and PMC-OA articles using both NER systems, are available under open licenses via the following GitHub repository: https://github.com/EsmaeilNourani/LSFO-expansion. This repository contains links to two associated GitHub repositories and a Zenodo project related to the study. LSFO is also availa-ble at BioPortal: https://bioportal.bioontology.org/ontologies/LSFO\n",
      "--------------------------------------------------\n",
      "ID: 74\n",
      "Titre: LOCAS: Multi-label mRNA Localization with Supervised Contrastive Learning\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614785v2\n",
      "DOI: 10.1101/2024.09.24.614785\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614785v2.full.pdf\n",
      "Résumé: Traditional methods for mRNA subcellular localization often fail to account for multiple compartmentalization. Recent multi-label models have improved performance, but still face challenges in capturing complex localization patterns. We introduce LOCAS (Localization with Supervised Contrastive Learning), which integrates an RNA language model to generate initial embeddings, employs supervised contrastive learning (SCL) to identify distinct RNA clusters, and uses a multi-label classification head (ML-Decoder) with cross-attention for accurate predictions. Through extensive ablation studies and multi-label overlapping threshold tuning, LOCAS achieves state-of-the-art performance across all metrics, providing a robust solution for RNA localization tasks.\n",
      "--------------------------------------------------\n",
      "ID: 75\n",
      "Titre: Cell-type-specific mapping of enhancers and target genes from single-cell multimodal data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614814v1\n",
      "DOI: 10.1101/2024.09.24.614814\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614814v1.full.pdf\n",
      "Résumé: Mapping enhancers and target genes in disease-related cell types has provided critical insights into the functional mechanisms of genetic variants identified by genomewide association studies (GWAS). However, most existing analyses rely on bulk data or cultured cell lines, which may fail to identify cell-type-specific enhancers and target genes. Recently, single-cell multimodal data measuring both gene expression and chromatin accessibility within the same cells have enabled the inference of enhancer-gene pairs in a cell-type-specific and context-specific manner. However, this task is challenged by the data9s high sparsity, sequencing depth variation, and the computational burden of analyzing a large number of enhancer-gene pairs. To address these challenges, we propose scMultiMap, a statistical method that infers enhancer-gene association from sparse multimodal counts using a joint latent-variable model. It adjusts for technical confounding, permits fast moment-based estimation and provides analytically derived p-values. In systematic analyses of blood and brain data, scMultiMap shows appropriate type I error control, high statistical power with greater reproducibility across independent datasets and stronger consistency with orthogonal data modalities. Meanwhile, its computational cost is less than 1% of existing methods. When applied to single-cell multimodal data from postmortem brain samples from Alzheimer9s disease (AD) patients and controls, scMultiMap gave the highest heritability enrichment in microglia and revealed new insights into the regulatory mechanisms of AD GWAS variants in microglia.\n",
      "--------------------------------------------------\n",
      "ID: 76\n",
      "Titre: Labels as a Feature: Network Homophily for Systematically Discovering human GPCR Drug-Target Interactions\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.03.29.586957v2\n",
      "DOI: 10.1101/2024.03.29.586957\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.03.29.586957v2.full.pdf\n",
      "Résumé: Machine learning (ML) has revolutionized drug discovery by enabling the exploration of vast, uncharted chemical spaces essential for discovering novel patentable drugs. Despite the critical role of human G protein-coupled receptors (hGPCRs) in FDA approved drugs, exhaustive in-distribution drug-target interaction (DTI) testing across all pairs of hGPCRs and known drugs is rare due to significant economic and technical challenges. This often leaves off-target effects unexplored, which poses a considerable risk to drug safety. In contrast to the traditional focus on out-of-distribution (OOD) exploration (drug discovery), we introduce a neighbourhood-to-prediction model termed Chemical Space Neural Networks (CSNN) that leverages network homophily and training-free graph neural networks (GNNs) with Labels as Features (LaF). We show that CSNN′s ability to make accurate predictions strongly correlates with network homophily. Thus, LaFs strongly increase a ML model′s capacity to enhance in distribution prediction accuracy, which we show by integrating labeled data during inference. We validate these advancements in a high-throughput yeast biosensing system (3773 DTIs, 539 compounds, 7 hGPCRs) to discover novel DTIs for FDA approved drugs and to expand the general understanding of how to build reliable predictors to guide experimental verification.\n",
      "--------------------------------------------------\n",
      "ID: 77\n",
      "Titre: Machine Learning-Enhanced Drug Discovery for BACE1: A Novel Approach to Alzheimer’s Therapeutics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614844v1\n",
      "DOI: 10.1101/2024.09.24.614844\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614844v1.full.pdf\n",
      "Résumé: Alzheimer9s disease (AD) remains a significant global health challenge, with BACE1, a critical therapeutic target due to its role in the amyloidogenic pathway. This study presents an innovative approach to drug development by integrating machine learning (ML) techniques to identify the potential BACE1 inhibitors. Utilizing a comprehensive dataset of protein-ligand interactions, the current study employed a machine learning model to generate novel ligands with high binding affinity and specificity for the BACE1 active site. The model9s efficacy was validated through molecular docking studies, which demonstrated superior binding affinities compared to existing FDA-approved inhibitors such as Atabecestat and Lanabecestat. The findings reveal that the top candidate, MLC10, exhibits a unique mechanism of action by promoting controlled flexibility in BACE1, contrasting with the rigid conformations induced by traditional inhibitors. This dynamic modulation enhances the enzyme9s inhibition, suggesting a promising avenue for therapeutic intervention. Furthermore, solvent-accessible surface area analyses indicate that MLC10 facilitates a more favourable protein conformation, potentially altering catalytic behaviour through increased solvent interactions. This work underscores the transformative potential of machine learning in drug discovery, paving the way for the development of next-generation BACE1 inhibitors. By harnessing computational techniques, the present work aims to address the limitations of current therapies and contribute to more effective treatments for Alzheimer9s disease, ultimately improving patient outcomes and quality of life.\n",
      "--------------------------------------------------\n",
      "ID: 78\n",
      "Titre: Particular sequence characteristics induce bias in the detection of polymorphic transposable element insertions\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.25.614865v1\n",
      "DOI: 10.1101/2024.09.25.614865\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.25.614865v1.full.pdf\n",
      "Résumé: Transposable elements (TEs) have an important role in genome evolution but are challenging for bioinformatics detection due to their repetitive nature and ability to move and replicate within genomes. New sequencing technologies now enable the characterization of nucleotide and structural variations within species. Among them, TE polymorphism is critical to identify as it may influence species adaptation or trigger diseases. Despite the development of numerous bioinformatic programs, identifying the most effective tool is challenging due to non-overlapping results and varying efficiency across studies. Benchmarking efforts have highlighted the limitations of these tools, often evaluated on either real or simulated data. However, real data may be incomplete or contain unannotated TEs, while simulated data may not accurately reflect real genomes. This study introduces a simulation method generating data based on real genomes to control all genomic parameters. Evaluating several TE polymorphic detection tools using data from Drosophila melanogaster and Arabidopsis thaliana, our study investigates factors like copy size, sequence divergence, and GC content that influence detection efficiency. Our results indicate that only a few programs perform satisfactorily and that all are sensitive to TE and genomic characteristics, that may differ according to the species considered. Using Bos taurus population data as a case study to identify polymorphic LTR-retrotransposon insertions, we found low-frequency insertions particularly challenging to detect due to a high number of false positives. Increased sequencing coverage improved sensitivity but reduced precision. Our work underscores the importance of selecting appropriate tools and thresholds according to the specific research questions.\n",
      "--------------------------------------------------\n",
      "ID: 79\n",
      "Titre: Generalizable Morphological Profiling of Cells by Interpretable Unsupervised Learning\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614684v1\n",
      "DOI: 10.1101/2024.09.24.614684\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614684v1.full.pdf\n",
      "Résumé: The intersection of advanced microscopy and machine learning is revolutionizing cell biology into a quantitative, datadriven science. While traditional morphological profiling of cells relies on laborintensive manual feature extraction susceptible to biases, deep learning offers promising alternatives but struggles with the interpretability of its black-box operation and dependency on extensive labeled data. We introduce MorphoGenie, an unsupervised deeplearning framework designed to address these challenges in singlecell morphological profiling. Enabling disentangled representation learning integrated with high-fidelity image reconstructions, MorphoGenie possesses a critical attribute to learn a compact, generalizable and interpretable latent space. This facilitates the extraction of biologically meaningful features without human annotation, additionally overcoming the \"curse of dimensionality\" inherent in manual methods. Unlike prior models, MorphoGenie introduces a systematic approach to mapping disentangled latent representations to fundamental hierarchical morphological attributes, ensuring both semantic and biological interpretability. Moreover, it adheres to the concept of combinatorial generalization, a core principle of human intelligence, which greatly enhances the model9s capacity to generalize across a broad spectrum of imaging modalities (e.g., quantitative phase imaging and fluorescence imaging) and experimental conditions (ranging from discrete cell type/state classification to continuous trajectory inference). The framework offers a new, generalized strategy for unbiased and comprehensive morphological profiling, potentially revealing insights into cellular behavior in health and disease that might be overlooked by expert visual examination.\n",
      "--------------------------------------------------\n",
      "ID: 80\n",
      "Titre: Local Mean Suppression Filter for Effective Background Identification in Fluorescence Images\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.25.614955v1\n",
      "DOI: 10.1101/2024.09.25.614955\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.25.614955v1.full.pdf\n",
      "Résumé: We present an easy-to-use, nonlinear filter for effective background identification in fluorescence microscopy images with dense and low-contrast foreground. The pixel-wise filtering is based on comparison of the pixel intensity with the mean intensity of pixels in its local neighborhood. The pixel is given a background or foreground label depending on whether its intensity is less than or greater than the mean respectively. Multiple labels are generated for the same pixel by computing mean expression values by varying neighborhood size. These labels are accumulated to decide the final pixel label. We demonstrate that the performance of our filter favorably compares with state-of-the-art image processing, machine learning, and deep learning methods. We present three use cases that demonstrate its effectiveness, and also show how it can be used in multiplexed fluorescence imaging contexts and as a denoising step in image segmentation. A fast implementation of the filter is available in Python~3 on GitHub.\n",
      "--------------------------------------------------\n",
      "ID: 81\n",
      "Titre: FerroEnrich: An Interactive web tool for computing Ferroptosis index and gene enrichment\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.25.615075v1\n",
      "DOI: 10.1101/2024.09.25.615075\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.25.615075v1.full.pdf\n",
      "Résumé: Ferroptosis is an iron-dependent form of controlled cell death and is characterized by the formation of lipid peroxides. Understanding gene expressions associated with ferroptosis is critical for determining its function in illnesses and potential therapeutic approaches. Despite its significance, no computational model is currently available to accurately quantify the ferroptosis incident. FerroEnrich is a sophisticated web-based tool built using R Shiny application to enumerate the occurrence of ferroptosis based on the relevant gene expressions. This tool available at https://ferroenrich.shinyapps.io/ferroenrich/ processes the input gene expression file to identify genes that are resistant or prone to ferroptosis, calculates ferroptosis index value with dynamic colored heatmap and gene network plot. This manuscript describes the design, operation and usability of FerroEnrich, including examples and a discussion of its potential impact on ferroptosis research. FerroEnrich is a vital tool for researchers, allowing them to explore and analyze complicated gene expression data related to ferroptosis.\n",
      "--------------------------------------------------\n",
      "ID: 82\n",
      "Titre: An explainable graph neural network approach for effectively integrating multi-omics with prior knowledge to identify biomarkers from interacting biological domains\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.08.23.609465v2\n",
      "DOI: 10.1101/2024.08.23.609465\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.08.23.609465v2.full.pdf\n",
      "Résumé: The rapid growth of multi-omics datasets, in addition to the wealth of existing biological prior knowledge, necessitates the development of effective methods for their integration. Such methods are essential for building predictive models and identifying disease-related molecular markers. We propose a framework for supervised integration of multi-omics data with biological priors represented as knowledge graphs. Our framework leverages graph neural networks (GNNs) to model the relationships among features from high-dimensional 9omics data and set transformers to integrate low-dimensional representations of 9omics features. Furthermore, our framework incorporates explainability methods to elucidate important biomarkers and extract interaction relationships between biological quantities of interest. We demonstrate the effectiveness of our approach by applying it to Alzheimer9s disease (AD) multi-omics data from the ROSMAP cohort, showing that the integration of transcriptomics and proteomics data with AD biological domain network priors improves the prediction accuracy of AD status and highlights robust AD biomarkers.\n",
      "--------------------------------------------------\n",
      "ID: 83\n",
      "Titre: Continual integration of single-cell multimodal data with MIRACLE\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.613833v2\n",
      "DOI: 10.1101/2024.09.24.613833\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.613833v2.full.pdf\n",
      "Résumé: Single-cell sequencing technologies have revolutionized our understanding of cellular heterogeneity and facilitated the construction of multi-omics cell atlases via data integration. However, updating these atlases with new data conventionally requires reintegration of all data and is computationally intensive, hindering timely updates and dynamic adjustments in biological and medical research. To address this challenge, we present Multimodal Integration with Continual Learning (MIRACLE), a novel online learning framework for the adaptive and efficient integration of single-cell multimodal data. MIRACLE employs dynamic architectures and data rehearsal strategies to support continual learning, allowing diverse data to be integrated while minimizing information loss over time. Our evaluations demonstrate that MIRACLE achieves accurate online integration with reduced computational requirements, effectively updating and expanding atlases with new cross-tissue and cross-modal data, and precisely identifying novel cell types and transferring labels across datasets. MIRACLE provides an efficient and flexible tool for single-cell community to integrate, share and explore biological knowledge from single-cell multimodal data.\n",
      "--------------------------------------------------\n",
      "ID: 84\n",
      "Titre: Forecasting dominance of SARS-CoV-2 lineages by anomaly detection using deep AutoEncoders\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.10.24.563721v4\n",
      "DOI: 10.1101/2023.10.24.563721\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.10.24.563721v4.full.pdf\n",
      "Résumé: The coronavirus disease of 2019 (COVID19) pandemic is characterized by sequential emergence of severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) variants, lineages, and sublineages, outcompeting previously circulating ones because of, among other factors, increased transmissibility and immune escape. We propose DeepAutoCoV, an unsupervised deep learning anomaly detection system to predict future dominant lineages (FDLs). We define FDLs as viral (sub)lineages that will constitute more than 10% of all the viral sequences added to the GISAID database on a given week. DeepAutoCoV is trained and validated by assembling global and country-specific data sets from over 16 million Spike protein sequences sampled over a period of about 4 years. DeepAutoCoV successfully flags FDLs at very low frequencies (0.01%,3%), with median lead times of 4-17 weeks, and predicts FDLs ~5 and ~25 times better than a baseline approach For example, the B.1.617.2 vaccine reference strain was flagged as FDL when its frequency was only 0.01%, more than a year before it was considered for an updated COVID-19 vaccine. Furthermore, DeepAutoCoV outputs interpretable results by pinpointing specific mutations potentially linked to increased fitness, and may provide significant insights for the optimization of public health pre-emptive intervention strategies.\n",
      "--------------------------------------------------\n",
      "ID: 85\n",
      "Titre: Protein Sequence Modelling with Bayesian Flow Networks\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614734v1\n",
      "DOI: 10.1101/2024.09.24.614734\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614734v1.full.pdf\n",
      "Résumé: Exploring the vast and largely uncharted territory of amino acid sequences is crucial for understanding complex protein functions and the engineering of novel therapeutic proteins. Whilst generative machine learning has advanced protein sequence modelling, no existing approach is proficient for both unconditional and conditional generation. In this work, we propose that Bayesian Flow Networks (BFNs),  a recently introduced framework for generative modelling, can address these challenges. We present ProtBFN, a 650M parameter model trained on protein sequences curated from UniProtKB, which generates natural-like, diverse, structurally coherent, and novel protein sequences, significantly outperforming leading autoregressive and discrete diffusion models. Further, we fine-tune ProtBFN on heavy chains from the Observed Antibody Space (OAS) to obtain an antibody-specific model, AbBFN, which we use to evaluate zero-shot conditional generation capabilities. AbBFN is found to be competitive with, or better than, antibody-specific BERT-style models, when applied to predicting individual framework or complimentary determining regions (CDR).\n",
      "--------------------------------------------------\n",
      "ID: 86\n",
      "Titre: RanBALL: An Ensemble Random Projection Model for Identifying Subtypes of B-cell Acute Lymphoblastic Leukemia\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614777v1\n",
      "DOI: 10.1101/2024.09.24.614777\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614777v1.full.pdf\n",
      "Résumé: As the most common pediatric malignancy, B-cell acute lymphoblastic leukemia (B-ALL) has multiple distinct subtypes characterized by recurrent and sporadic somatic and germline genetic alterations. Identification of B-ALL subtypes can facilitate risk stratification and enable tailored therapeutic approaches. Existing methods for B-ALL subtyping primarily depend on immunophenotypic, cytogenetic and genomic analyses, which would be costly, complicated, and laborious in clinical practice applications. To overcome these challenges, we present RanBALL (an Ensemble Random Projection-Based Model for Identifying B-Cell Acute Lymphoblastic Leukemia Subtypes), an accurate and cost-effective model for B-ALL subtype identification based on transcriptomic profiling only. RanBALL leverages random projection (RP) to construct an ensemble of dimension-reduced multi-class support vector machine (SVM) classifiers for B-ALL subtyping. Results based on 10 times 5-fold cross validation tests for >1700 B-ALL patients demonstrated that the proposed model achieved an accuracy of 93.35%, indicating promising prediction capabilities of RanBALL for B-ALL subtyping. The high accuracies of RanBALL suggested that our model could effectively capture underlying patterns of transcriptomic profiling for accurate B-ALL subtype identification. We believe RanBALL will facilitate the discovery of B-ALL subtype-specific marker genes and therapeutic targets, and eventually have consequential positive impacts on downstream risk stratification and tailored treatment design.\n",
      "--------------------------------------------------\n",
      "ID: 87\n",
      "Titre: scEMB: Learning context representation of genes based on large-scale single-cell transcriptomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614685v1\n",
      "DOI: 10.1101/2024.09.24.614685\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614685v1.full.pdf\n",
      "Résumé: Background: The rapid advancement of single-cell transcriptomic technologies has led to the curation of millions of cellular profiles, providing unprecedented insights into cellular heterogeneity across various tissues and developmental stages. This growing wealth of data presents an opportunity to uncover complex gene-gene relationships, yet also poses significant computational challenges. Results: We present scEMB, a transformer-based deep learning model developed to capture context-aware gene embeddings from large-scale single-cell transcriptomics data. Trained on over 30 million single-cell transcriptomes, scEMB utilizes an innovative binning strategy that integrates data across multiple platforms, effectively preserving both gene expression hierarchies and cell-type specificity. In downstream tasks such as batch integration, clustering, and cell type annotation, scEMB demonstrates superior performance compared to existing models like scGPT and Geneformer. Notably, scEMB excels in silico correlation analysis, accurately predicting gene perturbation effects in CRISPR-edited datasets and microglia state transition, identifying a few known Alzheimer9s disease (AD) risks genes in top gene list. Additionally, scEMB offers robust fine-tuning capabilities for domain-specific applications, making it a versatile tool for tackling diverse biological problems such as therapeutic target discovery and disease modeling. Conclusions: scEMB represents a powerful tool for extracting biologically meaningful insights from complex gene expression data. Its ability to model in silico perturbation effects and conduct correlation analyses in the embedding space highlights its potential to accelerate discoveries in precision medicine and therapeutic development.\n",
      "--------------------------------------------------\n",
      "ID: 88\n",
      "Titre: Analytical Assessment of Metagenomic Workflows for Pathogen Detection with NIST RM 8376 and Two Sample Matrices\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614717v1\n",
      "DOI: 10.1101/2024.09.24.614717\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614717v1.full.pdf\n",
      "Résumé: We assessed the analytical performance of metagenomic workflows using NIST Reference Material 8376 DNA from bacterial pathogens spiked into two simulated clinical samples: cerebral spinal fluid (CSF) and stool. Sequencing and taxonomic classification were used to generate signals for each sample and taxa of interest, and to estimate the limit of detection (LOD), the response function, and linear dynamic range. We found that the LODs for taxa spiked into CSF ranged from approximately (0.1 to 0.3) copy/μL, with a linearity of 0.96 to 0.99. For stool, the LODs ranged from (10 to 221) copy/μL, with a linearity of 0.99 to 1.01. Further, discriminating different E. coli strains proved to be workflow-dependent, as only one classifier:database combination of the three tested showed the ability to differentiate the two pathogenic and  commensal strains. Surprisingly, when we compared the response functions of the same taxa in the two different sample types, we found those functions to be the same, despite large differences in LODs. This suggests that the agnostic-diagnostic theory for metagenomics may apply to different target organisms and different sample types. Using RMs, we were able to generate quantitative analytical performance metrics for each workflow and sample set, enabling relatively rapid workflow screening before employing clinical samples. This makes these RMs a useful tool that will generate data needed to support translation of metagenomics into regulated use.\n",
      "--------------------------------------------------\n",
      "ID: 89\n",
      "Titre: BRDKRM: An Explainable Framework for Disease Modifying Drug Identification\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614653v1\n",
      "DOI: 10.1101/2024.09.24.614653\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614653v1.full.pdf\n",
      "Résumé: Background: Drug classification into disease-modifying (DM) and symptomatic (SYM) categories are crucial for clinical decision-making and therapeutic strategy development. Current methods often lack transparency and interpretability, limiting their utility in the biomedical domain. Method: We propose the Boundary Restricted Dynamic Key Route Mapping (BRDKRM) framework, which leverages the contextual overlap between disease and drug nodes in a heterogeneous graph. Our method aggregates genes from the top K shortest paths between diseases and\n",
      "known drugs, delineating disease neighborhood boundaries. We then classify unknown drug-disease treatments based on the average score of boundary genes in paths to unknown drugs. Results: BRDKRM achieved an accuracy of 86.78% in categorizing drug-disease treatments. We identified 530 genes playing roles in both modifying diseases and providing symptomatic relief. We demonstrated the method9s efficacy is demonstrated by using case studies on multiple sclerosis. Conclusion: BRDKRM offers an explainable approach to drug classification, bypassing extensive clinical trials. By providing biologically sound interpretations of drug classifications, our framework enhances understanding of therapeutic interventions and paves the way for more precise and efficient healthcare solutions.\n",
      "--------------------------------------------------\n",
      "ID: 90\n",
      "Titre: PlantConnectome: knowledge graph encompassing >70,000 plant articles\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.07.11.548541v3\n",
      "DOI: 10.1101/2023.07.11.548541\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.07.11.548541v3.full.pdf\n",
      "Résumé: One of the main quests of plant biology is understanding how genes and metabolites work together to form complex networks that drive plant growth, development, and responses to environmental stimuli. However, the ever-growing volume and diversity of scientific literature make it increasingly challenging to stay current with the latest advances in gene function studies. Here, we tackle the challenge by deploying the text-mining capacities of large language models to process over 71,000 plant biology abstracts. Our approach unveiled nearly 5 million functional relationships between a wide array of biological entities -genes, metabolites, tissues, and others - with a high accuracy of over 85%. We encapsulated these findings in PlantConnectome, a user-friendly database, and demonstrated its diverse utility by providing insights into gene regulatory networks, protein-protein interactions, and stress responses. We believe this innovative use of AI in the life sciences will allow plant scientists to keep up to date with the rapidly growing corpus of scientific literature. PlantConnectome is available at https://plant.connectome.tools/.\n",
      "--------------------------------------------------\n",
      "ID: 91\n",
      "Titre: Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2\n",
      "DOI: 10.1101/2024.02.29.582810\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2.full.pdf\n",
      "Résumé: The emergence of genomic language models (gLMs) offers an unsupervised approach to learning a wide diversity of cis- regulatory patterns in the non-coding genome without requiring labels of functional activity generated by wet-lab experiments. Previous evaluations have shown that pre-trained gLMs can be leveraged to improve predictive performance across a broad range of regulatory genomics tasks, albeit using relatively simple benchmark datasets and baseline models. Since the gLMs in these studies were tested upon fine-tuning their weights for each downstream task, determining whether gLM representations embody a foundational understanding of cis-regulatory biology remains an open question. Here we evaluate the representational power of pre-trained gLMs to predict and interpret cell-type-specific functional genomics data that span DNA and RNA regulation. Our findings suggest that probing the representations of pre-trained gLMs do not offer substantial advantages over conventional machine learning approaches that use one-hot encoded sequences. This work highlights a major gap with current gLMs, raising potential issues in conventional pre-training strategies for the non-coding genome.\n",
      "--------------------------------------------------\n",
      "ID: 92\n",
      "Titre: Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2\n",
      "DOI: 10.1101/2024.09.22.614322\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2.full.pdf\n",
      "Résumé: Single-cell RNA sequencing (scRNA-seq) in multi-condition experiments enables the systematic assessment of treatment effects. Analyzing scRNA-seq data relies on linear dimensionality reduction (DR) methods like principal component analysis (PCA). These methods decompose high-dimensional gene expression profiles into tractable factor representations and prototypical gene expression patterns (components), facilitating the study of cell type variation. However, integrating study covariates within linear DR frameworks remains a challenging task. We present scPCA, a flexible DR framework that jointly models cellular heterogeneity and conditioning variables, allowing it to recover an integrated factor representation and reveal transcriptional changes across conditions and components of the decomposition. We show that scPCA extracts an interpretable latent representation by analyzing unstimulated and IFN-beta-treated PBMCs, and showcase that the model may be employed to effectively address batch effects. We examine age-related changes in rodent lung cell populations, uncovering a previously unreported surge in Ccl5 expression in T cells. We illustrate how scPCA may be employed to identify coordinated transcriptional changes across multiple time-points in depolarized visual cortex neurons. Finally, we show that scPCA elucidates transcriptional shifts in CRISPR-Cas9 chordin knockout zebrafish fish single-cell data despite large difference cell abundance across conditions. Since scPCA introduces a general approach to account for conditioning variables in high-dimensional data, it may also be applicable to datasets other than scRNA-seq.\n",
      "--------------------------------------------------\n",
      "ID: 93\n",
      "Titre: varCADD: large sets of standing genetic variation enable genome-wide pathogenicity prediction\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614666v1\n",
      "DOI: 10.1101/2024.09.24.614666\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614666v1.full.pdf\n",
      "Résumé: Machine learning and artificial intelligence are increasingly being applied to identify phenotypically causal genetic variation. These data-driven methods require comprehensive training sets to deliver reliable results. However, large unbiased datasets for variant prioritization and effect predictions are rare as most of the available databases do not represent a broad ensemble of variant effects and are often biased towards coding genome, or even towards few well-studied genes. To overcome these issues, we propose several alternative training sets derived from subsets of human standing variation. Specifically, we use variants identified from whole-genome sequences of 71,156 individuals contained in gnomAD v3.0 and approximate the benign set with frequent and the deleterious set with rare standing variation. We apply the Combined Annotation Dependent Depletion framework (CADD) and train several alternative models using CADD v1.6. Using the NCBI ClinVar validation set, we demonstrate that the alternative models have state-of-the art accuracy, outperforming the widely used pathogenicity score CADD v1.6 in certain genomic regions. Being larger than conventional databases, including the evolutionary-derived training dataset of about 30 million variants in CADD, standing variation cover a broader range of genomic regions and rare instances of the applied annotations. For example, they cover more recent evolutionary changes common in gene regulatory regions, which are more challenging to assess with conventional tools. Finally, datasets derived from standing variation better represent allelic changes in the human genome and do not require extensive simulations and adaptations to annotations of the evolutionary-derived sequence alterations used for CADD training. We provide datasets as well as trained models to the community for further development and application.\n",
      "--------------------------------------------------\n",
      "ID: 94\n",
      "Titre: Diversity and abundance of ring nucleases in type III CRISPR-Cas loci\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614671v1\n",
      "DOI: 10.1101/2024.09.24.614671\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614671v1.full.pdf\n",
      "Résumé: Most type III CRISPR-Cas systems facilitate immune responses against invading mobile genetic elements such as phages by generating cyclic oligoadenylates (cOAs). Downstream effectors activated by cOAs are typically non-specific proteins that induce damage to essential cellular components, thereby preventing phage epidemics. Due to these toxic effects, it is crucial that the production and concentration of cOAs remain under tight regulatory control during infection-free periods or when deactivating the immune response after clearing an infection. Type III CRISPR loci often encode enzymes known as ring nucleases (RNs) that bind and degrade specific cOAs, while some effectors are auto-deactivating. Despite the discovery of several classes of RNs, a comprehensive bioinformatic analysis of type III CRISPR-Cas loci in this context is lacking. Here, we examined 38,742 prokaryotic genomes to provide a global overview of type III CRISPR loci, focussing on the known and predicted RNs. The candidate RNs Csx16 and Csx20 are confirmed as active enzymes, joining Crn1-3. Distributions and patterns of co-occurrence of RNs and associated effectors are explored, allowing the conclusion that a sizeable majority of type III CRISPR systems regulate cOA levels by degrading the signalling molecules, which has implications for cell fate following viral infection.\n",
      "--------------------------------------------------\n",
      "ID: 95\n",
      "Titre: Germinated brown rice research: bibliometric analysis of progress, hotspots and trends\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614820v1\n",
      "DOI: 10.1101/2024.09.24.614820\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614820v1.full.pdf\n",
      "Résumé: To gain a deeper understanding of global research trends and focal points in germinated brown rice, this article takes the relevant literature on germinated brown rice in the core database of Web of Science as the research object. By using bibliometric analysis, the literature on germinated brown rice published from 2003 to 2023 is deeply analyzed, and the global research progress, hotspots and future development trend of germinated brown rice are summarized. Since 2018, research on germinated brown rice has been rapidly developing, with a significant surge in interest since 2020. China leads in the number of publications, institutions, and core authors in this research area. The primary research topics include the nutritional value, physiological active components, and industrial applications of germinated brown rice. Current research frontiers involve identifying, evaluating, and enhancing bioactive substances in germinated brown rice for food applications. Research in this field remains active, and application scenarios are becoming increasingly diverse. Future studies may explore new directions in related equipment and ingredients.\n",
      "--------------------------------------------------\n",
      "ID: 96\n",
      "Titre: p53motifDB: integration of genomic information and tumor suppressor p53 binding motifs\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614594v1\n",
      "DOI: 10.1101/2024.09.24.614594\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.24.614594v1.full.pdf\n",
      "Résumé: The tumor suppressor gene TP53 encodes the DNA binding transcription factor p53 and is one of the most commonly mutated genes in human cancer. Tumor suppressor activity requires binding of p53 to its DNA response elements and subsequent transcriptional activation of a diverse set of target genes. Despite decades of close study, the logic underlying p53 interactions with its numerous potential genomic binding sites and target genes is not yet fully understood. Here, we present a database of DNA and chromatin-based information focused on putative p53 binding sites in the human genome to allow users to generate and test new hypotheses related to p53 activity in the genome. Users can query genomic locations based on experimentally observed p53 binding, regulatory element activity, genetic variation, evolutionary conservation, chromatin modification state, and chromatin structure. We present multiple use cases demonstrating the utility of this database for generating novel biological hypotheses, such as chromatin-based determinants of p53 binding and potential cell type-specific p53 activity. All database information is also available as a precompiled sqlite database for use in local analysis or as a Shiny web application.\n",
      "--------------------------------------------------\n",
      "ID: 97\n",
      "Titre: RNA-DCGen: Dual Constrained RNA Sequence Generation with LLM-Attack\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614570v1\n",
      "DOI: 10.1101/2024.09.23.614570\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.23.614570v1.full.pdf\n",
      "Résumé: Designing RNA sequences with specific properties is critical for developing personalized medications and therapeutics. While recent diffusion and flow-matching-based generative models have made strides in conditional sequence design, they face two key limitations: specialization for fixed constraint types, such as tertiary structures, and lack of flexibility in imposing additional conditions beyond the primary property of interest. To address these challenges, we introduce RNA-DCGen, a generalized framework for RNA sequence generation that is adaptable to any structural or functional properties through straightforward finetuning with an RNA language model (RNA-LM). Additionally, RNA-DCGen can enforce conditions on the generated sequences by fixing specific conserved regions. On RNA generation conditioned on RNA distance maps, RNA-DCGen generates sequences with an average R2 score of 0.625 compared to random sequences that score only 0.118 over 250 generations as judged by a separate more capable RNA-LM. When conditioned on RNA secondary structures, RNA-DCGen achieves an average F1 score of 0.4 against a random baseline of 0.006.\n",
      "--------------------------------------------------\n",
      "ID: 98\n",
      "Titre: Clustering-independent estimation of cell abundances in bulk tissues using single-cell RNA-seq data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.02.06.527318v2\n",
      "DOI: 10.1101/2023.02.06.527318\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.02.06.527318v2.full.pdf\n",
      "Résumé: Single-cell RNA-sequencing has transformed the study of biological tissues by enabling transcriptomic characterizations of their constituent cell states. Computational methods for gene expression deconvolution use this information to infer the cell composition of related tissues profiled at the bulk level. However, current deconvolution methods are restricted to discrete cell types and have limited power to make inferences about continuous cellular processes like cell differentiation or immune cell activation. We present ConDecon, a clustering-independent method for inferring the likelihood for each cell in a single-cell dataset to be present in a bulk tissue. ConDecon represents an improvement in phenotypic resolution and functionality with respect to regression-based methods. Using ConDecon, we discover the implication of neurodegenerative microglia inflammatory pathways in the mesenchymal transformation of pediatric ependymoma and characterize their spatial trajectories of activation. The generality of this approach enables the deconvolution of other data modalities such as bulk ATAC-seq data.\n",
      "--------------------------------------------------\n",
      "ID: 99\n",
      "Titre: SeaMoon: Prediction of molecular motions based on language models\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614585v1\n",
      "DOI: 10.1101/2024.09.23.614585\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.23.614585v1.full.pdf\n",
      "Résumé: How protein move and deform determines their interactions with the environment and is thus of utmost importance for cellular functioning. Following the revolution in single protein 3D structure prediction, researchers have focused on repurposing or developing deep learning models for sampling alternative protein conformations. In this work, we explored whether continuous compact representations of protein motions could be predicted directly from protein sequences, without exploiting nor sampling protein structures. Our approach, called SeaMoon, leverages protein Language Model (pLM) embeddings as input to a lightweight (≈1M trainable parameters) convolutional neural network. SeaMoon achieves a success rate of up to 40% when assessed against ≈1~000 collections of experimental conformations exhibiting a wide range of motions. SeaMoon capture motions not accessible to the normal mode analysis, an unsupervised physics-based method relying solely on a protein structure9s 3D geometry, and generalises to proteins that do not have any detectable sequence similarity to the training set. SeaMoon is easily retrainable with novel or updated pLMs.\n",
      "--------------------------------------------------\n",
      "ID: 100\n",
      "Titre: INSPIRE: interpretable, flexible and spatially-aware integration of multiple spatial transcriptomics datasets from diverse sources\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614539v1\n",
      "DOI: 10.1101/2024.09.23.614539\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.23.614539v1.full.pdf\n",
      "Résumé: Recent advances in spatial transcriptomics technologies have led to a growing number of diverse datasets, offering unprecedented opportunities to explore tissue organizations and functions within spatial contexts. However, it remains a significant challenge to effectively integrate and interpret these data, often originating from different samples, technologies, and developmental stages. In this paper, we present INSPIRE, a deep learning method for integrative analyses of multiple spatial transcriptomics datasets to address this challenge. With designs of graph neural networks and an adversarial learning mechanism, INSPIRE enables spatially informed and adaptable integration of data from varying sources. By incorporating non-negative matrix factorization, INSPIRE uncovers interpretable spatial factors with corresponding gene programs, revealing tissue architectures, cell type distributions and biological processes. We demonstrate the capabilities of INSPIRE by applying it to human cortex slices from different samples, mouse brain slices with complementary views, mouse hippocampus and embryo slices generated through different technologies, and spatiotemporal organogenesis atlases containing half a million spatial spots. INSPIRE shows superior performance in identifying detailed biological signals, effectively borrowing information across distinct profiling technologies, and elucidating dynamical changes during embryonic development. Furthermore, we utilize INSPIRE to build 3D models of tissues and whole organisms from multiple slices, demonstrating its power and versatility.\n",
      "--------------------------------------------------\n",
      "Les titres et les liens des articles ont été enregistrés et affichés dans la base de données SQLite.\n"
     ]
    }
   ],
   "source": [
    "# affiche articles stockés dans database \n",
    "c.execute(\"SELECT * FROM articles\")\n",
    "\n",
    "rows = c.fetchall() #récupere tous les results de la requete d'avant =tuple(id,title,linke ect) de chaque article\n",
    "\n",
    "print(\"=== Liste des articles enregistrés ===\\n\")\n",
    "\n",
    "for row in rows:\n",
    "    print(f\"ID: {row[0]}\")\n",
    "    print(f\"Titre: {row[1]}\")\n",
    "    print(f\"Lien: {row[2]}\")\n",
    "    print(f\"DOI: {row[3]}\")\n",
    "    print(f\"Date: {row[4]}\")\n",
    "    print(f\"Lien PDF: {row[5]}\")\n",
    "    print(f\"Résumé: {row[6]}\")  # Afficher l'abstract ici\n",
    "    print(\"-\" * 50)  # separateur visuel entre chaque article\n",
    "\n",
    "\n",
    "\n",
    "print(\"Les titres et les liens des articles ont été enregistrés et affichés dans la base de données SQLite.\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fermer la connexion à la base de données (quand tu as finis)\n",
    "#database.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Extraire les URLs des dépôts logiciels dans les abstracts ou les fichiers PDF "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. extraction des liens spécifiques aux dépôts logiciels : La fonction extract_software_links_from_text() utilise des expressions régulières spécifiques pour rechercher des URLs de dépôts logiciels populaires (GitHub, GitLab, Bitbucket, SourceForge, Zenodo).\n",
    "\n",
    "2.\tÉtape progressive d’extraction : Le script commence par analyser les résumés, puis les fichiers PDF si nécessaire, pour minimiser l’utilisation des ressources réseau.\n",
    "\n",
    "3.\tMise à jour propre de la base de données : Seuls les liens valides et nettoyés sont insérés dans la base de données.\n",
    "\n",
    "---> eviter de retraiter les mm articles donc verifier les urls et archiver \n",
    "\n",
    "1.\tVérification des articles traités : Avant de traiter chaque article, on vérifie s’il y a déjà des liens dans la colonne software_links. Si oui, l’article est ignoré.\n",
    "\t\n",
    "2.\tPause après chaque requête : Un délai d’une seconde est ajouté pour éviter de surcharger le serveur lors de l’extraction des liens.\n",
    "\n",
    "prochain vendredi : \n",
    "\n",
    "Étapes suivantes :\n",
    "\n",
    "\t1.\tVérification des URLs : Tester la validité des URLs en vérifiant leur accessibilité avec des requêtes HTTP.\n",
    "    \n",
    "\t2.\tArchivage via Software Heritage : Utiliser l’API Software Heritage pour vérifier si le dépôt est déjà archivé, puis l’archiver si nécessaire.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Liens de dépôt logiciel valides pour l'article 1 : ['https://doi.org/10.1101/2024.10.01.613179', 'https://academic.oup.com/ndt/article-pdf/32/1/', 'https://cdn.10xgenomics.com/image/upload/v1666737555/', 'https://journals.biologists.com/dev/', 'https://github.com/CostaLab/phlower', 'https://academic.oup.com/', 'https://journals.biologists.com/dev/article-pdf/146/12/dev178673/3112612/dev178673.pdf', 'https://phlower.readthedocs.io/en/latest/']\n",
      "Liens de dépôt logiciel valides pour l'article 2 : ['https://doi.org/10.1093/', 'https://doi.org/10.5281/ZENODO.6954035', 'https://doi.org/10.1038/s41586-020-2871-y', 'https://doi.org/10.1093/bioinformatics/btad074', 'https://doi.org/10.1128/jb.00619-08', 'https://www.biorxiv.org/content/early/2023/10/17/2023.09.22.558964.full.pdf', 'https://doi.org/10.1146/', 'https://doi.org/10.1016/j.cell.2020.05.023', 'https://github.com/pangenome/', 'https://doi.org/10.1101/2022.09.17', 'https://doi.org/10.1089/cmb.2017.0251', 'https://doi.org/10.1038/', 'https://doi.org/10.1038/nbt.4235', 'https://doi.org/10.1038/s41587-023-01793-w', 'https://doi.org/10.1101/gr.214155.116', 'https://doi.org/10.1093/bioinformatics/btaa777', 'https://doi.org/10.1038/ng.3117', 'https://doi.org/10.1038/s41588-023-01459-y', 'https://doi.org/10.1038/nbt.4227', 'https://doi.org/10', 'https://doi.org/10.1126/science.abn2937', 'https://doi.org/10.1038/s41586-023-05895-y', 'https://doi.org/10.1038/s41586-023-05896-x', 'https://github.com/marschall-lab/GFAffix', 'https://doi.org/10.1371/journal.pcbi.1009123', 'https://github.com/waveygang/wfmash', 'https://doi.org/10.1101/gr.123356', 'https://doi.org/10.1101/gr.214270.116', 'https://doi.org/10.1093/molbev/msw215', 'https://pggb.readthedocs.io/', 'https://doi.org/10.1038/s41586-023-05976-y', 'https://doi.org/10.1101/2023.09.22.558964', 'https://doi.org/', 'https://github.com/', 'https://doi.org/10.1093/bioinformatics/bty597', 'https://doi.org/10.1038/s41597-021-00947-2', 'https://doi.org/10.1186/s13059-019-1905-y', 'https://doi.org/10.1038/nbt.1754', 'https://doi.org/10.1101/', 'https://doi.org/10.1093/bioinformatics/', 'https://doi', 'https://doi.org/10.17863/', 'https://zenodo', 'https://doi.org/10.1038/s41592-023-01914-y', 'https://doi.org/10.1093/bioinformatics/btac743']\n",
      "Liens de dépôt logiciel valides pour l'article 3 : ['http://db.idrblab.net/ttd/', 'https://bioinfogp.cnb.csic.es/tools/Venny/index.html', 'https://david.ncifcrf.gov/', 'https://www.bioinformatics.com.cn/login/', 'https://www.genecards.org/', 'https://vina.scripts.edu/', 'https://disgenet.com/', 'https://omim.org/', 'http://string-db.org', 'https://www.uniprot.org/', 'http://tcmspw.com/tcmsp.php', 'https://www.pdb.org/', 'https://ccsb.scripts.edu/mgltools/downloads/', 'https://doi.org/10.1101/2024.09.30.615978', 'https://vina.scripps.edu', 'http://www.swisstargetprediction.ch/']\n",
      "Liens de dépôt logiciel valides pour l'article 4 : ['https://proceedings.neurips.cc/paper/2020/hash/d1e946f4e67db4b362ad23818a6fb78a-', 'https://github.com/facebookresearch/esm', 'http://arxiv.org/abs/2205.07023', 'https://github.com/flatironinstitute/DeepFRI', 'https://proceedings.neurips.cc/paper/2017/hash/8a20a8621978632d76c43dfd28b67767-', 'https://openreview.net/pdf?id=AAML7ivghpY', 'https://doi.org/10.1101/2024.09.30.615718']\n",
      "Liens de dépôt logiciel valides pour l'article 5 : ['https://www.uniprot.org/proteomes/UP000005640', 'https://doi.org/10.1101/2024.06.27.600973']\n",
      "Liens de dépôt logiciel valides pour l'article 6 : ['https://github.com/wz-create/AptaDiff', 'https://doi.org/10.1101/2023.11.25.568693', 'http://github.com/SheffieldML/']\n",
      "Liens de dépôt logiciel valides pour l'article 7 : ['https://www.nature.com/articles/s41598-', 'http://lipidworld.biomedcentral.com/', 'https://dx.plos.org/10.1371/journal.pone.0172774', 'https://www.who.int/news-room/fact-sheets/detail/cardiovascular-diseases-', 'https://training.cochrane.org/handbook', 'https://link.springer.com/10.1007/978-3-', 'https://www.tandfonline', 'https://www.nature.com/articles/nature25973', 'https://dx.plos.org/10', 'https://www.nature.com/articles/s41522-023-00430-3(accessed', 'https://linkinghub.elsevier.com/retrieve/pii/S235239641930800X', 'http://link.springer.com/', 'https://www.hindawi', 'https://www.diabetesatlas.org', 'https://dx.plos', 'https://www.frontiersin.org/articles/10.3389/fendo.2023.1119201/full', 'https://www.nature.com/articles/', 'https://www.nature.com/', 'https://www.nature', 'https://gut.bmj.com/lookup/doi/10.1136/gutjnl-2015-309618', 'http://academic', 'https://www.frontiersin.org/articles/10.3389/fmicb.2021', 'https://www.frontiersin.org/', 'https://doi.org/10.1101/2024.10.01.615711', 'https://www.who.int/news-room/fact-sheets/detail/diabetes', 'https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-', 'http://www.rstudio.com', 'https://linkinghub.elsevier.com/retrieve/pii/', 'https://onlinelibrary.wiley.com/', 'https://genomemedicine', 'https://www.frontiersin.org/articles/10.3389/fnut.2022', 'https://www', 'https://linkinghub.elsevier.com/', 'https://plotdigitizer.com/app', 'https://bmcmicrobiol.biomedcentral']\n",
      "Liens de dépôt logiciel valides pour l'article 8 : ['https://bmblx.bmi.osumc.edu/scgnn+', 'https://doi.org/10.1038/s41592-024-02201-', 'https://doi.org/10.1038/s41586-023-06139-9', 'https://doi.org/10.1101/2024.09.30.615735', 'https://doi.org/10.1038/s41592-024-02235-4']\n",
      "Liens de dépôt logiciel valides pour l'article 9 : ['https://www.youtube.com/@MaxQuantChannel', 'https://github.com/JurgenCox/mqtools7', 'https://www.maxquant.org/', 'https://groups.google.com/g/maxquant-list', 'https://groups.google.com/g/perseus-list', 'https://github.com/cox-labs/CoxLab_Bug_Reporting', 'https://doi.org/10.1101/2024.09.30.615788', 'https://www.maxquant.org/']\n",
      "Liens de dépôt logiciel valides pour l'article 10 : ['https://github.com/', 'https://doi.org/10.1101/2024.09.30.614425', 'https://github.com/vandeplaslab/full_profile']\n",
      "Liens de dépôt logiciel valides pour l'article 11 : ['https://github.com/martinpacesa/BindCraft/blob/main/notebooks/BindCraft.ipynb', 'https://github.com/martinpacesa/BindCraft', 'https://doi.org/10.1101/2024.09.30.615802']\n",
      "Liens de dépôt logiciel valides pour l'article 12 : ['https://doi.org/10.1038/msb.2012.26', 'https://doi.org/10.1056/nejmoa2007764', 'https://doi.org/10.1056/nejmoa2021436', 'https://doi.org/10.1021/ci300367a', 'https://doi.org/10.1186/s12918-018-0532-7', 'https://doi.org/10.1093/bioinformatics/btaa501', 'https://doi.org/10.1109/BIBM.2012.6392634', 'https://doi.org/10.1093/nar/gkh131', 'https://doi.org/10.1186/s12859-018-2379-8', 'https://doi.org/10.1021/acs.jchemed.5b00404', 'https://doi.org/10.1007/s10462-022-10306-1', 'https://doi.org/10.1001/jamanetworkopen.2018.4196', 'https://doi.org/10.1145/3269206.3271739', 'https://doi.org/10.1016/j.ijantimicag.2020.105949', 'https://doi.org/10.1145/3132847.3133021', 'https://doi.org/10.1145/2939672.2939754', 'http://arxiv.org/abs/1803.08375', 'https://doi.org/10.1016/j.ins.2021.09.008', 'https://doi.org/10.24963/ijcai.2018/521', 'https://doi.org/10.1007/978-3-319-93417-4_38', 'https://doi.org/10.1016/j.jbi.2021.103707', 'https://doi.org/10.1109/TCBB.2022.3172421', 'https://doi.org/10.1093/bioinformatics/btn479', 'https://doi.org/10.1101/2024.09.30.615812', 'https://doi.org/10.1093/database/bav061', 'https://doi.org/10.1093/bioinformatics/btx659', 'https://doi.org/10.1371/journal.pcbi.1002998', 'https://doi.org/10.1186/s12859-020-03724-x', 'https://doi.org/10.1200/JCO.2017.73.4186', 'https://doi.org/10.3390/make5020036', 'https://doi.org/10.30534/ijatcse/2020/175942020', 'https://doi.org/10.1016/j.jbi.2015.03.002', 'https://doi.org/10.1093/nar/gkw1092', 'https://doi.org/10.3390/ijms24098169', 'https://doi.org/10.1016/j.ymeth.2020.05.007', 'https://doi.org/10.1016/j.jbi.2018.11.005', 'https://doi.org/10.1145/3641288', 'https://doi.org/10.1145/3307339.3342161', 'https://doi.org/10.1145/3292500.3330931', 'https://doi.org/10.1109/TKDE.2008.239', 'https://doi.org/10.1038/srep12339', 'https://doi.org/10.1002/cpt.1434', 'https://doi.org/10.1016/j.isci.2024.109148', 'https://doi.org/10.1093/nar/gkx1037', 'https://doi.org/10.1093/nar/gkv951', 'https://doi.org/10.3390/ph16071050', 'https://doi.org/10.1371/journal.pone.0140816', 'https://doi.org/10.1093/bioinformatics/bty593', 'https://doi.org/10.1136/amiajnl-2013-002512', 'https://doi.org/10.1109/I4C57141.2022.10057747', 'https://doi.org/10.1145/2939672.2939753']\n",
      "Liens de dépôt logiciel valides pour l'article 13 : ['https://doi.org/10.1101/2024.09.30.615863']\n",
      "Liens de dépôt logiciel valides pour l'article 14 : ['https://doi.org/10.1101/2024.09.30.615766', 'https://github.com/DincalciLab/samurai', 'https://genebe.net/tools/liftover', 'https://github.com/DIncalciLab/SAMURAI_paper_scripts', 'https://github.com/DIncalciLab/samurai', 'https://42basepairs.com/browse/s3/gatk-test-data/cnv/somatic']\n",
      "Liens de dépôt logiciel valides pour l'article 15 : ['https://doi.org/10.1101/2024.09.30.615776']\n",
      "Liens de dépôt logiciel valides pour l'article 16 : ['http://neuroci.com/', 'https://doi.org/10.1101/2024.09.29.615661']\n",
      "Liens de dépôt logiciel valides pour l'article 17 : ['https://github.com/sarwanpasha/Binding-Site-Prediction', 'https://doi.org/10.1101/2023.08.23.554389']\n",
      "Liens de dépôt logiciel valides pour l'article 18 : ['https://doi.org/10.1101/2024.09.29.615743']\n",
      "Liens de dépôt logiciel valides pour l'article 19 : ['https://doi.org/10.1101/2024.09.30.615762', 'https://github.com/lingxusb/PlasmidGPT', 'https://www.addgene.org/', 'https://www.addgene.org/', 'https://genome.jgi.doe.gov/portal/IMG_PR/IMG_PR.home.html']\n",
      "Liens de dépôt logiciel valides pour l'article 20 : ['https://doi.org/10.1101/2024.09.30.615418', 'https://www.vlinderstichting.nl/vlinders/overzicht-vlinders/details-vlinder/groente-uil']\n",
      "Liens de dépôt logiciel valides pour l'article 21 : ['http://asia.ensembl.org/info/genome/genebuild/biotypes.html)[54]', 'https://github.com/jiwooo-', 'https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_38/', 'https://www.gtexportal.org/home/tissue/)[53]', 'http://gimlab.online/data/total_RNAseq/Supplementary_files.zip', 'https://doi.org/10.1101/2024.09.30.615750']\n",
      "Liens de dépôt logiciel valides pour l'article 22 : ['https://stix.colorado.edu', 'https://cancer.sanger.ac.uk/cosmic', 'https://doi.org/10.18434/MDS2-2336', 'https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1KG_ONT_VIENNA/README_1', 'https://www.biorxiv.org/content/10.1101/2024.04.18.590093v1', 'https://zenodo.org/records/10819636', 'https://github.com/zhengxinchang/excord-lr', 'https://zenodo.org/records/10819636', 'https://doi.org/10.5281/zenodo.13702318', 'https://gnomad.broadinstitute.org/help/sv-overview', 'https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1KG_ONT_VIENNA/t2t/', 'https://github.com/ryanlayer/stix', 'https://github.com/jnothman/UpSetPlot)was', 'https://ftp-', 'https://s3.amazonaws.com/1000g-', 'https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1KG_ONT_VIENNA/hg38/', 'https://github.com/ryanlayer/stix', 'https://github.com/behzodcu/staffr', 'https://ftp.ncbi.nlm.nih.gov/ReferenceSamples/giab/release/AshkenazimTrio/HG002_NA24385_son/N', 'https://doi.org/10.1101/2024.09.30.615931', 'https://human-']\n",
      "Liens de dépôt logiciel valides pour l'article 23 : ['https://doi.org/10.21203/rs.3.rs-1937079/v1', 'http://arxiv.org/abs/2302.12977', 'https://doi.org/10.1101/2024.09.29.615658', 'https://doi.org/10.4121/15022410.V1', 'https://doi.org/10.5281/zenodo.11236776', 'http://arxiv.org/abs/2308.06374', 'http://arxiv.org/abs/2305.19801', 'https://doi.org/10.5281/zenodo.7521476', 'https://doi.org/10.48550/arXiv.2110.15087', 'https://doi.org/10.48550/arXiv.2310.08365', 'https://doi.org/10.26434/chemrxiv-2024-kxlgm', 'https://doi.org/10.1101/2021.10.28.466262']\n",
      "Liens de dépôt logiciel valides pour l'article 24 : ['http://www.uniprot.org/', 'https://doi.org/10.1101/2024.08.21.609040', 'http://www.ebi.ac.uk/swissprot/']\n",
      "Liens de dépôt logiciel valides pour l'article 25 : ['https://networkx.github.io', 'https://github.com/IBM/Hestia-OOD', 'https://doi.org/10.1101/2024.03.14.584508', 'https://leashbio.substack.com/p/belka-results-suggest-computers-can']\n",
      "Liens de dépôt logiciel valides pour l'article 26 : ['https://smarter-database.readthedocs.io/en/latest/data-import.html,accessed:2024-04-17', 'https://github.com/cnr-ibba/SMARTER-database/issues,accessed:2024-08-08', 'https://docs.docker.com/compose/,accessed:2024-04-17', 'https://en.wikipedia.org/wiki/REST,accessed:2024-04-17', 'https://doi.org/10.1101/2024.09.01.610681', 'https://smarter-database.readthedocs.io/en/latest/index.html,accessed:2024-08-08', 'https://docs.anaconda.com/', 'https://www.ebi.ac.uk/eva/?Submit-Data,accessed:2024-04-16', 'https://webserver.ibba.cnr.it/smarter-api/usingsmarterapiR', 'http://www.fao.org/documents/card/en/c/cc3079en', 'https://angular.io/,accessed:2024-04-17', 'https://webserver.ibba.cnr.it/smarter-api/docs/,accessed:2024-04-17', 'https://cnr-ibba.github.io/r-smarter-api/)andthroughthedatabaseportalathttps://webserver.ibba.cnr.it/smarter/', 'http://genome.cshlp.org/lookup/doi/10.1101/gr.9.8.677', 'https://cnr-ibba.github.io/r-smarter-api/articles/smarterapi.html#collect-genotypes).Metadataandsample', 'https://gsejournal.biomedcentral.com/articles/10.1186/s12711-018-0427-5', 'https://www.ebi.ac.uk/ols4/ontologies/pato,accessed:2024-04-16', 'https://webserver.ibba.cnr.it/smarter/,accessed:2024-04-17', 'http://semver.org/.web', 'https://github.com/cnr-ibba/SMARTER-database', 'https://github.com/cnr-ibba/SMARTER-database,accessed:2024-05-20', 'https://www.mongodb.com/,accessed:2024-04-17', 'https://medium.com/@julianam.tyler/what-is-the-difference-between-odm-and-orm-267bbb7778b0', 'https://drivendata.github.io/cookiecutter-data-science/,accessed:2024-04-17', 'https://cnr-ibba.github.io/r-smarter-api/,rpackageversion', 'https://github.com/cnr-ibba/SMARTER-database', 'https://smarterproject.eu/,accessed:2024-04-16', 'https://github.com/EBIvariation/vcf-validator?tab=readme-ov-file#assembly-checker,accessed:2024-04-16', 'https://github.com/cnr-ibba/SMARTER-database/blob/master/HISTORY.rst,accessed:2024-08-08', 'https://www.cookiecutter.io/,accessed:2024-04-17', 'https://mongoengine-odm.readthedocs.io/,accessed:2024-04-17', 'https://smarter-database.readthedocs.io/en/latest/', 'https://www.illumina.com/documents/products/technotes/technote_topbot.pdf,accessed:']\n",
      "Liens de dépôt logiciel valides pour l'article 27 : ['https://doi.org/10.1093/', 'https://doi.org/10.1101/204438', 'https://onlinelibrary.wiley.com/doi/10.1002/advs.202300366', 'https://CRAN.R-project.org/', 'https://github.com/tagtag/TDbasedUFEHiC', 'https://doi.org/10.18129/B9.bioc.AnnotationHub.Rpackage', 'https://doi.org/10.1101/2024.02.11.579821', 'https://doi.org/10.1038/', 'https://doi.org/10.1038/s41586-020-2493-4', 'https://www.biorxiv.org/content/early/2017/11/14/204438.full.pdf', 'https://doi.org/10.1136/', 'https://doi.org/10', 'https://doi.org/10.1186/s13059-019-1658-7', 'https://www.biorxiv.org/content/early/2024/02/12/2024.02.11.579821.full.pdf', 'https://academic.oup.com/bioinformatics/article-', 'https://bioconductor.org/', 'https://doi.org/10.18129/', 'http://genome.cshlp.org/content/27/11/1939.full.pdf+html', 'https://doi.org/10.1002/advs', 'https://doi.org/10.1186/s13059-016-0896-1', 'https://ard.bmj.com/content/83/8/1048.full.pdf', 'https://bioconductor.org/packages/AnnotationHub', 'https://doi.org/10.1101/gr.220640.117', 'https://doi.org/10.1371/journal', 'https://doi.org/10.1101/2024.09.29.615651', 'https://bioconductor.org/packages/TDbasedUFE', 'https://academic.oup.com/nar/article-', 'https://doi.org/10.18129/B9.bioc.TDbasedUFE', 'https://doi.org/10.1093/nar/gkq1019']\n",
      "Liens de dépôt logiciel valides pour l'article 28 : ['https://orcid.org/0000-0003-3652-2540', 'https://github.com/ekernf01/perturbation_benchmarking', 'https://groups.google.com/forum/#!topic/cellnet_r/pXHt2J6ZH6I', 'https://github.com/const-ae/linear_perturbation_prediction-Paper/blob/main/benchmark/src/run_l', 'https://www.gsk.ai/causalbench-challenge/', 'http://pcahan1.github.io/cellnetr/', 'https://orcid.org/0000-0002-2310-8191', 'https://orcid.org/0000-0002-5287-627X', 'https://doi.org/10.1101/2023.07.28.551039', 'https://orcid.org/0000-0001-7013-1899', 'https://blog.newlimit.com/p/2023-year-in-review', 'https://orcid.org/0000-0003-3429-3174']\n",
      "Liens de dépôt logiciel valides pour l'article 29 : ['https://github.com/sehagler/drug', 'https://doi.org/10.4103%2Fijpvm', 'https://acsjournals.onlinelibrary.wiley.com/doi/abs/10', 'https://doi.org/10.1093/nar/gkab1028', 'https://doi.org/10', 'https://reactome', 'https://academic', 'https://doi.org/10.1101/2024.09.27.615497', 'https://www']\n",
      "Liens de dépôt logiciel valides pour l'article 30 : ['https://doi.org/10.1101/2024.09.16.613342', 'http://dx.doi.org/10', 'https://doi.org/10.1038/']\n",
      "Liens de dépôt logiciel valides pour l'article 31 : ['https://doi.org/10.1101/2023.11.24.568620', 'https://scholar.google.co']\n",
      "Liens de dépôt logiciel valides pour l'article 32 : ['https://doi.org/10.1021/', 'http://www.jstor', 'https://www.sciencedirect.com/science/article/pii/S0198885921000628', 'https://doi.org/10.1038/nrg.2016.49', 'https://doi.org/10.1021/acssynbio.9b00103', 'https://doi.org/10.1073/pnas.2104878118', 'https://doi.org/10.1038/', 'https://doi.org/10.1073/pnas.0910781107', 'https://doi.org/10.1038/s41586-021-04184-w', 'http://dx.doi.org/10.1039/D1CC04635G', 'https://scicore.unibas.ch', 'http://dx.doi.org/10.1039/', 'https://doi.org/10.1101/2024.09.25.615059', 'https://www.ncbi.nlm.nih.gov/pmc/articles/', 'https://doi.org/10', 'https://doi.org/10.1073/pnas.2311726121', 'http://www.jstor.org/stable/2957618', 'https://www.sciencedirect.com/science/article/pii/S2405471221004154', 'https://doi.org/10.1038/nbt1070', 'https://doi.org/10.1038/s41586-023-05696-3', 'https://www.mdpi.com/', 'https://doi.org/10.1073/pnas.1614437114', 'https://doi.org/10.1126/science.abn3445', 'https://doi.org/10.1038/s41592-019-0496-6', 'https://doi.org/10.1016/j.tips.2021.11.010', 'https://arxiv.org/abs/2405.05167', 'https://doi.org/10.1038/s41929-022-00821-3', 'https://doi.org/10.1073/pnas.170297297', 'https://proceedings.neurips.cc/paper/1993/file/', 'https://www.sciencedirect.com/science/article/pii/S0092867422011199', 'https://doi.org/10.1038/s41467-024-45630-3', 'https://doi.org/', 'https://www.sciencedirect.com/science/article/pii/S0958166999000518', 'http://www.sciencedirect.com/science/article/pii/S0076687916300374', 'http://dx.doi', 'https://doi.org/10.1016/j.cell.2020.08.012', 'https://doi.org/10.1162/neco.1996.8.5.1085', 'https://www.sciencedirect.com/science/', 'https://www.sciencedirect.com/', 'https://chat.openai.com', 'https://www.sciencedirect.com/science/article/']\n",
      "Liens de dépôt logiciel valides pour l'article 33 : ['https://data.leucegene.iricca/', 'https://doi.org/10.1101/2023.05.10.540140', 'https://alexdray86.github.io/CLIMB/build/index.htmlandhttps://alexdray86.github.io/CLIFF/']\n",
      "Liens de dépôt logiciel valides pour l'article 34 : ['https://github.com/r2dt-bio/r2dt-client', 'https://github.com/DasLab/draw_rna', 'https://r2dt.bio', 'http://rna.ucsc.edu/', 'https://github.com/rnacentral/r2dt-web', 'https://github.com/LDWLab/RNA2D-data-schema', 'https://rnacentral.org/r2dt', 'https://json-schema.org/', 'https://docs.r2dt.bio/en/latest/', 'https://doi.org/10.1101/2024.09.29.611006', 'https://hub.docker.com/r/rnacentral/r2dt', 'https://ldwlab.github.io/XRNA-React', 'https://chatgpt.com/g/', 'https://rnacanvas.app', 'https://github.com/PDBeurope/pdb-rna-viewer', 'https://r2dt.bio', 'https://github.com/LDWLab/RNA2D-data-schema/', 'https://docs.r2dt.bio', 'https://github.com/r2dt-bio/r2dt', 'https://github.com/r2dt-']\n",
      "Liens de dépôt logiciel valides pour l'article 35 : ['https://doi.org/10.1101/2024.09.28.615630', 'http://www.repeatmasker.org']\n",
      "Liens de dépôt logiciel valides pour l'article 36 : ['https://ice.synthego.com', 'https://doi.org/10.1101/2024.09.28.614980', 'http://bioinf.wehi.edu.au/software/elda/', 'http://bioinf.wehi.edu.au/software/elda/index.html']\n",
      "Liens de dépôt logiciel valides pour l'article 37 : ['https://www.youtube.com/watch?v=-yquJiNKlAE', 'https://doi.org/10.1101/2024.09.28.615626', 'https://crfm']\n",
      "Liens de dépôt logiciel valides pour l'article 38 : ['https://doi.org/10.46354/i3m.2024.emss.011', 'https://github.com/davecliff/ESCG', 'https://doi.org/10.46354/i3m.2024.emss.028', 'https://doi.org/10.1101/2024.09.27.615336']\n",
      "Liens de dépôt logiciel valides pour l'article 39 : ['https://doi.org/10.1101/2024.09.28.615583']\n",
      "Liens de dépôt logiciel valides pour l'article 40 : ['https://arxiv.org/abs/1412.6572', 'https://arxiv.org/abs/1911.07067', 'https://github.com/HySonLab/LightMed', 'https://arxiv.org/abs/2102.04306', 'https://arxiv.org/abs/1608.03983', 'https://arxiv.org/abs/2012.09501', 'https://doi.org/10.1101/2024.09.28.615584', 'https://arxiv.org/abs/1612.01105', 'https://arxiv.org/abs/1312.5851', 'https://arxiv.org/abs/1703.00523', 'https://arxiv.org/abs/1804.03999', 'https://arxiv.org/abs/1603.05027', 'https://arxiv', 'https://arxiv.org/abs/1409.1556', 'https://arxiv.org/abs/2006.04868']\n",
      "Liens de dépôt logiciel valides pour l'article 41 : ['https://github.com/molloy-lab/TREE-QMC', 'https://github.com/molloy-lab/tree-qmc-study/tree/main/han2024wtreeqmc', 'https://doi.org/10.1101/2024.09.27.615467']\n",
      "Liens de dépôt logiciel valides pour l'article 42 : ['https://github.com/wanglongxing2020/ML_code', 'https://www.proteinatlas.org/', 'https://dcc.icgc.org/', 'https://ualcan.path.uab.edu/index.html', 'https://doi.org/10.1101/2024.09.27.615369', 'https://www.ncbi.nlm.nih.gov/gds', 'https://portal.gdc.cancer.gov/']\n",
      "Liens de dépôt logiciel valides pour l'article 43 : ['https://arxiv.org/abs/1312.6114', 'https://github.com/yyLIU12138/INSTINCT', 'https://doi.org/10.5281/zenodo.7768714', 'https://arxiv.org/abs/2010.16061', 'https://arxiv.org/abs/1511.07289', 'https://arxiv.org/abs/1802.03426', 'https://arxiv.org/abs/1609.02907', 'https://github.com/Chen-Li-17/simCAS/tree/main/data', 'https://www.biorxiv.org/content/10.1101/2024.04.22.590602v1.abstract', 'https://instinct.readthedocs.io/en/latest/index.html', 'https://doi.org/10.1101/2024.05.26.595944']\n",
      "Liens de dépôt logiciel valides pour l'article 44 : ['https://doi.org/10.1101/2024.09.27.615367', 'https://github.com/OscarGVelasco/ClusterFoldSimilarity', 'https://doi.org/10.1016/j.stem.2020.07.006', 'https://link.springer.com/articles/10.1186/s13059-', 'https://www.immport.org/shared/study/SDY998', 'https://www.nature.com/articles/s41467-', 'https://www.nature.com/articles/s41587-023-01716-9', 'https://pubmed.ncbi.nlm.nih.gov/30980045/', 'https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02568-9', 'https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1010210', 'https://www.nature.com/articles/s41576-022-00449-w', 'https://www.tandfonline.com/doi/abs/10.1128/MCB.00362-09', 'https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1000446', 'https://onlinelibrary.wiley.com/doi/full/10.1002/ctm2.1068', 'https://pubmed.ncbi.nlm.nih.gov/27693023/', 'https://www.mdpi.com/1422-', 'https://pubmed.ncbi.nlm.nih.gov/28641533/', 'https://www.bioconductor.org/', 'http://www.cell.com/article/S1525001619304046/fulltext', 'https://molecularbrain.biomedcentral.com/articles/10.1186/s13041-019-0522-8', 'https://www.nature.com/articles/s41587-023-01766-z', 'https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02367-2', 'https://www.ncbi.nlm.nih.gov/books/NBK573422/', 'https://www.biorxiv.org/content/10.1101/404962v1', 'https://pubmed.ncbi.nlm.nih.gov/24218581/', 'https://link.springer.com/chapter/10.1007/978-1-4614-8090-7_8', 'https://doi.org/10.1101/202D4sc.a0ml91.27.615367', 'https://www.science.org/doi/10.1126/science.abl4896', 'https://www.mdpi.com/2218-273X/11/9/1361/htm', 'https://www.science.org/doi/10.1126/science.346.6206.237', 'https://www.nature.com/articles/s41593-020-00783-4', 'https://academic.oup.com/bioinformatics/article/34/23/4054/5040306', 'https://pubmed.ncbi.nlm.nih.gov/23312518/', 'https://onlinelibrary.wiley.com/doi/full/10.1002/glia.23771', 'https://www.nature.com/articles/nmeth.1937', 'https://atlas.fredhutch.org/data/nygc/multimodal/pbmc_multimodal.h5seurat', 'https://www.jneurosci.org/content/25/14/3628.short', 'https://bioconductor.org/packages/release/data/experiment/html/MouseGastrulationData.html', 'https://www.nature.com/articles/nbt.2859', 'https://pubmed.ncbi.nlm.nih.gov/20979621/', 'https://www.nature.com/articles/s41590-019-0378-1', 'https://link.springer.com/article/10.1007/s00401-009-0619-8', 'https://www.nature.com/articles/s41587-023-01767-y', 'https://www.sciencedirect.com/science/article/pii/S2049080114000545', 'https://doi.org/101146/annurev-biodatasci-020422-050645', 'https://www.10xgenomics.com/resources/datasets/10k-human-', 'https://www.biorxiv.org/content/10.1101/2023.11.05.565674v1', 'https://www.annualreviews.org/doi/abs/10.1146/annurev-', 'https://www.biorxiv.org/content/10.1101/2024.05.23.595582v1', 'https://www.nature.com/articles/ncb2843', 'https://pubmed.ncbi.nlm.nih.gov/36198237/', 'https://www.immport.org/shared/study/SDY998', 'https://www.cell.com/iscience/pdf/S2589-0042(19)30414-6.pdf', 'https://molecularneurodegeneration.biomedcentral.com/articles/10.1186/s13024-024-00703-1', 'https://journals.physiology.org/doi/10.1152/physrev.00042.2016', 'https://www.nature.com/articles/s41592-021-01336-8', 'https://academic.oup.com/cercor/article-abstract/13/3/252/355488', 'https://pubmed.ncbi.nlm.nih.gov/15494728/', 'https://www.nature.com/articles/s41586-019-0933-9', 'https://pubmed.ncbi.nlm.nih.gov/32139688/', 'https://doi.org/10.1016/j.neuron.2021.11.023', 'https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-1926-6', 'https://pubmed.ncbi.nlm.nih.gov/26160885/']\n",
      "Liens de dépôt logiciel valides pour l'article 45 : ['https://doi.org/10.1101/2024.09.27.615342', 'https://www.gov.uk/government/publications/', 'https://github.com/ukhsa-collaboration/within-patient-pOXA48-conjugation']\n",
      "Liens de dépôt logiciel valides pour l'article 46 : ['https://doi.org/10.1101/2024.09.29.615668', 'https://imigitlab.uni-muenster.de/published/glinda']\n",
      "Liens de dépôt logiciel valides pour l'article 47 : ['https://cellxgene.cziscience.com/collections/8191c283-0816-424b-9b61-c3e1d6258a77', 'https://huggingface.co/datasets/ctheodoris/Genecorpus-30M', 'https://doi.org:10.1038/s41592-021-01336-8', 'https://doi.org:10.1038/s41592-024-02201-0', 'https://doi.org:10.1016/j.cell.2020.12.016', 'https://doi.org:10.1093/bioinformatics/btae089', 'https://doi.org:10.1093/bib/bbaa269', 'https://doi.org:10.1186/s12967-023-04150-2', 'https://cellxgene.cziscience.com/collections/60358420-6055-411d-ba4f-', 'https://doi.org:10.1038/s41586-023-06139-9', 'https://doi.org:10.1038/nm1613', 'https://doi.org:10.1038/s41467-020-18873-z', 'https://web.stanford.edu/group/nusselab/cgi-bin/wnt/target_genes', 'https://doi.org:10.1038/s41467-022-32111-8', 'https://doi.org/10.1101/2024.09.27.615337', 'https://doi.org:10.1038/s41467-021-21246-9', 'https://www.gsea-', 'https://doi.org:10.1093/bioinformatics/btab036', 'https://doi.org:10.1038/s41586-022-05060-x', 'https://doi.org:10.1016/j.cell.2015.02.035', 'https://doi.org:10.1038/s41587-021-01139-4', 'https://doi.org:10.1038/s41592-024-02305-7']\n",
      "Liens de dépôt logiciel valides pour l'article 48 : ['https://doi.org/10.1101/2024.09.27.615366']\n",
      "Liens de dépôt logiciel valides pour l'article 49 : ['https://CRAN.R-project.org/package=ggpubr', 'https://github.com/akdess/CaSpER', 'https://github.com/AntonioDeFalco/SCEVAN', 'https://github.com/bioliyezhang/Benchmark_scRNASeq_by_scMultiomics', 'https://kharchenkolab.github.io/numbat/articles/numbat.html', 'https://doi.org/10.1101/2024.09.26.615284', 'https://github.com/broadinstitute/infercnv', 'https://github.com/navinlabcode/copykat']\n",
      "Liens de dépôt logiciel valides pour l'article 50 : ['https://doi.org/10.1101/2024.09.26.615264']\n",
      "Liens de dépôt logiciel valides pour l'article 51 : ['https://doi.org/10.1101/2024.08.07.606975']\n",
      "Liens de dépôt logiciel valides pour l'article 52 : ['https://doi.org/10.1101/2024.09.26.615248', 'https://github.com/zhouyflab/R-Predictor', 'https://github.com/zhenyaliu77/DeepLRR', 'https://itol.embl.de', 'https://deepsig.biocomp.unibo.it', 'https://github.com/labstructbioinf/DeepCoil', 'https://ftp.ncbi.nlm.nih.gov/blast/db/FASTA/', 'https://github.com/SwiftSeal/resistify', 'https://golgi.sandbox.google.com', 'https://services.healthtech.dtu.dk/services/TMHMM-2.0', 'https://github.com/ElofssonLab/scampi2', 'https://biotec.njau.edu.cn/rlkdb', 'https://github.com/eliza-m/NLRexpress', 'https://cb.csail.mit.edu/cb/paircoil2', 'https://github.com/fteufel/signalp-6.0', 'https://github.com/phytolrr/predict-phytolrr', 'http://www.enzim.hu/hmmtop', 'https://mybiosoftware.com/coils-2-2-prediction-coiled-coil-regions-proteins.html', 'https://phobius.sbc.su.se', 'https://github.com/eliza-m/NLRexpress', 'https://services.healthtech.dtu.dk/services/LipoP-1.0', 'https://github.com/slt666666/NLRtracker', 'https://www.megasoftware.net', 'https://github.com/BolognaBiocomp/coconat']\n",
      "Liens de dépôt logiciel valides pour l'article 53 : ['https://www.tensorflow.org/', 'https://www.st', 'https://mrimaster.com/magn', 'https://www.mathworks.com/matl', 'https://www.imatest', 'https://ww', 'https://builtin.com/machine-learning/adam-optimization', 'https://www.statisticshowto.com/nrms', 'https://www.kaggle.com/datasets/dsch', 'https://doi.org/10.1101/2024.08.02.606374', 'https://keras.io/', 'https://onlinelibrary.wiley.com/doi/10.100', 'https://uhlibraries.pressbooks.pub/buildingskillsfordatascience/', 'https://nifti.nimh.nih.gov/']\n",
      "Liens de dépôt logiciel valides pour l'article 54 : ['https://doi.org/10.1101/2024.06.03.597180', 'https://cellxgene.cziscience', 'https://doi.org/10.1101/2023.10.30.563174']\n",
      "Liens de dépôt logiciel valides pour l'article 55 : ['https://datadryad.org/stash/dataset/doi:10.5061/dryad.x3ffbg7mw', 'https://doi.org/10.1101/2023.08.28.554970', 'https://github.com/']\n",
      "Liens de dépôt logiciel valides pour l'article 56 : ['https://github.com/ronboger/conformal-protein-', 'https://doi.org/10.1101/2024.06.27.601042']\n",
      "Liens de dépôt logiciel valides pour l'article 57 : ['https://doi.org/10.1101/2024.09.27.615406', 'https://imetalab.ca/']\n",
      "Liens de dépôt logiciel valides pour l'article 58 : ['https://doi.org/10.1101/2024.09.26.615249', 'https://swissmodel.expasy.org', 'https://blast.ncbi.nlm.nih.gov', 'https://www.uniprot.org]']\n",
      "Liens de dépôt logiciel valides pour l'article 59 : ['https://github.com/Graylab/', 'https://doi.org/10.1101/2024.09.27.615401']\n",
      "Liens de dépôt logiciel valides pour l'article 60 : ['https://citypopulation.de(accessedJune4,2023', 'https://doi.org/10.1101/2024.09.26.614772', 'https://metagenomics-workshop.readthedocs.io/en/2014-11-uppsala/']\n",
      "Liens de dépôt logiciel valides pour l'article 61 : ['https://doi.org/10.1101/2024.09.27.614001', 'https://zenodo.org/records/10679280', 'https://www.uniprot.org', 'https://github.com/salesforce/progen/', 'https://conoserver.org/', 'https://zenodo.org/records/7309036', 'https://zenodo.org/records/10679280', 'https://www.uniprot.org/', 'https://github.com/xueww/ConoDL）.Additionally', 'https://github.com/xueww/ConoDL.The']\n",
      "Liens de dépôt logiciel valides pour l'article 62 : ['http://creativecommons.org/licenses', 'https://doi.org/10.3390/xxxxx', 'https://doi.org/10.1101/2024.09.26.615132']\n",
      "Liens de dépôt logiciel valides pour l'article 63 : ['http://bioinfo.ihb.ac.cn/fishcode', 'https://github.com/hliihbcas/FishCODE', 'http://bioinfo.ihb.ac.cn/fishcode', 'https://doi.org/10.1101/2024.09.25.614839']\n",
      "Liens de dépôt logiciel valides pour l'article 64 : ['https://doi.org/10.1002/imt2.69', 'https://www.genome.jp/kegg/', 'https://www.medrxiv.org/content/10.1101', 'http://ccb.jhu.edu/software/fqtrim/index.shtml', 'https://github.com/Wonderangela123/M2AE', 'https://doi.org/10.1101/2024.09.25.614767']\n",
      "Liens de dépôt logiciel valides pour l'article 65 : ['https://doi.org/10.1016/j.smim.2023.101727', 'https://doi.org/10.1074/mcp.T400003-MCP200', 'https://doi.org/10.1093/nar/gkac1052', 'https://github.com/y-256/libdivsufsort', 'https://doi.org/10.1007/978-1-0716-4007-4_11', 'https://github.com/IlyaGrebnov/libsais', 'https://doi.org/10.1093/genetics/iyad031', 'https://doi.org/10.1093/bioinformatics/btaa553', 'https://doi.org/10.1080/19490976.2019.1702431', 'https://pubs.acs.org/doi/full/10.1021/acs.jproteome.3c00091', 'https://doi.org/10.1038/s41467-021-27542-8', 'https://prosite.expasy.org/scanprosite/', 'https://www.uniprot.org/peptide-search', 'https://doi.org/10.1186/s40168-020-00967-x', 'https://doi.org/10.1021/acs.jproteome.8b00716', 'https://doi.org/10.1093/nar/gkac993', 'https://doi.org/10.1093/bioinformatics/btt484', 'https://doi.org/10.1038/75556', 'https://doi.org/10.1101/2024.09.26.615136']\n",
      "Liens de dépôt logiciel valides pour l'article 66 : ['https://github.com/bioasp/precursor).We', 'https://www.plafrim.fr', 'https://github.com/bioasp/seed2lp', 'http://borensteinlab.com/software_NetSeed.html', 'https://github.com/potassco/clingo-lpx', 'https://doi.org/10.57745/OS1JND', 'http://dx.doi.org/10.1038/s41579-020-00458-8', 'http://bioinformatics', 'https://journals.asm.org/journal/mmbr', 'https://doi.org/10.21105/joss.03021', 'https://doi.org/10.1007/', 'https://doi.org/10.1101/2023.06.07.544035', 'http://www.ncbi.nlm.nih.gov/pubmed/28861526http:', 'https://doi.org/10.1101/2024.09.26.615309']\n",
      "Liens de dépôt logiciel valides pour l'article 67 : ['https://doi.org/10.1101/2024.09.25.615070', 'https://github.com/aberer/RogueNaRok/blob/master/example/', 'http://tree.bio.ed.ac.uk/software/figtree/']\n",
      "Liens de dépôt logiciel valides pour l'article 68 : ['https://www.gsea-msigdb.org/gsea/msigdb/index.jsp', 'https://hrfrost.host.dartmouth.edu/SCGeneSetOpt/', 'https://satijalab.org/seurat/articles/pbmc3k', 'https://cf.10xgenomics.com/samples/cell/pbmc3k/pbmc3k', 'https://doi.org/10.1101/2024.09.25.615040', 'https://hrfrost.host.dartmouth.edu/SCGeneSetOpt/', 'https://www.proteinatlas.org/download/rna']\n",
      "Liens de dépôt logiciel valides pour l'article 69 : ['https://i3-', 'https://doi.org/10.1101/2024.09.25.614939', 'https://github.com/i3-unit/AnalyzAIRR', 'https://github.com/i3-']\n",
      "Liens de dépôt logiciel valides pour l'article 70 : ['https://doi.org/10.1101/2024.04.08.588507', 'https://github.com/cgob/codonDT_snakemake', 'https://github.com/vam-sin/ribogl/']\n",
      "Liens de dépôt logiciel valides pour l'article 71 : ['https://doi.org/10.1016/0925-2312(91)90023-5', 'https://doi.org/10.7717/peerj.6543', 'https://doi.org/10.1093/bib/bbad073', 'https://doi.org/10.1007/s42979-024-02743-2', 'https://doi.org/10.1109/CAI54212.2023.00017', 'https://doi.org/10.1046/j.1471-4159.2003.01948.x', 'https://doi.org/10.1056/NEJMoa1202753', 'https://doi.org/10.1038/s41576-018-0004-3', 'https://doi.org/10.1038/s41586-019-1195-2', 'https://doi.org/10.1016/j.mad.2020.111319', 'https://doi.org/10.1111/biom.13108', 'https://doi.org/10.1016/j.biopsych.2009.01.027', 'https://doi.org/10.1023/A:1021709817809', 'https://doi.org/10.1101/2024.09.25.614862', 'https://doi.org/10.1159/000369101', 'https://doi.org/10.1038/s41390-', 'https://doi.org/10.1038/nrn.2015.1', 'https://doi.org/10.1038/ng.439', 'https://doi.org/10.1002/ima.22970', 'https://www.nejm.org/doi/full/10.1056/nejm199802193380804', 'https://doi.org/10.1016/j.bbi.2014.11.002', 'https://doi.org/10.1016/j.bspc.2024.106023', 'https://doi.org/10.1038/s41588-020-0696-0', 'https://doi.org/10.1016/j.ijbiomac.2020.07.327', 'https://doi.org/10.1214/12-AOAS597', 'https://doi.org/10.1016/j.csbj.2023.02.021', 'https://doi.org/10.1093/bioinformatics/btp543', 'https://doi.org/10.1007/BF00994018', 'https://doi.org/10.1016/j.neuron.2013.12.003', 'https://doi.org/10.1038/s41598-023-30892-6', 'https://doi.org/10.1007/978-0-387-', 'https://doi.org/10.19026/rjaset.7.299', 'https://doi.org/10.1016/j.neunet.2023.10.040', 'https://doi.org/10.1002/sim.9953', 'https://doi.org/10.1109/DSMP.2018.8478522', 'https://doi.org/10.1212/WNL.0000000000009058', 'https://doi.org/10.1038/tp.2012.45', 'https://doi.org/10.1096/fasebj.9.5.7896005', 'https://doi.org/10.1007/s11227-023-05655-9', 'https://doi.org/10.1111/acel.14044', 'https://doi.org/10.1007/978-94-007-0753-5_1689', 'https://doi.org/10.1109/ACCESS.2024.3390186', 'https://doi.org/10.3389/fnins.2024.1277187', 'https://doi.org/10.1038/ng.803', 'https://doi.org/10.1097/NEN.0b013e3181af2e46', 'https://doi.org/10.1038/ncomms11398', 'https://doi.org/10.1109/ICDAR.1995.598994', 'https://doi.org/10.1007/s11910-014-0499-8', 'https://doi.org/10.1007/978-0-387-39940-9_565', 'https://doi.org/10.1016/j.cca.2024.117842', 'https://doi.org/10.1093/bib/bbab569', 'https://doi.org/10.1016/j.jalz.2018.02.018', 'https://doi.org/10.7326/0003-4819-109-4-298', 'https://doi.org/10.1186/s13195-024-01488-7', 'https://doi.org/10.1371/journal.pone.0248413', 'https://doi.org/10.1016/j.bspc.2023.105669', 'https://doi.org/10.1007/s40745-022-00452-2', 'https://doi.org/10.1038/s41551-018-0304-0', 'https://doi.org/10.1093/brain/awz384', 'https://doi.org/10.1016/j.compbiomed.2023.107814', 'https://doi.org/10.2147/DNND.S228939', 'https://doi.org/10.1038/s41598-024-56837-1', 'https://doi.org/10.1038/nmeth.2930', 'https://doi.org/10.1016/j.bbi.2011.01.017', 'https://doi.org/10.3233/JAD-2006-9406', 'https://doi.org/10.1016/j.compbiomed.2023.107303', 'https://doi.org/10.1016/j.arr.2016.01.002', 'https://doi.org/10.1212/WNL.0000000000002923', 'https://doi.org/10.3389/frdem.2024.1332928', 'https://doi.org/10.1007/s40484-016-', 'https://doi.org/10.1007/978-3-540-39964-3_62', 'https://github.com/wan-mlab/WIMOAD', 'https://doi.org/10.1016/j.pneurobio.2019.101662']\n",
      "Liens de dépôt logiciel valides pour l'article 72 : ['https://docs.hubmapconsortium.org/assays/rnaseq', 'https://docs.hubmapconsortium.org/assays/seqfish', 'https://mtmorgan.github.io/', 'https://docs.hubmapconsortium.org/assays/codex', 'https://doi.org/10.1101/2024.09.26.615227', 'https://github.com/christinehou11/HuBMAPR)and', 'https://docs.hubmapconsortium.org/data-submission', 'https://github.com/christinehou11/HuBMAPR', 'https://www.R-project.org/', 'https://academic.oup.com/nar/article-lookup/doi/', 'https://www.perlfoundation.org/artistic-license-20', 'https://www.nature.com/', 'https://portal.hubmapconsortium.org/collections', 'https://portal.hubmapconsortium.org/', 'https://github.com/hubmapconsortium', 'https://genomebiology.biomedcentral.com/articles/', 'https://www.ncbi', 'https://hubmapconsortium.org/policies/', 'https://docs.hubmapconsortium.org/assays/imc', 'https://www.globus.org/', 'https://bmcbioinformatics', 'https://www.nature.com/articles/nmeth.4197']\n",
      "Liens de dépôt logiciel valides pour l'article 73 : ['https://bioportal.bioontology.org/ontologies/LSFO', 'https://github.com/EsmaeilNourani/LSFO-expansion', 'https://zenodo.org/records/10450308', 'https://github.com/spyysalo/bert-vocab-eval', 'https://doi.org/10.1101/2024.06.13.598816', 'https://esmaeilnourani.github.io/lifestylefactors-annotation-', 'https://github.com/larsjuhljensen/tagger']\n",
      "Liens de dépôt logiciel valides pour l'article 74 : ['https://doi.org/10.1101/2024.09.24.614785']\n",
      "Liens de dépôt logiciel valides pour l'article 75 : ['https://doi.org/10.1101/2024.09.24.614814', 'https://www.10xgenomics.com/', 'https://github.com/', 'https://www.10xgenomics.com/datasets/']\n",
      "Liens de dépôt logiciel valides pour l'article 76 : ['https://www.sciencedirect.com/science/article/pii/S0010482524001604', 'https://www.nature.com/articles/s41573-019-0024-5)(June2019', 'https://biosig.lab.uq.edu.au/pdcsm_gpcr/and', 'https://www.annualreviews.org/content/journals/10.1146/annurev.soc.27', 'https://doi.org/10.1186/s13321-024-00870-9)(June2024', 'http://arxiv.org/abs/', 'https://www.nature.com/articles/s41589-024-01679-1)(Aug.2024', 'https://github.com/chainer/chainer-chemistry/blob/master/chainer_', 'http://arxiv.org/abs/2303', 'https://doi.org/10.1021/acsomega', 'https://doi.org/10.1007/s10822-016-9906-3)(Mar.2016', 'http://arxiv.org/abs/2106.06134', 'https://www.nature', 'https://ieeexplore.ieee.org/document/9681226)(May2023', 'https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv', 'https://www.nature.com/articles/s41586-021-03819-2)(Aug.2021', 'https://github.com/ashvardanian/SimSIMD', 'https://doi.org/10.1101/2024.03.29.586957', 'http://arxiv.org/abs/2102.05497', 'https://doi.org/10.1093/bioinformatics/btaa968)(June', 'https://doi.org/', 'http://arxiv.org/abs/2404.14928', 'http://arxiv.org/abs/2404.19288', 'https://doi', 'https://www.biorxiv']\n",
      "Liens de dépôt logiciel valides pour l'article 77 : ['https://www.biorxiv.org/content/10.1101/2023.06.29.543848v1', 'https://www.biorxiv.org/content/10.1101/2024.09.19.313992v1', 'https://doi.org/10.1101/2024.09.24.614844']\n",
      "Liens de dépôt logiciel valides pour l'article 78 : ['https://doi.org/10.1186/s13100-019-0197-9', 'https://doi.org/10.3390/ijms23094755', 'https://doi.org/10.1101/2024.09.25.s614i8m65', 'https://doi.org/10.1038/28653', 'https://doi.org/10.1371/journal.pgen.1002487', 'https://doi.org/10.3390/biology1020395', 'https://doi.org/10.1186/s13100-015-0055-3', 'https://doi.org/10.1101/gr.226993.117', 'https://doi.org/10.1186/1471-2148-7-66', 'https://doi.org/10.1093/nar/gkab010', 'https://doi.org/10.1038/nrg2165-c4', 'https://doi.org/10.1038/nature05260', 'https://doi.org/10.3389/fpls.2022.814718', 'https://doi.org/10.1093/bib/bbm048', 'https://doi.org/10.1534/g3.117.043893', 'https://github.com/PacificBiosciences/pbsv', 'https://doi.org/10.1371/journal.pone.0030008', 'https://doi.org/10.1111/mec.14963', 'https://doi.org/10.1007/s00239-', 'https://doi.org/10.1534/g3.112.005348', 'https://doi.org/10.1126/science.1178534', 'https://doi.org/10.1093/bib/bbw072', 'https://doi.org/10.1038/nrg2640', 'https://get.genotoul.fr/', 'https://doi.org/10.1093/genetics/iyab113', 'https://github.com/e-lerat/replicaTE', 'https://doi.org/10.1093/nar/gku323', 'https://doi.org/10.1093/gbe/evx050', 'https://doi.org/10.7554/eLife.20777', 'https://doi.org/10.1101/2024.09.25.614865', 'https://www.girinst.org/', 'https://doi.org/10.1101/gr.277383.122', 'https://doi.org/10.1186/s13059-018-1577-z', 'https://doi.org/10.1093/bioinformatics/bts697', 'https://doi.org/10.1186/s12864-015-1975-5', 'https://www.r-', 'https://doi.org/10.7717/peerj.2942', 'https://doi.org/10.3390/cells11162501', 'https://www.arabidopsis.org/', 'https://doi.org/10.1093/molbev/msw137', 'https://tehub.org/', 'https://doi.org/10.1093/bioinformatics/btr708', 'https://doi.org/10.1007/s00521-021-06188-z', 'https://doi.org/10.1093/bioinformatics/btz727']\n",
      "Liens de dépôt logiciel valides pour l'article 79 : ['https://doi.org/10.1101/2024.09.24.614684', 'https://github.com/rashmisrm/MorphoGenie']\n",
      "Liens de dépôt logiciel valides pour l'article 80 : ['https://github.com/uttamLab/LMSF', 'https://doi.org/10.1101/2024.09.25.614955', 'https://doi.org/10.7303/syn61804540']\n",
      "Liens de dépôt logiciel valides pour l'article 81 : ['https://ferroenrich.shinyapps.io/ferroenrich/', 'https://doi.org/10.1101/2024.09.25.615075']\n",
      "Liens de dépôt logiciel valides pour l'article 82 : ['https://www.synapse.org/Synapse:syn31534849', 'https://doi.org/10.1016/0022-3956(75)90026-', 'https://www.synapse.org/Synapse:syn30821562', 'https://doi.org/10.1101/2024.08.23.609465', 'https://adknowledgeportal.synapse.org', 'https://www.synapse.org/Synapse:syn51739831', 'https://www.synapse.org/Synapse:syn52331749']\n",
      "Liens de dépôt logiciel valides pour l'article 83 : ['https://doi.org/10.1101/2024.0h9t.t2p4s.:6//1a3t8la3s3.;f', 'https://www.10xgenomics.com/re', 'https://atlas.fredhutch.org/nygc/m', 'https://github.com/welch-lab/liger', 'https://huggingface.co/ctheodoris/Geneformer', 'https://singlecell.broadinstitute.org/single_cell/study/SCP1303/', 'https://github.com/jaydu1/scVAEIT', 'https://azimuth.hubmapconsortium.org', 'https://www.10xgenomics.com/resources/datasets/', 'https://github.com/YosefLab/', 'https://www.10xgenomics.com/resources/datasets/10-', 'https://github.com/SONGDONGYUAN1994/scsampler', 'https://github.com/melobio/', 'https://satijalab.org/seurat/articles/seurat5_sketch_analysis.html', 'https://www.10xgenomics.com/resources/datasets/pbmc-', 'https://github.com/theislab/trVAE', 'https://github.com/', 'https://atlas.fredhutch.org/nygc/multimodal-pbmc', 'https://github.com/theislab/multigrate', 'https://doi.org/10.1101/2024.09.24.613833', 'https://www.10xgenomics.com/resources/datasets/10-k-', 'https://github.com/jlakkis/']\n",
      "Liens de dépôt logiciel valides pour l'article 84 : ['https://github.com/simoRancati/DeepAutoCoV', 'https://www.pfizer.com/news/press-release/press-release-detail/pfizer-and-biontech-provide-update-o', 'https://doi.org/10.1101/2023.10.24.563721', 'https://arxiv.org/abs/2109.05019v4', 'https://data.who.int/dashboards/covid19/cases', 'https://arxiv.org/abs/1605.08695v2', 'https://www.semanticscholar.org/paper/22-A-Model-of-Evolutionary-Change-in-Proteins-O./ff3e2e9', 'https://www.who.int/news/item/18-05-2023-statement-on-the-antigen-composition-of-covid-19-vacci', 'https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-classifications.html']\n",
      "Liens de dépôt logiciel valides pour l'article 85 : ['https://huggingface.co/', 'https://github.com/jeffreyruffolo/AntiBERTy)andAbLang2(https:', 'https://github.com/instadeepai/protein-sequence-bfn', 'https://doi.org/10.1101/2024.09.24.614734']\n",
      "Liens de dépôt logiciel valides pour l'article 86 : ['https://doi.org/10.1007/978-1-60327-367-1_12', 'https://doi.org/10.1007/978-0-387-39940-9_565', 'https://link.springer.com/10.1007/978-1-62703-059-5', 'https://doi.org/10.1101/2024.09.24.614777', 'https://www.researchsquare.com/article/rs-2798895/v1', 'https://www.stjude.cloud', 'http://www.ams.org/conm/026/', 'http://arxiv.org/abs/1802.03426', 'https://github.com/wan-mlab/RanBALL', 'https://pecan.stjude.cloud/static/hg19/pan-all/BALL-1988S-HTSeq.zip', 'https://dl.acm.org/doi/10.1145/1150402.1150436', 'https://journals.lww.com/hemasphere/fulltext/2022/04000/monitoring_of_leukemia_clo', 'https://orcid.org/0000-0003-0661-2684', 'http://perspectivesinmedicine.cshlp.org/content/10/7/a034835.abstract', 'http://ftp.ensembl.org/pub/release-']\n",
      "Liens de dépôt logiciel valides pour l'article 87 : ['https://github.com/bowang-lab/scGPT', 'https://doi.org/10.1101/2024.09.24.614685', 'https://cellxgene.cziscience.com/']\n",
      "Liens de dépôt logiciel valides pour l'article 88 : ['https://doi.org/10.1101/2024.09.24.614717', 'https://www.nist.gov/publication']\n",
      "Liens de dépôt logiciel valides pour l'article 89 : ['https://doi.org/10.1101/2024.09.24.614653', 'https://doi']\n",
      "Liens de dépôt logiciel valides pour l'article 90 : ['https://plant.connectome.tools/api/alias/psad1”', 'https://arxiv.org/abs/2408.07222v1', 'https://doi.org/10.3390/ijms22073560', 'https://doi.org/10.3389/fpls.2015.00277', 'https://colab.research.google.com/drive/1gENxLK2172Bq1sV_dO6JhvORsFgaq0fS?us', 'https://plant2.connectome.tools/normal/heat', 'https://agris-knowledgebase.org/downloads.html', 'https://github.com/mutwil/plant_connectome_latest/', 'https://doi.org/10.1038/s41467-019-08746-5', 'https://figshare.com/ndownloader/files/49392538', 'https://doi.org/10.1093/mp/ssp043', 'https://doi.org/10.1016/j.atg.2016.10.003', 'https://doi.org/10.1073/pnas.1219229110', 'https://doi.org/10.1111/pbi.13583', 'https://doi.org/10.1073/pnas.0503392102', 'https://plant.connectome.tools/catalogue', 'https://doi.org/10.1002/pro.3978', 'https://doi.org/10.1371/journal.pone.0105726', 'https://doi.org/10.1093/jxb/erz517', 'https://doi.org/10.1083/jcb.201807166', 'https://doi.org/10.1111/j.1365-313X.2004.02011.x', 'https://figshare.com/ndownloader/files/49198933', 'https://doi.org/10.1016/j.jbi.2018.08.009', 'https://doi.org/10.1038/srep10533', 'https://doi.org/10.3390/plants11091130', 'https://plant.connectome.tools/', 'https://doi.org/10.1105/tpc.105.031542', 'https://doi.org/10.1038/nmeth.2340', 'https://doi.org/10.1016/j.tplants.2013.10.006', 'https://doi.org/10.1186/s12859-019-2594-y', 'https://github.com/mutwil/plant_connectome_latest.git', 'https://doi.org/10.3389/fpls.2016.01445', 'https://doi.org/10.1038/s41477-020-0605-7', 'https://doi.org/10.1093/nar/gkac1000', 'https://openai.com/api/pricing/', 'https://doi.org/10.1101/2023.07.11.548541', 'https://doi.org/10.1371/journal.pone.0055814', 'https://doi.org/10.3390/plants10010103', 'https://doi.org/10.1002/pld3.483', 'https://plant.connectome.tools/alias/psad1', 'https://plant.connectome.tools/alias/MYB46', 'https://doi.org/10.1016/j.cell.2015.05.002', 'https://plant.connectome.tools/', 'https://doi.org/10.1104/pp.109.145318', 'https://doi.org/10.1105/tpc.108.061325', 'https://doi.org/10.3389/fpls.2021.770794', 'https://plant2.connectome.tools/alias/TOC75', 'https://doi.org/10.1016/j.csbj.2024.09.010', 'https://figshare.com/ndownloader/files/49392595', 'https://doi.org/10.1093/nar/gkq1120']\n",
      "Liens de dépôt logiciel valides pour l'article 91 : ['https://doi.org/10.1101/2024.02.29.582810', 'https://github.com/amberT15/LLM_eval)as', 'https://doi.org/10.5281/zenodo.11583224']\n",
      "Liens de dépôt logiciel valides pour l'article 92 : ['https://doi.org/10.1101/2024.09.22.614322']\n",
      "Liens de dépôt logiciel valides pour l'article 93 : ['https://doi.org/10.1126/science.abn8197', 'https://doi.org/10.1016/j.cels.2017.11.003', 'https://doi.org/10.1093/nar/gkx1153', 'https://doi.org/10.1093/bioinformatics/btu703', 'https://doi.org/10.1038/gim.2017.26', 'https://doi.org/10.1038/s41588-018-0192-y', 'https://doi.org/10.1038/nrg3031', 'https://doi.org/10.1186/s12864-', 'https://hgdownload.soe.ucsc.edu/gbdb/hg38/gnomAD/vcf/gnomad.genomes.r3.0.sites.vcf.gz', 'https://doi.org/10.1101/mcs.a006196', 'https://doi.org/10.1038/nbt.1589', 'https://doi.org/10.1016/j.ajhg.2021.08.012', 'https://doi.org/10.1038/nature24018', 'https://doi.org/10.1038/sj.ejhg.5201987', 'https://doi.org/10.1038/s41435-022-00192-6', 'https://doi.org/10.1016/j.cell.2011.09.008', 'https://doi.org/10.1038/s41525-023-00386-5', 'https://doi.org/10.1086/513473', 'https://doi.org/10.1038/s41586-', 'https://doi.org/10.1038/s41588-018-0167-z', 'https://ftp.ncbi.nlm.nih.gov/pub/clinvar/tab_delimited/archive/', 'https://doi.org/10.1016/j.molcel.2018.10.037', 'https://doi.org/10.1038/nrg2779', 'https://doi.org/10.1038/nature15393', 'https://doi.org/10.1093/bioinformatics/btq671', 'https://doi.org/10.1038/s41598-018-29325-6', 'https://doi.org/10.1038/ng.3810', 'https://doi.org/10.1186/s13229-015-', 'https://doi.org/10.1086/282718', 'https://doi.org/10.1093/bioinformatics/btu356', 'https://doi.org/10.1101/2024.09.24.614666', 'https://doi.org/10.1038/s10038-020-00845-2', 'https://doi.org/10.1186/s13073-021-00835-9', 'https://doi.org/10.1186/s12859-023-05294-0', 'https://doi.org/10.1038/nature17661', 'https://doi.org/10.1038/s41467-022-31872-6', 'https://doi.org/10.1093/bioinformatics/btv009', 'https://doi.org/10.7554/eLife.46922', 'https://hgdownload.soe', 'https://doi.org/10.1038/s41559-017-0447-5', 'https://doi.org/10.1016/j.neuron.2018.01.015', 'https://cadd.bihealth.org/static/ReleaseNotes_CADD_v1.6.pdf', 'https://doi.org/10.1002/humu.24309', 'https://doi.org/10.1038/gim.2017.14', 'https://doi.org/10.1371/journal.pcbi.1000322', 'https://doi.org/10.1038/s41467-019-11526-w', 'https://doi.org/10.1038/nature19057', 'https://doi.org/10.1136/jmedgenet-2014-102437', 'https://doi.org/10.1186/gb-2011-12-9-r84', 'https://doi.org/10.1056/NEJMoa2209046', 'https://doi.org/10.1146/annurev-genom-031714-125740', 'https://doi.org/10.1038/nrg3118', 'https://doi.org/10.1093/gbe/evad066', 'https://doi.org/10.1073/pnas.1318948111', 'https://doi.org/10.1093/hmg/11.20.2417', 'https://doi.org/10.1126/science.abj6987', 'https://doi.org/10.1146/annurev.ge.23.120189.002005', 'https://doi.org/10.1038/s41586-018-0461-z', 'https://doi.org/10.1038/ng.2892', 'https://doi.org/10.1038/nrg2986', 'https://doi.org/10.1161/CIRCGENETICS.111.960864', 'https://doi.org/10.1093/hmg/ddu400', 'https://doi.org/10.1093/nar/gky1016', 'https://doi.org/10.1186/s13073-022-01073-3', 'https://doi.org/10.1101/2022.03.20.485034', 'https://doi.org/10.1146/annurev-genom-083118-014857', 'https://doi.org/10.1038/s41525-022-00301-4', 'https://doi.org/10.1038/s41586-020-2859-', 'https://doi.org/10.1093/database/baad026', 'https://doi.org/10.1073/pnas.1114759108', 'https://doi.org/10.1093/nar/gkad989']\n",
      "Liens de dépôt logiciel valides pour l'article 94 : ['https://doi.org/10.1101/2024.09.24.614671', 'https://gephi.org/', 'https://f1000research.com/articles/10-33', 'https://www.biorxiv.org/content/10.1101/2024.05.08', 'https://vihoikka.github.io/rn_browser', 'https://github.com/vihoikka/ring_nucleases', 'http://www.rstudio.com/']\n",
      "Liens de dépôt logiciel valides pour l'article 95 : ['https://doi.org/10.1101/2024.09.24.614820', 'https://charticulator.com/app/index.html']\n",
      "Liens de dépôt logiciel valides pour l'article 96 : ['https://doi.org/10.1101/2024.09.24.614594', 'https://p53motifdb.its.albany.edu/', 'http://www.repeatmasker.org', 'https://CRAN.R-project.org/package=dplyr', 'https://shiny.posit.co/', 'https://CRAN.R-project.org/package=tidyr', 'https://p53motifDB.its.albany.edu']\n",
      "Liens de dépôt logiciel valides pour l'article 97 : ['https://doi.org/10.1101/2024.09.23.614570', 'https://yanglab.qd.sdu.edu.cn/trRosettaRNA/download/']\n",
      "Liens de dépôt logiciel valides pour l'article 98 : ['https://scpca.alexslemonade.org/', 'https://github.com/favilaco/deconv_benchmark', 'https://github.com/amitfrish/scBio', 'https://doi.org/10.1101/2023.02.06.527318']\n",
      "Liens de dépôt logiciel valides pour l'article 99 : ['https://doi.org/10.1101/2024.09.23.614585', 'https://github']\n",
      "Liens de dépôt logiciel valides pour l'article 100 : ['https://www.10xgenomics.com/datasets/mouse-brain-serial-section-2-sagittal-posterior-', 'https://doi.brainimagelibrary.org/', 'https://doi.org/', 'https://github.com/jiazhao97/INSPIRE', 'https://singlecell.broadinstitute', 'https://db.cngb.org/stomics/mosta/', 'https://www.10xgenomics.com/datasets/mouse-brain-section-coronal-1-standard-1-', 'https://doi.org/10.1038/s41586-024-07069-w', 'https://crukci.shinyapps.io/', 'https://doi', 'https://www.10xgenomics.com/datasets', 'https://doi.org/10.1101/2024.09.23.614539', 'https://www.10xgenomics.com/datasets/mouse-brain-serial-section-2-sagittal-anterior-1-', 'https://github.com/jiazhao97/INSPIRE']\n"
     ]
    }
   ],
   "source": [
    "import sqlite3\n",
    "import requests\n",
    "import pdfplumber\n",
    "import io\n",
    "import re\n",
    "import time\n",
    "\n",
    "# Connexion à la base de données SQLite\n",
    "database = sqlite3.connect('bioinformatics_articles.db')\n",
    "c = database.cursor()\n",
    "\n",
    "# Ajouter une colonne pour stocker les liens de dépôt logiciel si elle n'existe pas\n",
    "try:\n",
    "    c.execute(\"ALTER TABLE articles ADD COLUMN software_links TEXT\")\n",
    "    database.commit()\n",
    "except sqlite3.OperationalError:\n",
    "    pass  # La colonne existe déjà\n",
    "\n",
    "# Fonction pour vérifier si un lien est valide\n",
    "def is_valid_link(link):\n",
    "    return link.startswith('http://') or link.startswith('https://')\n",
    "\n",
    "# Fonction pour nettoyer les liens partiels ou incorrects\n",
    "def clean_link(link):\n",
    "    link = link.strip(').;,')  # Supprimer les parenthèses et les points en fin de lien\n",
    "    return link if is_valid_link(link) else None\n",
    "\n",
    "# Fonction pour extraire les liens à partir d'un texte\n",
    "def extract_links_from_text(text):\n",
    "    return re.findall(r'https?://\\S+', text)\n",
    "\n",
    "# Fonction pour extraire les liens de dépôt logiciel\n",
    "def extract_software_links():\n",
    "    # Récupérer tous les articles de la base de données\n",
    "    c.execute(\"SELECT id, title, pdf_link, abstract, software_links FROM articles\")\n",
    "    articles = c.fetchall()\n",
    "\n",
    "    for article in articles:\n",
    "        article_id, title, pdf_link, abstract, existing_links = article\n",
    "\n",
    "        # Si l'article a déjà des liens de dépôt logiciel, on l'ignore\n",
    "        if existing_links:\n",
    "            print(f\"Article {article_id} déjà traité, on passe au suivant.\")\n",
    "            continue\n",
    "\n",
    "        software_links_set = set()  # Utiliser un set pour éliminer les doublons\n",
    "\n",
    "        # Vérifier le lien PDF\n",
    "        if pdf_link and pdf_link != 'Lien PDF non disponible':\n",
    "            try:\n",
    "                # Télécharger le fichier PDF\n",
    "                response = requests.get(pdf_link)\n",
    "                response.raise_for_status()  # Vérifie si la requête a réussi\n",
    "\n",
    "                # Utiliser pdfplumber pour extraire le texte du PDF\n",
    "                with pdfplumber.open(io.BytesIO(response.content)) as pdf:\n",
    "                    for page in pdf.pages:\n",
    "                        text = page.extract_text()\n",
    "                        if text:\n",
    "                            # Rechercher des liens de dépôt logiciel dans le texte\n",
    "                            software_links_set.update(extract_links_from_text(text))\n",
    "\n",
    "            except Exception as e:\n",
    "                print(f\"Erreur lors de l'ouverture du PDF pour l'article {article_id}: {e}\")\n",
    "\n",
    "        # Si aucun lien n'a été trouvé dans le PDF, vérifier l'abstract\n",
    "        if not software_links_set and abstract != 'Abstract non disponible':\n",
    "            software_links_set.update(extract_links_from_text(abstract))\n",
    "\n",
    "        # Nettoyage des liens et validation\n",
    "        valid_links = [clean_link(link) for link in software_links_set if clean_link(link)]\n",
    "\n",
    "        # Mise à jour de la base de données avec les liens de dépôt logiciel\n",
    "        if valid_links:\n",
    "            print(f\"Liens de dépôt logiciel valides pour l'article {article_id} : {valid_links}\")\n",
    "            c.execute(\"UPDATE articles SET software_links = ? WHERE id = ?\", (', '.join(valid_links), article_id))\n",
    "        else:\n",
    "            print(f\"Aucun lien de dépôt logiciel trouvé pour l'article {article_id}\")\n",
    "\n",
    "        time.sleep(1)  # Pause pour éviter de surcharger le serveur\n",
    "\n",
    "    # Sauvegarder les modifications dans la base de données\n",
    "    database.commit()\n",
    "\n",
    "# Appeler la fonction d'extraction\n",
    "extract_software_links()\n",
    "\n",
    "# Fermer la connexion à la base de données\n",
    "#database.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
