{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Installation de bibliotheques \n",
    "\n",
    "**pip install beautifulsoup4**\n",
    "\n",
    "**pip install requests**\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import sqlite3\n",
    "import time\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Scraper les articles bioRxiv liés à la bioinformatique "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = 'https://www.biorxiv.org'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<sqlite3.Cursor at 0x7fd3d81cb5e0>"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# creer base de données SQLite \"bioinformatics_articles\" \n",
    "database = sqlite3.connect('bioinformatics_articles.db')\n",
    "c = database.cursor() #ici le curseur 'c' permettra d'executer/parcourir les requetes SQL sur database\n",
    "\n",
    "\n",
    "c.execute(\"DROP TABLE IF EXISTS articles\")\n",
    "\n",
    "# create table 'articles' avec col id,title et le lien \n",
    "c.execute('''CREATE TABLE IF NOT EXISTS articles\n",
    "             (id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "             title TEXT, \n",
    "             link TEXT,\n",
    "             doi TEXT UNIQUE,\n",
    "             date TEXT,\n",
    "             pdf_link TEXT,\n",
    "             abstract TEXT)''')\n",
    "#id c'est un identifiant unique généré automatiquement "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Parcourir les articles  de la collection Bioinformatics contenant 2153 pages d'articles et enregistrer leurs informations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scraping page 1 sur 3\n",
      "Nombre total d'articles trouvés sur la page 1 : 10\n",
      "Article ajouté : Enhanced prediction of protein functional identity through the integration of sequence and structural features\n",
      "Titre: Enhanced prediction of protein functional identity through the integration of sequence and structural features\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615718v1\n",
      "DOI: 10.1101/2024.09.30.615718\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615718v1.full.pdf+html\n",
      "Résumé:Although over 300 million protein sequences are registered in a reference sequence database, only 0.2% have experimentally determined functions. This suggests that many valuable proteins, potentially catalyzing novel enzymatic reactions, remain undiscovered among the vast number of function-unknown proteins. In this study, we developed a method to predict whether two proteins catalyze the same enzymatic reaction by analyzing sequence and structural similarities, utilizing structural models predicted by AlphaFold2. We performed pocket detection and domain decomposition for each structural model. The similarity between protein pairs was assessed using features such as full-length sequence similarity, domain structural similarity, and pocket similarity. We developed several models using conventional machine learning algorithms and found that the LightGBM-based model outperformed the models. Our method also surpassed existing approaches, including those based solely on full-length sequence similarity and state-of-the-art deep learning models. Feature importance analysis revealed that domain sequence identity, calculated through structural alignment, had the greatest influence on the prediction. Therefore, our findings demonstrate that integrating sequence and structural information improves the accuracy of protein function prediction.\n",
      "\n",
      "Article ajouté : Phage display enables machine learning discovery of cancer antigen specific TCRs\n",
      "Titre: Phage display enables machine learning discovery of cancer antigen specific TCRs\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.06.27.600973v2\n",
      "DOI: 10.1101/2024.06.27.600973\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.06.27.600973v2.full.pdf+html\n",
      "Résumé:T cells targeting epitopes in infectious diseases or cancer play a central role in spontaneous and therapy-induced immune responses. T-cell epitope recognition is mediated by the binding of the T-Cell Receptor (TCR) and TCRs recognizing clinically relevant epitopes are promising for T-cell based therapies. Starting from one of the few known TCRs targeting the cancer-testis antigen NY-ESO-1 157-165 epitope, we built large phage display libraries of TCRs with randomized Complementary Determining Region 3 of the β chain. The TCR libraries were panned against the NY-ESO-1 epitope, which enabled us to collect thousands of epitope-specific TCR sequences. We then trained a machine learning TCR-epitope interaction predictor with this data and could identify several epitope-specific TCRs directly from TCR repertoires. Cellular binding and functional assays revealed that the predicted TCRs displayed activity towards the NY-ESO-1 epitope and no detectable cross-reactivity with self-peptides. Overall, our work demonstrates how display technologies combined with machine learning models of TCR-epitope recognition can effectively leverage large TCR repertoires for TCR discovery.\n",
      "\n",
      "Article ajouté : AptaDiff: de novo design and optimization of aptamers based on diffusion models\n",
      "Titre: AptaDiff: de novo design and optimization of aptamers based on diffusion models\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.11.25.568693v2\n",
      "DOI: 10.1101/2023.11.25.568693\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.11.25.568693v2.full.pdf+html\n",
      "Résumé:Aptamers are single-stranded nucleic acid ligands, featuring high affinity and specificity to target molecules. Traditionally they are identified from large DNA/RNA libraries using in vitro methods, like Systematic Evolution of Ligands by Exponential Enrichment (SELEX). However, these libraries capture only a small fraction of theoretical sequence space, and various aptamer candidates are constrained by actual sequencing capabilities from the experiment. Addressing this, we proposed AptaDiff, the first in silico aptamer design and optimization method based on the diffusion model. Our Aptadiff can generate aptamers beyond the constraints of high-throughput sequencing data, leveraging motif-dependent latent embeddings from variational autoencoder, and can optimize aptamers by affinity-guided aptamer generation according to Bayesian optimization. Comparative evaluations revealed AptaDiff9s superiority over existing aptamer generation methods in terms of quality and fidelity across four high-throughput screening data targeting distinct proteins. Moreover, Surface Plasmon Resonance (SPR) experiments were conducted to validate the binding affinity of aptamers generated through Bayesian optimization for two target proteins. The results unveiled a significant boost of 87.9% and 60.2% in RU values, along with a 3.6-fold and 2.4-fold decrease in KD values for the respective target proteins. Notably, the optimized aptamers demonstrated superior binding affinity compared to top experimental candidates selected through SELEX, underscoring the promising outcomes of our AptaDiff in accelerating the discovery of superior aptamers.\n",
      "\n",
      "Article ajouté : Gut microbiota and type 2 diabetes associations: A meta-analysis of 16s studies and their methodological challenges\n",
      "Titre: Gut microbiota and type 2 diabetes associations: A meta-analysis of 16s studies and their methodological challenges\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.10.01.615711v1\n",
      "DOI: 10.1101/2024.10.01.615711\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.10.01.615711v1.full.pdf+html\n",
      "Résumé:Diabetes mellitus is a prevalent chronic non-communicable disease, and recent studies have explored the link between gut microbiota and its development. Despite some evidence suggesting an association, the influence of gut microbiota on type 2 diabetes (DM2) remains unclear. A systematic search of PubMed (Janeiro 2016 to 10 December 2023) using the keywords: (16S and diabetes or DM2) and (gut microbiota and diabetes or DM2). The studies included compared gut microbiome diversity between diabetic and non-diabetic adults using 16S rRNA sequencing, excluding children, interventions, and type 1 diabetes. Alpha diversity indices and bacterial mean abundance were analyzed, with statistical assessments using a random-effects model and I2 for heterogeneity. Thirteen studies met the criteria, with the Shannon index being the most commonly used measure. Results showed significant heterogeneity (I2 > 75%) and no notable differences between diabetic and non-diabetic groups. Other indices, such as Chao1 and phylogenetic whole tree, similarly showed no consistent differences. Taxonomic analysis also failed to find phyla consistently correlated with DM2, with variability across studies. The relationship between gut microbiota and diabetes remains uncertain due to technical and biological factors that are often overlooked. The inconsistencies across studies highlight the low reproducibility common in microbiota research.\n",
      "\n",
      "Article ajouté : scGNN+: Adapting ChatGPT for Seamless Tutorial and Code Optimization\n",
      "Titre: scGNN+: Adapting ChatGPT for Seamless Tutorial and Code Optimization\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615735v1\n",
      "DOI: 10.1101/2024.09.30.615735\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615735v1.full.pdf+html\n",
      "Résumé:Foundation models have transformed AI by leveraging large-scale data to efficiently perform diverse tasks, and their applications in bioinformatics are primarily focused on data-centric tasks like cell type annotation and gene expression analysis. However, their potential extends beyond data analysis, offering significant opportunities in software development and optimization, such as code refinement, tutorial generation, and advanced visualization. For example, models like OpenAI Codex can suggest optimized code snippets and generate well-documented, reproducible workflows, enhancing accessibility and reducing computational complexity. Despite these advantages, the use of foundation models for improving computational tool engineering in single-cell research remains underutilized. To address this gap, we developed scGNN+, a web-based platform that combines the power of graph neural networks with the capabilities of ChatGPT to enhance reproducibility, code optimization, and visualization. scGNN+ further simplifies the process for users by generating standardized, well-annotated code, making complex procedures more accessible to non-programmers. Additionally, ChatGPT integration allows users to create high-quality, customizable visualizations through natural language prompts, improving data interpretation and presentation. Ultimately, scGNN+ offers a user-friendly, reproducible, and optimized solution for single-cell research, leveraging the full potential of foundation models in bioinformatics. scGNN+ is publicly available at https://bmblx.bmi.osumc.edu/scgnn+.\n",
      "\n",
      "Article ajouté : Isobaric labeling update in MaxQuant\n",
      "Titre: Isobaric labeling update in MaxQuant\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615788v1\n",
      "DOI: 10.1101/2024.09.30.615788\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615788v1.full.pdf+html\n",
      "Résumé:We present an update of the MaxQuant software for isobaric labeling data and evaluate its performance on benchmark datasets. Impurity correction factors can be applied to labels mixing C- and N-type reporter ions, such as TMT Pro. Application to a single-cell multi-species mixture benchmark shows high accuracy of the impurity-corrected results. TMT data recorded with FAIMS separation can be analyzed directly in MaxQuant without splitting the raw data into separate files per FAIMS voltage. Weighted median normalization, is applied to several datasets, including large-scale human body atlas data. In the benchmark datasets the weighted median normalization either removes or strongly reduces the batch effects between different TMT plexes and results in clustering by biology. In datasets including a reference channel, we find that weighted median normalization performs as well or better when the reference channel is ignored and only the sample channel intensities are used, suggesting that the measurement of a reference channel is unnecessary when using weighted median normalization in MaxQuant. We demonstrate that MaxQuant including the weighted median normalization performs well on multi-notch MS3 data, as well as on phosphorylation data. MaxQuant is freely available for any purpose and can be downloaded from https://www.maxquant.org/.\n",
      "\n",
      "Article ajouté : Preserving Full Spectrum Information in Imaging Mass Spectrometry Data Reduction\n",
      "Titre: Preserving Full Spectrum Information in Imaging Mass Spectrometry Data Reduction\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.614425v1\n",
      "DOI: 10.1101/2024.09.30.614425\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.614425v1.full.pdf+html\n",
      "Résumé:Imaging Mass Spectrometry (IMS) has become an important tool for molecular characterization of biological tissue. However, IMS experiments tend to yield large datasets, routinely recording over 200,000 ion intensity values per mass spectrum and more than 100,000 pixels, i.e., spectra, per dataset. Traditionally, IMS data size challenges have been addressed by feature selection or extraction, such as by peak picking and peak integration. Selective data reduction techniques such as peak picking only retain certain parts of a mass spectrum, and often these describe only medium-to-high-abundance species. Since lower-intensity peaks and, for example, near-isobar species are sometimes missed, selective methods can potentially bias downstream analysis towards a subset of species in the data rather than considering all species measured. Results: We present an alternative to selective data reduction of IMS data that achieves similar data size reduction while better conserving the ion intensity profiles across all recorded m/z-bins, thereby preserving full spectrum information. Our method utilizes a low-rank matrix completion model combined with a randomized sparse-format-aware algorithm to approximate IMS datasets. This representation offers reduced dimensionality and a data footprint comparable to peak picking, but also retains complete spectral profiles, enabling comprehensive analysis and compression. We demonstrate improved preservation of lower signal-to-noise-ratio signals and near-isobars, mitigation of selection bias, and reduced information loss compared to current state-of-the art data reduction methods in IMS.\n",
      "\n",
      "Article ajouté : BindCraft: one-shot design of functional protein binders\n",
      "Titre: BindCraft: one-shot design of functional protein binders\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615802v1\n",
      "DOI: 10.1101/2024.09.30.615802\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615802v1.full.pdf+html\n",
      "Résumé:Protein-protein interactions (PPIs) are at the core of all key biological processes. However, the complexity of the structural features that determine PPIs makes their design challenging. We present BindCraft, an open-source and automated pipeline for de novo protein binder design with experimental success rates of 10-100%. BindCraft leverages the trained deep learning weights of AlphaFold2 to generate nanomolar binders without the need for high-throughput screening or experimental optimization, even in the absence of known binding sites. We successfully designed binders against a diverse set of challenging targets, including cell-surface receptors, common allergens, de novo designed proteins, and multi-domain nucleases, such as CRISPR-Cas9. We showcase their functional and therapeutic potential by demonstrating that designed binders can reduce IgE binding to birch allergen in patient-derived samples. This work represents a significant advancement towards a \"one design-one binder\" approach in computational design, with immense potential in therapeutics, diagnostics, and biotechnology.\n",
      "\n",
      "Article ajouté : UniGEN-DDI: Computing drug-drug interactions using a unified graph embedding network\n",
      "Titre: UniGEN-DDI: Computing drug-drug interactions using a unified graph embedding network\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615812v1\n",
      "DOI: 10.1101/2024.09.30.615812\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615812v1.full.pdf+html\n",
      "Résumé:Finding drug-drug interaction is crucial for patient safety and treatment efficacy. Two drugs may show a synergistic effect but may sometimes cause a severe health issue, including lethality. Wet lab studies are often performed to understand such interactions but are limited by cost and time. However, the biochemical data generated can be explored to compute unknown interactions. Here, we developed a computational model named UniGEN-DDI (Unified Graph Embedding Network for Drug-Drug Interaction) for the estimation of interactions between drugs. It is a simple unified network model containing the biochemical information of drug association developed using the compiled data from DrugBank 5.1.0. The feature learning of the drugs was carried out using a combination of GraphSAGE and Node2Vec algorithms, which were found efficient in extracting diverse features. The simple architecture of our model led to a significant reduction in computational time compared to the baselines, while maintaining a high prediction accuracy. The model performed well on the data, which was equally distributed between interacting and non-interacting drugs. As a more challenging evaluation, we performed non-overlapping splitting of the data based on the drug action on different parts of the body, and our model performed well in both interaction estimation and time efficiency. Our model successfully identified the 12 unknown drug interactions in DrugBank 5.1.0, which were updated in DrugBank 6.0.\n",
      "\n",
      "Article ajouté : Divergence of the Individual repeats in the leucine-rich repeat domains of human Toll-like receptors explain their diversity and functional adaptations.\n",
      "Titre: Divergence of the Individual repeats in the leucine-rich repeat domains of human Toll-like receptors explain their diversity and functional adaptations.\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615863v1\n",
      "DOI: 10.1101/2024.09.30.615863\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615863v1.full.pdf+html\n",
      "Résumé:The receptor domains of Toll-like receptors (TLRs) are characterized by a solenoid-like structure composed of tandem repeats of α/β units known as Leucine Rich Repeats (LRRs). LRR proteins form large paralogous families, with nearly 400 in the human genome alone, all sharing similar semi-regular solenoid-like structures. Despite this structural similarity, they exhibit remarkable diversity in binding specificity. For TLR receptors, this includes a range of pathogen-associated molecular patterns (PAMPs), while other LRR proteins bind an extensive array of ligands, including proteins, DNA, RNA, and small molecules. The LRR domains contain repeats that have similar, yet not identical, 3D structures and patterns of conserved residues. Through in-depth analysis of sequence and structural conservation in individual repeats of human TLRs, we demonstrate that even subtle variations between these repeats alter the local solenoid structure, leading to significant functional changes. Variations in repeat length and defining patterns result in local changes in curvature and the emergence of structural features such as loops, cavities, or specific interaction interfaces. Understanding how divergence in LRR repeats influences their functional roles can provide deeper insights into their binding mechanisms, including interactions with unknown ligands, both in humans and across the diverse phylogenetic spectrum of animals that rely on their TLR repertoires for immune activation.\n",
      "\n",
      "Scraping page 2 sur 3\n",
      "Nombre total d'articles trouvés sur la page 2 : 10\n",
      "Article ajouté : RAPDOR: Using Jensen-Shannon Distance for the computational analysis of complex proteomics datasets\n",
      "Titre: RAPDOR: Using Jensen-Shannon Distance for the computational analysis of complex proteomics datasets\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615781v1\n",
      "DOI: 10.1101/2024.09.30.615781\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615781v1.full.pdf+html\n",
      "Résumé:The computational analysis of large proteomics datasets, such as those from gradient profiling or spatially resolved proteomics, is often as crucial as the experimental design. We present RAPDOR, a tool for intuitive analyzing and visualizing such datasets, based on the Jensen-Shannon distance and subsequent analysis of similarities between replicates, applied to three datasets. First, we examined the in-gradient distribution profiles of protein complexes with or without RNase treatment (GradR) to identify the set of RNA-binding proteins (RBPs) in the cyanobacterium Synechocystis sp. PCC 6803. RBPs play pivotal regulatory and structural roles; although numerous RBPs have been identified, the complete set is unknown for any species. RAPDOR identified 80 potential RBPs, including ribosomal proteins, likely RNA-modifying enzymes, and several proteins not previously associated with RNA binding. High-ranking putative RBPs, such as the universal stress protein Sll1388, or the translation inhibitor LrtA/RaiA, were predicted by RAPDOR but not the TriPepSVM algorithm, indicating uncharacterized RBP domains. These data are available online at https://synecho-rapdor.biologie.uni-freiburg.de, providing a comprehensive resource for RNase-sensitive protein complexes in cyanobacteria. We then show by reanalyzing existing datasets, that RAPDOR is effective in examining the intracellular redistribution of proteins under stress conditions. RAPDOR is a generic, non-parametric tool for the intuitive and versatile analysis of highly complex data sets such as the study of protein distributions using fractionation protocols.\n",
      "\n",
      "Article ajouté : SAMURAI: Shallow Analysis of Copy nuMber alterations Using a Reproducible And Integrated bioinformatics pipeline\n",
      "Titre: SAMURAI: Shallow Analysis of Copy nuMber alterations Using a Reproducible And Integrated bioinformatics pipeline\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615766v1\n",
      "DOI: 10.1101/2024.09.30.615766\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615766v1.full.pdf+html\n",
      "Résumé:Shallow whole-genome sequencing (sWGS) offers a cost-effective approach to detect  copy number alterations (CNAs). However, there remains a gap for a standardized workflow specifically designed for sWGS analysis. To address this need, in this work we present SAMURAI  a bioinformatics pipeline specifically designed for analyzing CNAs from sWGS data in a standardized and reproducible manner. SAMURAI is built using established community standards, ensuring portability, scalability, and reproducibility. The pipeline features a modular design with independent blocks for data pre-processing, copy number analysis, and customized reporting. Users can select workflows tailored for either solid or liquid biopsy analysis (e.g., circulating tumor DNA), with specific tools integrated for each sample type. The final report generated by SAMURAI provides detailed results to facilitate data interpretation and potential downstream analyses.To demonstrate its robustness, SAMURAI was validated using simulated and real-world datasets. The pipeline achieved high concordance with ground truth data and maintained consistent performance across various scenarios. By promoting standardization and offering a versatile workflow, SAMURAI empowers researchers in diverse environments to reliably analyze CNAs from sWGS data. This, in turn, holds promise for advancements in precision medicine.\n",
      "\n",
      "Article ajouté : Capturing the Hierarchically Assorted Protein-protein Interaction Modules of Mammalian Cell\n",
      "Titre: Capturing the Hierarchically Assorted Protein-protein Interaction Modules of Mammalian Cell\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615776v1\n",
      "DOI: 10.1101/2024.09.30.615776\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615776v1.full.pdf+html\n",
      "Résumé:Proteins are organized into modules by both functions and physical interactions within compartments of an eukaryotic cell. The in vivo chemical crosslinking mass spectrometry (XL-MS) data collected from organelles, the whole cells and tissues are able to provide unique information about both protein-protein interaction (PPI) and the intensity of PPI. In the present study, we have retrieved 55,982 crosslinked peptides (XL-peptides) from the XL-MS databases, out of which 6,356 in vivo PPIs were identified. Introduction of the MONET software into analysis of 4,526 hetero PPIs revealed a total of 402 protein modules, including 15, 58 and 163 stable protein complex(s), condensate-forming protein module(s) and intrinsically disordered region (IDR)-containing protein module(s), respectively. The application of ChatGPT in analysis of these modules unexpectedly identified 4 vesicle-related modules. Together, these modules were assorted into 6 communities (module of modules) and 3 systems (module of communities) hierarchically. The bioinformatic analysis found that the three systems are corresponding to three compartments of eukaryotic cell: nuclei, mitochondria, endoplasmic reticulum (ER), respectively. This study presents a novel and comprehensive biomolecular modulome of a mammalian cell, which captures putative protein compositions of protein complexes, protein condensates and vesicles and provides a hierarchical protein organization and function within compartments of mammalian cell.\n",
      "\n",
      "Article ajouté : Separating biological variance from noise by applying EM algorithm to modified General Linear Model\n",
      "Titre: Separating biological variance from noise by applying EM algorithm to modified General Linear Model\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615661v1\n",
      "DOI: 10.1101/2024.09.29.615661\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615661v1.full.pdf+html\n",
      "Résumé:Introduction: The General Linear Model (GLM) has been widely used in research, where error term has been treated as noise. However, compelling evidence suggests that in biological systems, the target variables may possess their innate variances. \n",
      "Methods: A modified GLM was proposed to explicitly model biological variance and non-biological noise. Employing the Expectation and Maximization (EM) scheme can distinguish biological variance from noise, termed EMSEV (EM for Separating Variances). The performance of EMSEV was evaluated by varying noise levels, dimensions of the design matrix, and covariance structures of the target variables.\n",
      "Results: The deviation between EMSEV outputs and the pre-defined distribution parameters increased with noise level. With a proper initial guess, when the noise magnitude and the variance of the target variables were similar, there were deviations of 3% and 10 to 16% in the estimated mean and covariance of the target variables, respectively, along with a 1.7% deviation in noise estimation.\n",
      "Conclusion: EMSEV appears promising for distinguishing signal variance from noise in biological systems. The potential applications and implications in biological science and statistical inference are discussed.\n",
      "\n",
      "Article ajouté : Advancing Protein-DNA Binding Site Prediction: Integrating Sequence Models and Machine Learning Classifiers\n",
      "Titre: Advancing Protein-DNA Binding Site Prediction: Integrating Sequence Models and Machine Learning Classifiers\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.08.23.554389v2\n",
      "DOI: 10.1101/2023.08.23.554389\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.08.23.554389v2.full.pdf+html\n",
      "Résumé:Predicting protein-DNA binding sites is a challenging computational problem that has led to the development of advanced algorithms and techniques in the field of bioinformatics. Identifying the specific residues where proteins bind to DNA is of paramount importance, as it enables the modeling of their interactions and facilitates downstream studies. Nevertheless, the development of accurate and efficient computational methods for this task remains a persistent challenge. Accurate prediction of protein-DNA binding sites has far-reaching implications for understanding molecular mechanisms, disease processes, drug discovery, and synthetic biology applications. It helps bridge the gap between genomics and functional biology, enabling researchers to uncover the intricacies of cellular processes and advance our knowledge of the biological world. The method used to predict DNA binding residues in this study is a potent combination of conventional bioinformatics tools, protein language models, and cutting-edge machine learning and deep learning classifiers. On a dataset of protein-DNA binding sites, our model is meticulously trained, and it is then rigorously examined using several experiments. As indicated by higher predictive behavior with AUC values on two benchmark datasets, the results show superior performance when compared to existing models. The suggested model has a strong capacity for generalization and shows specificity for DNA-binding sites. We further demonstrated the adaptability of our model as a universal framework for binding site prediction by training it on a variety of protein-ligand binding site datasets. In conclusion, our innovative approach for predicting protein-DNA binding residues holds great promise in advancing our understanding of molecular interactions, thus paving the way for several groundbreaking applications in the field of molecular biology and genetics. Our approach demonstrated efficacy and versatility underscore its potential for driving transformative discoveries in biomolecular research.\n",
      "\n",
      "Article ajouté : Exploring the link between extended red blood cell parameters and platelet indices in voluntary blood donors\n",
      "Titre: Exploring the link between extended red blood cell parameters and platelet indices in voluntary blood donors\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615743v1\n",
      "DOI: 10.1101/2024.09.29.615743\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615743v1.full.pdf+html\n",
      "Résumé:Background: As regular blood donors are prone to iron deficiency, importance of extended red blood cell (eRBC) parameters in identifying donors with depleted iron stores was investigated.Thrombocytosis has been well documented in patients affected with IDA. Thus, significance of eRBC parameters in identifying iron deficiency associated thrombocytosis was also examined in this cohort. Methods: Blood samples were collected in EDTA tubes from consenting donors for analyses of routine haematological and eRBC parameters. Serum samples were isolated for estimation of iron parameters. Results: Iron deficient donors had significantly altered eRBC parameters. Among them, Ret-He with a cut-off of ≥32 pg had high AUC (0.822) and showed relatively high sensitivity & specificity in detecting iron deficiency. Combination of Ret-He with CCI increased sensitivity & specificity to 90.6% and 98.2%, in detection of donors affected with iron restricted erythropoiesis. This cohort had increased platelet counts, which showed significant association with Ret-He (β= -0.373), RBC-He (β= -0.384), CCI (β= 0.384), Hypo-He (β=0.494) and Micro-R (β= 0.299). Elevated platelet counts also showed significant correlations with these eRBC parameters, which was absent in iron replete donors. Conclusions: eRBC parameters are sensitive indicators of non-anaemic iron deficiency, which may be enhanced by combining them. Their significant association with elevated platelet counts in iron deficient donors, highlights their importance in reflecting iron deficiency associated thrombocytosis.\n",
      "\n",
      "Article ajouté : PlasmidGPT: a generative framework for plasmid design and annotation\n",
      "Titre: PlasmidGPT: a generative framework for plasmid design and annotation\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615762v1\n",
      "DOI: 10.1101/2024.09.30.615762\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615762v1.full.pdf+html\n",
      "Résumé:We introduce PlasmidGPT, a generative language model pretrained on 153k engineered plasmid sequences from Addgene. PlasmidGPT generates de novo sequences that share similar characteristics with engineered plasmids but show low sequence identity to the training data. We demonstrate its ability to generate plasmids in a controlled manner based on the input sequence or specific design constraint. Moreover, our model learns informative embeddings of both engineered and natural plasmids, allowing for efficient prediction of a wide range of sequence-related attributes.\n",
      "\n",
      "Article ajouté : Non-detection during excursions by citizen scientists modeled as a function of weather, season, list length, and individual preferences\n",
      "Titre: Non-detection during excursions by citizen scientists modeled as a function of weather, season, list length, and individual preferences\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615418v1\n",
      "DOI: 10.1101/2024.09.30.615418\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615418v1.full.pdf+html\n",
      "Résumé:INTRODUCTION: Citizen science is an increasingly valuable source of information about biodiversity. It is challenging to use this information for analysis of distribution and trends. The lack of a protocol leads to bias in observations and therefore data are not representative. The bias is a consequence of unequal detection probabilities, caused by different preferences and habits of citizen scientists. METHODS: We propose to incorporate characteristics of these excursions in analyses of data collected by citizen scientists to improve estimates of the probability that a species is not detected and reported, even though it does occur. By limiting these models to areas that are known to be occupied, detection can be modeled separately without considering variation in occupancy. We apply this idea to 150 common species in the Southwest Delta of The Netherlands, and illustrate the data selection, the modeling process and the results using four species. RESULTS: The strongest features to predict detection are the number of species during a visit (list length), earlier observations of the target species by the same observer, and the day of year. We compare three approaches to predict the total non-detection probability that takes all visits to an area into account. Predictions based on only the number of visits were outperformed by predictions that also take the list length into account. Our predictions based on all features combined consistently beat both other approaches, across all 10 species groups that were compared. DISCUSSION: We thus show that explicitly modelling the characteristics of all visits to an occupied area results in estimation of non-detection probabilities, while providing insight into the causes of detection and reporting bias. Furthermore, predictions of our model provide a basis for quantifying the sampling effort in each area, which is a promising first step to correct bias in citizen science data when aiming to map a species′ distribution.\n",
      "\n",
      "Article ajouté : Unlocking the Potential of Low Quality Total RNA-seq Data: A Stepwise Mapping Approach for Improved Quantitative Analyses\n",
      "Titre: Unlocking the Potential of Low Quality Total RNA-seq Data: A Stepwise Mapping Approach for Improved Quantitative Analyses\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615750v1\n",
      "DOI: 10.1101/2024.09.30.615750\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615750v1.full.pdf+html\n",
      "Résumé:High-throughput sequencing assays face persistent challenges when analyzing low-quality RNAs, often assessed by the RNA integrity number (RIN). Current preprocessing methods and pipelines designed for mRNA-seq presume high-quality RNAs, overlooking the nuanced complexities arising from degraded transcripts in low-quality samples. This study questions the applicability of standard analysis pipelines, especially when sequencing low-quality total RNAs, which are sometimes the sole recourse for specific inquiries. To address this, we conducted a comprehensive investigation using large sequencing reads obtained from blood biospecimens with varying RINs. Introducing a novel mapping approach, termed 9stepwise mapping9, our systematic comparative analyses propose an optimal practice for total RNA-seq data analysis. Contrary to conventional mapping procedures, the 9stepwise mapping9 approach unveils additional transcriptome information, crucial for stable differential expression analysis, even with total RNA-seq data from specimens with relatively low RINs. Our method proves particularly valuable when analyzing limited specimens with low RNA quality.\n",
      "\n",
      "Article ajouté : STIX: Long-reads based Accurate Structural Variation Annotation at Population Scale\n",
      "Titre: STIX: Long-reads based Accurate Structural Variation Annotation at Population Scale\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615931v1\n",
      "DOI: 10.1101/2024.09.30.615931\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615931v1.full.pdf+html\n",
      "Résumé:The prioritization of Structural Variants (SV), which is needed to rank and identify potential pathogenic alleles, is still in its infancy. This is exemplified over gnomAD only being able to annotate 33.5% of GIAB SVs. To overcome this, we present the first long-read based annotation resource for both GRCh38 and CHM13-T2T reference using STIX. In contrast to previous methods, STIX indexes SV-informative long-reads themselves, can thus be easily extended and accurately annotate all SV types including insertions. STIX successfully annotated 95.9% of GIAB Tier1 SVs. STIX further improved cancer based SV prioritization by highlighting 3,563 SV from COSMIC being common in the population. We further showcase that mosaic SV can be independently gained and may be widely spread throughout the population. This highlights the need for accurate SV population frequency annotation to further facilitate the adoption of SV via long-read sequencing in medical research and clinical applications.\n",
      "\n",
      "Scraping page 3 sur 3\n",
      "Nombre total d'articles trouvés sur la page 3 : 10\n",
      "Article ajouté : A patient-centric knowledge graph approach to prioritize mutants for selective anti-cancer targeting\n",
      "Titre: A patient-centric knowledge graph approach to prioritize mutants for selective anti-cancer targeting\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615658v1\n",
      "DOI: 10.1101/2024.09.29.615658\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615658v1.full.pdf+html\n",
      "Résumé:Personalized oncology has revolutionized cancer treatment by targeting specific genetic aberrations in tumors. However, the identification of suitable targets for anti-cancer therapies remains a challenge. In this study, we introduce a knowledge graph approach to prioritize cancer mutations with clinical, functional, and structural significance as potential therapeutic targets. Focusing on the human kinome, we integrate protein-protein interaction and patient-centric mutation networks to construct a comprehensive network enriched with transcriptomic, structural, and drug response data, together covering five layers of information. Moreover, we make the constructed knowledge graph publicly available, along with a plethora of scripts to facilitate further annotation and expansion of the network. Interactive visualization resources are also provided, ensuring accessibility for researchers regardless of computational expertise and enabling detailed analysis by cancer type and individual layers of information. This comprehensive resource has the potential to identify relevant mutations for targeted therapeutic interventions, thereby advancing personalized oncology and improving patient outcomes.\n",
      "\n",
      "Article ajouté : DeepPWM-BindingNet: Unleashing Binding Prediction with Combined Sequence and PWM Features\n",
      "Titre: DeepPWM-BindingNet: Unleashing Binding Prediction with Combined Sequence and PWM Features\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.08.21.609040v2\n",
      "DOI: 10.1101/2024.08.21.609040\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.08.21.609040v2.full.pdf+html\n",
      "Résumé:A crucial challenge in molecular biology is the prediction of DNA-protein binding interactions, which has applications in the study of gene regulation and genome functionality. In this paper, we present a novel deep-learning framework to predict DNA-protein binding interactions with increased precision and interoperability. Our proposed framework DeepPWM-BindingNet leverages the rich information encoded in Position Weight Matrices (PWMs), which capture the sequence-specific binding preferences of proteins. These PWM-derived features are seamlessly integrated into a hybrid model of convolutional recurrent neural networks (CRNNs) that extracts hierarchical features from DNA sequences and protein structures. The sequential dependencies within the sequences are captured by recurrent layers. By incorporating PWM-derived features, the model9s interpretability is improved, enabling researchers to learn more about the underlying binding mechanisms. The model9s capacity to locate crucial binding sites is improved by the incorporation of an attention mechanism that highlights crucial regions. Experiments on diverse DNA-protein interaction datasets demonstrate the proposed approach improves the predictive performance. The proposed model holds significant potential in deciphering intricate DNA-protein interactions, ultimately advancing our comprehension of gene regulation mechanisms.\n",
      "\n",
      "Article ajouté : A new framework for evaluating model out-of-distribution for the biochemical domain\n",
      "Titre: A new framework for evaluating model out-of-distribution for the biochemical domain\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.03.14.584508v2\n",
      "DOI: 10.1101/2024.03.14.584508\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.03.14.584508v2.full.pdf+html\n",
      "Résumé:Quantifying model generalization to out-of-distribution data has been a longstanding challenge in machine learning. Addressing this issue is crucial for leveraging machine learning in scientific discovery, where models must generalize to new molecules or materials. Current methods typically split data into train and test sets using various criteria, like temporal, sequence identity, scaffold, or random cross-validation; before evaluating model performance. However, with so many splitting criteria available, existing approaches offer limited guidance on selecting the most appropriate one, and they do not provide mechanisms for incorporating prior knowledge about the target deployment distribution(s). To tackle this problem, we have developed a novel metric, AU-GOOD, which quantifies expected model performance under conditions of increasing dissimilarity between train and test sets, while also accounting for prior knowledge about the target deployment distribution(s), when available. This metric is broadly applicable to biochemical entities, including proteins, small molecules, nucleic acids, or cells; as long as a relevant similarity function is defined for them. Recognizing the wide range of similarity functions used in biochemistry, we propose criteria to guide the selection of the most appropriate metric for partitioning. We also introduce a new partitioning algorithm that generates more challenging test sets, and we propose statistical methods for comparing models based on AU-GOOD. Finally, we demonstrate the insights that can be gained from this framework by applying it to two different use cases: developing predictors for pharmaceutical properties of small molecules, and using protein language models as embeddings to build biophysical property predictors. Source code is available at https://github.com/IBM/Hestia-OOD.\n",
      "\n",
      "Article ajouté : SMARTER-database: a tool to integrate SNP array datasets for sheep and goat breeds\n",
      "Titre: SMARTER-database: a tool to integrate SNP array datasets for sheep and goat breeds\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.01.610681v2\n",
      "DOI: 10.1101/2024.09.01.610681\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.01.610681v2.full.pdf+html\n",
      "Résumé:Underutilized sheep and goat breeds have the ability to adapt to challenging environments due to their genetic composition. Integrating publicly available genomic datasets with new data will facilitate genetic diversity analyses; however, this process is complicated by important data discrepancies, such as outdated assembly versions or different data formats. Here we present the SMARTER-database, a collection of tools and scripts to standardize genomic data and metadata mainly from SNP chips arrays on global small ruminant populations with a focus on reproducibility. SMARTER-database harmonizes genotypes for about 12,000 sheep and 6,000 goats to a uniform coding and assembly version. Users can access the genotype data via FTP and interact with the metadata through a web interface or programmatically using their custom scripts, enabling efficient filtering and selection of samples. These tools will empower researchers to focus on the crucial aspects of adaptation and contribute to livestock sustainability, leveraging the rich dataset provided by the SMARTER-database.\n",
      "\n",
      "Article ajouté : Novel AI-powered computational method using tensor decomposition can discover the common optimal bin sizes when integrating multiple Hi-C datasets\n",
      "Titre: Novel AI-powered computational method using tensor decomposition can discover the common optimal bin sizes when integrating multiple Hi-C datasets\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615651v1\n",
      "DOI: 10.1101/2024.09.29.615651\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615651v1.full.pdf+html\n",
      "Résumé:iscovering the optimal bin sizes (or resolutions) for the integration of multiple Hi-C datasets is difficult, since bin sizes must be common over multiple datasets whereas dependence of quality upon bin sizes can vary from dataset to dataset. Moreover, we are not supposed to seek common structures in the smaller bin sizes than optimal bin sizes if we cannot suppress the appearance of phase transition-like phenomena even after increasing the number of mapped short reads per bins, since it might mean that there are no common structures any more in finer resolutions and individual Hi-C datasets might have to be analyzed separately. Thus, quality assessments of individual datasets have the limited ability to determine the best bin size common for all datasets. In this study, we propose, the first-ever to our knowledge, a novel adoption of tensor decomposition (TD) based unsupervised feature extraction (FE) to choose the optimal bin sizes for the integration of multiple Hi-C datasets. The use of TD-based unsupervised FE can exhibit something like phase-transition phenomena, by which we can automatically estimate the possible smallest bin size (or the highest resolution) empirically without manually setting some threshold value by hand for the integration of multiple Hi-C datasets.\n",
      "\n",
      "Article ajouté : A systematic comparison of computational methods for expression forecasting\n",
      "Titre: A systematic comparison of computational methods for expression forecasting\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.07.28.551039v2\n",
      "DOI: 10.1101/2023.07.28.551039\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.07.28.551039v2.full.pdf+html\n",
      "Résumé:Expression forecasting methods use machine learning models to predict how a cell will alter its transcriptome upon perturbation. Such methods are enticing because they promise to answer pressing questions in fields ranging from developmental genetics to cell fate engineering and because they are a fast, cheap, and accessible complement to the corresponding experiments. However, the absolute and relative accuracy of these methods is poorly characterized, limiting their informed use, their improvement, and the interpretation of their predictions. To address these issues, we created a benchmarking platform that combines a panel of 11 large-scale perturbation datasets with an expression forecasting software engine that encompasses or interfaces to a wide variety of methods. We used our platform to systematically assess methods, parameters, and sources of auxiliary data, finding that performance strongly depends on the choice of metric, and especially for simple metrics like mean squared error, it is uncommon for expression forecasting methods to out-perform simple baselines. Our platform will serve as a resource to improve methods and to identify contexts in which expression forecasting can succeed.\n",
      "\n",
      "Article ajouté : Quantifying Cancer Drug Targetability across Biochemical Pathways\n",
      "Titre: Quantifying Cancer Drug Targetability across Biochemical Pathways\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.615497v1\n",
      "DOI: 10.1101/2024.09.27.615497\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.615497v1.full.pdf+html\n",
      "Résumé:The evaluation of cancer drug treatments within the complex context of biological networks and pathways is challenging. Drugs can be \"promiscuous,\" targeting multiple genes or pathways, not just the ones of interest. This complexity makes it difficult to assess the full impact of a drug solely based on individual knowledgebases. We make the impact more comprehensible through the characterization of drug targetability in terms of a combination of biochemical pathway data from the Reactome Knowledgebase and drug-target interaction data from the Cancer Targetome. This leads to a graphical representation of drug targetability. We further improve comprehensibility by developing metrics to quantify the graph structures in order to evaluate drug target specificity and navigate the hierarchical nature of biological networks and identify candidate treatment options to increase specificity and minimize off-target effects. We have developed software that calculates these metrics for drugs in the Cancer Targetome. The software is available for research purposes only at https://github.com/sehagler/drug_targetability.\n",
      "\n",
      "Article ajouté : Deep learning-based predictions of gene perturbation effects do not yet outperform simple linear methods\n",
      "Titre: Deep learning-based predictions of gene perturbation effects do not yet outperform simple linear methods\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.16.613342v3\n",
      "DOI: 10.1101/2024.09.16.613342\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.16.613342v3.full.pdf+html\n",
      "Résumé:Advanced deep-learning methods, such as transformer-based foundation models, promise to learn representations of biology that can be employed to predict in silico the outcome of unseen experiments, such as the effect of genetic perturbations on the transcriptomes of human cells. To see whether current models already reach this goal, we benchmarked two state-of-the-art foundation models and one popular graph-based deep learning framework against deliberately simplistic linear models in two important use cases: For combinatorial perturbations of two genes for which only data for the individual single perturbations have been seen, we find that a simple additive model outperformed the deep learning-based approaches. Also, for perturbations of genes that have not yet been seen, but which may be \"interpolated\" from biological similarity or network context, a simple linear model performed as good as the deep learning-based approaches. While the promise of deep neural networks for the representation of biological systems and prediction of experimental outcomes is plausible, our work highlights the need for critical benchmarking to direct research efforts that aim to bring transfer learning to biology.\n",
      "\n",
      "Article ajouté : An Efficient Computational Method to Create Positive NIPT Samples with Autosomal Trisomy\n",
      "Titre: An Efficient Computational Method to Create Positive NIPT Samples with Autosomal Trisomy\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.11.24.568620v4\n",
      "DOI: 10.1101/2023.11.24.568620\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.11.24.568620v4.full.pdf+html\n",
      "Résumé:Noninvasive prenatal test (NIPT) has been widely used for screening trisomy on chromosomes 13 (T13), 18 (T18), and 21 (T21). However, the false negative rate of NIPT algorithms has not been thoroughly evaluated due to the lack of positive samples. In this study, we present an efficient computational approach to create positive samples with autosomal trisomy from negative samples. We applied the approach to establish a low coverage dataset of 1440 positive samples with T13, T18, and T21 aberrations for both mosaic and non-mosaic conditions. We examined the performance of WisecondorX and its improvement, called VINIPT, on both negative and positive datasets. Experiments showed that WisecondorX and VINIPT with a z-score threshold of 3.3 were able to detect all non-mosaic samples with T13, T18, and T21 aberrations (i.e., the sensitivity of 100%). Using a lower z-score threshold of 2.58 when analyzing mosaic samples, both WisecondorX and VINIPT have the overall sensitivity of 99.7% on detecting T13, T18, and T21 aberrations from mosaic samples. WisecondorX has the specificity of 98.5% for the non-mosaic analysis, but a considerably lower specificity of 95% for the mosaic analysis. VINIPT has a much better specificity than WisecondorX, i.e., 99.9% for the non-mosaic analysis, and 98.2% for the mosaic analysis.  The results suggest that VINIPT can play as a powerful NIPT tool for the low coverage data.\n",
      "\n",
      "Article ajouté : Impact of Interval Censoring on Data Accuracy and Machine Learning Performance in Biological High-Throughput Screening\n",
      "Titre: Impact of Interval Censoring on Data Accuracy and Machine Learning Performance in Biological High-Throughput Screening\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.25.615059v2\n",
      "DOI: 10.1101/2024.09.25.615059\n",
      "Date de publication: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.25.615059v2.full.pdf+html\n",
      "Résumé:High-throughput screening (HTS) combined with deep mutational scanning (DMS) and next-generation DNA sequencing (NGS) have great potential to accelerate discovery and optimization of biological therapeutics. Typical workflows involve generation of a mutagenized variant library, screening/selection of variants based on phenotypic fitness, and comprehensive analysis of binned variant populations by NGS. However, in such cases, the HTS data are subject to interval censoring, where each fitness value is calculated based on the assignment of variants to bins. Such censoring leads to increased uncertainty, which can impact data accuracy and, consequently, the performance of machine learning (ML) algorithms tasked with predicting sequence-fitness pairings. Here, we investigated the impact of interval censoring on data quality and ML performance in biological HTS experiments. We theoretically analyzed the impact of data censoring and propose a dimensionless number, theRatio of Discretization(RD), to assist in optimizing HTS parameters such as the bin width and the sampling size. This approach can be used to minimize errors in fitness prediction by ML and to improve the reliability of these methods. These findings are not limited to biological HTS techniques and can be applied to other systems where interval censoring is an advantageous measurement strategy.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "## Nombre de pages à scraper (exemple : 10 pages à la fois)\n",
    "from pickle import TRUE\n",
    "\n",
    "\n",
    "total_pages = 3\n",
    "\n",
    "# boucle qui parcour chaque page de la collection Bioinformatics\n",
    "for page in range(1, total_pages + 1):\n",
    "    print(f\"Scraping page {page} sur {total_pages}\")\n",
    "\n",
    "    # requete qui recup la page bioinfo actuelle\n",
    "    response = requests.get(f'{url}/collection/bioinformatics?page={page}') #? mean numero de la {page}\n",
    "    soup = BeautifulSoup(response.content, 'html.parser')\n",
    "\n",
    "    # Trouver les liens vers les articles sur cette page\n",
    "    articles = soup.find_all('a', class_='highwire-cite-linked-title')  # find liens hypertextes avec classe CSS\n",
    "\n",
    "    print(f'Nombre total d\\'articles trouvés sur la page {page} : {len(articles)}')\n",
    "\n",
    "    # Parcourir tous les articles de la page et enregistrer leurs titres et liens\n",
    "    for article in articles:\n",
    "        title = article.text.strip()  # extraire le titre dans la balise <a>, accessible via .text.\n",
    "        link = url + article['href']  # extraire le lien , href=destination du lien de mon article\n",
    "\n",
    "        # Requête pour accéder à la page de l'article\n",
    "        article_response = requests.get(link)\n",
    "        article_soup = BeautifulSoup(article_response.content, 'html.parser')\n",
    "\n",
    "        # extraire le DOI (dans une balise meta ou dans l'URL)\n",
    "        doi_tag = article_soup.find('meta', {'name': 'citation_doi'})  # balise <meta> qui a l’attribut name=\"citation_doi\".\n",
    "        doi = doi_tag['content'] if doi_tag else 'DOI non disponible'  # si balise existe -> attribut content contient la valeur doi\n",
    "\n",
    "        # extraire la date de publication (dans une balise meta ou time)et la formater au format ISO 8601\n",
    "        date_tag = article_soup.find('meta', {'name': 'citation_publication_date'})\n",
    "        date = date_tag['content'] if date_tag else 'Date non disponible'\n",
    "        # Remplacer les barres obliques par des tirets avant la conversion\n",
    "        if date != 'Date non disponible':\n",
    "            date = date.replace('/', '-')  # Remplacer '/' par '-'\n",
    "            date = datetime.strptime(date, '%Y-%m-%d').date().isoformat()\n",
    "\n",
    "        # extraire le lien vers le fichier PDF\n",
    "        # Trouver tous les liens et filtrer ceux qui contiennent 'PDF'\n",
    "        pdf_links = article_soup.find_all('a')\n",
    "        pdf_link = 'Lien PDF non disponible'\n",
    "        for pdf in pdf_links:\n",
    "            if pdf and 'PDF' in pdf.text:\n",
    "                pdf_link = url + pdf['href']  # Construire le lien complet\n",
    "                break  # Quitter la boucle une fois le lien trouvé\n",
    "        \n",
    "        \n",
    "         # Extraire l'abstract\n",
    "        abstract_tag = article_soup.find('meta', {'name': 'citation_abstract'})\n",
    "        if abstract_tag:\n",
    "            abstract_html = abstract_tag['content']\n",
    "            clean_abstract = BeautifulSoup(abstract_html, \"html.parser\").get_text(strip=True)\n",
    "        else:\n",
    "            clean_abstract = 'Abstract non disponible'\n",
    "\n",
    "\n",
    "        # tout inserer dans la base de données SQLite\n",
    "        # Vérifier si l'article existe déjà dans la base de données (vérification sur DOI et date)\n",
    "        c.execute(\"SELECT * FROM articles WHERE doi = ? AND date = ?\", (doi, date))\n",
    "        result = c.fetchone() # recupère la 1ere ligne du result donc si mm doi/date trouvé retourne cette ligne en tuple sinon NONE\n",
    "\n",
    "        if result:\n",
    "            print(f\"L'article avec DOI {doi} existe déjà, ignoré.\")\n",
    "        else:\n",
    "            # Insérer l'article dans la base de données\n",
    "            c.execute(\"INSERT INTO articles (title, link, doi, date, pdf_link, abstract) VALUES (?, ?, ?, ?, ?, ?)\", \n",
    "                      (title, link, doi, date, pdf_link, clean_abstract))\n",
    "            print(f\"Article ajouté : {title}\")\n",
    "\n",
    "        # Afficher les informations extraites\n",
    "        print(f'Titre: {title}\\nLien: {link}\\nDOI: {doi}\\nDate de publication: {date}\\nLien PDF: {pdf_link}\\nRésumé:{clean_abstract}\\n')\n",
    "\n",
    "\n",
    "        # pause d'une seconde pour éviter de surcharger le serveur\n",
    "        time.sleep(1)\n",
    "        \n",
    "    # Sauvegarder les modifications dans la base de données après chaque page\n",
    "    database.commit()\n",
    "   \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Affichage de mes articles dans la base de données"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "=== Liste des articles enregistrés ===\n",
      "\n",
      "ID: 1\n",
      "Titre: Enhanced prediction of protein functional identity through the integration of sequence and structural features\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615718v1\n",
      "DOI: 10.1101/2024.09.30.615718\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615718v1.full.pdf+html\n",
      "Résumé: Although over 300 million protein sequences are registered in a reference sequence database, only 0.2% have experimentally determined functions. This suggests that many valuable proteins, potentially catalyzing novel enzymatic reactions, remain undiscovered among the vast number of function-unknown proteins. In this study, we developed a method to predict whether two proteins catalyze the same enzymatic reaction by analyzing sequence and structural similarities, utilizing structural models predicted by AlphaFold2. We performed pocket detection and domain decomposition for each structural model. The similarity between protein pairs was assessed using features such as full-length sequence similarity, domain structural similarity, and pocket similarity. We developed several models using conventional machine learning algorithms and found that the LightGBM-based model outperformed the models. Our method also surpassed existing approaches, including those based solely on full-length sequence similarity and state-of-the-art deep learning models. Feature importance analysis revealed that domain sequence identity, calculated through structural alignment, had the greatest influence on the prediction. Therefore, our findings demonstrate that integrating sequence and structural information improves the accuracy of protein function prediction.\n",
      "--------------------------------------------------\n",
      "ID: 2\n",
      "Titre: Phage display enables machine learning discovery of cancer antigen specific TCRs\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.06.27.600973v2\n",
      "DOI: 10.1101/2024.06.27.600973\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.06.27.600973v2.full.pdf+html\n",
      "Résumé: T cells targeting epitopes in infectious diseases or cancer play a central role in spontaneous and therapy-induced immune responses. T-cell epitope recognition is mediated by the binding of the T-Cell Receptor (TCR) and TCRs recognizing clinically relevant epitopes are promising for T-cell based therapies. Starting from one of the few known TCRs targeting the cancer-testis antigen NY-ESO-1 157-165 epitope, we built large phage display libraries of TCRs with randomized Complementary Determining Region 3 of the β chain. The TCR libraries were panned against the NY-ESO-1 epitope, which enabled us to collect thousands of epitope-specific TCR sequences. We then trained a machine learning TCR-epitope interaction predictor with this data and could identify several epitope-specific TCRs directly from TCR repertoires. Cellular binding and functional assays revealed that the predicted TCRs displayed activity towards the NY-ESO-1 epitope and no detectable cross-reactivity with self-peptides. Overall, our work demonstrates how display technologies combined with machine learning models of TCR-epitope recognition can effectively leverage large TCR repertoires for TCR discovery.\n",
      "--------------------------------------------------\n",
      "ID: 3\n",
      "Titre: AptaDiff: de novo design and optimization of aptamers based on diffusion models\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.11.25.568693v2\n",
      "DOI: 10.1101/2023.11.25.568693\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.11.25.568693v2.full.pdf+html\n",
      "Résumé: Aptamers are single-stranded nucleic acid ligands, featuring high affinity and specificity to target molecules. Traditionally they are identified from large DNA/RNA libraries using in vitro methods, like Systematic Evolution of Ligands by Exponential Enrichment (SELEX). However, these libraries capture only a small fraction of theoretical sequence space, and various aptamer candidates are constrained by actual sequencing capabilities from the experiment. Addressing this, we proposed AptaDiff, the first in silico aptamer design and optimization method based on the diffusion model. Our Aptadiff can generate aptamers beyond the constraints of high-throughput sequencing data, leveraging motif-dependent latent embeddings from variational autoencoder, and can optimize aptamers by affinity-guided aptamer generation according to Bayesian optimization. Comparative evaluations revealed AptaDiff9s superiority over existing aptamer generation methods in terms of quality and fidelity across four high-throughput screening data targeting distinct proteins. Moreover, Surface Plasmon Resonance (SPR) experiments were conducted to validate the binding affinity of aptamers generated through Bayesian optimization for two target proteins. The results unveiled a significant boost of 87.9% and 60.2% in RU values, along with a 3.6-fold and 2.4-fold decrease in KD values for the respective target proteins. Notably, the optimized aptamers demonstrated superior binding affinity compared to top experimental candidates selected through SELEX, underscoring the promising outcomes of our AptaDiff in accelerating the discovery of superior aptamers.\n",
      "--------------------------------------------------\n",
      "ID: 4\n",
      "Titre: Gut microbiota and type 2 diabetes associations: A meta-analysis of 16s studies and their methodological challenges\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.10.01.615711v1\n",
      "DOI: 10.1101/2024.10.01.615711\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.10.01.615711v1.full.pdf+html\n",
      "Résumé: Diabetes mellitus is a prevalent chronic non-communicable disease, and recent studies have explored the link between gut microbiota and its development. Despite some evidence suggesting an association, the influence of gut microbiota on type 2 diabetes (DM2) remains unclear. A systematic search of PubMed (Janeiro 2016 to 10 December 2023) using the keywords: (16S and diabetes or DM2) and (gut microbiota and diabetes or DM2). The studies included compared gut microbiome diversity between diabetic and non-diabetic adults using 16S rRNA sequencing, excluding children, interventions, and type 1 diabetes. Alpha diversity indices and bacterial mean abundance were analyzed, with statistical assessments using a random-effects model and I2 for heterogeneity. Thirteen studies met the criteria, with the Shannon index being the most commonly used measure. Results showed significant heterogeneity (I2 > 75%) and no notable differences between diabetic and non-diabetic groups. Other indices, such as Chao1 and phylogenetic whole tree, similarly showed no consistent differences. Taxonomic analysis also failed to find phyla consistently correlated with DM2, with variability across studies. The relationship between gut microbiota and diabetes remains uncertain due to technical and biological factors that are often overlooked. The inconsistencies across studies highlight the low reproducibility common in microbiota research.\n",
      "--------------------------------------------------\n",
      "ID: 5\n",
      "Titre: scGNN+: Adapting ChatGPT for Seamless Tutorial and Code Optimization\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615735v1\n",
      "DOI: 10.1101/2024.09.30.615735\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615735v1.full.pdf+html\n",
      "Résumé: Foundation models have transformed AI by leveraging large-scale data to efficiently perform diverse tasks, and their applications in bioinformatics are primarily focused on data-centric tasks like cell type annotation and gene expression analysis. However, their potential extends beyond data analysis, offering significant opportunities in software development and optimization, such as code refinement, tutorial generation, and advanced visualization. For example, models like OpenAI Codex can suggest optimized code snippets and generate well-documented, reproducible workflows, enhancing accessibility and reducing computational complexity. Despite these advantages, the use of foundation models for improving computational tool engineering in single-cell research remains underutilized. To address this gap, we developed scGNN+, a web-based platform that combines the power of graph neural networks with the capabilities of ChatGPT to enhance reproducibility, code optimization, and visualization. scGNN+ further simplifies the process for users by generating standardized, well-annotated code, making complex procedures more accessible to non-programmers. Additionally, ChatGPT integration allows users to create high-quality, customizable visualizations through natural language prompts, improving data interpretation and presentation. Ultimately, scGNN+ offers a user-friendly, reproducible, and optimized solution for single-cell research, leveraging the full potential of foundation models in bioinformatics. scGNN+ is publicly available at https://bmblx.bmi.osumc.edu/scgnn+.\n",
      "--------------------------------------------------\n",
      "ID: 6\n",
      "Titre: Isobaric labeling update in MaxQuant\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615788v1\n",
      "DOI: 10.1101/2024.09.30.615788\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615788v1.full.pdf+html\n",
      "Résumé: We present an update of the MaxQuant software for isobaric labeling data and evaluate its performance on benchmark datasets. Impurity correction factors can be applied to labels mixing C- and N-type reporter ions, such as TMT Pro. Application to a single-cell multi-species mixture benchmark shows high accuracy of the impurity-corrected results. TMT data recorded with FAIMS separation can be analyzed directly in MaxQuant without splitting the raw data into separate files per FAIMS voltage. Weighted median normalization, is applied to several datasets, including large-scale human body atlas data. In the benchmark datasets the weighted median normalization either removes or strongly reduces the batch effects between different TMT plexes and results in clustering by biology. In datasets including a reference channel, we find that weighted median normalization performs as well or better when the reference channel is ignored and only the sample channel intensities are used, suggesting that the measurement of a reference channel is unnecessary when using weighted median normalization in MaxQuant. We demonstrate that MaxQuant including the weighted median normalization performs well on multi-notch MS3 data, as well as on phosphorylation data. MaxQuant is freely available for any purpose and can be downloaded from https://www.maxquant.org/.\n",
      "--------------------------------------------------\n",
      "ID: 7\n",
      "Titre: Preserving Full Spectrum Information in Imaging Mass Spectrometry Data Reduction\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.614425v1\n",
      "DOI: 10.1101/2024.09.30.614425\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.614425v1.full.pdf+html\n",
      "Résumé: Imaging Mass Spectrometry (IMS) has become an important tool for molecular characterization of biological tissue. However, IMS experiments tend to yield large datasets, routinely recording over 200,000 ion intensity values per mass spectrum and more than 100,000 pixels, i.e., spectra, per dataset. Traditionally, IMS data size challenges have been addressed by feature selection or extraction, such as by peak picking and peak integration. Selective data reduction techniques such as peak picking only retain certain parts of a mass spectrum, and often these describe only medium-to-high-abundance species. Since lower-intensity peaks and, for example, near-isobar species are sometimes missed, selective methods can potentially bias downstream analysis towards a subset of species in the data rather than considering all species measured. Results: We present an alternative to selective data reduction of IMS data that achieves similar data size reduction while better conserving the ion intensity profiles across all recorded m/z-bins, thereby preserving full spectrum information. Our method utilizes a low-rank matrix completion model combined with a randomized sparse-format-aware algorithm to approximate IMS datasets. This representation offers reduced dimensionality and a data footprint comparable to peak picking, but also retains complete spectral profiles, enabling comprehensive analysis and compression. We demonstrate improved preservation of lower signal-to-noise-ratio signals and near-isobars, mitigation of selection bias, and reduced information loss compared to current state-of-the art data reduction methods in IMS.\n",
      "--------------------------------------------------\n",
      "ID: 8\n",
      "Titre: BindCraft: one-shot design of functional protein binders\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615802v1\n",
      "DOI: 10.1101/2024.09.30.615802\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615802v1.full.pdf+html\n",
      "Résumé: Protein-protein interactions (PPIs) are at the core of all key biological processes. However, the complexity of the structural features that determine PPIs makes their design challenging. We present BindCraft, an open-source and automated pipeline for de novo protein binder design with experimental success rates of 10-100%. BindCraft leverages the trained deep learning weights of AlphaFold2 to generate nanomolar binders without the need for high-throughput screening or experimental optimization, even in the absence of known binding sites. We successfully designed binders against a diverse set of challenging targets, including cell-surface receptors, common allergens, de novo designed proteins, and multi-domain nucleases, such as CRISPR-Cas9. We showcase their functional and therapeutic potential by demonstrating that designed binders can reduce IgE binding to birch allergen in patient-derived samples. This work represents a significant advancement towards a \"one design-one binder\" approach in computational design, with immense potential in therapeutics, diagnostics, and biotechnology.\n",
      "--------------------------------------------------\n",
      "ID: 9\n",
      "Titre: UniGEN-DDI: Computing drug-drug interactions using a unified graph embedding network\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615812v1\n",
      "DOI: 10.1101/2024.09.30.615812\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615812v1.full.pdf+html\n",
      "Résumé: Finding drug-drug interaction is crucial for patient safety and treatment efficacy. Two drugs may show a synergistic effect but may sometimes cause a severe health issue, including lethality. Wet lab studies are often performed to understand such interactions but are limited by cost and time. However, the biochemical data generated can be explored to compute unknown interactions. Here, we developed a computational model named UniGEN-DDI (Unified Graph Embedding Network for Drug-Drug Interaction) for the estimation of interactions between drugs. It is a simple unified network model containing the biochemical information of drug association developed using the compiled data from DrugBank 5.1.0. The feature learning of the drugs was carried out using a combination of GraphSAGE and Node2Vec algorithms, which were found efficient in extracting diverse features. The simple architecture of our model led to a significant reduction in computational time compared to the baselines, while maintaining a high prediction accuracy. The model performed well on the data, which was equally distributed between interacting and non-interacting drugs. As a more challenging evaluation, we performed non-overlapping splitting of the data based on the drug action on different parts of the body, and our model performed well in both interaction estimation and time efficiency. Our model successfully identified the 12 unknown drug interactions in DrugBank 5.1.0, which were updated in DrugBank 6.0.\n",
      "--------------------------------------------------\n",
      "ID: 10\n",
      "Titre: Divergence of the Individual repeats in the leucine-rich repeat domains of human Toll-like receptors explain their diversity and functional adaptations.\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615863v1\n",
      "DOI: 10.1101/2024.09.30.615863\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615863v1.full.pdf+html\n",
      "Résumé: The receptor domains of Toll-like receptors (TLRs) are characterized by a solenoid-like structure composed of tandem repeats of α/β units known as Leucine Rich Repeats (LRRs). LRR proteins form large paralogous families, with nearly 400 in the human genome alone, all sharing similar semi-regular solenoid-like structures. Despite this structural similarity, they exhibit remarkable diversity in binding specificity. For TLR receptors, this includes a range of pathogen-associated molecular patterns (PAMPs), while other LRR proteins bind an extensive array of ligands, including proteins, DNA, RNA, and small molecules. The LRR domains contain repeats that have similar, yet not identical, 3D structures and patterns of conserved residues. Through in-depth analysis of sequence and structural conservation in individual repeats of human TLRs, we demonstrate that even subtle variations between these repeats alter the local solenoid structure, leading to significant functional changes. Variations in repeat length and defining patterns result in local changes in curvature and the emergence of structural features such as loops, cavities, or specific interaction interfaces. Understanding how divergence in LRR repeats influences their functional roles can provide deeper insights into their binding mechanisms, including interactions with unknown ligands, both in humans and across the diverse phylogenetic spectrum of animals that rely on their TLR repertoires for immune activation.\n",
      "--------------------------------------------------\n",
      "ID: 11\n",
      "Titre: RAPDOR: Using Jensen-Shannon Distance for the computational analysis of complex proteomics datasets\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615781v1\n",
      "DOI: 10.1101/2024.09.30.615781\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615781v1.full.pdf+html\n",
      "Résumé: The computational analysis of large proteomics datasets, such as those from gradient profiling or spatially resolved proteomics, is often as crucial as the experimental design. We present RAPDOR, a tool for intuitive analyzing and visualizing such datasets, based on the Jensen-Shannon distance and subsequent analysis of similarities between replicates, applied to three datasets. First, we examined the in-gradient distribution profiles of protein complexes with or without RNase treatment (GradR) to identify the set of RNA-binding proteins (RBPs) in the cyanobacterium Synechocystis sp. PCC 6803. RBPs play pivotal regulatory and structural roles; although numerous RBPs have been identified, the complete set is unknown for any species. RAPDOR identified 80 potential RBPs, including ribosomal proteins, likely RNA-modifying enzymes, and several proteins not previously associated with RNA binding. High-ranking putative RBPs, such as the universal stress protein Sll1388, or the translation inhibitor LrtA/RaiA, were predicted by RAPDOR but not the TriPepSVM algorithm, indicating uncharacterized RBP domains. These data are available online at https://synecho-rapdor.biologie.uni-freiburg.de, providing a comprehensive resource for RNase-sensitive protein complexes in cyanobacteria. We then show by reanalyzing existing datasets, that RAPDOR is effective in examining the intracellular redistribution of proteins under stress conditions. RAPDOR is a generic, non-parametric tool for the intuitive and versatile analysis of highly complex data sets such as the study of protein distributions using fractionation protocols.\n",
      "--------------------------------------------------\n",
      "ID: 12\n",
      "Titre: SAMURAI: Shallow Analysis of Copy nuMber alterations Using a Reproducible And Integrated bioinformatics pipeline\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615766v1\n",
      "DOI: 10.1101/2024.09.30.615766\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615766v1.full.pdf+html\n",
      "Résumé: Shallow whole-genome sequencing (sWGS) offers a cost-effective approach to detect  copy number alterations (CNAs). However, there remains a gap for a standardized workflow specifically designed for sWGS analysis. To address this need, in this work we present SAMURAI  a bioinformatics pipeline specifically designed for analyzing CNAs from sWGS data in a standardized and reproducible manner. SAMURAI is built using established community standards, ensuring portability, scalability, and reproducibility. The pipeline features a modular design with independent blocks for data pre-processing, copy number analysis, and customized reporting. Users can select workflows tailored for either solid or liquid biopsy analysis (e.g., circulating tumor DNA), with specific tools integrated for each sample type. The final report generated by SAMURAI provides detailed results to facilitate data interpretation and potential downstream analyses.To demonstrate its robustness, SAMURAI was validated using simulated and real-world datasets. The pipeline achieved high concordance with ground truth data and maintained consistent performance across various scenarios. By promoting standardization and offering a versatile workflow, SAMURAI empowers researchers in diverse environments to reliably analyze CNAs from sWGS data. This, in turn, holds promise for advancements in precision medicine.\n",
      "--------------------------------------------------\n",
      "ID: 13\n",
      "Titre: Capturing the Hierarchically Assorted Protein-protein Interaction Modules of Mammalian Cell\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615776v1\n",
      "DOI: 10.1101/2024.09.30.615776\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615776v1.full.pdf+html\n",
      "Résumé: Proteins are organized into modules by both functions and physical interactions within compartments of an eukaryotic cell. The in vivo chemical crosslinking mass spectrometry (XL-MS) data collected from organelles, the whole cells and tissues are able to provide unique information about both protein-protein interaction (PPI) and the intensity of PPI. In the present study, we have retrieved 55,982 crosslinked peptides (XL-peptides) from the XL-MS databases, out of which 6,356 in vivo PPIs were identified. Introduction of the MONET software into analysis of 4,526 hetero PPIs revealed a total of 402 protein modules, including 15, 58 and 163 stable protein complex(s), condensate-forming protein module(s) and intrinsically disordered region (IDR)-containing protein module(s), respectively. The application of ChatGPT in analysis of these modules unexpectedly identified 4 vesicle-related modules. Together, these modules were assorted into 6 communities (module of modules) and 3 systems (module of communities) hierarchically. The bioinformatic analysis found that the three systems are corresponding to three compartments of eukaryotic cell: nuclei, mitochondria, endoplasmic reticulum (ER), respectively. This study presents a novel and comprehensive biomolecular modulome of a mammalian cell, which captures putative protein compositions of protein complexes, protein condensates and vesicles and provides a hierarchical protein organization and function within compartments of mammalian cell.\n",
      "--------------------------------------------------\n",
      "ID: 14\n",
      "Titre: Separating biological variance from noise by applying EM algorithm to modified General Linear Model\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615661v1\n",
      "DOI: 10.1101/2024.09.29.615661\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615661v1.full.pdf+html\n",
      "Résumé: Introduction: The General Linear Model (GLM) has been widely used in research, where error term has been treated as noise. However, compelling evidence suggests that in biological systems, the target variables may possess their innate variances. \n",
      "Methods: A modified GLM was proposed to explicitly model biological variance and non-biological noise. Employing the Expectation and Maximization (EM) scheme can distinguish biological variance from noise, termed EMSEV (EM for Separating Variances). The performance of EMSEV was evaluated by varying noise levels, dimensions of the design matrix, and covariance structures of the target variables.\n",
      "Results: The deviation between EMSEV outputs and the pre-defined distribution parameters increased with noise level. With a proper initial guess, when the noise magnitude and the variance of the target variables were similar, there were deviations of 3% and 10 to 16% in the estimated mean and covariance of the target variables, respectively, along with a 1.7% deviation in noise estimation.\n",
      "Conclusion: EMSEV appears promising for distinguishing signal variance from noise in biological systems. The potential applications and implications in biological science and statistical inference are discussed.\n",
      "--------------------------------------------------\n",
      "ID: 15\n",
      "Titre: Advancing Protein-DNA Binding Site Prediction: Integrating Sequence Models and Machine Learning Classifiers\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.08.23.554389v2\n",
      "DOI: 10.1101/2023.08.23.554389\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.08.23.554389v2.full.pdf+html\n",
      "Résumé: Predicting protein-DNA binding sites is a challenging computational problem that has led to the development of advanced algorithms and techniques in the field of bioinformatics. Identifying the specific residues where proteins bind to DNA is of paramount importance, as it enables the modeling of their interactions and facilitates downstream studies. Nevertheless, the development of accurate and efficient computational methods for this task remains a persistent challenge. Accurate prediction of protein-DNA binding sites has far-reaching implications for understanding molecular mechanisms, disease processes, drug discovery, and synthetic biology applications. It helps bridge the gap between genomics and functional biology, enabling researchers to uncover the intricacies of cellular processes and advance our knowledge of the biological world. The method used to predict DNA binding residues in this study is a potent combination of conventional bioinformatics tools, protein language models, and cutting-edge machine learning and deep learning classifiers. On a dataset of protein-DNA binding sites, our model is meticulously trained, and it is then rigorously examined using several experiments. As indicated by higher predictive behavior with AUC values on two benchmark datasets, the results show superior performance when compared to existing models. The suggested model has a strong capacity for generalization and shows specificity for DNA-binding sites. We further demonstrated the adaptability of our model as a universal framework for binding site prediction by training it on a variety of protein-ligand binding site datasets. In conclusion, our innovative approach for predicting protein-DNA binding residues holds great promise in advancing our understanding of molecular interactions, thus paving the way for several groundbreaking applications in the field of molecular biology and genetics. Our approach demonstrated efficacy and versatility underscore its potential for driving transformative discoveries in biomolecular research.\n",
      "--------------------------------------------------\n",
      "ID: 16\n",
      "Titre: Exploring the link between extended red blood cell parameters and platelet indices in voluntary blood donors\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615743v1\n",
      "DOI: 10.1101/2024.09.29.615743\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615743v1.full.pdf+html\n",
      "Résumé: Background: As regular blood donors are prone to iron deficiency, importance of extended red blood cell (eRBC) parameters in identifying donors with depleted iron stores was investigated.Thrombocytosis has been well documented in patients affected with IDA. Thus, significance of eRBC parameters in identifying iron deficiency associated thrombocytosis was also examined in this cohort. Methods: Blood samples were collected in EDTA tubes from consenting donors for analyses of routine haematological and eRBC parameters. Serum samples were isolated for estimation of iron parameters. Results: Iron deficient donors had significantly altered eRBC parameters. Among them, Ret-He with a cut-off of ≥32 pg had high AUC (0.822) and showed relatively high sensitivity & specificity in detecting iron deficiency. Combination of Ret-He with CCI increased sensitivity & specificity to 90.6% and 98.2%, in detection of donors affected with iron restricted erythropoiesis. This cohort had increased platelet counts, which showed significant association with Ret-He (β= -0.373), RBC-He (β= -0.384), CCI (β= 0.384), Hypo-He (β=0.494) and Micro-R (β= 0.299). Elevated platelet counts also showed significant correlations with these eRBC parameters, which was absent in iron replete donors. Conclusions: eRBC parameters are sensitive indicators of non-anaemic iron deficiency, which may be enhanced by combining them. Their significant association with elevated platelet counts in iron deficient donors, highlights their importance in reflecting iron deficiency associated thrombocytosis.\n",
      "--------------------------------------------------\n",
      "ID: 17\n",
      "Titre: PlasmidGPT: a generative framework for plasmid design and annotation\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615762v1\n",
      "DOI: 10.1101/2024.09.30.615762\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615762v1.full.pdf+html\n",
      "Résumé: We introduce PlasmidGPT, a generative language model pretrained on 153k engineered plasmid sequences from Addgene. PlasmidGPT generates de novo sequences that share similar characteristics with engineered plasmids but show low sequence identity to the training data. We demonstrate its ability to generate plasmids in a controlled manner based on the input sequence or specific design constraint. Moreover, our model learns informative embeddings of both engineered and natural plasmids, allowing for efficient prediction of a wide range of sequence-related attributes.\n",
      "--------------------------------------------------\n",
      "ID: 18\n",
      "Titre: Non-detection during excursions by citizen scientists modeled as a function of weather, season, list length, and individual preferences\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615418v1\n",
      "DOI: 10.1101/2024.09.30.615418\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615418v1.full.pdf+html\n",
      "Résumé: INTRODUCTION: Citizen science is an increasingly valuable source of information about biodiversity. It is challenging to use this information for analysis of distribution and trends. The lack of a protocol leads to bias in observations and therefore data are not representative. The bias is a consequence of unequal detection probabilities, caused by different preferences and habits of citizen scientists. METHODS: We propose to incorporate characteristics of these excursions in analyses of data collected by citizen scientists to improve estimates of the probability that a species is not detected and reported, even though it does occur. By limiting these models to areas that are known to be occupied, detection can be modeled separately without considering variation in occupancy. We apply this idea to 150 common species in the Southwest Delta of The Netherlands, and illustrate the data selection, the modeling process and the results using four species. RESULTS: The strongest features to predict detection are the number of species during a visit (list length), earlier observations of the target species by the same observer, and the day of year. We compare three approaches to predict the total non-detection probability that takes all visits to an area into account. Predictions based on only the number of visits were outperformed by predictions that also take the list length into account. Our predictions based on all features combined consistently beat both other approaches, across all 10 species groups that were compared. DISCUSSION: We thus show that explicitly modelling the characteristics of all visits to an occupied area results in estimation of non-detection probabilities, while providing insight into the causes of detection and reporting bias. Furthermore, predictions of our model provide a basis for quantifying the sampling effort in each area, which is a promising first step to correct bias in citizen science data when aiming to map a species′ distribution.\n",
      "--------------------------------------------------\n",
      "ID: 19\n",
      "Titre: Unlocking the Potential of Low Quality Total RNA-seq Data: A Stepwise Mapping Approach for Improved Quantitative Analyses\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615750v1\n",
      "DOI: 10.1101/2024.09.30.615750\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615750v1.full.pdf+html\n",
      "Résumé: High-throughput sequencing assays face persistent challenges when analyzing low-quality RNAs, often assessed by the RNA integrity number (RIN). Current preprocessing methods and pipelines designed for mRNA-seq presume high-quality RNAs, overlooking the nuanced complexities arising from degraded transcripts in low-quality samples. This study questions the applicability of standard analysis pipelines, especially when sequencing low-quality total RNAs, which are sometimes the sole recourse for specific inquiries. To address this, we conducted a comprehensive investigation using large sequencing reads obtained from blood biospecimens with varying RINs. Introducing a novel mapping approach, termed 9stepwise mapping9, our systematic comparative analyses propose an optimal practice for total RNA-seq data analysis. Contrary to conventional mapping procedures, the 9stepwise mapping9 approach unveils additional transcriptome information, crucial for stable differential expression analysis, even with total RNA-seq data from specimens with relatively low RINs. Our method proves particularly valuable when analyzing limited specimens with low RNA quality.\n",
      "--------------------------------------------------\n",
      "ID: 20\n",
      "Titre: STIX: Long-reads based Accurate Structural Variation Annotation at Population Scale\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.30.615931v1\n",
      "DOI: 10.1101/2024.09.30.615931\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.30.615931v1.full.pdf+html\n",
      "Résumé: The prioritization of Structural Variants (SV), which is needed to rank and identify potential pathogenic alleles, is still in its infancy. This is exemplified over gnomAD only being able to annotate 33.5% of GIAB SVs. To overcome this, we present the first long-read based annotation resource for both GRCh38 and CHM13-T2T reference using STIX. In contrast to previous methods, STIX indexes SV-informative long-reads themselves, can thus be easily extended and accurately annotate all SV types including insertions. STIX successfully annotated 95.9% of GIAB Tier1 SVs. STIX further improved cancer based SV prioritization by highlighting 3,563 SV from COSMIC being common in the population. We further showcase that mosaic SV can be independently gained and may be widely spread throughout the population. This highlights the need for accurate SV population frequency annotation to further facilitate the adoption of SV via long-read sequencing in medical research and clinical applications.\n",
      "--------------------------------------------------\n",
      "ID: 21\n",
      "Titre: A patient-centric knowledge graph approach to prioritize mutants for selective anti-cancer targeting\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615658v1\n",
      "DOI: 10.1101/2024.09.29.615658\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615658v1.full.pdf+html\n",
      "Résumé: Personalized oncology has revolutionized cancer treatment by targeting specific genetic aberrations in tumors. However, the identification of suitable targets for anti-cancer therapies remains a challenge. In this study, we introduce a knowledge graph approach to prioritize cancer mutations with clinical, functional, and structural significance as potential therapeutic targets. Focusing on the human kinome, we integrate protein-protein interaction and patient-centric mutation networks to construct a comprehensive network enriched with transcriptomic, structural, and drug response data, together covering five layers of information. Moreover, we make the constructed knowledge graph publicly available, along with a plethora of scripts to facilitate further annotation and expansion of the network. Interactive visualization resources are also provided, ensuring accessibility for researchers regardless of computational expertise and enabling detailed analysis by cancer type and individual layers of information. This comprehensive resource has the potential to identify relevant mutations for targeted therapeutic interventions, thereby advancing personalized oncology and improving patient outcomes.\n",
      "--------------------------------------------------\n",
      "ID: 22\n",
      "Titre: DeepPWM-BindingNet: Unleashing Binding Prediction with Combined Sequence and PWM Features\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.08.21.609040v2\n",
      "DOI: 10.1101/2024.08.21.609040\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.08.21.609040v2.full.pdf+html\n",
      "Résumé: A crucial challenge in molecular biology is the prediction of DNA-protein binding interactions, which has applications in the study of gene regulation and genome functionality. In this paper, we present a novel deep-learning framework to predict DNA-protein binding interactions with increased precision and interoperability. Our proposed framework DeepPWM-BindingNet leverages the rich information encoded in Position Weight Matrices (PWMs), which capture the sequence-specific binding preferences of proteins. These PWM-derived features are seamlessly integrated into a hybrid model of convolutional recurrent neural networks (CRNNs) that extracts hierarchical features from DNA sequences and protein structures. The sequential dependencies within the sequences are captured by recurrent layers. By incorporating PWM-derived features, the model9s interpretability is improved, enabling researchers to learn more about the underlying binding mechanisms. The model9s capacity to locate crucial binding sites is improved by the incorporation of an attention mechanism that highlights crucial regions. Experiments on diverse DNA-protein interaction datasets demonstrate the proposed approach improves the predictive performance. The proposed model holds significant potential in deciphering intricate DNA-protein interactions, ultimately advancing our comprehension of gene regulation mechanisms.\n",
      "--------------------------------------------------\n",
      "ID: 23\n",
      "Titre: A new framework for evaluating model out-of-distribution for the biochemical domain\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.03.14.584508v2\n",
      "DOI: 10.1101/2024.03.14.584508\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.03.14.584508v2.full.pdf+html\n",
      "Résumé: Quantifying model generalization to out-of-distribution data has been a longstanding challenge in machine learning. Addressing this issue is crucial for leveraging machine learning in scientific discovery, where models must generalize to new molecules or materials. Current methods typically split data into train and test sets using various criteria, like temporal, sequence identity, scaffold, or random cross-validation; before evaluating model performance. However, with so many splitting criteria available, existing approaches offer limited guidance on selecting the most appropriate one, and they do not provide mechanisms for incorporating prior knowledge about the target deployment distribution(s). To tackle this problem, we have developed a novel metric, AU-GOOD, which quantifies expected model performance under conditions of increasing dissimilarity between train and test sets, while also accounting for prior knowledge about the target deployment distribution(s), when available. This metric is broadly applicable to biochemical entities, including proteins, small molecules, nucleic acids, or cells; as long as a relevant similarity function is defined for them. Recognizing the wide range of similarity functions used in biochemistry, we propose criteria to guide the selection of the most appropriate metric for partitioning. We also introduce a new partitioning algorithm that generates more challenging test sets, and we propose statistical methods for comparing models based on AU-GOOD. Finally, we demonstrate the insights that can be gained from this framework by applying it to two different use cases: developing predictors for pharmaceutical properties of small molecules, and using protein language models as embeddings to build biophysical property predictors. Source code is available at https://github.com/IBM/Hestia-OOD.\n",
      "--------------------------------------------------\n",
      "ID: 24\n",
      "Titre: SMARTER-database: a tool to integrate SNP array datasets for sheep and goat breeds\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.01.610681v2\n",
      "DOI: 10.1101/2024.09.01.610681\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.01.610681v2.full.pdf+html\n",
      "Résumé: Underutilized sheep and goat breeds have the ability to adapt to challenging environments due to their genetic composition. Integrating publicly available genomic datasets with new data will facilitate genetic diversity analyses; however, this process is complicated by important data discrepancies, such as outdated assembly versions or different data formats. Here we present the SMARTER-database, a collection of tools and scripts to standardize genomic data and metadata mainly from SNP chips arrays on global small ruminant populations with a focus on reproducibility. SMARTER-database harmonizes genotypes for about 12,000 sheep and 6,000 goats to a uniform coding and assembly version. Users can access the genotype data via FTP and interact with the metadata through a web interface or programmatically using their custom scripts, enabling efficient filtering and selection of samples. These tools will empower researchers to focus on the crucial aspects of adaptation and contribute to livestock sustainability, leveraging the rich dataset provided by the SMARTER-database.\n",
      "--------------------------------------------------\n",
      "ID: 25\n",
      "Titre: Novel AI-powered computational method using tensor decomposition can discover the common optimal bin sizes when integrating multiple Hi-C datasets\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.29.615651v1\n",
      "DOI: 10.1101/2024.09.29.615651\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.29.615651v1.full.pdf+html\n",
      "Résumé: iscovering the optimal bin sizes (or resolutions) for the integration of multiple Hi-C datasets is difficult, since bin sizes must be common over multiple datasets whereas dependence of quality upon bin sizes can vary from dataset to dataset. Moreover, we are not supposed to seek common structures in the smaller bin sizes than optimal bin sizes if we cannot suppress the appearance of phase transition-like phenomena even after increasing the number of mapped short reads per bins, since it might mean that there are no common structures any more in finer resolutions and individual Hi-C datasets might have to be analyzed separately. Thus, quality assessments of individual datasets have the limited ability to determine the best bin size common for all datasets. In this study, we propose, the first-ever to our knowledge, a novel adoption of tensor decomposition (TD) based unsupervised feature extraction (FE) to choose the optimal bin sizes for the integration of multiple Hi-C datasets. The use of TD-based unsupervised FE can exhibit something like phase-transition phenomena, by which we can automatically estimate the possible smallest bin size (or the highest resolution) empirically without manually setting some threshold value by hand for the integration of multiple Hi-C datasets.\n",
      "--------------------------------------------------\n",
      "ID: 26\n",
      "Titre: A systematic comparison of computational methods for expression forecasting\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.07.28.551039v2\n",
      "DOI: 10.1101/2023.07.28.551039\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.07.28.551039v2.full.pdf+html\n",
      "Résumé: Expression forecasting methods use machine learning models to predict how a cell will alter its transcriptome upon perturbation. Such methods are enticing because they promise to answer pressing questions in fields ranging from developmental genetics to cell fate engineering and because they are a fast, cheap, and accessible complement to the corresponding experiments. However, the absolute and relative accuracy of these methods is poorly characterized, limiting their informed use, their improvement, and the interpretation of their predictions. To address these issues, we created a benchmarking platform that combines a panel of 11 large-scale perturbation datasets with an expression forecasting software engine that encompasses or interfaces to a wide variety of methods. We used our platform to systematically assess methods, parameters, and sources of auxiliary data, finding that performance strongly depends on the choice of metric, and especially for simple metrics like mean squared error, it is uncommon for expression forecasting methods to out-perform simple baselines. Our platform will serve as a resource to improve methods and to identify contexts in which expression forecasting can succeed.\n",
      "--------------------------------------------------\n",
      "ID: 27\n",
      "Titre: Quantifying Cancer Drug Targetability across Biochemical Pathways\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.27.615497v1\n",
      "DOI: 10.1101/2024.09.27.615497\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.27.615497v1.full.pdf+html\n",
      "Résumé: The evaluation of cancer drug treatments within the complex context of biological networks and pathways is challenging. Drugs can be \"promiscuous,\" targeting multiple genes or pathways, not just the ones of interest. This complexity makes it difficult to assess the full impact of a drug solely based on individual knowledgebases. We make the impact more comprehensible through the characterization of drug targetability in terms of a combination of biochemical pathway data from the Reactome Knowledgebase and drug-target interaction data from the Cancer Targetome. This leads to a graphical representation of drug targetability. We further improve comprehensibility by developing metrics to quantify the graph structures in order to evaluate drug target specificity and navigate the hierarchical nature of biological networks and identify candidate treatment options to increase specificity and minimize off-target effects. We have developed software that calculates these metrics for drugs in the Cancer Targetome. The software is available for research purposes only at https://github.com/sehagler/drug_targetability.\n",
      "--------------------------------------------------\n",
      "ID: 28\n",
      "Titre: Deep learning-based predictions of gene perturbation effects do not yet outperform simple linear methods\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.16.613342v3\n",
      "DOI: 10.1101/2024.09.16.613342\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.16.613342v3.full.pdf+html\n",
      "Résumé: Advanced deep-learning methods, such as transformer-based foundation models, promise to learn representations of biology that can be employed to predict in silico the outcome of unseen experiments, such as the effect of genetic perturbations on the transcriptomes of human cells. To see whether current models already reach this goal, we benchmarked two state-of-the-art foundation models and one popular graph-based deep learning framework against deliberately simplistic linear models in two important use cases: For combinatorial perturbations of two genes for which only data for the individual single perturbations have been seen, we find that a simple additive model outperformed the deep learning-based approaches. Also, for perturbations of genes that have not yet been seen, but which may be \"interpolated\" from biological similarity or network context, a simple linear model performed as good as the deep learning-based approaches. While the promise of deep neural networks for the representation of biological systems and prediction of experimental outcomes is plausible, our work highlights the need for critical benchmarking to direct research efforts that aim to bring transfer learning to biology.\n",
      "--------------------------------------------------\n",
      "ID: 29\n",
      "Titre: An Efficient Computational Method to Create Positive NIPT Samples with Autosomal Trisomy\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.11.24.568620v4\n",
      "DOI: 10.1101/2023.11.24.568620\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2023.11.24.568620v4.full.pdf+html\n",
      "Résumé: Noninvasive prenatal test (NIPT) has been widely used for screening trisomy on chromosomes 13 (T13), 18 (T18), and 21 (T21). However, the false negative rate of NIPT algorithms has not been thoroughly evaluated due to the lack of positive samples. In this study, we present an efficient computational approach to create positive samples with autosomal trisomy from negative samples. We applied the approach to establish a low coverage dataset of 1440 positive samples with T13, T18, and T21 aberrations for both mosaic and non-mosaic conditions. We examined the performance of WisecondorX and its improvement, called VINIPT, on both negative and positive datasets. Experiments showed that WisecondorX and VINIPT with a z-score threshold of 3.3 were able to detect all non-mosaic samples with T13, T18, and T21 aberrations (i.e., the sensitivity of 100%). Using a lower z-score threshold of 2.58 when analyzing mosaic samples, both WisecondorX and VINIPT have the overall sensitivity of 99.7% on detecting T13, T18, and T21 aberrations from mosaic samples. WisecondorX has the specificity of 98.5% for the non-mosaic analysis, but a considerably lower specificity of 95% for the mosaic analysis. VINIPT has a much better specificity than WisecondorX, i.e., 99.9% for the non-mosaic analysis, and 98.2% for the mosaic analysis.  The results suggest that VINIPT can play as a powerful NIPT tool for the low coverage data.\n",
      "--------------------------------------------------\n",
      "ID: 30\n",
      "Titre: Impact of Interval Censoring on Data Accuracy and Machine Learning Performance in Biological High-Throughput Screening\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.25.615059v2\n",
      "DOI: 10.1101/2024.09.25.615059\n",
      "Date: 2024-01-01\n",
      "Lien PDF: https://www.biorxiv.org/content/10.1101/2024.09.25.615059v2.full.pdf+html\n",
      "Résumé: High-throughput screening (HTS) combined with deep mutational scanning (DMS) and next-generation DNA sequencing (NGS) have great potential to accelerate discovery and optimization of biological therapeutics. Typical workflows involve generation of a mutagenized variant library, screening/selection of variants based on phenotypic fitness, and comprehensive analysis of binned variant populations by NGS. However, in such cases, the HTS data are subject to interval censoring, where each fitness value is calculated based on the assignment of variants to bins. Such censoring leads to increased uncertainty, which can impact data accuracy and, consequently, the performance of machine learning (ML) algorithms tasked with predicting sequence-fitness pairings. Here, we investigated the impact of interval censoring on data quality and ML performance in biological HTS experiments. We theoretically analyzed the impact of data censoring and propose a dimensionless number, theRatio of Discretization(RD), to assist in optimizing HTS parameters such as the bin width and the sampling size. This approach can be used to minimize errors in fitness prediction by ML and to improve the reliability of these methods. These findings are not limited to biological HTS techniques and can be applied to other systems where interval censoring is an advantageous measurement strategy.\n",
      "--------------------------------------------------\n",
      "Les titres et les liens des articles ont été enregistrés et affichés dans la base de données SQLite.\n"
     ]
    }
   ],
   "source": [
    "# affiche articles stockés dans database \n",
    "c.execute(\"SELECT * FROM articles\")\n",
    "\n",
    "rows = c.fetchall() #récupere tous les results de la requete d'avant =tuple(id,title,linke ect) de chaque article\n",
    "\n",
    "print(\"=== Liste des articles enregistrés ===\\n\")\n",
    "\n",
    "for row in rows:\n",
    "    print(f\"ID: {row[0]}\")\n",
    "    print(f\"Titre: {row[1]}\")\n",
    "    print(f\"Lien: {row[2]}\")\n",
    "    print(f\"DOI: {row[3]}\")\n",
    "    print(f\"Date: {row[4]}\")\n",
    "    print(f\"Lien PDF: {row[5]}\")\n",
    "    print(f\"Résumé: {row[6]}\")  # Afficher l'abstract ici\n",
    "    print(\"-\" * 50)  # separateur visuel entre chaque article\n",
    "\n",
    "\n",
    "\n",
    "print(\"Les titres et les liens des articles ont été enregistrés et affichés dans la base de données SQLite.\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fermer la connexion à la base de données (quand tu as finis)\n",
    "#database.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
