{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Installation de bibliotheques \n",
    "\n",
    "**pip install beautifulsoup4**\n",
    "\n",
    "**pip install requests**\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import sqlite3\n",
    "import time"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Scraper les articles bioRxiv liés à la bioinformatique "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = 'https://www.biorxiv.org'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<sqlite3.Cursor at 0x7f85401b1b90>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# creer base de données SQLite \"bioinformatics_articles\" \n",
    "database = sqlite3.connect('bioinformatics_articles.db')\n",
    "c = database.cursor() #ici le curseur 'c' permettra d'executer/parcourir les requetes SQL sur database\n",
    "\n",
    "\n",
    "c.execute(\"DROP TABLE IF EXISTS articles\")\n",
    "\n",
    "# create table 'articles' avec col id,title et le lien \n",
    "c.execute('''CREATE TABLE IF NOT EXISTS articles\n",
    "             (id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "             title TEXT, \n",
    "             link TEXT,\n",
    "             doi TEXT,\n",
    "             date TEXT,\n",
    "             pdf_link TEXT)''')\n",
    "#id c'est un identifiant unique généré automatiquement "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Parcourir les articles  de la collection biofino contenant 2153 pages et nregistrer leurs informations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scraping page 1 sur 2\n",
      "Nombre total d'articles trouvés sur la page 1 : 10\n",
      "Titre: Continual integration of single-cell multimodal data with MIRACLE\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.613833v2\n",
      "DOI: 10.1101/2024.09.24.613833\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: Forecasting dominance of SARS-CoV-2 lineages by anomaly detection using deep AutoEncoders\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.10.24.563721v4\n",
      "DOI: 10.1101/2023.10.24.563721\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: Protein Sequence Modelling with Bayesian Flow Networks\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614734v1\n",
      "DOI: 10.1101/2024.09.24.614734\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: RanBALL: An Ensemble Random Projection Model for Identifying Subtypes of B-cell Acute Lymphoblastic Leukemia\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614777v1\n",
      "DOI: 10.1101/2024.09.24.614777\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: scEMB: Learning context representation of genes based on large-scale single-cell transcriptomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614685v1\n",
      "DOI: 10.1101/2024.09.24.614685\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: Analytical Assessment of Metagenomic Workflows for Pathogen Detection with NIST RM 8376 and Two Sample Matrices\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614717v1\n",
      "DOI: 10.1101/2024.09.24.614717\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: BRDKRM: An Explainable Framework for Disease Modifying Drug Identification\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614653v1\n",
      "DOI: 10.1101/2024.09.24.614653\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: PlantConnectome: knowledge graph encompassing >70,000 plant articles\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.07.11.548541v3\n",
      "DOI: 10.1101/2023.07.11.548541\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2\n",
      "DOI: 10.1101/2024.02.29.582810\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2\n",
      "DOI: 10.1101/2024.09.22.614322\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Scraping page 2 sur 2\n",
      "Nombre total d'articles trouvés sur la page 2 : 10\n",
      "Titre: varCADD: large sets of standing genetic variation enable genome-wide pathogenicity prediction\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614666v1\n",
      "DOI: 10.1101/2024.09.24.614666\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: Diversity and abundance of ring nucleases in type III CRISPR-Cas loci\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614671v1\n",
      "DOI: 10.1101/2024.09.24.614671\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: Germinated brown rice research: bibliometric analysis of progress, hotspots and trends\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614820v1\n",
      "DOI: 10.1101/2024.09.24.614820\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: p53motifDB: integration of genomic information and tumor suppressor p53 binding motifs\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614594v1\n",
      "DOI: 10.1101/2024.09.24.614594\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: RNA-DCGen: Dual Constrained RNA Sequence Generation with LLM-Attack\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614570v1\n",
      "DOI: 10.1101/2024.09.23.614570\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: Clustering-independent estimation of cell abundances in bulk tissues using single-cell RNA-seq data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.02.06.527318v2\n",
      "DOI: 10.1101/2023.02.06.527318\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: SeaMoon: Prediction of molecular motions based on language models\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614585v1\n",
      "DOI: 10.1101/2024.09.23.614585\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: INSPIRE: interpretable, flexible and spatially-aware integration of multiple spatial transcriptomics datasets from diverse sources\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614539v1\n",
      "DOI: 10.1101/2024.09.23.614539\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: Clustering individuals using INMTD: a novel versatile multi-view embedding framework integrating omics and imaging data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614478v1\n",
      "DOI: 10.1101/2024.09.23.614478\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n",
      "Titre: MedGraphNet: Leveraging Multi-Relational Graph Neural Networks and Text Knowledge for Biomedical Predictions\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614782v1\n",
      "DOI: 10.1101/2024.09.24.614782\n",
      "Date de publication: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# Nombre de pages à scraper (2153 au total but bcp de temps teste sur 2 page fait envirosn 2min)\n",
    "total_pages = 2\n",
    "\n",
    "# boucle qui parcour chaque page de la collection Bioinformatics\n",
    "for page in range(1, total_pages + 1):\n",
    "    print(f\"Scraping page {page} sur {total_pages}\")\n",
    "\n",
    "    # requete qui recup la page bioinfo actuelle\n",
    "    response = requests.get(f'{url}/collection/bioinformatics?page={page}') #? mean numero de la {page}\n",
    "    soup = BeautifulSoup(response.content, 'html.parser')\n",
    "\n",
    "    # Trouver les liens vers les articles sur cette page\n",
    "    articles = soup.find_all('a', class_='highwire-cite-linked-title')  # find liens hypertextes avec classe CSS\n",
    "\n",
    "    print(f'Nombre total d\\'articles trouvés sur la page {page} : {len(articles)}')\n",
    "\n",
    "    # Parcourir tous les articles de la page et enregistrer leurs titres et liens\n",
    "    for article in articles:\n",
    "        title = article.text.strip()  # extraire le titre dans la balise <a>, accessible via .text.\n",
    "        link = url + article['href']  # extraire le lien , href=destination du lien de mon article\n",
    "\n",
    "        # Requête pour accéder à la page de l'article\n",
    "        article_response = requests.get(link)\n",
    "        article_soup = BeautifulSoup(article_response.content, 'html.parser')\n",
    "\n",
    "        # extraire le DOI (dans une balise meta ou dans l'URL)\n",
    "        doi_tag = article_soup.find('meta', {'name': 'citation_doi'})  # balise <meta> qui a l’attribut name=\"citation_doi\".\n",
    "        doi = doi_tag['content'] if doi_tag else 'DOI non disponible'  # si balise existe -> attribut content contient la valeur doi\n",
    "\n",
    "        # extraire la date de publication (dans une balise meta ou time)\n",
    "        date_tag = article_soup.find('meta', {'name': 'citation_publication_date'})\n",
    "        date = date_tag['content'] if date_tag else 'Date non disponible'\n",
    "\n",
    "        # extraire le lien vers le fichier PDF\n",
    "        pdf_tag = article_soup.find('a', string='Download PDF')\n",
    "        pdf_link = url + pdf_tag['href'] if pdf_tag else 'Lien PDF non disponible'\n",
    "\n",
    "        # tout inserer dans la base de données SQLite\n",
    "        c.execute(\n",
    "            \"INSERT INTO articles (title, link, doi, date, pdf_link) VALUES (?, ?, ?, ?, ?)\",\n",
    "            (title, link, doi, date, pdf_link)\n",
    "        )\n",
    "\n",
    "        # Afficher les informations extraites\n",
    "        print(f'Titre: {title}\\nLien: {link}\\nDOI: {doi}\\nDate de publication: {date}\\nLien PDF: {pdf_link}\\n')\n",
    "\n",
    "        # pause d'une seconde pour éviter de surcharger le serveur\n",
    "        time.sleep(1)\n",
    "        \n",
    "    # Sauvegarder les modifications dans la base de données après chaque page\n",
    "    database.commit()\n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Affichage de mes articles dans la base de données"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "=== Liste des articles enregistrés ===\n",
      "\n",
      "ID: 1\n",
      "Titre: Continual integration of single-cell multimodal data with MIRACLE\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.613833v2\n",
      "DOI: 10.1101/2024.09.24.613833\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 2\n",
      "Titre: Forecasting dominance of SARS-CoV-2 lineages by anomaly detection using deep AutoEncoders\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.10.24.563721v4\n",
      "DOI: 10.1101/2023.10.24.563721\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 3\n",
      "Titre: Protein Sequence Modelling with Bayesian Flow Networks\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614734v1\n",
      "DOI: 10.1101/2024.09.24.614734\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 4\n",
      "Titre: RanBALL: An Ensemble Random Projection Model for Identifying Subtypes of B-cell Acute Lymphoblastic Leukemia\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614777v1\n",
      "DOI: 10.1101/2024.09.24.614777\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 5\n",
      "Titre: scEMB: Learning context representation of genes based on large-scale single-cell transcriptomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614685v1\n",
      "DOI: 10.1101/2024.09.24.614685\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 6\n",
      "Titre: Analytical Assessment of Metagenomic Workflows for Pathogen Detection with NIST RM 8376 and Two Sample Matrices\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614717v1\n",
      "DOI: 10.1101/2024.09.24.614717\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 7\n",
      "Titre: BRDKRM: An Explainable Framework for Disease Modifying Drug Identification\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614653v1\n",
      "DOI: 10.1101/2024.09.24.614653\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 8\n",
      "Titre: PlantConnectome: knowledge graph encompassing >70,000 plant articles\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.07.11.548541v3\n",
      "DOI: 10.1101/2023.07.11.548541\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 9\n",
      "Titre: Evaluating the representational power of pre-trained DNA language models for regulatory genomics\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.02.29.582810v2\n",
      "DOI: 10.1101/2024.02.29.582810\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 10\n",
      "Titre: Joint Modeling of Cellular Heterogeneity and Condition Effects with scPCA in Single-Cell RNA-Seq\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.22.614322v2\n",
      "DOI: 10.1101/2024.09.22.614322\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 11\n",
      "Titre: varCADD: large sets of standing genetic variation enable genome-wide pathogenicity prediction\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614666v1\n",
      "DOI: 10.1101/2024.09.24.614666\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 12\n",
      "Titre: Diversity and abundance of ring nucleases in type III CRISPR-Cas loci\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614671v1\n",
      "DOI: 10.1101/2024.09.24.614671\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 13\n",
      "Titre: Germinated brown rice research: bibliometric analysis of progress, hotspots and trends\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614820v1\n",
      "DOI: 10.1101/2024.09.24.614820\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 14\n",
      "Titre: p53motifDB: integration of genomic information and tumor suppressor p53 binding motifs\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614594v1\n",
      "DOI: 10.1101/2024.09.24.614594\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 15\n",
      "Titre: RNA-DCGen: Dual Constrained RNA Sequence Generation with LLM-Attack\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614570v1\n",
      "DOI: 10.1101/2024.09.23.614570\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 16\n",
      "Titre: Clustering-independent estimation of cell abundances in bulk tissues using single-cell RNA-seq data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2023.02.06.527318v2\n",
      "DOI: 10.1101/2023.02.06.527318\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 17\n",
      "Titre: SeaMoon: Prediction of molecular motions based on language models\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614585v1\n",
      "DOI: 10.1101/2024.09.23.614585\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 18\n",
      "Titre: INSPIRE: interpretable, flexible and spatially-aware integration of multiple spatial transcriptomics datasets from diverse sources\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614539v1\n",
      "DOI: 10.1101/2024.09.23.614539\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 19\n",
      "Titre: Clustering individuals using INMTD: a novel versatile multi-view embedding framework integrating omics and imaging data\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.23.614478v1\n",
      "DOI: 10.1101/2024.09.23.614478\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "ID: 20\n",
      "Titre: MedGraphNet: Leveraging Multi-Relational Graph Neural Networks and Text Knowledge for Biomedical Predictions\n",
      "Lien: https://www.biorxiv.org/content/10.1101/2024.09.24.614782v1\n",
      "DOI: 10.1101/2024.09.24.614782\n",
      "Date: 2024/01/01\n",
      "Lien PDF: Lien PDF non disponible\n",
      "--------------------------------------------------\n",
      "Les titres et les liens des articles ont été enregistrés et affichés dans la base de données SQLite.\n"
     ]
    }
   ],
   "source": [
    "# affiche articles stockés dans database\n",
    "c.execute(\"SELECT * FROM articles\")\n",
    "\n",
    "rows = c.fetchall() #récupere tous les results de la requete d'avant =tuple(id,title,link) de chaque article\n",
    "\n",
    "print(\"=== Liste des articles enregistrés ===\\n\")\n",
    "\n",
    "for row in rows:\n",
    "    print(f\"ID: {row[0]}\")\n",
    "    print(f\"Titre: {row[1]}\")\n",
    "    print(f\"Lien: {row[2]}\")\n",
    "    print(f\"DOI: {row[3]}\")\n",
    "    print(f\"Date: {row[4]}\")\n",
    "    print(f\"Lien PDF: {row[5]}\")\n",
    "    print(\"-\" * 50)  # separateur visuel entre chaque article\n",
    "\n",
    "\n",
    "\n",
    "print(\"Les titres et les liens des articles ont été enregistrés et affichés dans la base de données SQLite.\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Fermer la connexion à la base de données (quand tu as finis)\n",
    "#database.close() "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
