{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Installation de bibliotheques \n",
    "\n",
    "**pip install beautifulsoup4**\n",
    "\n",
    "**pip install requests**\n",
    "\n",
    "**pip install PyPDF2**\n",
    "\n",
    "**pip install pdfplumber**\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import sqlite3\n",
    "import time\n",
    "from datetime import datetime\n",
    "\n",
    "import pdfplumber\n",
    "import re, io\n",
    "import PyPDF2\n",
    "from io import BytesIO\n",
    "import os\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Scraper les articles bioRxiv liés à la bioinformatique "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = 'https://www.biorxiv.org'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<sqlite3.Cursor at 0x7fd4109d3f80>"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# creer base de données SQLite \"bioinformatics_articles\" \n",
    "database = sqlite3.connect('bioinformatics_articles.db')\n",
    "c = database.cursor() #ici le curseur 'c' permettra d'executer/parcourir les requetes SQL sur database\n",
    "\n",
    "\n",
    "c.execute(\"DROP TABLE IF EXISTS articles\")\n",
    "\n",
    "# create table 'articles' avec col id,title et le lien \n",
    "c.execute('''CREATE TABLE IF NOT EXISTS articles\n",
    "             (id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "             title TEXT, \n",
    "             link TEXT,\n",
    "             doi TEXT UNIQUE,\n",
    "             date TEXT,\n",
    "             pdf_link TEXT,\n",
    "             abstract TEXT)''')\n",
    "#id c'est un identifiant unique généré automatiquement "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Parcourir les articles  de la collection Bioinformatics contenant 2153 pages d'articles et enregistrer leurs informations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "## Nombre de pages à scraper (exemple : 10 pages à la fois)\n",
    "from pickle import TRUE\n",
    "\n",
    "\n",
    "total_pages = 15\n",
    "\n",
    "# boucle qui parcour chaque page de la collection Bioinformatics\n",
    "for page in range(1, total_pages + 1):\n",
    "    print(f\"Scraping page {page} sur {total_pages}\")\n",
    "\n",
    "    # requete qui recup la page bioinfo actuelle\n",
    "    response = requests.get(f'{url}/collection/bioinformatics?page={page}') #? mean numero de la {page}\n",
    "    soup = BeautifulSoup(response.content, 'html.parser')\n",
    "\n",
    "    # Trouver les liens vers les articles sur cette page\n",
    "    articles = soup.find_all('a', class_='highwire-cite-linked-title')  # find liens hypertextes avec classe CSS\n",
    "\n",
    "    print(f'Nombre total d\\'articles trouvés sur la page {page} : {len(articles)}')\n",
    "\n",
    "    # Parcourir tous les articles de la page et enregistrer leurs titres et liens\n",
    "    for article in articles:\n",
    "        title = article.text.strip()  # extraire le titre dans la balise <a>, accessible via .text.\n",
    "        link = url + article['href']  # extraire le lien , href=destination du lien de mon article\n",
    "\n",
    "        # Requête pour accéder à la page de l'article\n",
    "        article_response = requests.get(link)\n",
    "        article_soup = BeautifulSoup(article_response.content, 'html.parser')\n",
    "\n",
    "        # extraire le DOI (dans une balise meta ou dans l'URL)\n",
    "        doi_tag = article_soup.find('meta', {'name': 'citation_doi'})  # balise <meta> qui a l’attribut name=\"citation_doi\".\n",
    "        doi = doi_tag['content'] if doi_tag else 'DOI non disponible'  # si balise existe -> attribut content contient la valeur doi\n",
    "\n",
    "        # extraire la date de publication (dans une balise meta ou time)et la formater au format ISO 8601\n",
    "        date_tag = article_soup.find('meta', {'name': 'citation_publication_date'})\n",
    "        date = date_tag['content'] if date_tag else 'Date non disponible'\n",
    "        # Remplacer les barres obliques par des tirets avant la conversion\n",
    "        if date != 'Date non disponible':\n",
    "            date = date.replace('/', '-')  # Remplacer '/' par '-'\n",
    "            date = datetime.strptime(date, '%Y-%m-%d').date().isoformat()\n",
    "\n",
    "        # extraire le lien vers le fichier PDF\n",
    "        pdf_links = article_soup.find_all('a')\n",
    "        pdf_link = None\n",
    "        for pdf in pdf_links:\n",
    "            if pdf and 'PDF' in pdf.text:\n",
    "                potential_pdf_link = url + pdf['href']\n",
    "                if potential_pdf_link.endswith('.pdf'):  # Vérifier si le lien se termine par .pdf\n",
    "                    pdf_link = potential_pdf_link\n",
    "                    break  # Quitter la boucle une fois le lien trouvé\n",
    "\n",
    "        # En cas de lien invalide, assigner une valeur par défaut\n",
    "        if not pdf_link:\n",
    "            pdf_link = 'Lien PDF non disponible'\n",
    "\n",
    "        \n",
    "        \n",
    "         # Extraire l'abstract\n",
    "        abstract_tag = article_soup.find('meta', {'name': 'citation_abstract'})\n",
    "        if abstract_tag:\n",
    "            abstract_html = abstract_tag['content']\n",
    "            clean_abstract = BeautifulSoup(abstract_html, \"html.parser\").get_text(strip=True)\n",
    "        else:\n",
    "            clean_abstract = 'Abstract non disponible'\n",
    "\n",
    "\n",
    "        # tout inserer dans la base de données SQLite\n",
    "        # Vérifier si l'article existe déjà dans la base de données (vérification sur DOI et date)\n",
    "        c.execute(\"SELECT * FROM articles WHERE doi = ? AND date = ?\", (doi, date))\n",
    "        result = c.fetchone() # recupère la 1ere ligne du result donc si mm doi/date trouvé retourne cette ligne en tuple sinon NONE\n",
    "\n",
    "        if result:\n",
    "            print(f\"L'article avec DOI {doi} existe déjà, ignoré.\")\n",
    "        else:\n",
    "            # Insérer l'article dans la base de données\n",
    "            c.execute(\"INSERT INTO articles (title, link, doi, date, pdf_link, abstract) VALUES (?, ?, ?, ?, ?, ?)\", \n",
    "                      (title, link, doi, date, pdf_link, clean_abstract))\n",
    "            print(f\"Article ajouté : {title}\")\n",
    "\n",
    "        # Afficher les informations extraites\n",
    "        #print(f'Titre: {title}\\nLien: {link}\\nDOI: {doi}\\nDate de publication: {date}\\nLien PDF: {pdf_link}\\nRésumé:{clean_abstract}\\n')\n",
    "        # pause d'une seconde pour éviter de surcharger le serveur\n",
    "        time.sleep(1)\n",
    "        \n",
    "    # Sauvegarder les modifications dans la base de données après chaque page\n",
    "    database.commit()\n",
    "   \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Affichage de mes articles dans la base de données (A supprimer c'est juste pr que je vois bien l'ajout)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "''' affiche articles stockés dans database \n",
    "c.execute(\"SELECT * FROM articles\")\n",
    "\n",
    "rows = c.fetchall() #récupere tous les results de la requete d'avant =tuple(id,title,linke ect) de chaque article\n",
    "\n",
    "print(\"=== Liste des articles enregistrés ===\\n\")\n",
    "\n",
    "for row in rows:\n",
    "    print(f\"ID: {row[0]}\")\n",
    "    print(f\"Titre: {row[1]}\")\n",
    "    print(f\"Lien: {row[2]}\")\n",
    "    print(f\"DOI: {row[3]}\")\n",
    "    print(f\"Date: {row[4]}\")\n",
    "    print(f\"Lien PDF: {row[5]}\")\n",
    "    print(f\"Résumé: {row[6]}\")  # Afficher l'abstract ici\n",
    "    print(\"-\" * 50)  # separateur visuel entre chaque article\n",
    "\n",
    "\n",
    "\n",
    "print(\"Les titres et les liens des articles ont été enregistrés et affichés dans la base de données SQLite.\")\n",
    "\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fermer la connexion à la base de données (quand tu as finis)\n",
    "#database.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Extraire les URLs des dépôts logiciels dans les abstracts ou les fichiers PDF "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1.\tExtraction des liens spécifiques aux dépôts logiciels :\n",
    "\n",
    "La fonction extract_links_from_text() utilise des expressions régulières pour rechercher des URLs dans les textes des résumés et fichiers PDF, notamment des liens vers des dépôts logiciels populaires comme GitHub, GitLab, Bitbucket, SourceForge, ou Zenodo.\n",
    "\n",
    "2.\tÉtape progressive d’extraction:\n",
    "\n",
    "Le script commence par analyser les résumés. Si aucun lien n’est trouvé, il analyse alors le contenu des fichiers PDF des articles pour minimiser les ressources réseau et système.\n",
    "\n",
    "3.\tMise à jour propre de la base de données\n",
    "\n",
    "Seuls les liens valides, nettoyés (pour éviter les erreurs de formatage), et non redondants sont insérés dans la base de données dans la colonne software_links.\n",
    "\n",
    "Évite de retraiter les mêmes articles\n",
    "\n",
    "1.\tVérification des articles traités:\n",
    "\n",
    "Avant de traiter chaque article, le script vérifie si des liens existent déjà dans la colonne software_links. Si des liens sont déjà présents, l’article est ignoré, ce qui permet de ne pas retraiter les articles déjà analysés.\n",
    "\n",
    "2.\tPause après chaque requête:\n",
    "\n",
    "Une pause d’une seconde est introduite après chaque requête pour éviter de surcharger les serveurs web lors de l’extraction des fichiers PDF ou des liens.\n",
    "\n",
    "\n",
    "# suite du projet  :\n",
    "\n",
    "1.\t Vérification des URLs:\n",
    "\n",
    "Tester la validité des URLs extraites en effectuant des requêtes HTTP pour s’assurer qu’elles sont accessibles et correspondent à des dépôts logiciels actifs.\n",
    "\n",
    "2.\tArchivage via Software Heritage:\n",
    "\n",
    "Utiliser l’API de Software Heritage pour vérifier si le dépôt logiciel est déjà archivé dans leur base de données. Si le dépôt n’est pas encore archivé, soumettre automatiquement une demande d’archivage via l’API.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Liens de dépôt logiciel valides pour l'article 1 : ['https://doi.org/10.1101/2024.10.01.613179', 'https://academic.oup.com/ndt/article-pdf/32/1/', 'https://cdn.10xgenomics.com/image/upload/v1666737555/', 'https://journals.biologists.com/dev/', 'https://github.com/CostaLab/phlower', 'https://academic.oup.com/', 'https://journals.biologists.com/dev/article-pdf/146/12/dev178673/3112612/dev178673.pdf', 'https://phlower.readthedocs.io/en/latest/']\n",
      "Liens de dépôt logiciel valides pour l'article 2 : ['https://doi.org/10.1093/', 'https://doi.org/10.5281/ZENODO.6954035', 'https://doi.org/10.1038/s41586-020-2871-y', 'https://doi.org/10.1093/bioinformatics/btad074', 'https://doi.org/10.1128/jb.00619-08', 'https://www.biorxiv.org/content/early/2023/10/17/2023.09.22.558964.full.pdf', 'https://doi.org/10.1146/', 'https://doi.org/10.1016/j.cell.2020.05.023', 'https://github.com/pangenome/', 'https://doi.org/10.1101/2022.09.17', 'https://doi.org/10.1089/cmb.2017.0251', 'https://doi.org/10.1038/', 'https://doi.org/10.1038/nbt.4235', 'https://doi.org/10.1038/s41587-023-01793-w', 'https://doi.org/10.1101/gr.214155.116', 'https://doi.org/10.1093/bioinformatics/btaa777', 'https://doi.org/10.1038/ng.3117', 'https://doi.org/10.1038/s41588-023-01459-y', 'https://doi.org/10.1038/nbt.4227', 'https://doi.org/10', 'https://doi.org/10.1126/science.abn2937', 'https://doi.org/10.1038/s41586-023-05895-y', 'https://doi.org/10.1038/s41586-023-05896-x', 'https://github.com/marschall-lab/GFAffix', 'https://doi.org/10.1371/journal.pcbi.1009123', 'https://github.com/waveygang/wfmash', 'https://doi.org/10.1101/gr.123356', 'https://doi.org/10.1101/gr.214270.116', 'https://doi.org/10.1093/molbev/msw215', 'https://pggb.readthedocs.io/', 'https://doi.org/10.1038/s41586-023-05976-y', 'https://doi.org/10.1101/2023.09.22.558964', 'https://doi.org/', 'https://github.com/', 'https://doi.org/10.1093/bioinformatics/bty597', 'https://doi.org/10.1038/s41597-021-00947-2', 'https://doi.org/10.1186/s13059-019-1905-y', 'https://doi.org/10.1038/nbt.1754', 'https://doi.org/10.1101/', 'https://doi.org/10.1093/bioinformatics/', 'https://doi', 'https://doi.org/10.17863/', 'https://zenodo', 'https://doi.org/10.1038/s41592-023-01914-y', 'https://doi.org/10.1093/bioinformatics/btac743']\n",
      "Liens de dépôt logiciel valides pour l'article 3 : ['http://db.idrblab.net/ttd/', 'https://bioinfogp.cnb.csic.es/tools/Venny/index.html', 'https://david.ncifcrf.gov/', 'https://www.bioinformatics.com.cn/login/', 'https://www.genecards.org/', 'https://vina.scripts.edu/', 'https://disgenet.com/', 'https://omim.org/', 'http://string-db.org', 'https://www.uniprot.org/', 'http://tcmspw.com/tcmsp.php', 'https://www.pdb.org/', 'https://ccsb.scripts.edu/mgltools/downloads/', 'https://doi.org/10.1101/2024.09.30.615978', 'https://vina.scripps.edu', 'http://www.swisstargetprediction.ch/']\n",
      "Liens de dépôt logiciel valides pour l'article 4 : ['https://proceedings.neurips.cc/paper/2020/hash/d1e946f4e67db4b362ad23818a6fb78a-', 'https://github.com/facebookresearch/esm', 'http://arxiv.org/abs/2205.07023', 'https://github.com/flatironinstitute/DeepFRI', 'https://proceedings.neurips.cc/paper/2017/hash/8a20a8621978632d76c43dfd28b67767-', 'https://openreview.net/pdf?id=AAML7ivghpY', 'https://doi.org/10.1101/2024.09.30.615718']\n",
      "Liens de dépôt logiciel valides pour l'article 5 : ['https://www.uniprot.org/proteomes/UP000005640', 'https://doi.org/10.1101/2024.06.27.600973']\n",
      "Liens de dépôt logiciel valides pour l'article 6 : ['https://github.com/wz-create/AptaDiff', 'https://doi.org/10.1101/2023.11.25.568693', 'http://github.com/SheffieldML/']\n",
      "Liens de dépôt logiciel valides pour l'article 7 : ['https://www.nature.com/articles/s41598-', 'http://lipidworld.biomedcentral.com/', 'https://dx.plos.org/10.1371/journal.pone.0172774', 'https://www.who.int/news-room/fact-sheets/detail/cardiovascular-diseases-', 'https://training.cochrane.org/handbook', 'https://link.springer.com/10.1007/978-3-', 'https://www.tandfonline', 'https://www.nature.com/articles/nature25973', 'https://dx.plos.org/10', 'https://www.nature.com/articles/s41522-023-00430-3(accessed', 'https://linkinghub.elsevier.com/retrieve/pii/S235239641930800X', 'http://link.springer.com/', 'https://www.hindawi', 'https://www.diabetesatlas.org', 'https://dx.plos', 'https://www.frontiersin.org/articles/10.3389/fendo.2023.1119201/full', 'https://www.nature.com/articles/', 'https://www.nature.com/', 'https://www.nature', 'https://gut.bmj.com/lookup/doi/10.1136/gutjnl-2015-309618', 'http://academic', 'https://www.frontiersin.org/articles/10.3389/fmicb.2021', 'https://www.frontiersin.org/', 'https://doi.org/10.1101/2024.10.01.615711', 'https://www.who.int/news-room/fact-sheets/detail/diabetes', 'https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-', 'http://www.rstudio.com', 'https://linkinghub.elsevier.com/retrieve/pii/', 'https://onlinelibrary.wiley.com/', 'https://genomemedicine', 'https://www.frontiersin.org/articles/10.3389/fnut.2022', 'https://www', 'https://linkinghub.elsevier.com/', 'https://plotdigitizer.com/app', 'https://bmcmicrobiol.biomedcentral']\n",
      "Liens de dépôt logiciel valides pour l'article 8 : ['https://bmblx.bmi.osumc.edu/scgnn+', 'https://doi.org/10.1038/s41592-024-02201-', 'https://doi.org/10.1038/s41586-023-06139-9', 'https://doi.org/10.1101/2024.09.30.615735', 'https://doi.org/10.1038/s41592-024-02235-4']\n",
      "Liens de dépôt logiciel valides pour l'article 9 : ['https://www.youtube.com/@MaxQuantChannel', 'https://github.com/JurgenCox/mqtools7', 'https://www.maxquant.org/', 'https://groups.google.com/g/maxquant-list', 'https://groups.google.com/g/perseus-list', 'https://github.com/cox-labs/CoxLab_Bug_Reporting', 'https://doi.org/10.1101/2024.09.30.615788', 'https://www.maxquant.org/']\n",
      "Liens de dépôt logiciel valides pour l'article 10 : ['https://github.com/', 'https://doi.org/10.1101/2024.09.30.614425', 'https://github.com/vandeplaslab/full_profile']\n",
      "Liens de dépôt logiciel valides pour l'article 11 : ['https://github.com/martinpacesa/BindCraft/blob/main/notebooks/BindCraft.ipynb', 'https://github.com/martinpacesa/BindCraft', 'https://doi.org/10.1101/2024.09.30.615802']\n",
      "Liens de dépôt logiciel valides pour l'article 12 : ['https://doi.org/10.1038/msb.2012.26', 'https://doi.org/10.1056/nejmoa2007764', 'https://doi.org/10.1056/nejmoa2021436', 'https://doi.org/10.1021/ci300367a', 'https://doi.org/10.1186/s12918-018-0532-7', 'https://doi.org/10.1093/bioinformatics/btaa501', 'https://doi.org/10.1109/BIBM.2012.6392634', 'https://doi.org/10.1093/nar/gkh131', 'https://doi.org/10.1186/s12859-018-2379-8', 'https://doi.org/10.1021/acs.jchemed.5b00404', 'https://doi.org/10.1007/s10462-022-10306-1', 'https://doi.org/10.1001/jamanetworkopen.2018.4196', 'https://doi.org/10.1145/3269206.3271739', 'https://doi.org/10.1016/j.ijantimicag.2020.105949', 'https://doi.org/10.1145/3132847.3133021', 'https://doi.org/10.1145/2939672.2939754', 'http://arxiv.org/abs/1803.08375', 'https://doi.org/10.1016/j.ins.2021.09.008', 'https://doi.org/10.24963/ijcai.2018/521', 'https://doi.org/10.1007/978-3-319-93417-4_38', 'https://doi.org/10.1016/j.jbi.2021.103707', 'https://doi.org/10.1109/TCBB.2022.3172421', 'https://doi.org/10.1093/bioinformatics/btn479', 'https://doi.org/10.1101/2024.09.30.615812', 'https://doi.org/10.1093/database/bav061', 'https://doi.org/10.1093/bioinformatics/btx659', 'https://doi.org/10.1371/journal.pcbi.1002998', 'https://doi.org/10.1186/s12859-020-03724-x', 'https://doi.org/10.1200/JCO.2017.73.4186', 'https://doi.org/10.3390/make5020036', 'https://doi.org/10.30534/ijatcse/2020/175942020', 'https://doi.org/10.1016/j.jbi.2015.03.002', 'https://doi.org/10.1093/nar/gkw1092', 'https://doi.org/10.3390/ijms24098169', 'https://doi.org/10.1016/j.ymeth.2020.05.007', 'https://doi.org/10.1016/j.jbi.2018.11.005', 'https://doi.org/10.1145/3641288', 'https://doi.org/10.1145/3307339.3342161', 'https://doi.org/10.1145/3292500.3330931', 'https://doi.org/10.1109/TKDE.2008.239', 'https://doi.org/10.1038/srep12339', 'https://doi.org/10.1002/cpt.1434', 'https://doi.org/10.1016/j.isci.2024.109148', 'https://doi.org/10.1093/nar/gkx1037', 'https://doi.org/10.1093/nar/gkv951', 'https://doi.org/10.3390/ph16071050', 'https://doi.org/10.1371/journal.pone.0140816', 'https://doi.org/10.1093/bioinformatics/bty593', 'https://doi.org/10.1136/amiajnl-2013-002512', 'https://doi.org/10.1109/I4C57141.2022.10057747', 'https://doi.org/10.1145/2939672.2939753']\n",
      "Liens de dépôt logiciel valides pour l'article 13 : ['https://doi.org/10.1101/2024.09.30.615863']\n",
      "Liens de dépôt logiciel valides pour l'article 14 : ['https://doi.org/10.1101/2024.09.30.615766', 'https://github.com/DincalciLab/samurai', 'https://genebe.net/tools/liftover', 'https://github.com/DIncalciLab/SAMURAI_paper_scripts', 'https://github.com/DIncalciLab/samurai', 'https://42basepairs.com/browse/s3/gatk-test-data/cnv/somatic']\n",
      "Liens de dépôt logiciel valides pour l'article 15 : ['https://doi.org/10.1101/2024.09.30.615776']\n",
      "Liens de dépôt logiciel valides pour l'article 16 : ['http://neuroci.com/', 'https://doi.org/10.1101/2024.09.29.615661']\n",
      "Liens de dépôt logiciel valides pour l'article 17 : ['https://github.com/sarwanpasha/Binding-Site-Prediction', 'https://doi.org/10.1101/2023.08.23.554389']\n",
      "Liens de dépôt logiciel valides pour l'article 18 : ['https://doi.org/10.1101/2024.09.29.615743']\n",
      "Liens de dépôt logiciel valides pour l'article 19 : ['https://doi.org/10.1101/2024.09.30.615762', 'https://github.com/lingxusb/PlasmidGPT', 'https://www.addgene.org/', 'https://www.addgene.org/', 'https://genome.jgi.doe.gov/portal/IMG_PR/IMG_PR.home.html']\n",
      "Liens de dépôt logiciel valides pour l'article 20 : ['https://doi.org/10.1101/2024.09.30.615418', 'https://www.vlinderstichting.nl/vlinders/overzicht-vlinders/details-vlinder/groente-uil']\n",
      "Liens de dépôt logiciel valides pour l'article 21 : ['http://asia.ensembl.org/info/genome/genebuild/biotypes.html)[54]', 'https://github.com/jiwooo-', 'https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_38/', 'https://www.gtexportal.org/home/tissue/)[53]', 'http://gimlab.online/data/total_RNAseq/Supplementary_files.zip', 'https://doi.org/10.1101/2024.09.30.615750']\n",
      "Liens de dépôt logiciel valides pour l'article 22 : ['https://stix.colorado.edu', 'https://cancer.sanger.ac.uk/cosmic', 'https://doi.org/10.18434/MDS2-2336', 'https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1KG_ONT_VIENNA/README_1', 'https://www.biorxiv.org/content/10.1101/2024.04.18.590093v1', 'https://zenodo.org/records/10819636', 'https://github.com/zhengxinchang/excord-lr', 'https://zenodo.org/records/10819636', 'https://doi.org/10.5281/zenodo.13702318', 'https://gnomad.broadinstitute.org/help/sv-overview', 'https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1KG_ONT_VIENNA/t2t/', 'https://github.com/ryanlayer/stix', 'https://github.com/jnothman/UpSetPlot)was', 'https://ftp-', 'https://s3.amazonaws.com/1000g-', 'https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1KG_ONT_VIENNA/hg38/', 'https://github.com/ryanlayer/stix', 'https://github.com/behzodcu/staffr', 'https://ftp.ncbi.nlm.nih.gov/ReferenceSamples/giab/release/AshkenazimTrio/HG002_NA24385_son/N', 'https://doi.org/10.1101/2024.09.30.615931', 'https://human-']\n",
      "Liens de dépôt logiciel valides pour l'article 23 : ['https://doi.org/10.21203/rs.3.rs-1937079/v1', 'http://arxiv.org/abs/2302.12977', 'https://doi.org/10.1101/2024.09.29.615658', 'https://doi.org/10.4121/15022410.V1', 'https://doi.org/10.5281/zenodo.11236776', 'http://arxiv.org/abs/2308.06374', 'http://arxiv.org/abs/2305.19801', 'https://doi.org/10.5281/zenodo.7521476', 'https://doi.org/10.48550/arXiv.2110.15087', 'https://doi.org/10.48550/arXiv.2310.08365', 'https://doi.org/10.26434/chemrxiv-2024-kxlgm', 'https://doi.org/10.1101/2021.10.28.466262']\n",
      "Liens de dépôt logiciel valides pour l'article 24 : ['http://www.uniprot.org/', 'https://doi.org/10.1101/2024.08.21.609040', 'http://www.ebi.ac.uk/swissprot/']\n",
      "Liens de dépôt logiciel valides pour l'article 25 : ['https://networkx.github.io', 'https://github.com/IBM/Hestia-OOD', 'https://doi.org/10.1101/2024.03.14.584508', 'https://leashbio.substack.com/p/belka-results-suggest-computers-can']\n",
      "Liens de dépôt logiciel valides pour l'article 26 : ['https://smarter-database.readthedocs.io/en/latest/data-import.html,accessed:2024-04-17', 'https://github.com/cnr-ibba/SMARTER-database/issues,accessed:2024-08-08', 'https://docs.docker.com/compose/,accessed:2024-04-17', 'https://en.wikipedia.org/wiki/REST,accessed:2024-04-17', 'https://doi.org/10.1101/2024.09.01.610681', 'https://smarter-database.readthedocs.io/en/latest/index.html,accessed:2024-08-08', 'https://docs.anaconda.com/', 'https://www.ebi.ac.uk/eva/?Submit-Data,accessed:2024-04-16', 'https://webserver.ibba.cnr.it/smarter-api/usingsmarterapiR', 'http://www.fao.org/documents/card/en/c/cc3079en', 'https://angular.io/,accessed:2024-04-17', 'https://webserver.ibba.cnr.it/smarter-api/docs/,accessed:2024-04-17', 'https://cnr-ibba.github.io/r-smarter-api/)andthroughthedatabaseportalathttps://webserver.ibba.cnr.it/smarter/', 'http://genome.cshlp.org/lookup/doi/10.1101/gr.9.8.677', 'https://cnr-ibba.github.io/r-smarter-api/articles/smarterapi.html#collect-genotypes).Metadataandsample', 'https://gsejournal.biomedcentral.com/articles/10.1186/s12711-018-0427-5', 'https://www.ebi.ac.uk/ols4/ontologies/pato,accessed:2024-04-16', 'https://webserver.ibba.cnr.it/smarter/,accessed:2024-04-17', 'http://semver.org/.web', 'https://github.com/cnr-ibba/SMARTER-database', 'https://github.com/cnr-ibba/SMARTER-database,accessed:2024-05-20', 'https://www.mongodb.com/,accessed:2024-04-17', 'https://medium.com/@julianam.tyler/what-is-the-difference-between-odm-and-orm-267bbb7778b0', 'https://drivendata.github.io/cookiecutter-data-science/,accessed:2024-04-17', 'https://cnr-ibba.github.io/r-smarter-api/,rpackageversion', 'https://github.com/cnr-ibba/SMARTER-database', 'https://smarterproject.eu/,accessed:2024-04-16', 'https://github.com/EBIvariation/vcf-validator?tab=readme-ov-file#assembly-checker,accessed:2024-04-16', 'https://github.com/cnr-ibba/SMARTER-database/blob/master/HISTORY.rst,accessed:2024-08-08', 'https://www.cookiecutter.io/,accessed:2024-04-17', 'https://mongoengine-odm.readthedocs.io/,accessed:2024-04-17', 'https://smarter-database.readthedocs.io/en/latest/', 'https://www.illumina.com/documents/products/technotes/technote_topbot.pdf,accessed:']\n",
      "Liens de dépôt logiciel valides pour l'article 27 : ['https://doi.org/10.1093/', 'https://doi.org/10.1101/204438', 'https://onlinelibrary.wiley.com/doi/10.1002/advs.202300366', 'https://CRAN.R-project.org/', 'https://github.com/tagtag/TDbasedUFEHiC', 'https://doi.org/10.18129/B9.bioc.AnnotationHub.Rpackage', 'https://doi.org/10.1101/2024.02.11.579821', 'https://doi.org/10.1038/', 'https://doi.org/10.1038/s41586-020-2493-4', 'https://www.biorxiv.org/content/early/2017/11/14/204438.full.pdf', 'https://doi.org/10.1136/', 'https://doi.org/10', 'https://doi.org/10.1186/s13059-019-1658-7', 'https://www.biorxiv.org/content/early/2024/02/12/2024.02.11.579821.full.pdf', 'https://academic.oup.com/bioinformatics/article-', 'https://bioconductor.org/', 'https://doi.org/10.18129/', 'http://genome.cshlp.org/content/27/11/1939.full.pdf+html', 'https://doi.org/10.1002/advs', 'https://doi.org/10.1186/s13059-016-0896-1', 'https://ard.bmj.com/content/83/8/1048.full.pdf', 'https://bioconductor.org/packages/AnnotationHub', 'https://doi.org/10.1101/gr.220640.117', 'https://doi.org/10.1371/journal', 'https://doi.org/10.1101/2024.09.29.615651', 'https://bioconductor.org/packages/TDbasedUFE', 'https://academic.oup.com/nar/article-', 'https://doi.org/10.18129/B9.bioc.TDbasedUFE', 'https://doi.org/10.1093/nar/gkq1019']\n",
      "Liens de dépôt logiciel valides pour l'article 28 : ['https://orcid.org/0000-0003-3652-2540', 'https://github.com/ekernf01/perturbation_benchmarking', 'https://groups.google.com/forum/#!topic/cellnet_r/pXHt2J6ZH6I', 'https://github.com/const-ae/linear_perturbation_prediction-Paper/blob/main/benchmark/src/run_l', 'https://www.gsk.ai/causalbench-challenge/', 'http://pcahan1.github.io/cellnetr/', 'https://orcid.org/0000-0002-2310-8191', 'https://orcid.org/0000-0002-5287-627X', 'https://doi.org/10.1101/2023.07.28.551039', 'https://orcid.org/0000-0001-7013-1899', 'https://blog.newlimit.com/p/2023-year-in-review', 'https://orcid.org/0000-0003-3429-3174']\n",
      "Liens de dépôt logiciel valides pour l'article 29 : ['https://github.com/sehagler/drug', 'https://doi.org/10.4103%2Fijpvm', 'https://acsjournals.onlinelibrary.wiley.com/doi/abs/10', 'https://doi.org/10.1093/nar/gkab1028', 'https://doi.org/10', 'https://reactome', 'https://academic', 'https://doi.org/10.1101/2024.09.27.615497', 'https://www']\n",
      "Liens de dépôt logiciel valides pour l'article 30 : ['https://doi.org/10.1101/2024.09.16.613342', 'http://dx.doi.org/10', 'https://doi.org/10.1038/']\n",
      "Liens de dépôt logiciel valides pour l'article 31 : ['https://doi.org/10.1101/2023.11.24.568620', 'https://scholar.google.co']\n",
      "Liens de dépôt logiciel valides pour l'article 32 : ['https://doi.org/10.1021/', 'http://www.jstor', 'https://www.sciencedirect.com/science/article/pii/S0198885921000628', 'https://doi.org/10.1038/nrg.2016.49', 'https://doi.org/10.1021/acssynbio.9b00103', 'https://doi.org/10.1073/pnas.2104878118', 'https://doi.org/10.1038/', 'https://doi.org/10.1073/pnas.0910781107', 'https://doi.org/10.1038/s41586-021-04184-w', 'http://dx.doi.org/10.1039/D1CC04635G', 'https://scicore.unibas.ch', 'http://dx.doi.org/10.1039/', 'https://doi.org/10.1101/2024.09.25.615059', 'https://www.ncbi.nlm.nih.gov/pmc/articles/', 'https://doi.org/10', 'https://doi.org/10.1073/pnas.2311726121', 'http://www.jstor.org/stable/2957618', 'https://www.sciencedirect.com/science/article/pii/S2405471221004154', 'https://doi.org/10.1038/nbt1070', 'https://doi.org/10.1038/s41586-023-05696-3', 'https://www.mdpi.com/', 'https://doi.org/10.1073/pnas.1614437114', 'https://doi.org/10.1126/science.abn3445', 'https://doi.org/10.1038/s41592-019-0496-6', 'https://doi.org/10.1016/j.tips.2021.11.010', 'https://arxiv.org/abs/2405.05167', 'https://doi.org/10.1038/s41929-022-00821-3', 'https://doi.org/10.1073/pnas.170297297', 'https://proceedings.neurips.cc/paper/1993/file/', 'https://www.sciencedirect.com/science/article/pii/S0092867422011199', 'https://doi.org/10.1038/s41467-024-45630-3', 'https://doi.org/', 'https://www.sciencedirect.com/science/article/pii/S0958166999000518', 'http://www.sciencedirect.com/science/article/pii/S0076687916300374', 'http://dx.doi', 'https://doi.org/10.1016/j.cell.2020.08.012', 'https://doi.org/10.1162/neco.1996.8.5.1085', 'https://www.sciencedirect.com/science/', 'https://www.sciencedirect.com/', 'https://chat.openai.com', 'https://www.sciencedirect.com/science/article/']\n",
      "Liens de dépôt logiciel valides pour l'article 33 : ['https://data.leucegene.iricca/', 'https://doi.org/10.1101/2023.05.10.540140', 'https://alexdray86.github.io/CLIMB/build/index.htmlandhttps://alexdray86.github.io/CLIFF/']\n",
      "Liens de dépôt logiciel valides pour l'article 34 : ['https://github.com/r2dt-bio/r2dt-client', 'https://github.com/DasLab/draw_rna', 'https://r2dt.bio', 'http://rna.ucsc.edu/', 'https://github.com/rnacentral/r2dt-web', 'https://github.com/LDWLab/RNA2D-data-schema', 'https://rnacentral.org/r2dt', 'https://json-schema.org/', 'https://docs.r2dt.bio/en/latest/', 'https://doi.org/10.1101/2024.09.29.611006', 'https://hub.docker.com/r/rnacentral/r2dt', 'https://ldwlab.github.io/XRNA-React', 'https://chatgpt.com/g/', 'https://rnacanvas.app', 'https://github.com/PDBeurope/pdb-rna-viewer', 'https://r2dt.bio', 'https://github.com/LDWLab/RNA2D-data-schema/', 'https://docs.r2dt.bio', 'https://github.com/r2dt-bio/r2dt', 'https://github.com/r2dt-']\n",
      "Liens de dépôt logiciel valides pour l'article 35 : ['https://doi.org/10.1101/2024.09.28.615630', 'http://www.repeatmasker.org']\n",
      "Liens de dépôt logiciel valides pour l'article 36 : ['https://ice.synthego.com', 'https://doi.org/10.1101/2024.09.28.614980', 'http://bioinf.wehi.edu.au/software/elda/', 'http://bioinf.wehi.edu.au/software/elda/index.html']\n",
      "Liens de dépôt logiciel valides pour l'article 37 : ['https://www.youtube.com/watch?v=-yquJiNKlAE', 'https://doi.org/10.1101/2024.09.28.615626', 'https://crfm']\n",
      "Liens de dépôt logiciel valides pour l'article 38 : ['https://doi.org/10.46354/i3m.2024.emss.011', 'https://github.com/davecliff/ESCG', 'https://doi.org/10.46354/i3m.2024.emss.028', 'https://doi.org/10.1101/2024.09.27.615336']\n",
      "Liens de dépôt logiciel valides pour l'article 39 : ['https://doi.org/10.1101/2024.09.28.615583']\n",
      "Liens de dépôt logiciel valides pour l'article 40 : ['https://arxiv.org/abs/1412.6572', 'https://arxiv.org/abs/1911.07067', 'https://github.com/HySonLab/LightMed', 'https://arxiv.org/abs/2102.04306', 'https://arxiv.org/abs/1608.03983', 'https://arxiv.org/abs/2012.09501', 'https://doi.org/10.1101/2024.09.28.615584', 'https://arxiv.org/abs/1612.01105', 'https://arxiv.org/abs/1312.5851', 'https://arxiv.org/abs/1703.00523', 'https://arxiv.org/abs/1804.03999', 'https://arxiv.org/abs/1603.05027', 'https://arxiv', 'https://arxiv.org/abs/1409.1556', 'https://arxiv.org/abs/2006.04868']\n",
      "Liens de dépôt logiciel valides pour l'article 41 : ['https://github.com/molloy-lab/TREE-QMC', 'https://github.com/molloy-lab/tree-qmc-study/tree/main/han2024wtreeqmc', 'https://doi.org/10.1101/2024.09.27.615467']\n",
      "Liens de dépôt logiciel valides pour l'article 42 : ['https://github.com/wanglongxing2020/ML_code', 'https://www.proteinatlas.org/', 'https://dcc.icgc.org/', 'https://ualcan.path.uab.edu/index.html', 'https://doi.org/10.1101/2024.09.27.615369', 'https://www.ncbi.nlm.nih.gov/gds', 'https://portal.gdc.cancer.gov/']\n",
      "Liens de dépôt logiciel valides pour l'article 43 : ['https://arxiv.org/abs/1312.6114', 'https://github.com/yyLIU12138/INSTINCT', 'https://doi.org/10.5281/zenodo.7768714', 'https://arxiv.org/abs/2010.16061', 'https://arxiv.org/abs/1511.07289', 'https://arxiv.org/abs/1802.03426', 'https://arxiv.org/abs/1609.02907', 'https://github.com/Chen-Li-17/simCAS/tree/main/data', 'https://www.biorxiv.org/content/10.1101/2024.04.22.590602v1.abstract', 'https://instinct.readthedocs.io/en/latest/index.html', 'https://doi.org/10.1101/2024.05.26.595944']\n",
      "Liens de dépôt logiciel valides pour l'article 44 : ['https://doi.org/10.1101/2024.09.27.615367', 'https://github.com/OscarGVelasco/ClusterFoldSimilarity', 'https://doi.org/10.1016/j.stem.2020.07.006', 'https://link.springer.com/articles/10.1186/s13059-', 'https://www.immport.org/shared/study/SDY998', 'https://www.nature.com/articles/s41467-', 'https://www.nature.com/articles/s41587-023-01716-9', 'https://pubmed.ncbi.nlm.nih.gov/30980045/', 'https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02568-9', 'https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1010210', 'https://www.nature.com/articles/s41576-022-00449-w', 'https://www.tandfonline.com/doi/abs/10.1128/MCB.00362-09', 'https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1000446', 'https://onlinelibrary.wiley.com/doi/full/10.1002/ctm2.1068', 'https://pubmed.ncbi.nlm.nih.gov/27693023/', 'https://www.mdpi.com/1422-', 'https://pubmed.ncbi.nlm.nih.gov/28641533/', 'https://www.bioconductor.org/', 'http://www.cell.com/article/S1525001619304046/fulltext', 'https://molecularbrain.biomedcentral.com/articles/10.1186/s13041-019-0522-8', 'https://www.nature.com/articles/s41587-023-01766-z', 'https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02367-2', 'https://www.ncbi.nlm.nih.gov/books/NBK573422/', 'https://www.biorxiv.org/content/10.1101/404962v1', 'https://pubmed.ncbi.nlm.nih.gov/24218581/', 'https://link.springer.com/chapter/10.1007/978-1-4614-8090-7_8', 'https://doi.org/10.1101/202D4sc.a0ml91.27.615367', 'https://www.science.org/doi/10.1126/science.abl4896', 'https://www.mdpi.com/2218-273X/11/9/1361/htm', 'https://www.science.org/doi/10.1126/science.346.6206.237', 'https://www.nature.com/articles/s41593-020-00783-4', 'https://academic.oup.com/bioinformatics/article/34/23/4054/5040306', 'https://pubmed.ncbi.nlm.nih.gov/23312518/', 'https://onlinelibrary.wiley.com/doi/full/10.1002/glia.23771', 'https://www.nature.com/articles/nmeth.1937', 'https://atlas.fredhutch.org/data/nygc/multimodal/pbmc_multimodal.h5seurat', 'https://www.jneurosci.org/content/25/14/3628.short', 'https://bioconductor.org/packages/release/data/experiment/html/MouseGastrulationData.html', 'https://www.nature.com/articles/nbt.2859', 'https://pubmed.ncbi.nlm.nih.gov/20979621/', 'https://www.nature.com/articles/s41590-019-0378-1', 'https://link.springer.com/article/10.1007/s00401-009-0619-8', 'https://www.nature.com/articles/s41587-023-01767-y', 'https://www.sciencedirect.com/science/article/pii/S2049080114000545', 'https://doi.org/101146/annurev-biodatasci-020422-050645', 'https://www.10xgenomics.com/resources/datasets/10k-human-', 'https://www.biorxiv.org/content/10.1101/2023.11.05.565674v1', 'https://www.annualreviews.org/doi/abs/10.1146/annurev-', 'https://www.biorxiv.org/content/10.1101/2024.05.23.595582v1', 'https://www.nature.com/articles/ncb2843', 'https://pubmed.ncbi.nlm.nih.gov/36198237/', 'https://www.immport.org/shared/study/SDY998', 'https://www.cell.com/iscience/pdf/S2589-0042(19)30414-6.pdf', 'https://molecularneurodegeneration.biomedcentral.com/articles/10.1186/s13024-024-00703-1', 'https://journals.physiology.org/doi/10.1152/physrev.00042.2016', 'https://www.nature.com/articles/s41592-021-01336-8', 'https://academic.oup.com/cercor/article-abstract/13/3/252/355488', 'https://pubmed.ncbi.nlm.nih.gov/15494728/', 'https://www.nature.com/articles/s41586-019-0933-9', 'https://pubmed.ncbi.nlm.nih.gov/32139688/', 'https://doi.org/10.1016/j.neuron.2021.11.023', 'https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-1926-6', 'https://pubmed.ncbi.nlm.nih.gov/26160885/']\n",
      "Liens de dépôt logiciel valides pour l'article 45 : ['https://doi.org/10.1101/2024.09.27.615342', 'https://www.gov.uk/government/publications/', 'https://github.com/ukhsa-collaboration/within-patient-pOXA48-conjugation']\n",
      "Liens de dépôt logiciel valides pour l'article 46 : ['https://doi.org/10.1101/2024.09.29.615668', 'https://imigitlab.uni-muenster.de/published/glinda']\n",
      "Liens de dépôt logiciel valides pour l'article 47 : ['https://cellxgene.cziscience.com/collections/8191c283-0816-424b-9b61-c3e1d6258a77', 'https://huggingface.co/datasets/ctheodoris/Genecorpus-30M', 'https://doi.org:10.1038/s41592-021-01336-8', 'https://doi.org:10.1038/s41592-024-02201-0', 'https://doi.org:10.1016/j.cell.2020.12.016', 'https://doi.org:10.1093/bioinformatics/btae089', 'https://doi.org:10.1093/bib/bbaa269', 'https://doi.org:10.1186/s12967-023-04150-2', 'https://cellxgene.cziscience.com/collections/60358420-6055-411d-ba4f-', 'https://doi.org:10.1038/s41586-023-06139-9', 'https://doi.org:10.1038/nm1613', 'https://doi.org:10.1038/s41467-020-18873-z', 'https://web.stanford.edu/group/nusselab/cgi-bin/wnt/target_genes', 'https://doi.org:10.1038/s41467-022-32111-8', 'https://doi.org/10.1101/2024.09.27.615337', 'https://doi.org:10.1038/s41467-021-21246-9', 'https://www.gsea-', 'https://doi.org:10.1093/bioinformatics/btab036', 'https://doi.org:10.1038/s41586-022-05060-x', 'https://doi.org:10.1016/j.cell.2015.02.035', 'https://doi.org:10.1038/s41587-021-01139-4', 'https://doi.org:10.1038/s41592-024-02305-7']\n",
      "Liens de dépôt logiciel valides pour l'article 48 : ['https://doi.org/10.1101/2024.09.27.615366']\n",
      "Liens de dépôt logiciel valides pour l'article 49 : ['https://CRAN.R-project.org/package=ggpubr', 'https://github.com/akdess/CaSpER', 'https://github.com/AntonioDeFalco/SCEVAN', 'https://github.com/bioliyezhang/Benchmark_scRNASeq_by_scMultiomics', 'https://kharchenkolab.github.io/numbat/articles/numbat.html', 'https://doi.org/10.1101/2024.09.26.615284', 'https://github.com/broadinstitute/infercnv', 'https://github.com/navinlabcode/copykat']\n",
      "Liens de dépôt logiciel valides pour l'article 50 : ['https://doi.org/10.1101/2024.09.26.615264']\n",
      "Liens de dépôt logiciel valides pour l'article 51 : ['https://doi.org/10.1101/2024.08.07.606975']\n",
      "Liens de dépôt logiciel valides pour l'article 52 : ['https://doi.org/10.1101/2024.09.26.615248', 'https://github.com/zhouyflab/R-Predictor', 'https://github.com/zhenyaliu77/DeepLRR', 'https://itol.embl.de', 'https://deepsig.biocomp.unibo.it', 'https://github.com/labstructbioinf/DeepCoil', 'https://ftp.ncbi.nlm.nih.gov/blast/db/FASTA/', 'https://github.com/SwiftSeal/resistify', 'https://golgi.sandbox.google.com', 'https://services.healthtech.dtu.dk/services/TMHMM-2.0', 'https://github.com/ElofssonLab/scampi2', 'https://biotec.njau.edu.cn/rlkdb', 'https://github.com/eliza-m/NLRexpress', 'https://cb.csail.mit.edu/cb/paircoil2', 'https://github.com/fteufel/signalp-6.0', 'https://github.com/phytolrr/predict-phytolrr', 'http://www.enzim.hu/hmmtop', 'https://mybiosoftware.com/coils-2-2-prediction-coiled-coil-regions-proteins.html', 'https://phobius.sbc.su.se', 'https://github.com/eliza-m/NLRexpress', 'https://services.healthtech.dtu.dk/services/LipoP-1.0', 'https://github.com/slt666666/NLRtracker', 'https://www.megasoftware.net', 'https://github.com/BolognaBiocomp/coconat']\n",
      "Liens de dépôt logiciel valides pour l'article 53 : ['https://www.tensorflow.org/', 'https://www.st', 'https://mrimaster.com/magn', 'https://www.mathworks.com/matl', 'https://www.imatest', 'https://ww', 'https://builtin.com/machine-learning/adam-optimization', 'https://www.statisticshowto.com/nrms', 'https://www.kaggle.com/datasets/dsch', 'https://doi.org/10.1101/2024.08.02.606374', 'https://keras.io/', 'https://onlinelibrary.wiley.com/doi/10.100', 'https://uhlibraries.pressbooks.pub/buildingskillsfordatascience/', 'https://nifti.nimh.nih.gov/']\n",
      "Liens de dépôt logiciel valides pour l'article 54 : ['https://doi.org/10.1101/2024.06.03.597180', 'https://cellxgene.cziscience', 'https://doi.org/10.1101/2023.10.30.563174']\n",
      "Liens de dépôt logiciel valides pour l'article 55 : ['https://datadryad.org/stash/dataset/doi:10.5061/dryad.x3ffbg7mw', 'https://doi.org/10.1101/2023.08.28.554970', 'https://github.com/']\n",
      "Liens de dépôt logiciel valides pour l'article 56 : ['https://github.com/ronboger/conformal-protein-', 'https://doi.org/10.1101/2024.06.27.601042']\n",
      "Liens de dépôt logiciel valides pour l'article 57 : ['https://doi.org/10.1101/2024.09.27.615406', 'https://imetalab.ca/']\n",
      "Liens de dépôt logiciel valides pour l'article 58 : ['https://doi.org/10.1101/2024.09.26.615249', 'https://swissmodel.expasy.org', 'https://blast.ncbi.nlm.nih.gov', 'https://www.uniprot.org]']\n",
      "Liens de dépôt logiciel valides pour l'article 59 : ['https://github.com/Graylab/', 'https://doi.org/10.1101/2024.09.27.615401']\n",
      "Liens de dépôt logiciel valides pour l'article 60 : ['https://citypopulation.de(accessedJune4,2023', 'https://doi.org/10.1101/2024.09.26.614772', 'https://metagenomics-workshop.readthedocs.io/en/2014-11-uppsala/']\n",
      "Liens de dépôt logiciel valides pour l'article 61 : ['https://doi.org/10.1101/2024.09.27.614001', 'https://zenodo.org/records/10679280', 'https://www.uniprot.org', 'https://github.com/salesforce/progen/', 'https://conoserver.org/', 'https://zenodo.org/records/7309036', 'https://zenodo.org/records/10679280', 'https://www.uniprot.org/', 'https://github.com/xueww/ConoDL）.Additionally', 'https://github.com/xueww/ConoDL.The']\n",
      "Liens de dépôt logiciel valides pour l'article 62 : ['http://creativecommons.org/licenses', 'https://doi.org/10.3390/xxxxx', 'https://doi.org/10.1101/2024.09.26.615132']\n",
      "Liens de dépôt logiciel valides pour l'article 63 : ['http://bioinfo.ihb.ac.cn/fishcode', 'https://github.com/hliihbcas/FishCODE', 'http://bioinfo.ihb.ac.cn/fishcode', 'https://doi.org/10.1101/2024.09.25.614839']\n",
      "Liens de dépôt logiciel valides pour l'article 64 : ['https://doi.org/10.1002/imt2.69', 'https://www.genome.jp/kegg/', 'https://www.medrxiv.org/content/10.1101', 'http://ccb.jhu.edu/software/fqtrim/index.shtml', 'https://github.com/Wonderangela123/M2AE', 'https://doi.org/10.1101/2024.09.25.614767']\n",
      "Liens de dépôt logiciel valides pour l'article 65 : ['https://doi.org/10.1016/j.smim.2023.101727', 'https://doi.org/10.1074/mcp.T400003-MCP200', 'https://doi.org/10.1093/nar/gkac1052', 'https://github.com/y-256/libdivsufsort', 'https://doi.org/10.1007/978-1-0716-4007-4_11', 'https://github.com/IlyaGrebnov/libsais', 'https://doi.org/10.1093/genetics/iyad031', 'https://doi.org/10.1093/bioinformatics/btaa553', 'https://doi.org/10.1080/19490976.2019.1702431', 'https://pubs.acs.org/doi/full/10.1021/acs.jproteome.3c00091', 'https://doi.org/10.1038/s41467-021-27542-8', 'https://prosite.expasy.org/scanprosite/', 'https://www.uniprot.org/peptide-search', 'https://doi.org/10.1186/s40168-020-00967-x', 'https://doi.org/10.1021/acs.jproteome.8b00716', 'https://doi.org/10.1093/nar/gkac993', 'https://doi.org/10.1093/bioinformatics/btt484', 'https://doi.org/10.1038/75556', 'https://doi.org/10.1101/2024.09.26.615136']\n",
      "Liens de dépôt logiciel valides pour l'article 66 : ['https://github.com/bioasp/precursor).We', 'https://www.plafrim.fr', 'https://github.com/bioasp/seed2lp', 'http://borensteinlab.com/software_NetSeed.html', 'https://github.com/potassco/clingo-lpx', 'https://doi.org/10.57745/OS1JND', 'http://dx.doi.org/10.1038/s41579-020-00458-8', 'http://bioinformatics', 'https://journals.asm.org/journal/mmbr', 'https://doi.org/10.21105/joss.03021', 'https://doi.org/10.1007/', 'https://doi.org/10.1101/2023.06.07.544035', 'http://www.ncbi.nlm.nih.gov/pubmed/28861526http:', 'https://doi.org/10.1101/2024.09.26.615309']\n",
      "Liens de dépôt logiciel valides pour l'article 67 : ['https://doi.org/10.1101/2024.09.25.615070', 'https://github.com/aberer/RogueNaRok/blob/master/example/', 'http://tree.bio.ed.ac.uk/software/figtree/']\n",
      "Liens de dépôt logiciel valides pour l'article 68 : ['https://www.gsea-msigdb.org/gsea/msigdb/index.jsp', 'https://hrfrost.host.dartmouth.edu/SCGeneSetOpt/', 'https://satijalab.org/seurat/articles/pbmc3k', 'https://cf.10xgenomics.com/samples/cell/pbmc3k/pbmc3k', 'https://doi.org/10.1101/2024.09.25.615040', 'https://hrfrost.host.dartmouth.edu/SCGeneSetOpt/', 'https://www.proteinatlas.org/download/rna']\n",
      "Liens de dépôt logiciel valides pour l'article 69 : ['https://i3-', 'https://doi.org/10.1101/2024.09.25.614939', 'https://github.com/i3-unit/AnalyzAIRR', 'https://github.com/i3-']\n",
      "Liens de dépôt logiciel valides pour l'article 70 : ['https://doi.org/10.1101/2024.04.08.588507', 'https://github.com/cgob/codonDT_snakemake', 'https://github.com/vam-sin/ribogl/']\n",
      "Liens de dépôt logiciel valides pour l'article 71 : ['https://doi.org/10.1016/0925-2312(91)90023-5', 'https://doi.org/10.7717/peerj.6543', 'https://doi.org/10.1093/bib/bbad073', 'https://doi.org/10.1007/s42979-024-02743-2', 'https://doi.org/10.1109/CAI54212.2023.00017', 'https://doi.org/10.1046/j.1471-4159.2003.01948.x', 'https://doi.org/10.1056/NEJMoa1202753', 'https://doi.org/10.1038/s41576-018-0004-3', 'https://doi.org/10.1038/s41586-019-1195-2', 'https://doi.org/10.1016/j.mad.2020.111319', 'https://doi.org/10.1111/biom.13108', 'https://doi.org/10.1016/j.biopsych.2009.01.027', 'https://doi.org/10.1023/A:1021709817809', 'https://doi.org/10.1101/2024.09.25.614862', 'https://doi.org/10.1159/000369101', 'https://doi.org/10.1038/s41390-', 'https://doi.org/10.1038/nrn.2015.1', 'https://doi.org/10.1038/ng.439', 'https://doi.org/10.1002/ima.22970', 'https://www.nejm.org/doi/full/10.1056/nejm199802193380804', 'https://doi.org/10.1016/j.bbi.2014.11.002', 'https://doi.org/10.1016/j.bspc.2024.106023', 'https://doi.org/10.1038/s41588-020-0696-0', 'https://doi.org/10.1016/j.ijbiomac.2020.07.327', 'https://doi.org/10.1214/12-AOAS597', 'https://doi.org/10.1016/j.csbj.2023.02.021', 'https://doi.org/10.1093/bioinformatics/btp543', 'https://doi.org/10.1007/BF00994018', 'https://doi.org/10.1016/j.neuron.2013.12.003', 'https://doi.org/10.1038/s41598-023-30892-6', 'https://doi.org/10.1007/978-0-387-', 'https://doi.org/10.19026/rjaset.7.299', 'https://doi.org/10.1016/j.neunet.2023.10.040', 'https://doi.org/10.1002/sim.9953', 'https://doi.org/10.1109/DSMP.2018.8478522', 'https://doi.org/10.1212/WNL.0000000000009058', 'https://doi.org/10.1038/tp.2012.45', 'https://doi.org/10.1096/fasebj.9.5.7896005', 'https://doi.org/10.1007/s11227-023-05655-9', 'https://doi.org/10.1111/acel.14044', 'https://doi.org/10.1007/978-94-007-0753-5_1689', 'https://doi.org/10.1109/ACCESS.2024.3390186', 'https://doi.org/10.3389/fnins.2024.1277187', 'https://doi.org/10.1038/ng.803', 'https://doi.org/10.1097/NEN.0b013e3181af2e46', 'https://doi.org/10.1038/ncomms11398', 'https://doi.org/10.1109/ICDAR.1995.598994', 'https://doi.org/10.1007/s11910-014-0499-8', 'https://doi.org/10.1007/978-0-387-39940-9_565', 'https://doi.org/10.1016/j.cca.2024.117842', 'https://doi.org/10.1093/bib/bbab569', 'https://doi.org/10.1016/j.jalz.2018.02.018', 'https://doi.org/10.7326/0003-4819-109-4-298', 'https://doi.org/10.1186/s13195-024-01488-7', 'https://doi.org/10.1371/journal.pone.0248413', 'https://doi.org/10.1016/j.bspc.2023.105669', 'https://doi.org/10.1007/s40745-022-00452-2', 'https://doi.org/10.1038/s41551-018-0304-0', 'https://doi.org/10.1093/brain/awz384', 'https://doi.org/10.1016/j.compbiomed.2023.107814', 'https://doi.org/10.2147/DNND.S228939', 'https://doi.org/10.1038/s41598-024-56837-1', 'https://doi.org/10.1038/nmeth.2930', 'https://doi.org/10.1016/j.bbi.2011.01.017', 'https://doi.org/10.3233/JAD-2006-9406', 'https://doi.org/10.1016/j.compbiomed.2023.107303', 'https://doi.org/10.1016/j.arr.2016.01.002', 'https://doi.org/10.1212/WNL.0000000000002923', 'https://doi.org/10.3389/frdem.2024.1332928', 'https://doi.org/10.1007/s40484-016-', 'https://doi.org/10.1007/978-3-540-39964-3_62', 'https://github.com/wan-mlab/WIMOAD', 'https://doi.org/10.1016/j.pneurobio.2019.101662']\n",
      "Liens de dépôt logiciel valides pour l'article 72 : ['https://docs.hubmapconsortium.org/assays/rnaseq', 'https://docs.hubmapconsortium.org/assays/seqfish', 'https://mtmorgan.github.io/', 'https://docs.hubmapconsortium.org/assays/codex', 'https://doi.org/10.1101/2024.09.26.615227', 'https://github.com/christinehou11/HuBMAPR)and', 'https://docs.hubmapconsortium.org/data-submission', 'https://github.com/christinehou11/HuBMAPR', 'https://www.R-project.org/', 'https://academic.oup.com/nar/article-lookup/doi/', 'https://www.perlfoundation.org/artistic-license-20', 'https://www.nature.com/', 'https://portal.hubmapconsortium.org/collections', 'https://portal.hubmapconsortium.org/', 'https://github.com/hubmapconsortium', 'https://genomebiology.biomedcentral.com/articles/', 'https://www.ncbi', 'https://hubmapconsortium.org/policies/', 'https://docs.hubmapconsortium.org/assays/imc', 'https://www.globus.org/', 'https://bmcbioinformatics', 'https://www.nature.com/articles/nmeth.4197']\n",
      "Liens de dépôt logiciel valides pour l'article 73 : ['https://bioportal.bioontology.org/ontologies/LSFO', 'https://github.com/EsmaeilNourani/LSFO-expansion', 'https://zenodo.org/records/10450308', 'https://github.com/spyysalo/bert-vocab-eval', 'https://doi.org/10.1101/2024.06.13.598816', 'https://esmaeilnourani.github.io/lifestylefactors-annotation-', 'https://github.com/larsjuhljensen/tagger']\n",
      "Liens de dépôt logiciel valides pour l'article 74 : ['https://doi.org/10.1101/2024.09.24.614785']\n",
      "Liens de dépôt logiciel valides pour l'article 75 : ['https://doi.org/10.1101/2024.09.24.614814', 'https://www.10xgenomics.com/', 'https://github.com/', 'https://www.10xgenomics.com/datasets/']\n",
      "Liens de dépôt logiciel valides pour l'article 76 : ['https://www.sciencedirect.com/science/article/pii/S0010482524001604', 'https://www.nature.com/articles/s41573-019-0024-5)(June2019', 'https://biosig.lab.uq.edu.au/pdcsm_gpcr/and', 'https://www.annualreviews.org/content/journals/10.1146/annurev.soc.27', 'https://doi.org/10.1186/s13321-024-00870-9)(June2024', 'http://arxiv.org/abs/', 'https://www.nature.com/articles/s41589-024-01679-1)(Aug.2024', 'https://github.com/chainer/chainer-chemistry/blob/master/chainer_', 'http://arxiv.org/abs/2303', 'https://doi.org/10.1021/acsomega', 'https://doi.org/10.1007/s10822-016-9906-3)(Mar.2016', 'http://arxiv.org/abs/2106.06134', 'https://www.nature', 'https://ieeexplore.ieee.org/document/9681226)(May2023', 'https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv', 'https://www.nature.com/articles/s41586-021-03819-2)(Aug.2021', 'https://github.com/ashvardanian/SimSIMD', 'https://doi.org/10.1101/2024.03.29.586957', 'http://arxiv.org/abs/2102.05497', 'https://doi.org/10.1093/bioinformatics/btaa968)(June', 'https://doi.org/', 'http://arxiv.org/abs/2404.14928', 'http://arxiv.org/abs/2404.19288', 'https://doi', 'https://www.biorxiv']\n",
      "Liens de dépôt logiciel valides pour l'article 77 : ['https://www.biorxiv.org/content/10.1101/2023.06.29.543848v1', 'https://www.biorxiv.org/content/10.1101/2024.09.19.313992v1', 'https://doi.org/10.1101/2024.09.24.614844']\n",
      "Liens de dépôt logiciel valides pour l'article 78 : ['https://doi.org/10.1186/s13100-019-0197-9', 'https://doi.org/10.3390/ijms23094755', 'https://doi.org/10.1101/2024.09.25.s614i8m65', 'https://doi.org/10.1038/28653', 'https://doi.org/10.1371/journal.pgen.1002487', 'https://doi.org/10.3390/biology1020395', 'https://doi.org/10.1186/s13100-015-0055-3', 'https://doi.org/10.1101/gr.226993.117', 'https://doi.org/10.1186/1471-2148-7-66', 'https://doi.org/10.1093/nar/gkab010', 'https://doi.org/10.1038/nrg2165-c4', 'https://doi.org/10.1038/nature05260', 'https://doi.org/10.3389/fpls.2022.814718', 'https://doi.org/10.1093/bib/bbm048', 'https://doi.org/10.1534/g3.117.043893', 'https://github.com/PacificBiosciences/pbsv', 'https://doi.org/10.1371/journal.pone.0030008', 'https://doi.org/10.1111/mec.14963', 'https://doi.org/10.1007/s00239-', 'https://doi.org/10.1534/g3.112.005348', 'https://doi.org/10.1126/science.1178534', 'https://doi.org/10.1093/bib/bbw072', 'https://doi.org/10.1038/nrg2640', 'https://get.genotoul.fr/', 'https://doi.org/10.1093/genetics/iyab113', 'https://github.com/e-lerat/replicaTE', 'https://doi.org/10.1093/nar/gku323', 'https://doi.org/10.1093/gbe/evx050', 'https://doi.org/10.7554/eLife.20777', 'https://doi.org/10.1101/2024.09.25.614865', 'https://www.girinst.org/', 'https://doi.org/10.1101/gr.277383.122', 'https://doi.org/10.1186/s13059-018-1577-z', 'https://doi.org/10.1093/bioinformatics/bts697', 'https://doi.org/10.1186/s12864-015-1975-5', 'https://www.r-', 'https://doi.org/10.7717/peerj.2942', 'https://doi.org/10.3390/cells11162501', 'https://www.arabidopsis.org/', 'https://doi.org/10.1093/molbev/msw137', 'https://tehub.org/', 'https://doi.org/10.1093/bioinformatics/btr708', 'https://doi.org/10.1007/s00521-021-06188-z', 'https://doi.org/10.1093/bioinformatics/btz727']\n",
      "Liens de dépôt logiciel valides pour l'article 79 : ['https://doi.org/10.1101/2024.09.24.614684', 'https://github.com/rashmisrm/MorphoGenie']\n",
      "Liens de dépôt logiciel valides pour l'article 80 : ['https://github.com/uttamLab/LMSF', 'https://doi.org/10.1101/2024.09.25.614955', 'https://doi.org/10.7303/syn61804540']\n",
      "Liens de dépôt logiciel valides pour l'article 81 : ['https://ferroenrich.shinyapps.io/ferroenrich/', 'https://doi.org/10.1101/2024.09.25.615075']\n",
      "Liens de dépôt logiciel valides pour l'article 82 : ['https://www.synapse.org/Synapse:syn31534849', 'https://doi.org/10.1016/0022-3956(75)90026-', 'https://www.synapse.org/Synapse:syn30821562', 'https://doi.org/10.1101/2024.08.23.609465', 'https://adknowledgeportal.synapse.org', 'https://www.synapse.org/Synapse:syn51739831', 'https://www.synapse.org/Synapse:syn52331749']\n",
      "Liens de dépôt logiciel valides pour l'article 83 : ['https://doi.org/10.1101/2024.0h9t.t2p4s.:6//1a3t8la3s3.;f', 'https://www.10xgenomics.com/re', 'https://atlas.fredhutch.org/nygc/m', 'https://github.com/welch-lab/liger', 'https://huggingface.co/ctheodoris/Geneformer', 'https://singlecell.broadinstitute.org/single_cell/study/SCP1303/', 'https://github.com/jaydu1/scVAEIT', 'https://azimuth.hubmapconsortium.org', 'https://www.10xgenomics.com/resources/datasets/', 'https://github.com/YosefLab/', 'https://www.10xgenomics.com/resources/datasets/10-', 'https://github.com/SONGDONGYUAN1994/scsampler', 'https://github.com/melobio/', 'https://satijalab.org/seurat/articles/seurat5_sketch_analysis.html', 'https://www.10xgenomics.com/resources/datasets/pbmc-', 'https://github.com/theislab/trVAE', 'https://github.com/', 'https://atlas.fredhutch.org/nygc/multimodal-pbmc', 'https://github.com/theislab/multigrate', 'https://doi.org/10.1101/2024.09.24.613833', 'https://www.10xgenomics.com/resources/datasets/10-k-', 'https://github.com/jlakkis/']\n",
      "Liens de dépôt logiciel valides pour l'article 84 : ['https://github.com/simoRancati/DeepAutoCoV', 'https://www.pfizer.com/news/press-release/press-release-detail/pfizer-and-biontech-provide-update-o', 'https://doi.org/10.1101/2023.10.24.563721', 'https://arxiv.org/abs/2109.05019v4', 'https://data.who.int/dashboards/covid19/cases', 'https://arxiv.org/abs/1605.08695v2', 'https://www.semanticscholar.org/paper/22-A-Model-of-Evolutionary-Change-in-Proteins-O./ff3e2e9', 'https://www.who.int/news/item/18-05-2023-statement-on-the-antigen-composition-of-covid-19-vacci', 'https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-classifications.html']\n",
      "Liens de dépôt logiciel valides pour l'article 85 : ['https://huggingface.co/', 'https://github.com/jeffreyruffolo/AntiBERTy)andAbLang2(https:', 'https://github.com/instadeepai/protein-sequence-bfn', 'https://doi.org/10.1101/2024.09.24.614734']\n",
      "Liens de dépôt logiciel valides pour l'article 86 : ['https://doi.org/10.1007/978-1-60327-367-1_12', 'https://doi.org/10.1007/978-0-387-39940-9_565', 'https://link.springer.com/10.1007/978-1-62703-059-5', 'https://doi.org/10.1101/2024.09.24.614777', 'https://www.researchsquare.com/article/rs-2798895/v1', 'https://www.stjude.cloud', 'http://www.ams.org/conm/026/', 'http://arxiv.org/abs/1802.03426', 'https://github.com/wan-mlab/RanBALL', 'https://pecan.stjude.cloud/static/hg19/pan-all/BALL-1988S-HTSeq.zip', 'https://dl.acm.org/doi/10.1145/1150402.1150436', 'https://journals.lww.com/hemasphere/fulltext/2022/04000/monitoring_of_leukemia_clo', 'https://orcid.org/0000-0003-0661-2684', 'http://perspectivesinmedicine.cshlp.org/content/10/7/a034835.abstract', 'http://ftp.ensembl.org/pub/release-']\n",
      "Liens de dépôt logiciel valides pour l'article 87 : ['https://github.com/bowang-lab/scGPT', 'https://doi.org/10.1101/2024.09.24.614685', 'https://cellxgene.cziscience.com/']\n",
      "Liens de dépôt logiciel valides pour l'article 88 : ['https://doi.org/10.1101/2024.09.24.614717', 'https://www.nist.gov/publication']\n",
      "Liens de dépôt logiciel valides pour l'article 89 : ['https://doi.org/10.1101/2024.09.24.614653', 'https://doi']\n",
      "Liens de dépôt logiciel valides pour l'article 90 : ['https://plant.connectome.tools/api/alias/psad1”', 'https://arxiv.org/abs/2408.07222v1', 'https://doi.org/10.3390/ijms22073560', 'https://doi.org/10.3389/fpls.2015.00277', 'https://colab.research.google.com/drive/1gENxLK2172Bq1sV_dO6JhvORsFgaq0fS?us', 'https://plant2.connectome.tools/normal/heat', 'https://agris-knowledgebase.org/downloads.html', 'https://github.com/mutwil/plant_connectome_latest/', 'https://doi.org/10.1038/s41467-019-08746-5', 'https://figshare.com/ndownloader/files/49392538', 'https://doi.org/10.1093/mp/ssp043', 'https://doi.org/10.1016/j.atg.2016.10.003', 'https://doi.org/10.1073/pnas.1219229110', 'https://doi.org/10.1111/pbi.13583', 'https://doi.org/10.1073/pnas.0503392102', 'https://plant.connectome.tools/catalogue', 'https://doi.org/10.1002/pro.3978', 'https://doi.org/10.1371/journal.pone.0105726', 'https://doi.org/10.1093/jxb/erz517', 'https://doi.org/10.1083/jcb.201807166', 'https://doi.org/10.1111/j.1365-313X.2004.02011.x', 'https://figshare.com/ndownloader/files/49198933', 'https://doi.org/10.1016/j.jbi.2018.08.009', 'https://doi.org/10.1038/srep10533', 'https://doi.org/10.3390/plants11091130', 'https://plant.connectome.tools/', 'https://doi.org/10.1105/tpc.105.031542', 'https://doi.org/10.1038/nmeth.2340', 'https://doi.org/10.1016/j.tplants.2013.10.006', 'https://doi.org/10.1186/s12859-019-2594-y', 'https://github.com/mutwil/plant_connectome_latest.git', 'https://doi.org/10.3389/fpls.2016.01445', 'https://doi.org/10.1038/s41477-020-0605-7', 'https://doi.org/10.1093/nar/gkac1000', 'https://openai.com/api/pricing/', 'https://doi.org/10.1101/2023.07.11.548541', 'https://doi.org/10.1371/journal.pone.0055814', 'https://doi.org/10.3390/plants10010103', 'https://doi.org/10.1002/pld3.483', 'https://plant.connectome.tools/alias/psad1', 'https://plant.connectome.tools/alias/MYB46', 'https://doi.org/10.1016/j.cell.2015.05.002', 'https://plant.connectome.tools/', 'https://doi.org/10.1104/pp.109.145318', 'https://doi.org/10.1105/tpc.108.061325', 'https://doi.org/10.3389/fpls.2021.770794', 'https://plant2.connectome.tools/alias/TOC75', 'https://doi.org/10.1016/j.csbj.2024.09.010', 'https://figshare.com/ndownloader/files/49392595', 'https://doi.org/10.1093/nar/gkq1120']\n",
      "Liens de dépôt logiciel valides pour l'article 91 : ['https://doi.org/10.1101/2024.02.29.582810', 'https://github.com/amberT15/LLM_eval)as', 'https://doi.org/10.5281/zenodo.11583224']\n",
      "Liens de dépôt logiciel valides pour l'article 92 : ['https://doi.org/10.1101/2024.09.22.614322']\n",
      "Liens de dépôt logiciel valides pour l'article 93 : ['https://doi.org/10.1126/science.abn8197', 'https://doi.org/10.1016/j.cels.2017.11.003', 'https://doi.org/10.1093/nar/gkx1153', 'https://doi.org/10.1093/bioinformatics/btu703', 'https://doi.org/10.1038/gim.2017.26', 'https://doi.org/10.1038/s41588-018-0192-y', 'https://doi.org/10.1038/nrg3031', 'https://doi.org/10.1186/s12864-', 'https://hgdownload.soe.ucsc.edu/gbdb/hg38/gnomAD/vcf/gnomad.genomes.r3.0.sites.vcf.gz', 'https://doi.org/10.1101/mcs.a006196', 'https://doi.org/10.1038/nbt.1589', 'https://doi.org/10.1016/j.ajhg.2021.08.012', 'https://doi.org/10.1038/nature24018', 'https://doi.org/10.1038/sj.ejhg.5201987', 'https://doi.org/10.1038/s41435-022-00192-6', 'https://doi.org/10.1016/j.cell.2011.09.008', 'https://doi.org/10.1038/s41525-023-00386-5', 'https://doi.org/10.1086/513473', 'https://doi.org/10.1038/s41586-', 'https://doi.org/10.1038/s41588-018-0167-z', 'https://ftp.ncbi.nlm.nih.gov/pub/clinvar/tab_delimited/archive/', 'https://doi.org/10.1016/j.molcel.2018.10.037', 'https://doi.org/10.1038/nrg2779', 'https://doi.org/10.1038/nature15393', 'https://doi.org/10.1093/bioinformatics/btq671', 'https://doi.org/10.1038/s41598-018-29325-6', 'https://doi.org/10.1038/ng.3810', 'https://doi.org/10.1186/s13229-015-', 'https://doi.org/10.1086/282718', 'https://doi.org/10.1093/bioinformatics/btu356', 'https://doi.org/10.1101/2024.09.24.614666', 'https://doi.org/10.1038/s10038-020-00845-2', 'https://doi.org/10.1186/s13073-021-00835-9', 'https://doi.org/10.1186/s12859-023-05294-0', 'https://doi.org/10.1038/nature17661', 'https://doi.org/10.1038/s41467-022-31872-6', 'https://doi.org/10.1093/bioinformatics/btv009', 'https://doi.org/10.7554/eLife.46922', 'https://hgdownload.soe', 'https://doi.org/10.1038/s41559-017-0447-5', 'https://doi.org/10.1016/j.neuron.2018.01.015', 'https://cadd.bihealth.org/static/ReleaseNotes_CADD_v1.6.pdf', 'https://doi.org/10.1002/humu.24309', 'https://doi.org/10.1038/gim.2017.14', 'https://doi.org/10.1371/journal.pcbi.1000322', 'https://doi.org/10.1038/s41467-019-11526-w', 'https://doi.org/10.1038/nature19057', 'https://doi.org/10.1136/jmedgenet-2014-102437', 'https://doi.org/10.1186/gb-2011-12-9-r84', 'https://doi.org/10.1056/NEJMoa2209046', 'https://doi.org/10.1146/annurev-genom-031714-125740', 'https://doi.org/10.1038/nrg3118', 'https://doi.org/10.1093/gbe/evad066', 'https://doi.org/10.1073/pnas.1318948111', 'https://doi.org/10.1093/hmg/11.20.2417', 'https://doi.org/10.1126/science.abj6987', 'https://doi.org/10.1146/annurev.ge.23.120189.002005', 'https://doi.org/10.1038/s41586-018-0461-z', 'https://doi.org/10.1038/ng.2892', 'https://doi.org/10.1038/nrg2986', 'https://doi.org/10.1161/CIRCGENETICS.111.960864', 'https://doi.org/10.1093/hmg/ddu400', 'https://doi.org/10.1093/nar/gky1016', 'https://doi.org/10.1186/s13073-022-01073-3', 'https://doi.org/10.1101/2022.03.20.485034', 'https://doi.org/10.1146/annurev-genom-083118-014857', 'https://doi.org/10.1038/s41525-022-00301-4', 'https://doi.org/10.1038/s41586-020-2859-', 'https://doi.org/10.1093/database/baad026', 'https://doi.org/10.1073/pnas.1114759108', 'https://doi.org/10.1093/nar/gkad989']\n",
      "Liens de dépôt logiciel valides pour l'article 94 : ['https://doi.org/10.1101/2024.09.24.614671', 'https://gephi.org/', 'https://f1000research.com/articles/10-33', 'https://www.biorxiv.org/content/10.1101/2024.05.08', 'https://vihoikka.github.io/rn_browser', 'https://github.com/vihoikka/ring_nucleases', 'http://www.rstudio.com/']\n",
      "Liens de dépôt logiciel valides pour l'article 95 : ['https://doi.org/10.1101/2024.09.24.614820', 'https://charticulator.com/app/index.html']\n",
      "Liens de dépôt logiciel valides pour l'article 96 : ['https://doi.org/10.1101/2024.09.24.614594', 'https://p53motifdb.its.albany.edu/', 'http://www.repeatmasker.org', 'https://CRAN.R-project.org/package=dplyr', 'https://shiny.posit.co/', 'https://CRAN.R-project.org/package=tidyr', 'https://p53motifDB.its.albany.edu']\n",
      "Liens de dépôt logiciel valides pour l'article 97 : ['https://doi.org/10.1101/2024.09.23.614570', 'https://yanglab.qd.sdu.edu.cn/trRosettaRNA/download/']\n",
      "Liens de dépôt logiciel valides pour l'article 98 : ['https://scpca.alexslemonade.org/', 'https://github.com/favilaco/deconv_benchmark', 'https://github.com/amitfrish/scBio', 'https://doi.org/10.1101/2023.02.06.527318']\n",
      "Liens de dépôt logiciel valides pour l'article 99 : ['https://doi.org/10.1101/2024.09.23.614585', 'https://github']\n",
      "Liens de dépôt logiciel valides pour l'article 100 : ['https://www.10xgenomics.com/datasets/mouse-brain-serial-section-2-sagittal-posterior-', 'https://doi.brainimagelibrary.org/', 'https://doi.org/', 'https://github.com/jiazhao97/INSPIRE', 'https://singlecell.broadinstitute', 'https://db.cngb.org/stomics/mosta/', 'https://www.10xgenomics.com/datasets/mouse-brain-section-coronal-1-standard-1-', 'https://doi.org/10.1038/s41586-024-07069-w', 'https://crukci.shinyapps.io/', 'https://doi', 'https://www.10xgenomics.com/datasets', 'https://doi.org/10.1101/2024.09.23.614539', 'https://www.10xgenomics.com/datasets/mouse-brain-serial-section-2-sagittal-anterior-1-', 'https://github.com/jiazhao97/INSPIRE']\n"
     ]
    }
   ],
   "source": [
    "#pdfplumber : Utilisé pour extraire du texte à partir des fichiers PDF.\n",
    "# io : Fournit des outils pour manipuler des fichiers en mémoire (ici, le PDF téléchargé).\n",
    "# re : Module pour travailler avec des expressions régulières, utilisé pour extraire des URLs dans le texte.\n",
    "\n",
    "\n",
    "\n",
    "# Connexion à la base de données SQLite\n",
    "database = sqlite3.connect('bioinformatics_articles.db')\n",
    "c = database.cursor()\n",
    "\n",
    "# Ajouter une colonne software_links pour stocker les liens de dépôt logiciel si elle n'existe pas\n",
    "try:\n",
    "    c.execute(\"ALTER TABLE articles ADD COLUMN software_links TEXT\")\n",
    "    database.commit()\n",
    "except sqlite3.OperationalError:  # colonne existe déjà \n",
    "    pass  # La colonne existe déjà\n",
    "\n",
    "# Fonction pour vérifier si un lien est valide\n",
    "def is_valid_link(link):\n",
    "    return link.startswith('http://') or link.startswith('https://')\n",
    "\n",
    "# Fonction pour nettoyer les liens partiels ou incorrects\n",
    "def clean_link(link):\n",
    "    link = link.strip(').;,')  # Supprimer les parenthèses et les points en fin de lien\n",
    "    return link if is_valid_link(link) else None\n",
    "\n",
    "# Fonction pour extraire les liens à partir d'un texte\n",
    "def extract_links_from_text(text):\n",
    "    return re.findall(r'https?://\\S+', text)\n",
    "\n",
    "\n",
    "#fonction principale\n",
    "# Fonction pour extraire les liens de dépôt logiciel\n",
    "def extract_software_links():\n",
    "    # Récupérer tous les articles de la base de données\n",
    "    c.execute(\"SELECT id, title, pdf_link, abstract, software_links FROM articles\")\n",
    "    articles = c.fetchall() #récupère tous les articles \n",
    "\n",
    "    for article in articles:\n",
    "        article_id, title, pdf_link, abstract, existing_links = article\n",
    "\n",
    "        # Si l'article a déjà des liens de dépôt logiciel, on l'ignore\n",
    "        if existing_links:\n",
    "            print(f\"Article {article_id} déjà traité, on passe au suivant.\")\n",
    "            continue\n",
    "\n",
    "        software_links_set = set()  # Utiliser un set pour éliminer les doublons\n",
    "\n",
    "            \n",
    "            #extraire lien a partir du pdf\n",
    "        # Vérifier le lien PDF\n",
    "        if pdf_link and pdf_link != 'Lien PDF non disponible':\n",
    "            try:\n",
    "                # Télécharger le fichier PDF\n",
    "                response = requests.get(pdf_link)\n",
    "                response.raise_for_status()  # Vérifie si la requête a réussi\n",
    "\n",
    "                # Utiliser pdfplumber pour extraire le texte du PDF\n",
    "                with pdfplumber.open(io.BytesIO(response.content)) as pdf:\n",
    "                    for page in pdf.pages:\n",
    "                        text = page.extract_text()\n",
    "                        if text:\n",
    "                            # Rechercher des liens de dépôt logiciel dans le texte\n",
    "                            software_links_set.update(extract_links_from_text(text))\n",
    "\n",
    "            except Exception as e:\n",
    "                print(f\"Erreur lors de l'ouverture du PDF pour l'article {article_id}: {e}\")\n",
    "\n",
    "        # Si aucun lien n'a été trouvé dans le PDF, vérifier l'abstract\n",
    "        if not software_links_set and abstract != 'Abstract non disponible':\n",
    "            software_links_set.update(extract_links_from_text(abstract))\n",
    "\n",
    "        # Nettoyage des liens et validation\n",
    "        valid_links = [clean_link(link) for link in software_links_set if clean_link(link)]\n",
    "            #lien trouvés sont nettoyés et valide par clean_link() et is_valid_link().\n",
    "\n",
    "        # Mise à jour de la base de données avec les liens de dépôt logiciel\n",
    "        #liens valides trouvés, ajoutés à la base de données dans la colonne software_links \n",
    "        if valid_links:\n",
    "            print(f\"Liens de dépôt logiciel valides pour l'article {article_id} : {valid_links}\")\n",
    "            c.execute(\"UPDATE articles SET software_links = ? WHERE id = ?\", (', '.join(valid_links), article_id))\n",
    "        else:\n",
    "            print(f\"Aucun lien de dépôt logiciel trouvé pour l'article {article_id}\")\n",
    "\n",
    "        time.sleep(1)  # Pause pour éviter de surcharger le serveur\n",
    "\n",
    "    # Sauvegarder les modifications dans la base de données\n",
    "    database.commit()\n",
    "\n",
    "# Appeler la fonction d'extraction\n",
    "extract_software_links()\n",
    "\n",
    "# Fermer la connexion à la base de données\n",
    "#database.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
